{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Módulos"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "En Python, cada script o archivo de código fuente, se denominan módulos. Estos módulos, a la vez, pueden formar parte de paquetes. Un paquete, es una carpeta que contiene archivos `.py`. Por ejemplo, si guardáramos el contenido de la función para obtener codones a partir de un string, y le ponemos la extensión `py` y lo guardamos como `get_codons.py` sería un script de python, si el script está en la misma carpeta del notebook yo podría importarlo así:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import get_codons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Eso me permite reutilizar mi código. Afortunadamente python contienen módulos `built-in`, métodos integrados. Además, podemos instalar nuevos paquetes que contienen módulos con `pip`, el instalador oficial de Python, o con `conda`, el gestor de paquetes de Anaconda Inc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import calendar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     July 2020\n",
      "Mo Tu We Th Fr Sa Su\n",
      "       1  2  3  4  5\n",
      " 6  7  8  9 10 11 12\n",
      "13 14 15 16 17 18 19\n",
      "20 21 22 23 24 25 26\n",
      "27 28 29 30 31\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(calendar.month(2020, 7))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Es posible también, abreviar los namespaces mediante un alias. Para ello, durante la importación, se asigna la palabra clave as seguida del alias con el cuál nos referiremos en el futuro a ese namespace importado:\n",
    "\n",
    "- `import modulo`\n",
    "- `import modulo as m`\n",
    "- `import paquete.modulo1 as pm`\n",
    "- `import paquete.subpaquete.modulo1 as psm`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import calendar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(calendar.month(2020, 7))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from calendar import month"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     July 2020\n",
      "Mo Tu We Th Fr Sa Su\n",
      "       1  2  3  4  5\n",
      " 6  7  8  9 10 11 12\n",
      "13 14 15 16 17 18 19\n",
      "20 21 22 23 24 25 26\n",
      "27 28 29 30 31\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(month(2020, 7))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import calendar as c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     July 2020\n",
      "Mo Tu We Th Fr Sa Su\n",
      "       1  2  3  4  5\n",
      " 6  7  8  9 10 11 12\n",
      "13 14 15 16 17 18 19\n",
      "20 21 22 23 24 25 26\n",
      "27 28 29 30 31\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(c.month(2020, 7))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<center><img src=\"imgs/pandas.png\" width=700 height=700/></center>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import atgenomics as atg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function read_excel in module pandas.io.excel:\n",
      "\n",
      "read_excel(io, sheetname=0, header=0, skiprows=None, skip_footer=0, index_col=None, names=None, parse_cols=None, parse_dates=False, date_parser=None, na_values=None, thousands=None, convert_float=True, has_index_names=None, converters=None, true_values=None, false_values=None, engine=None, squeeze=False, **kwds)\n",
      "    Read an Excel table into a pandas DataFrame\n",
      "    \n",
      "    Parameters\n",
      "    ----------\n",
      "    io : string, path object (pathlib.Path or py._path.local.LocalPath),\n",
      "        file-like object, pandas ExcelFile, or xlrd workbook.\n",
      "        The string could be a URL. Valid URL schemes include http, ftp, s3,\n",
      "        and file. For file URLs, a host is expected. For instance, a local\n",
      "        file could be file://localhost/path/to/workbook.xlsx\n",
      "    sheetname : string, int, mixed list of strings/ints, or None, default 0\n",
      "    \n",
      "        Strings are used for sheet names, Integers are used in zero-indexed\n",
      "        sheet positions.\n",
      "    \n",
      "        Lists of strings/integers are used to request multiple sheets.\n",
      "    \n",
      "        Specify None to get all sheets.\n",
      "    \n",
      "        str|int -> DataFrame is returned.\n",
      "        list|None -> Dict of DataFrames is returned, with keys representing\n",
      "        sheets.\n",
      "    \n",
      "        Available Cases\n",
      "    \n",
      "        * Defaults to 0 -> 1st sheet as a DataFrame\n",
      "        * 1 -> 2nd sheet as a DataFrame\n",
      "        * \"Sheet1\" -> 1st sheet as a DataFrame\n",
      "        * [0,1,\"Sheet5\"] -> 1st, 2nd & 5th sheet as a dictionary of DataFrames\n",
      "        * None -> All sheets as a dictionary of DataFrames\n",
      "    \n",
      "    header : int, list of ints, default 0\n",
      "        Row (0-indexed) to use for the column labels of the parsed\n",
      "        DataFrame. If a list of integers is passed those row positions will\n",
      "        be combined into a ``MultiIndex``\n",
      "    skiprows : list-like\n",
      "        Rows to skip at the beginning (0-indexed)\n",
      "    skip_footer : int, default 0\n",
      "        Rows at the end to skip (0-indexed)\n",
      "    index_col : int, list of ints, default None\n",
      "        Column (0-indexed) to use as the row labels of the DataFrame.\n",
      "        Pass None if there is no such column.  If a list is passed,\n",
      "        those columns will be combined into a ``MultiIndex``\n",
      "    names : array-like, default None\n",
      "        List of column names to use. If file contains no header row,\n",
      "        then you should explicitly pass header=None\n",
      "    converters : dict, default None\n",
      "        Dict of functions for converting values in certain columns. Keys can\n",
      "        either be integers or column labels, values are functions that take one\n",
      "        input argument, the Excel cell content, and return the transformed\n",
      "        content.\n",
      "    true_values : list, default None\n",
      "        Values to consider as True\n",
      "    \n",
      "        .. versionadded:: 0.19.0\n",
      "    \n",
      "    false_values : list, default None\n",
      "        Values to consider as False\n",
      "    \n",
      "        .. versionadded:: 0.19.0\n",
      "    \n",
      "    parse_cols : int or list, default None\n",
      "        * If None then parse all columns,\n",
      "        * If int then indicates last column to be parsed\n",
      "        * If list of ints then indicates list of column numbers to be parsed\n",
      "        * If string then indicates comma separated list of column names and\n",
      "          column ranges (e.g. \"A:E\" or \"A,C,E:F\")\n",
      "    squeeze : boolean, default False\n",
      "        If the parsed data only contains one column then return a Series\n",
      "    na_values : scalar, str, list-like, or dict, default None\n",
      "        Additional strings to recognize as NA/NaN. If dict passed, specific\n",
      "        per-column NA values. By default the following values are interpreted\n",
      "        as NaN: '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN', '-nan',\n",
      "    '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL', 'NaN', 'nan'.\n",
      "    thousands : str, default None\n",
      "        Thousands separator for parsing string columns to numeric.  Note that\n",
      "        this parameter is only necessary for columns stored as TEXT in Excel,\n",
      "        any numeric columns will automatically be parsed, regardless of display\n",
      "        format.\n",
      "    keep_default_na : bool, default True\n",
      "        If na_values are specified and keep_default_na is False the default NaN\n",
      "        values are overridden, otherwise they're appended to.\n",
      "    verbose : boolean, default False\n",
      "        Indicate number of NA values placed in non-numeric columns\n",
      "    engine: string, default None\n",
      "        If io is not a buffer or path, this must be set to identify io.\n",
      "        Acceptable values are None or xlrd\n",
      "    convert_float : boolean, default True\n",
      "        convert integral floats to int (i.e., 1.0 --> 1). If False, all numeric\n",
      "        data will be read in as floats: Excel stores all numbers as floats\n",
      "        internally\n",
      "    has_index_names : boolean, default None\n",
      "        DEPRECATED: for version 0.17+ index names will be automatically\n",
      "        inferred based on index_col.  To read Excel output from 0.16.2 and\n",
      "        prior that had saved index names, use True.\n",
      "    \n",
      "    Returns\n",
      "    -------\n",
      "    parsed : DataFrame or Dict of DataFrames\n",
      "        DataFrame from the passed in Excel file.  See notes in sheetname\n",
      "        argument for more information on when a Dict of Dataframes is returned.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(pd.read_excel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function read_table in module pandas.io.parsers:\n",
      "\n",
      "read_table(filepath_or_buffer, sep='\\t', delimiter=None, header='infer', names=None, index_col=None, usecols=None, squeeze=False, prefix=None, mangle_dupe_cols=True, dtype=None, engine=None, converters=None, true_values=None, false_values=None, skipinitialspace=False, skiprows=None, nrows=None, na_values=None, keep_default_na=True, na_filter=True, verbose=False, skip_blank_lines=True, parse_dates=False, infer_datetime_format=False, keep_date_col=False, date_parser=None, dayfirst=False, iterator=False, chunksize=None, compression='infer', thousands=None, decimal=b'.', lineterminator=None, quotechar='\"', quoting=0, escapechar=None, comment=None, encoding=None, dialect=None, tupleize_cols=False, error_bad_lines=True, warn_bad_lines=True, skipfooter=0, skip_footer=0, doublequote=True, delim_whitespace=False, as_recarray=False, compact_ints=False, use_unsigned=False, low_memory=True, buffer_lines=None, memory_map=False, float_precision=None)\n",
      "    Read general delimited file into DataFrame\n",
      "    \n",
      "    Also supports optionally iterating or breaking of the file\n",
      "    into chunks.\n",
      "    \n",
      "    Additional help can be found in the `online docs for IO Tools\n",
      "    <http://pandas.pydata.org/pandas-docs/stable/io.html>`_.\n",
      "    \n",
      "    Parameters\n",
      "    ----------\n",
      "    filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or any object with a read() method (such as a file handle or StringIO)\n",
      "        The string could be a URL. Valid URL schemes include http, ftp, s3, and\n",
      "        file. For file URLs, a host is expected. For instance, a local file could\n",
      "        be file ://localhost/path/to/table.csv\n",
      "    sep : str, default \\t (tab-stop)\n",
      "        Delimiter to use. If sep is None, will try to automatically determine\n",
      "        this. Separators longer than 1 character and different from ``'\\s+'`` will\n",
      "        be interpreted as regular expressions, will force use of the python parsing\n",
      "        engine and will ignore quotes in the data. Regex example: ``'\\r\\t'``\n",
      "    delimiter : str, default ``None``\n",
      "        Alternative argument name for sep.\n",
      "    delim_whitespace : boolean, default False\n",
      "        Specifies whether or not whitespace (e.g. ``' '`` or ``'    '``) will be\n",
      "        used as the sep. Equivalent to setting ``sep='\\s+'``. If this option\n",
      "        is set to True, nothing should be passed in for the ``delimiter``\n",
      "        parameter.\n",
      "    \n",
      "        .. versionadded:: 0.18.1 support for the Python parser.\n",
      "    \n",
      "    header : int or list of ints, default 'infer'\n",
      "        Row number(s) to use as the column names, and the start of the data.\n",
      "        Default behavior is as if set to 0 if no ``names`` passed, otherwise\n",
      "        ``None``. Explicitly pass ``header=0`` to be able to replace existing\n",
      "        names. The header can be a list of integers that specify row locations for\n",
      "        a multi-index on the columns e.g. [0,1,3]. Intervening rows that are not\n",
      "        specified will be skipped (e.g. 2 in this example is skipped). Note that\n",
      "        this parameter ignores commented lines and empty lines if\n",
      "        ``skip_blank_lines=True``, so header=0 denotes the first line of data\n",
      "        rather than the first line of the file.\n",
      "    names : array-like, default None\n",
      "        List of column names to use. If file contains no header row, then you\n",
      "        should explicitly pass header=None. Duplicates in this list are not\n",
      "        allowed unless mangle_dupe_cols=True, which is the default.\n",
      "    index_col : int or sequence or False, default None\n",
      "        Column to use as the row labels of the DataFrame. If a sequence is given, a\n",
      "        MultiIndex is used. If you have a malformed file with delimiters at the end\n",
      "        of each line, you might consider index_col=False to force pandas to _not_\n",
      "        use the first column as the index (row names)\n",
      "    usecols : array-like, default None\n",
      "        Return a subset of the columns. All elements in this array must either\n",
      "        be positional (i.e. integer indices into the document columns) or strings\n",
      "        that correspond to column names provided either by the user in `names` or\n",
      "        inferred from the document header row(s). For example, a valid `usecols`\n",
      "        parameter would be [0, 1, 2] or ['foo', 'bar', 'baz']. Using this parameter\n",
      "        results in much faster parsing time and lower memory usage.\n",
      "    as_recarray : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version. Please call\n",
      "        `pd.read_csv(...).to_records()` instead.\n",
      "    \n",
      "        Return a NumPy recarray instead of a DataFrame after parsing the data.\n",
      "        If set to True, this option takes precedence over the `squeeze` parameter.\n",
      "        In addition, as row indices are not available in such a format, the\n",
      "        `index_col` parameter will be ignored.\n",
      "    squeeze : boolean, default False\n",
      "        If the parsed data only contains one column then return a Series\n",
      "    prefix : str, default None\n",
      "        Prefix to add to column numbers when no header, e.g. 'X' for X0, X1, ...\n",
      "    mangle_dupe_cols : boolean, default True\n",
      "        Duplicate columns will be specified as 'X.0'...'X.N', rather than\n",
      "        'X'...'X'. Passing in False will cause data to be overwritten if there\n",
      "        are duplicate names in the columns.\n",
      "    dtype : Type name or dict of column -> type, default None\n",
      "        Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}\n",
      "        (Unsupported with engine='python'). Use `str` or `object` to preserve and\n",
      "        not interpret dtype.\n",
      "    engine : {'c', 'python'}, optional\n",
      "        Parser engine to use. The C engine is faster while the python engine is\n",
      "        currently more feature-complete.\n",
      "    converters : dict, default None\n",
      "        Dict of functions for converting values in certain columns. Keys can either\n",
      "        be integers or column labels\n",
      "    true_values : list, default None\n",
      "        Values to consider as True\n",
      "    false_values : list, default None\n",
      "        Values to consider as False\n",
      "    skipinitialspace : boolean, default False\n",
      "        Skip spaces after delimiter.\n",
      "    skiprows : list-like or integer, default None\n",
      "        Line numbers to skip (0-indexed) or number of lines to skip (int)\n",
      "        at the start of the file\n",
      "    skipfooter : int, default 0\n",
      "        Number of lines at bottom of file to skip (Unsupported with engine='c')\n",
      "    skip_footer : int, default 0\n",
      "        DEPRECATED: use the `skipfooter` parameter instead, as they are identical\n",
      "    nrows : int, default None\n",
      "        Number of rows of file to read. Useful for reading pieces of large files\n",
      "    na_values : scalar, str, list-like, or dict, default None\n",
      "        Additional strings to recognize as NA/NaN. If dict passed, specific\n",
      "        per-column NA values.  By default the following values are interpreted as\n",
      "        NaN: '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN', '-nan',\n",
      "    '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL', 'NaN', 'nan'`.\n",
      "    keep_default_na : bool, default True\n",
      "        If na_values are specified and keep_default_na is False the default NaN\n",
      "        values are overridden, otherwise they're appended to.\n",
      "    na_filter : boolean, default True\n",
      "        Detect missing value markers (empty strings and the value of na_values). In\n",
      "        data without any NAs, passing na_filter=False can improve the performance\n",
      "        of reading a large file\n",
      "    verbose : boolean, default False\n",
      "        Indicate number of NA values placed in non-numeric columns\n",
      "    skip_blank_lines : boolean, default True\n",
      "        If True, skip over blank lines rather than interpreting as NaN values\n",
      "    parse_dates : boolean or list of ints or names or list of lists or dict, default False\n",
      "    \n",
      "        * boolean. If True -> try parsing the index.\n",
      "        * list of ints or names. e.g. If [1, 2, 3] -> try parsing columns 1, 2, 3\n",
      "          each as a separate date column.\n",
      "        * list of lists. e.g.  If [[1, 3]] -> combine columns 1 and 3 and parse as\n",
      "            a single date column.\n",
      "        * dict, e.g. {'foo' : [1, 3]} -> parse columns 1, 3 as date and call result\n",
      "          'foo'\n",
      "    \n",
      "        Note: A fast-path exists for iso8601-formatted dates.\n",
      "    infer_datetime_format : boolean, default False\n",
      "        If True and parse_dates is enabled, pandas will attempt to infer the format\n",
      "        of the datetime strings in the columns, and if it can be inferred, switch\n",
      "        to a faster method of parsing them. In some cases this can increase the\n",
      "        parsing speed by ~5-10x.\n",
      "    keep_date_col : boolean, default False\n",
      "        If True and parse_dates specifies combining multiple columns then\n",
      "        keep the original columns.\n",
      "    date_parser : function, default None\n",
      "        Function to use for converting a sequence of string columns to an array of\n",
      "        datetime instances. The default uses ``dateutil.parser.parser`` to do the\n",
      "        conversion. Pandas will try to call date_parser in three different ways,\n",
      "        advancing to the next if an exception occurs: 1) Pass one or more arrays\n",
      "        (as defined by parse_dates) as arguments; 2) concatenate (row-wise) the\n",
      "        string values from the columns defined by parse_dates into a single array\n",
      "        and pass that; and 3) call date_parser once for each row using one or more\n",
      "        strings (corresponding to the columns defined by parse_dates) as arguments.\n",
      "    dayfirst : boolean, default False\n",
      "        DD/MM format dates, international and European format\n",
      "    iterator : boolean, default False\n",
      "        Return TextFileReader object for iteration or getting chunks with\n",
      "        ``get_chunk()``.\n",
      "    chunksize : int, default None\n",
      "        Return TextFileReader object for iteration. `See IO Tools docs for more\n",
      "        information\n",
      "        <http://pandas.pydata.org/pandas-docs/stable/io.html#io-chunking>`_ on\n",
      "        ``iterator`` and ``chunksize``.\n",
      "    compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default 'infer'\n",
      "        For on-the-fly decompression of on-disk data. If 'infer', then use gzip,\n",
      "        bz2, zip or xz if filepath_or_buffer is a string ending in '.gz', '.bz2',\n",
      "        '.zip', or 'xz', respectively, and no decompression otherwise. If using\n",
      "        'zip', the ZIP file must contain only one data file to be read in.\n",
      "        Set to None for no decompression.\n",
      "    \n",
      "        .. versionadded:: 0.18.1 support for 'zip' and 'xz' compression.\n",
      "    \n",
      "    thousands : str, default None\n",
      "        Thousands separator\n",
      "    decimal : str, default '.'\n",
      "        Character to recognize as decimal point (e.g. use ',' for European data).\n",
      "    float_precision : string, default None\n",
      "        Specifies which converter the C engine should use for floating-point\n",
      "        values. The options are `None` for the ordinary converter,\n",
      "        `high` for the high-precision converter, and `round_trip` for the\n",
      "        round-trip converter.\n",
      "    lineterminator : str (length 1), default None\n",
      "        Character to break file into lines. Only valid with C parser.\n",
      "    quotechar : str (length 1), optional\n",
      "        The character used to denote the start and end of a quoted item. Quoted\n",
      "        items can include the delimiter and it will be ignored.\n",
      "    quoting : int or csv.QUOTE_* instance, default 0\n",
      "        Control field quoting behavior per ``csv.QUOTE_*`` constants. Use one of\n",
      "        QUOTE_MINIMAL (0), QUOTE_ALL (1), QUOTE_NONNUMERIC (2) or QUOTE_NONE (3).\n",
      "    doublequote : boolean, default ``True``\n",
      "       When quotechar is specified and quoting is not ``QUOTE_NONE``, indicate\n",
      "       whether or not to interpret two consecutive quotechar elements INSIDE a\n",
      "       field as a single ``quotechar`` element.\n",
      "    escapechar : str (length 1), default None\n",
      "        One-character string used to escape delimiter when quoting is QUOTE_NONE.\n",
      "    comment : str, default None\n",
      "        Indicates remainder of line should not be parsed. If found at the beginning\n",
      "        of a line, the line will be ignored altogether. This parameter must be a\n",
      "        single character. Like empty lines (as long as ``skip_blank_lines=True``),\n",
      "        fully commented lines are ignored by the parameter `header` but not by\n",
      "        `skiprows`. For example, if comment='#', parsing '#empty\\na,b,c\\n1,2,3'\n",
      "        with `header=0` will result in 'a,b,c' being\n",
      "        treated as the header.\n",
      "    encoding : str, default None\n",
      "        Encoding to use for UTF when reading/writing (ex. 'utf-8'). `List of Python\n",
      "        standard encodings\n",
      "        <https://docs.python.org/3/library/codecs.html#standard-encodings>`_\n",
      "    dialect : str or csv.Dialect instance, default None\n",
      "        If None defaults to Excel dialect. Ignored if sep longer than 1 char\n",
      "        See csv.Dialect documentation for more details\n",
      "    tupleize_cols : boolean, default False\n",
      "        Leave a list of tuples on columns as is (default is to convert to\n",
      "        a Multi Index on the columns)\n",
      "    error_bad_lines : boolean, default True\n",
      "        Lines with too many fields (e.g. a csv line with too many commas) will by\n",
      "        default cause an exception to be raised, and no DataFrame will be returned.\n",
      "        If False, then these \"bad lines\" will dropped from the DataFrame that is\n",
      "        returned. (Only valid with C parser)\n",
      "    warn_bad_lines : boolean, default True\n",
      "        If error_bad_lines is False, and warn_bad_lines is True, a warning for each\n",
      "        \"bad line\" will be output. (Only valid with C parser).\n",
      "    low_memory : boolean, default True\n",
      "        Internally process the file in chunks, resulting in lower memory use\n",
      "        while parsing, but possibly mixed type inference.  To ensure no mixed\n",
      "        types either set False, or specify the type with the `dtype` parameter.\n",
      "        Note that the entire file is read into a single DataFrame regardless,\n",
      "        use the `chunksize` or `iterator` parameter to return the data in chunks.\n",
      "        (Only valid with C parser)\n",
      "    buffer_lines : int, default None\n",
      "        DEPRECATED: this argument will be removed in a future version because its\n",
      "        value is not respected by the parser\n",
      "    compact_ints : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version\n",
      "    \n",
      "        If compact_ints is True, then for any column that is of integer dtype,\n",
      "        the parser will attempt to cast it as the smallest integer dtype possible,\n",
      "        either signed or unsigned depending on the specification from the\n",
      "        `use_unsigned` parameter.\n",
      "    use_unsigned : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version\n",
      "    \n",
      "        If integer columns are being compacted (i.e. `compact_ints=True`), specify\n",
      "        whether the column should be compacted to the smallest signed or unsigned\n",
      "        integer dtype.\n",
      "    memory_map : boolean, default False\n",
      "        If a filepath is provided for `filepath_or_buffer`, map the file object\n",
      "        directly onto memory and access the data directly from there. Using this\n",
      "        option can improve performance because there is no longer any I/O overhead.\n",
      "    \n",
      "    Returns\n",
      "    -------\n",
      "    result : DataFrame or TextParser\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(pd.read_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function read_csv in module pandas.io.parsers:\n",
      "\n",
      "read_csv(filepath_or_buffer, sep=',', delimiter=None, header='infer', names=None, index_col=None, usecols=None, squeeze=False, prefix=None, mangle_dupe_cols=True, dtype=None, engine=None, converters=None, true_values=None, false_values=None, skipinitialspace=False, skiprows=None, nrows=None, na_values=None, keep_default_na=True, na_filter=True, verbose=False, skip_blank_lines=True, parse_dates=False, infer_datetime_format=False, keep_date_col=False, date_parser=None, dayfirst=False, iterator=False, chunksize=None, compression='infer', thousands=None, decimal=b'.', lineterminator=None, quotechar='\"', quoting=0, escapechar=None, comment=None, encoding=None, dialect=None, tupleize_cols=False, error_bad_lines=True, warn_bad_lines=True, skipfooter=0, skip_footer=0, doublequote=True, delim_whitespace=False, as_recarray=False, compact_ints=False, use_unsigned=False, low_memory=True, buffer_lines=None, memory_map=False, float_precision=None)\n",
      "    Read CSV (comma-separated) file into DataFrame\n",
      "    \n",
      "    Also supports optionally iterating or breaking of the file\n",
      "    into chunks.\n",
      "    \n",
      "    Additional help can be found in the `online docs for IO Tools\n",
      "    <http://pandas.pydata.org/pandas-docs/stable/io.html>`_.\n",
      "    \n",
      "    Parameters\n",
      "    ----------\n",
      "    filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or any object with a read() method (such as a file handle or StringIO)\n",
      "        The string could be a URL. Valid URL schemes include http, ftp, s3, and\n",
      "        file. For file URLs, a host is expected. For instance, a local file could\n",
      "        be file ://localhost/path/to/table.csv\n",
      "    sep : str, default ','\n",
      "        Delimiter to use. If sep is None, will try to automatically determine\n",
      "        this. Separators longer than 1 character and different from ``'\\s+'`` will\n",
      "        be interpreted as regular expressions, will force use of the python parsing\n",
      "        engine and will ignore quotes in the data. Regex example: ``'\\r\\t'``\n",
      "    delimiter : str, default ``None``\n",
      "        Alternative argument name for sep.\n",
      "    delim_whitespace : boolean, default False\n",
      "        Specifies whether or not whitespace (e.g. ``' '`` or ``'    '``) will be\n",
      "        used as the sep. Equivalent to setting ``sep='\\s+'``. If this option\n",
      "        is set to True, nothing should be passed in for the ``delimiter``\n",
      "        parameter.\n",
      "    \n",
      "        .. versionadded:: 0.18.1 support for the Python parser.\n",
      "    \n",
      "    header : int or list of ints, default 'infer'\n",
      "        Row number(s) to use as the column names, and the start of the data.\n",
      "        Default behavior is as if set to 0 if no ``names`` passed, otherwise\n",
      "        ``None``. Explicitly pass ``header=0`` to be able to replace existing\n",
      "        names. The header can be a list of integers that specify row locations for\n",
      "        a multi-index on the columns e.g. [0,1,3]. Intervening rows that are not\n",
      "        specified will be skipped (e.g. 2 in this example is skipped). Note that\n",
      "        this parameter ignores commented lines and empty lines if\n",
      "        ``skip_blank_lines=True``, so header=0 denotes the first line of data\n",
      "        rather than the first line of the file.\n",
      "    names : array-like, default None\n",
      "        List of column names to use. If file contains no header row, then you\n",
      "        should explicitly pass header=None. Duplicates in this list are not\n",
      "        allowed unless mangle_dupe_cols=True, which is the default.\n",
      "    index_col : int or sequence or False, default None\n",
      "        Column to use as the row labels of the DataFrame. If a sequence is given, a\n",
      "        MultiIndex is used. If you have a malformed file with delimiters at the end\n",
      "        of each line, you might consider index_col=False to force pandas to _not_\n",
      "        use the first column as the index (row names)\n",
      "    usecols : array-like, default None\n",
      "        Return a subset of the columns. All elements in this array must either\n",
      "        be positional (i.e. integer indices into the document columns) or strings\n",
      "        that correspond to column names provided either by the user in `names` or\n",
      "        inferred from the document header row(s). For example, a valid `usecols`\n",
      "        parameter would be [0, 1, 2] or ['foo', 'bar', 'baz']. Using this parameter\n",
      "        results in much faster parsing time and lower memory usage.\n",
      "    as_recarray : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version. Please call\n",
      "        `pd.read_csv(...).to_records()` instead.\n",
      "    \n",
      "        Return a NumPy recarray instead of a DataFrame after parsing the data.\n",
      "        If set to True, this option takes precedence over the `squeeze` parameter.\n",
      "        In addition, as row indices are not available in such a format, the\n",
      "        `index_col` parameter will be ignored.\n",
      "    squeeze : boolean, default False\n",
      "        If the parsed data only contains one column then return a Series\n",
      "    prefix : str, default None\n",
      "        Prefix to add to column numbers when no header, e.g. 'X' for X0, X1, ...\n",
      "    mangle_dupe_cols : boolean, default True\n",
      "        Duplicate columns will be specified as 'X.0'...'X.N', rather than\n",
      "        'X'...'X'. Passing in False will cause data to be overwritten if there\n",
      "        are duplicate names in the columns.\n",
      "    dtype : Type name or dict of column -> type, default None\n",
      "        Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}\n",
      "        (Unsupported with engine='python'). Use `str` or `object` to preserve and\n",
      "        not interpret dtype.\n",
      "    engine : {'c', 'python'}, optional\n",
      "        Parser engine to use. The C engine is faster while the python engine is\n",
      "        currently more feature-complete.\n",
      "    converters : dict, default None\n",
      "        Dict of functions for converting values in certain columns. Keys can either\n",
      "        be integers or column labels\n",
      "    true_values : list, default None\n",
      "        Values to consider as True\n",
      "    false_values : list, default None\n",
      "        Values to consider as False\n",
      "    skipinitialspace : boolean, default False\n",
      "        Skip spaces after delimiter.\n",
      "    skiprows : list-like or integer, default None\n",
      "        Line numbers to skip (0-indexed) or number of lines to skip (int)\n",
      "        at the start of the file\n",
      "    skipfooter : int, default 0\n",
      "        Number of lines at bottom of file to skip (Unsupported with engine='c')\n",
      "    skip_footer : int, default 0\n",
      "        DEPRECATED: use the `skipfooter` parameter instead, as they are identical\n",
      "    nrows : int, default None\n",
      "        Number of rows of file to read. Useful for reading pieces of large files\n",
      "    na_values : scalar, str, list-like, or dict, default None\n",
      "        Additional strings to recognize as NA/NaN. If dict passed, specific\n",
      "        per-column NA values.  By default the following values are interpreted as\n",
      "        NaN: '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN', '-nan',\n",
      "    '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL', 'NaN', 'nan'`.\n",
      "    keep_default_na : bool, default True\n",
      "        If na_values are specified and keep_default_na is False the default NaN\n",
      "        values are overridden, otherwise they're appended to.\n",
      "    na_filter : boolean, default True\n",
      "        Detect missing value markers (empty strings and the value of na_values). In\n",
      "        data without any NAs, passing na_filter=False can improve the performance\n",
      "        of reading a large file\n",
      "    verbose : boolean, default False\n",
      "        Indicate number of NA values placed in non-numeric columns\n",
      "    skip_blank_lines : boolean, default True\n",
      "        If True, skip over blank lines rather than interpreting as NaN values\n",
      "    parse_dates : boolean or list of ints or names or list of lists or dict, default False\n",
      "    \n",
      "        * boolean. If True -> try parsing the index.\n",
      "        * list of ints or names. e.g. If [1, 2, 3] -> try parsing columns 1, 2, 3\n",
      "          each as a separate date column.\n",
      "        * list of lists. e.g.  If [[1, 3]] -> combine columns 1 and 3 and parse as\n",
      "            a single date column.\n",
      "        * dict, e.g. {'foo' : [1, 3]} -> parse columns 1, 3 as date and call result\n",
      "          'foo'\n",
      "    \n",
      "        Note: A fast-path exists for iso8601-formatted dates.\n",
      "    infer_datetime_format : boolean, default False\n",
      "        If True and parse_dates is enabled, pandas will attempt to infer the format\n",
      "        of the datetime strings in the columns, and if it can be inferred, switch\n",
      "        to a faster method of parsing them. In some cases this can increase the\n",
      "        parsing speed by ~5-10x.\n",
      "    keep_date_col : boolean, default False\n",
      "        If True and parse_dates specifies combining multiple columns then\n",
      "        keep the original columns.\n",
      "    date_parser : function, default None\n",
      "        Function to use for converting a sequence of string columns to an array of\n",
      "        datetime instances. The default uses ``dateutil.parser.parser`` to do the\n",
      "        conversion. Pandas will try to call date_parser in three different ways,\n",
      "        advancing to the next if an exception occurs: 1) Pass one or more arrays\n",
      "        (as defined by parse_dates) as arguments; 2) concatenate (row-wise) the\n",
      "        string values from the columns defined by parse_dates into a single array\n",
      "        and pass that; and 3) call date_parser once for each row using one or more\n",
      "        strings (corresponding to the columns defined by parse_dates) as arguments.\n",
      "    dayfirst : boolean, default False\n",
      "        DD/MM format dates, international and European format\n",
      "    iterator : boolean, default False\n",
      "        Return TextFileReader object for iteration or getting chunks with\n",
      "        ``get_chunk()``.\n",
      "    chunksize : int, default None\n",
      "        Return TextFileReader object for iteration. `See IO Tools docs for more\n",
      "        information\n",
      "        <http://pandas.pydata.org/pandas-docs/stable/io.html#io-chunking>`_ on\n",
      "        ``iterator`` and ``chunksize``.\n",
      "    compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default 'infer'\n",
      "        For on-the-fly decompression of on-disk data. If 'infer', then use gzip,\n",
      "        bz2, zip or xz if filepath_or_buffer is a string ending in '.gz', '.bz2',\n",
      "        '.zip', or 'xz', respectively, and no decompression otherwise. If using\n",
      "        'zip', the ZIP file must contain only one data file to be read in.\n",
      "        Set to None for no decompression.\n",
      "    \n",
      "        .. versionadded:: 0.18.1 support for 'zip' and 'xz' compression.\n",
      "    \n",
      "    thousands : str, default None\n",
      "        Thousands separator\n",
      "    decimal : str, default '.'\n",
      "        Character to recognize as decimal point (e.g. use ',' for European data).\n",
      "    float_precision : string, default None\n",
      "        Specifies which converter the C engine should use for floating-point\n",
      "        values. The options are `None` for the ordinary converter,\n",
      "        `high` for the high-precision converter, and `round_trip` for the\n",
      "        round-trip converter.\n",
      "    lineterminator : str (length 1), default None\n",
      "        Character to break file into lines. Only valid with C parser.\n",
      "    quotechar : str (length 1), optional\n",
      "        The character used to denote the start and end of a quoted item. Quoted\n",
      "        items can include the delimiter and it will be ignored.\n",
      "    quoting : int or csv.QUOTE_* instance, default 0\n",
      "        Control field quoting behavior per ``csv.QUOTE_*`` constants. Use one of\n",
      "        QUOTE_MINIMAL (0), QUOTE_ALL (1), QUOTE_NONNUMERIC (2) or QUOTE_NONE (3).\n",
      "    doublequote : boolean, default ``True``\n",
      "       When quotechar is specified and quoting is not ``QUOTE_NONE``, indicate\n",
      "       whether or not to interpret two consecutive quotechar elements INSIDE a\n",
      "       field as a single ``quotechar`` element.\n",
      "    escapechar : str (length 1), default None\n",
      "        One-character string used to escape delimiter when quoting is QUOTE_NONE.\n",
      "    comment : str, default None\n",
      "        Indicates remainder of line should not be parsed. If found at the beginning\n",
      "        of a line, the line will be ignored altogether. This parameter must be a\n",
      "        single character. Like empty lines (as long as ``skip_blank_lines=True``),\n",
      "        fully commented lines are ignored by the parameter `header` but not by\n",
      "        `skiprows`. For example, if comment='#', parsing '#empty\\na,b,c\\n1,2,3'\n",
      "        with `header=0` will result in 'a,b,c' being\n",
      "        treated as the header.\n",
      "    encoding : str, default None\n",
      "        Encoding to use for UTF when reading/writing (ex. 'utf-8'). `List of Python\n",
      "        standard encodings\n",
      "        <https://docs.python.org/3/library/codecs.html#standard-encodings>`_\n",
      "    dialect : str or csv.Dialect instance, default None\n",
      "        If None defaults to Excel dialect. Ignored if sep longer than 1 char\n",
      "        See csv.Dialect documentation for more details\n",
      "    tupleize_cols : boolean, default False\n",
      "        Leave a list of tuples on columns as is (default is to convert to\n",
      "        a Multi Index on the columns)\n",
      "    error_bad_lines : boolean, default True\n",
      "        Lines with too many fields (e.g. a csv line with too many commas) will by\n",
      "        default cause an exception to be raised, and no DataFrame will be returned.\n",
      "        If False, then these \"bad lines\" will dropped from the DataFrame that is\n",
      "        returned. (Only valid with C parser)\n",
      "    warn_bad_lines : boolean, default True\n",
      "        If error_bad_lines is False, and warn_bad_lines is True, a warning for each\n",
      "        \"bad line\" will be output. (Only valid with C parser).\n",
      "    low_memory : boolean, default True\n",
      "        Internally process the file in chunks, resulting in lower memory use\n",
      "        while parsing, but possibly mixed type inference.  To ensure no mixed\n",
      "        types either set False, or specify the type with the `dtype` parameter.\n",
      "        Note that the entire file is read into a single DataFrame regardless,\n",
      "        use the `chunksize` or `iterator` parameter to return the data in chunks.\n",
      "        (Only valid with C parser)\n",
      "    buffer_lines : int, default None\n",
      "        DEPRECATED: this argument will be removed in a future version because its\n",
      "        value is not respected by the parser\n",
      "    compact_ints : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version\n",
      "    \n",
      "        If compact_ints is True, then for any column that is of integer dtype,\n",
      "        the parser will attempt to cast it as the smallest integer dtype possible,\n",
      "        either signed or unsigned depending on the specification from the\n",
      "        `use_unsigned` parameter.\n",
      "    use_unsigned : boolean, default False\n",
      "        DEPRECATED: this argument will be removed in a future version\n",
      "    \n",
      "        If integer columns are being compacted (i.e. `compact_ints=True`), specify\n",
      "        whether the column should be compacted to the smallest signed or unsigned\n",
      "        integer dtype.\n",
      "    memory_map : boolean, default False\n",
      "        If a filepath is provided for `filepath_or_buffer`, map the file object\n",
      "        directly onto memory and access the data directly from there. Using this\n",
      "        option can improve performance because there is no longer any I/O overhead.\n",
      "    \n",
      "    Returns\n",
      "    -------\n",
      "    result : DataFrame or TextParser\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(pd.read_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/IGC.annotation.tsv.gz', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 15.2M  100 15.2M    0     0  2026k      0  0:00:07  0:00:07 --:--:-- 2055k\n"
     ]
    }
   ],
   "source": [
    "# Para notebooks de Azure\n",
    "! mkdir -p data\n",
    "! curl https://data-unix.s3-us-west-1.amazonaws.com/IGC.annotation.tsv.gz -o data/IGC.annotation.tsv.gz\n",
    "df = pd.read_csv(\"data/IGC.annotation.tsv.gz\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Para Notebooks de Jupyter instalado de manera local\n",
    "df = pd.read_csv('https://data-unix.s3-us-west-1.amazonaws.com/IGC.annotation.tsv.gz', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.frame.DataFrame"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5209933</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4932</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6811315</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>372</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00335</td>\n",
       "      <td>COG3411</td>\n",
       "      <td>0.187056</td>\n",
       "      <td>0.195327</td>\n",
       "      <td>Energy Metabolism</td>\n",
       "      <td>Energy production and conversion</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7221353</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8950791</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>177</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG125034</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2441669</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "      <td>984</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Coprobacillus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.112865</td>\n",
       "      <td>0.113084</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene ID                Gene Name  Gene Length Gene Completeness  \\\n",
       "0  5209933         MH0396_GL0114156          549    Lack both ends   \n",
       "1  6811315         MH0012_GL0174453          372          Complete   \n",
       "2  7221353         MH0389_GL0170585          330       Lack 5'-end   \n",
       "3  8950791         DOM015_GL0050638          177       Lack 5'-end   \n",
       "4  2441669  469596.HMPREF9488_01493          984          Complete   \n",
       "\n",
       "  Cohort Origin Taxonomic Annotation(Phylum Level)  \\\n",
       "0           EUR                            unknown   \n",
       "1           EUR                            unknown   \n",
       "2           EUR                            unknown   \n",
       "3           CHN                            unknown   \n",
       "4            SP                         Firmicutes   \n",
       "\n",
       "  Taxonomic Annotation(Genus Level) KEGG Annotation eggNOG Annotation  \\\n",
       "0                           unknown         unknown           COG4932   \n",
       "1                           unknown          K00335           COG3411   \n",
       "2                           unknown         unknown           unknown   \n",
       "3                           unknown         unknown         NOG125034   \n",
       "4                     Coprobacillus         unknown           unknown   \n",
       "\n",
       "   Sample Occurence Frequency  Individual Occurence Frequency  \\\n",
       "0                    0.008682                        0.009346   \n",
       "1                    0.187056                        0.195327   \n",
       "2                    0.015785                        0.018692   \n",
       "3                    0.008682                        0.008411   \n",
       "4                    0.112865                        0.113084   \n",
       "\n",
       "  KEGG Functional Categories            eggNOG Functional Categories  \\\n",
       "0                    unknown  Cell wall/membrane/envelope biogenesis   \n",
       "1          Energy Metabolism        Energy production and conversion   \n",
       "2                    unknown                                 unknown   \n",
       "3                    unknown                        Function unknown   \n",
       "4                    unknown                                 unknown   \n",
       "\n",
       "  Cohort Assembled  \n",
       "0              EUR  \n",
       "1      EUR;CHN;USA  \n",
       "2              EUR  \n",
       "3              CHN  \n",
       "4      EUR;CHN;USA  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>499995</th>\n",
       "      <td>278939</td>\n",
       "      <td>765560005-stool1_revised_scaffold5052_1_gene73628</td>\n",
       "      <td>2349</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00571</td>\n",
       "      <td>COG1002</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>0.152336</td>\n",
       "      <td>Genetic Information Processing</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499996</th>\n",
       "      <td>691723</td>\n",
       "      <td>638754422-stool2_revised_scaffold23317_1_gene1940</td>\n",
       "      <td>1698</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499997</th>\n",
       "      <td>4924959</td>\n",
       "      <td>MH0161_GL0035075</td>\n",
       "      <td>579</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.054459</td>\n",
       "      <td>0.064486</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499998</th>\n",
       "      <td>7655040</td>\n",
       "      <td>764042746-stool2_revised_scaffold22088_1_gene3...</td>\n",
       "      <td>288</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499999</th>\n",
       "      <td>5457503</td>\n",
       "      <td>V1.UC36-0_GL0134515</td>\n",
       "      <td>525</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "499995   278939  765560005-stool1_revised_scaffold5052_1_gene73628   \n",
       "499996   691723  638754422-stool2_revised_scaffold23317_1_gene1940   \n",
       "499997  4924959                                   MH0161_GL0035075   \n",
       "499998  7655040  764042746-stool2_revised_scaffold22088_1_gene3...   \n",
       "499999  5457503                                V1.UC36-0_GL0134515   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "499995         2349       Lack 5'-end           USA   \n",
       "499996         1698          Complete           USA   \n",
       "499997          579          Complete           EUR   \n",
       "499998          288          Complete           USA   \n",
       "499999          525       Lack 5'-end           EUR   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "499995                            unknown                           unknown   \n",
       "499996                            unknown                           unknown   \n",
       "499997                            unknown                           unknown   \n",
       "499998                            unknown                           unknown   \n",
       "499999                            unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "499995          K00571           COG1002                    0.153907   \n",
       "499996         unknown           unknown                    0.001579   \n",
       "499997         unknown           unknown                    0.054459   \n",
       "499998         unknown           unknown                    0.003946   \n",
       "499999         unknown           unknown                    0.008682   \n",
       "\n",
       "        Individual Occurence Frequency      KEGG Functional Categories  \\\n",
       "499995                        0.152336  Genetic Information Processing   \n",
       "499996                        0.000935                         unknown   \n",
       "499997                        0.064486                         unknown   \n",
       "499998                        0.002804                         unknown   \n",
       "499999                        0.007477                         unknown   \n",
       "\n",
       "       eggNOG Functional Categories Cohort Assembled  \n",
       "499995           Defense mechanisms          EUR;USA  \n",
       "499996                      unknown              USA  \n",
       "499997                      unknown          EUR;CHN  \n",
       "499998                      unknown              USA  \n",
       "499999                      unknown              EUR  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(500000, 14)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene ID                                 int64\n",
       "Gene Name                              object\n",
       "Gene Length                             int64\n",
       "Gene Completeness                      object\n",
       "Cohort Origin                          object\n",
       "Taxonomic Annotation(Phylum Level)     object\n",
       "Taxonomic Annotation(Genus Level)      object\n",
       "KEGG Annotation                        object\n",
       "eggNOG Annotation                      object\n",
       "Sample Occurence Frequency            float64\n",
       "Individual Occurence Frequency        float64\n",
       "KEGG Functional Categories             object\n",
       "eggNOG Functional Categories           object\n",
       "Cohort Assembled                       object\n",
       "dtype: object"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 500000 entries, 0 to 499999\n",
      "Data columns (total 14 columns):\n",
      "Gene ID                               500000 non-null int64\n",
      "Gene Name                             500000 non-null object\n",
      "Gene Length                           500000 non-null int64\n",
      "Gene Completeness                     499941 non-null object\n",
      "Cohort Origin                         500000 non-null object\n",
      "Taxonomic Annotation(Phylum Level)    500000 non-null object\n",
      "Taxonomic Annotation(Genus Level)     500000 non-null object\n",
      "KEGG Annotation                       500000 non-null object\n",
      "eggNOG Annotation                     500000 non-null object\n",
      "Sample Occurence Frequency            500000 non-null float64\n",
      "Individual Occurence Frequency        500000 non-null float64\n",
      "KEGG Functional Categories            500000 non-null object\n",
      "eggNOG Functional Categories          500000 non-null object\n",
      "Cohort Assembled                      491802 non-null object\n",
      "dtypes: float64(2), int64(2), object(10)\n",
      "memory usage: 53.4+ MB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>5.000000e+05</td>\n",
       "      <td>500000.000000</td>\n",
       "      <td>500000.000000</td>\n",
       "      <td>500000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.939795e+06</td>\n",
       "      <td>753.319838</td>\n",
       "      <td>0.077174</td>\n",
       "      <td>0.076986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>2.851155e+06</td>\n",
       "      <td>686.058789</td>\n",
       "      <td>0.139305</td>\n",
       "      <td>0.138812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>6.000000e+00</td>\n",
       "      <td>102.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>2.471293e+06</td>\n",
       "      <td>312.000000</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.005607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>4.947035e+06</td>\n",
       "      <td>579.000000</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>0.020561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>7.402792e+06</td>\n",
       "      <td>975.000000</td>\n",
       "      <td>0.077348</td>\n",
       "      <td>0.076636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>9.879844e+06</td>\n",
       "      <td>41277.000000</td>\n",
       "      <td>0.998421</td>\n",
       "      <td>0.999065</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene ID    Gene Length  Sample Occurence Frequency  \\\n",
       "count  5.000000e+05  500000.000000               500000.000000   \n",
       "mean   4.939795e+06     753.319838                    0.077174   \n",
       "std    2.851155e+06     686.058789                    0.139305   \n",
       "min    6.000000e+00     102.000000                    0.000000   \n",
       "25%    2.471293e+06     312.000000                    0.005525   \n",
       "50%    4.947035e+06     579.000000                    0.019732   \n",
       "75%    7.402792e+06     975.000000                    0.077348   \n",
       "max    9.879844e+06   41277.000000                    0.998421   \n",
       "\n",
       "       Individual Occurence Frequency  \n",
       "count                   500000.000000  \n",
       "mean                         0.076986  \n",
       "std                          0.138812  \n",
       "min                          0.000000  \n",
       "25%                          0.005607  \n",
       "50%                          0.020561  \n",
       "75%                          0.076636  \n",
       "max                          0.999065  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Gene ID</th>\n",
       "      <td>500000.0</td>\n",
       "      <td>4.939795e+06</td>\n",
       "      <td>2.851155e+06</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.471293e+06</td>\n",
       "      <td>4.947035e+06</td>\n",
       "      <td>7.402792e+06</td>\n",
       "      <td>9.879844e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Length</th>\n",
       "      <td>500000.0</td>\n",
       "      <td>7.533198e+02</td>\n",
       "      <td>6.860588e+02</td>\n",
       "      <td>102.0</td>\n",
       "      <td>3.120000e+02</td>\n",
       "      <td>5.790000e+02</td>\n",
       "      <td>9.750000e+02</td>\n",
       "      <td>4.127700e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <td>500000.0</td>\n",
       "      <td>7.717352e-02</td>\n",
       "      <td>1.393053e-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.524862e-03</td>\n",
       "      <td>1.973165e-02</td>\n",
       "      <td>7.734807e-02</td>\n",
       "      <td>9.984215e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <td>500000.0</td>\n",
       "      <td>7.698580e-02</td>\n",
       "      <td>1.388122e-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.607477e-03</td>\n",
       "      <td>2.056075e-02</td>\n",
       "      <td>7.663551e-02</td>\n",
       "      <td>9.990654e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   count          mean           std    min  \\\n",
       "Gene ID                         500000.0  4.939795e+06  2.851155e+06    6.0   \n",
       "Gene Length                     500000.0  7.533198e+02  6.860588e+02  102.0   \n",
       "Sample Occurence Frequency      500000.0  7.717352e-02  1.393053e-01    0.0   \n",
       "Individual Occurence Frequency  500000.0  7.698580e-02  1.388122e-01    0.0   \n",
       "\n",
       "                                         25%           50%           75%  \\\n",
       "Gene ID                         2.471293e+06  4.947035e+06  7.402792e+06   \n",
       "Gene Length                     3.120000e+02  5.790000e+02  9.750000e+02   \n",
       "Sample Occurence Frequency      5.524862e-03  1.973165e-02  7.734807e-02   \n",
       "Individual Occurence Frequency  5.607477e-03  2.056075e-02  7.663551e-02   \n",
       "\n",
       "                                         max  \n",
       "Gene ID                         9.879844e+06  \n",
       "Gene Length                     4.127700e+04  \n",
       "Sample Occurence Frequency      9.984215e-01  \n",
       "Individual Occurence Frequency  9.990654e-01  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "description = df.describe().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "description.to_csv('data/description.tsv', sep='\\t', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Gene ID</th>\n",
       "      <td>100.0</td>\n",
       "      <td>4.975205e+06</td>\n",
       "      <td>2.903049e+06</td>\n",
       "      <td>189878.000000</td>\n",
       "      <td>2.227776e+06</td>\n",
       "      <td>5.258160e+06</td>\n",
       "      <td>7.551792e+06</td>\n",
       "      <td>9.668785e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Length</th>\n",
       "      <td>100.0</td>\n",
       "      <td>7.247700e+02</td>\n",
       "      <td>5.476866e+02</td>\n",
       "      <td>117.000000</td>\n",
       "      <td>2.977500e+02</td>\n",
       "      <td>5.445000e+02</td>\n",
       "      <td>1.035750e+03</td>\n",
       "      <td>2.646000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <td>100.0</td>\n",
       "      <td>1.143094e-01</td>\n",
       "      <td>1.764729e-01</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>6.314128e-03</td>\n",
       "      <td>2.683504e-02</td>\n",
       "      <td>1.168114e-01</td>\n",
       "      <td>7.884767e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <td>100.0</td>\n",
       "      <td>1.129252e-01</td>\n",
       "      <td>1.748951e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.607477e-03</td>\n",
       "      <td>2.523364e-02</td>\n",
       "      <td>1.142523e-01</td>\n",
       "      <td>7.822430e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                count          mean           std  \\\n",
       "Gene ID                         100.0  4.975205e+06  2.903049e+06   \n",
       "Gene Length                     100.0  7.247700e+02  5.476866e+02   \n",
       "Sample Occurence Frequency      100.0  1.143094e-01  1.764729e-01   \n",
       "Individual Occurence Frequency  100.0  1.129252e-01  1.748951e-01   \n",
       "\n",
       "                                          min           25%           50%  \\\n",
       "Gene ID                         189878.000000  2.227776e+06  5.258160e+06   \n",
       "Gene Length                        117.000000  2.977500e+02  5.445000e+02   \n",
       "Sample Occurence Frequency           0.000789  6.314128e-03  2.683504e-02   \n",
       "Individual Occurence Frequency       0.000000  5.607477e-03  2.523364e-02   \n",
       "\n",
       "                                         75%           max  \n",
       "Gene ID                         7.551792e+06  9.668785e+06  \n",
       "Gene Length                     1.035750e+03  2.646000e+03  \n",
       "Sample Occurence Frequency      1.168114e-01  7.884767e-01  \n",
       "Individual Occurence Frequency  1.142523e-01  7.822430e-01  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(100).describe().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.T.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df2 = df.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df2.to_csv('data/dataframe_inver.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene ID', 'Gene Name', 'Gene Length', 'Gene Completeness',\n",
       "       'Cohort Origin', 'Taxonomic Annotation(Phylum Level)',\n",
       "       'Taxonomic Annotation(Genus Level)', 'KEGG Annotation',\n",
       "       'eggNOG Annotation', 'Sample Occurence Frequency',\n",
       "       'Individual Occurence Frequency', 'KEGG Functional Categories',\n",
       "       'eggNOG Functional Categories', 'Cohort Assembled'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "tuple"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.series.Series"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df.dtypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.indexes.base.Index"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         5209933\n",
       "1         6811315\n",
       "2         7221353\n",
       "3         8950791\n",
       "4         2441669\n",
       "5         4807882\n",
       "6          693534\n",
       "7          647891\n",
       "8         7928769\n",
       "9         8920818\n",
       "10        1821827\n",
       "11        5707244\n",
       "12        9553442\n",
       "13        5831026\n",
       "14        8222582\n",
       "15        5870802\n",
       "16        5203442\n",
       "17        6226899\n",
       "18        8094620\n",
       "19        6068154\n",
       "20        9139876\n",
       "21        9031743\n",
       "22        4137075\n",
       "23        3650803\n",
       "24        2312660\n",
       "25        7219798\n",
       "26        1293228\n",
       "27        3320656\n",
       "28        2811850\n",
       "29        5500609\n",
       "           ...   \n",
       "499970    4238583\n",
       "499971    6992342\n",
       "499972     186663\n",
       "499973    4517482\n",
       "499974    5233095\n",
       "499975    3484802\n",
       "499976    3280488\n",
       "499977    8177332\n",
       "499978    9145946\n",
       "499979    7115654\n",
       "499980    7451388\n",
       "499981    8734258\n",
       "499982    7786169\n",
       "499983     556756\n",
       "499984    4147487\n",
       "499985    9067856\n",
       "499986    6567495\n",
       "499987    6439309\n",
       "499988    5254719\n",
       "499989    6961318\n",
       "499990    6294127\n",
       "499991    9631397\n",
       "499992    8040629\n",
       "499993    7363211\n",
       "499994    2951767\n",
       "499995     278939\n",
       "499996     691723\n",
       "499997    4924959\n",
       "499998    7655040\n",
       "499999    5457503\n",
       "Name: Gene ID, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene ID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    5209933\n",
       "1    6811315\n",
       "2    7221353\n",
       "3    8950791\n",
       "4    2441669\n",
       "Name: Gene ID, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Gene ID'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Gene ID'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5209933</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6811315</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7221353</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8950791</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2441669</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene ID                Gene Name\n",
       "0  5209933         MH0396_GL0114156\n",
       "1  6811315         MH0012_GL0174453\n",
       "2  7221353         MH0389_GL0170585\n",
       "3  8950791         DOM015_GL0050638\n",
       "4  2441669  469596.HMPREF9488_01493"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['Gene ID', 'Gene Name']].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Gene ID</th>\n",
       "      <td>5209933</td>\n",
       "      <td>6811315</td>\n",
       "      <td>7221353</td>\n",
       "      <td>8950791</td>\n",
       "      <td>2441669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Name</th>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          0                 1                 2  \\\n",
       "Gene ID             5209933           6811315           7221353   \n",
       "Gene Name  MH0396_GL0114156  MH0012_GL0174453  MH0389_GL0170585   \n",
       "\n",
       "                          3                        4  \n",
       "Gene ID             8950791                  2441669  \n",
       "Gene Name  DOM015_GL0050638  469596.HMPREF9488_01493  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['Gene ID', 'Gene Name']].head().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=500000, step=1)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.indexes.range.RangeIndex"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on method sort_index in module pandas.core.frame:\n",
      "\n",
      "sort_index(axis=0, level=None, ascending=True, inplace=False, kind='quicksort', na_position='last', sort_remaining=True, by=None) method of pandas.core.frame.DataFrame instance\n",
      "    Sort object by labels (along an axis)\n",
      "    \n",
      "    Parameters\n",
      "    ----------\n",
      "    axis : index, columns to direct sorting\n",
      "    level : int or level name or list of ints or list of level names\n",
      "        if not None, sort on values in specified index level(s)\n",
      "    ascending : boolean, default True\n",
      "        Sort ascending vs. descending\n",
      "    inplace : bool, default False\n",
      "        if True, perform operation in-place\n",
      "    kind : {'quicksort', 'mergesort', 'heapsort'}, default 'quicksort'\n",
      "         Choice of sorting algorithm. See also ndarray.np.sort for more\n",
      "         information.  `mergesort` is the only stable algorithm. For\n",
      "         DataFrames, this option is only applied when sorting on a single\n",
      "         column or label.\n",
      "    na_position : {'first', 'last'}, default 'last'\n",
      "         `first` puts NaNs at the beginning, `last` puts NaNs at the end\n",
      "    sort_remaining : bool, default True\n",
      "        if true and sorting by level and index is multilevel, sort by other\n",
      "        levels too (in order) after sorting by specified level\n",
      "    \n",
      "    Returns\n",
      "    -------\n",
      "    sorted_obj : DataFrame\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(df.sort_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5209933</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4932</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6811315</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>372</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00335</td>\n",
       "      <td>COG3411</td>\n",
       "      <td>0.187056</td>\n",
       "      <td>0.195327</td>\n",
       "      <td>Energy Metabolism</td>\n",
       "      <td>Energy production and conversion</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7221353</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8950791</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>177</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG125034</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2441669</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "      <td>984</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Coprobacillus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.112865</td>\n",
       "      <td>0.113084</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4807882</td>\n",
       "      <td>V1.FI14_GL0101784</td>\n",
       "      <td>594</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>693534</td>\n",
       "      <td>MH0448_GL0045910</td>\n",
       "      <td>1695</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>647891</td>\n",
       "      <td>MH0433_GL0005093</td>\n",
       "      <td>1740</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7928769</td>\n",
       "      <td>O2.UC48-1_GL0076618</td>\n",
       "      <td>261</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8920818</td>\n",
       "      <td>764588959-stool1_revised_C775588_1_gene178391</td>\n",
       "      <td>180</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1821827</td>\n",
       "      <td>MH0333_GL0003962</td>\n",
       "      <td>1146</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1073</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>0.011215</td>\n",
       "      <td>unknown</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5707244</td>\n",
       "      <td>V1.FI15_GL0072437</td>\n",
       "      <td>501</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07095</td>\n",
       "      <td>COG0622</td>\n",
       "      <td>0.096290</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9553442</td>\n",
       "      <td>MH0149_GL0074986</td>\n",
       "      <td>126</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.335438</td>\n",
       "      <td>0.330841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5831026</td>\n",
       "      <td>MH0419_GL0063398</td>\n",
       "      <td>486</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG77285</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.030841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>8222582</td>\n",
       "      <td>MH0309_GL0024067</td>\n",
       "      <td>234</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG291728</td>\n",
       "      <td>0.048145</td>\n",
       "      <td>0.051402</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5870802</td>\n",
       "      <td>MH0096_GL0115416</td>\n",
       "      <td>480</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01520</td>\n",
       "      <td>COG0756</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5203442</td>\n",
       "      <td>MH0188_GL0079716</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.086819</td>\n",
       "      <td>0.090654</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6226899</td>\n",
       "      <td>MH0192_GL0144006</td>\n",
       "      <td>438</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01493</td>\n",
       "      <td>COG2131</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>8094620</td>\n",
       "      <td>O2.UC47-2_GL0030825</td>\n",
       "      <td>246</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG256796</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6068154</td>\n",
       "      <td>MH0040_GL0048712</td>\n",
       "      <td>456</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Ruminococcus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.284136</td>\n",
       "      <td>0.285047</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9139876</td>\n",
       "      <td>159753524-stool2_revised_C530648_1_gene94409</td>\n",
       "      <td>162</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9031743</td>\n",
       "      <td>160765029-stool1_revised_C1108515_1_gene40264</td>\n",
       "      <td>171</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4137075</td>\n",
       "      <td>O2.UC48-0_GL0276256</td>\n",
       "      <td>681</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K06206</td>\n",
       "      <td>COG1489</td>\n",
       "      <td>0.046567</td>\n",
       "      <td>0.034579</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3650803</td>\n",
       "      <td>MH0012_GL0124464</td>\n",
       "      <td>753</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02049</td>\n",
       "      <td>COG1116</td>\n",
       "      <td>0.326756</td>\n",
       "      <td>0.332710</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Inorganic ion transport and metabolism</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2312660</td>\n",
       "      <td>158256496-stool1_revised_C920453_1_gene161597</td>\n",
       "      <td>1014</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG277117</td>\n",
       "      <td>0.085241</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>7219798</td>\n",
       "      <td>MH0355_GL0176135</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01897</td>\n",
       "      <td>COG0318</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.037383</td>\n",
       "      <td>Lipid Metabolism</td>\n",
       "      <td>Lipid transport and metabolism;Secondary metab...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1293228</td>\n",
       "      <td>V1.UC50-1_GL0030242</td>\n",
       "      <td>1329</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01972</td>\n",
       "      <td>COG0272</td>\n",
       "      <td>0.024467</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Replication, recombination and repair</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3320656</td>\n",
       "      <td>O2.UC24-1_GL0155079</td>\n",
       "      <td>810</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG69716</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2811850</td>\n",
       "      <td>T2D-63A_GL0089788</td>\n",
       "      <td>906</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG40611</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5500609</td>\n",
       "      <td>MH0174_GL0034307</td>\n",
       "      <td>519</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499970</th>\n",
       "      <td>4238583</td>\n",
       "      <td>MH0284_GL0171112</td>\n",
       "      <td>666</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.152328</td>\n",
       "      <td>0.147664</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499971</th>\n",
       "      <td>6992342</td>\n",
       "      <td>MH0392_GL0212082</td>\n",
       "      <td>354</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499972</th>\n",
       "      <td>186663</td>\n",
       "      <td>V1.FI20_GL0146385</td>\n",
       "      <td>2661</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02355</td>\n",
       "      <td>COG3688;COG0480</td>\n",
       "      <td>0.727703</td>\n",
       "      <td>0.721495</td>\n",
       "      <td>Translation</td>\n",
       "      <td>Translation, ribosomal structure and biogenesi...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499973</th>\n",
       "      <td>4517482</td>\n",
       "      <td>DOF003_GL0027603</td>\n",
       "      <td>630</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG275003</td>\n",
       "      <td>0.233623</td>\n",
       "      <td>0.229907</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499974</th>\n",
       "      <td>5233095</td>\n",
       "      <td>MH0204_GL0162383</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499975</th>\n",
       "      <td>3484802</td>\n",
       "      <td>MH0188_GL0112429</td>\n",
       "      <td>780</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499976</th>\n",
       "      <td>3280488</td>\n",
       "      <td>MH0028_GL0020655</td>\n",
       "      <td>816</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>K09690</td>\n",
       "      <td>COG1682</td>\n",
       "      <td>0.361484</td>\n",
       "      <td>0.362617</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Carbohydrate transport and metabolism;Cell wal...</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499977</th>\n",
       "      <td>8177332</td>\n",
       "      <td>411485.FAEPRAM212_00318</td>\n",
       "      <td>240</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.783741</td>\n",
       "      <td>0.778505</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499978</th>\n",
       "      <td>9145946</td>\n",
       "      <td>MH0117_GL0078602</td>\n",
       "      <td>159</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499979</th>\n",
       "      <td>7115654</td>\n",
       "      <td>DOF004_GL0014088</td>\n",
       "      <td>342</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499980</th>\n",
       "      <td>7451388</td>\n",
       "      <td>MH0222_GL0182909</td>\n",
       "      <td>306</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499981</th>\n",
       "      <td>8734258</td>\n",
       "      <td>470145.BACCOP_04248</td>\n",
       "      <td>195</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.016822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499982</th>\n",
       "      <td>7786169</td>\n",
       "      <td>MH0157_GL0106576</td>\n",
       "      <td>273</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499983</th>\n",
       "      <td>556756</td>\n",
       "      <td>SZEY-101A_GL0065406</td>\n",
       "      <td>1839</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499984</th>\n",
       "      <td>4147487</td>\n",
       "      <td>MH0062_GL0030772</td>\n",
       "      <td>678</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07658</td>\n",
       "      <td>COG0745</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>0.244860</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>Transcription;Signal transduction mechanisms</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499985</th>\n",
       "      <td>9067856</td>\n",
       "      <td>160400887-stool1_revised_C1382776_1_gene10717</td>\n",
       "      <td>168</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499986</th>\n",
       "      <td>6567495</td>\n",
       "      <td>MH0065_GL0023744</td>\n",
       "      <td>399</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499987</th>\n",
       "      <td>6439309</td>\n",
       "      <td>MH0158_GL0048733</td>\n",
       "      <td>414</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03666</td>\n",
       "      <td>COG1923</td>\n",
       "      <td>0.440410</td>\n",
       "      <td>0.445794</td>\n",
       "      <td>Folding, Sorting and Degradation</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499988</th>\n",
       "      <td>5254719</td>\n",
       "      <td>159753524-stool2_revised_C538848_1_gene38208</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03531</td>\n",
       "      <td>COG0206</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499989</th>\n",
       "      <td>6961318</td>\n",
       "      <td>MH0333_GL0111005</td>\n",
       "      <td>357</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007893</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499990</th>\n",
       "      <td>6294127</td>\n",
       "      <td>V1.UC26-4_GL0011854</td>\n",
       "      <td>432</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07024</td>\n",
       "      <td>COG0561</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499991</th>\n",
       "      <td>9631397</td>\n",
       "      <td>MH0280_GL0114474</td>\n",
       "      <td>120</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.037096</td>\n",
       "      <td>0.033645</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499992</th>\n",
       "      <td>8040629</td>\n",
       "      <td>159611913-stool1_revised_C786324_1_gene54480</td>\n",
       "      <td>252</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.018942</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499993</th>\n",
       "      <td>7363211</td>\n",
       "      <td>MH0260_GL0023264</td>\n",
       "      <td>315</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499994</th>\n",
       "      <td>2951767</td>\n",
       "      <td>V1.UC26-4_GL0007902</td>\n",
       "      <td>879</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG47185</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499995</th>\n",
       "      <td>278939</td>\n",
       "      <td>765560005-stool1_revised_scaffold5052_1_gene73628</td>\n",
       "      <td>2349</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00571</td>\n",
       "      <td>COG1002</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>0.152336</td>\n",
       "      <td>Genetic Information Processing</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499996</th>\n",
       "      <td>691723</td>\n",
       "      <td>638754422-stool2_revised_scaffold23317_1_gene1940</td>\n",
       "      <td>1698</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499997</th>\n",
       "      <td>4924959</td>\n",
       "      <td>MH0161_GL0035075</td>\n",
       "      <td>579</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.054459</td>\n",
       "      <td>0.064486</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499998</th>\n",
       "      <td>7655040</td>\n",
       "      <td>764042746-stool2_revised_scaffold22088_1_gene3...</td>\n",
       "      <td>288</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499999</th>\n",
       "      <td>5457503</td>\n",
       "      <td>V1.UC36-0_GL0134515</td>\n",
       "      <td>525</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "0       5209933                                   MH0396_GL0114156   \n",
       "1       6811315                                   MH0012_GL0174453   \n",
       "2       7221353                                   MH0389_GL0170585   \n",
       "3       8950791                                   DOM015_GL0050638   \n",
       "4       2441669                            469596.HMPREF9488_01493   \n",
       "5       4807882                                  V1.FI14_GL0101784   \n",
       "6        693534                                   MH0448_GL0045910   \n",
       "7        647891                                   MH0433_GL0005093   \n",
       "8       7928769                                O2.UC48-1_GL0076618   \n",
       "9       8920818      764588959-stool1_revised_C775588_1_gene178391   \n",
       "10      1821827                                   MH0333_GL0003962   \n",
       "11      5707244                                  V1.FI15_GL0072437   \n",
       "12      9553442                                   MH0149_GL0074986   \n",
       "13      5831026                                   MH0419_GL0063398   \n",
       "14      8222582                                   MH0309_GL0024067   \n",
       "15      5870802                                   MH0096_GL0115416   \n",
       "16      5203442                                   MH0188_GL0079716   \n",
       "17      6226899                                   MH0192_GL0144006   \n",
       "18      8094620                                O2.UC47-2_GL0030825   \n",
       "19      6068154                                   MH0040_GL0048712   \n",
       "20      9139876       159753524-stool2_revised_C530648_1_gene94409   \n",
       "21      9031743      160765029-stool1_revised_C1108515_1_gene40264   \n",
       "22      4137075                                O2.UC48-0_GL0276256   \n",
       "23      3650803                                   MH0012_GL0124464   \n",
       "24      2312660      158256496-stool1_revised_C920453_1_gene161597   \n",
       "25      7219798                                   MH0355_GL0176135   \n",
       "26      1293228                                V1.UC50-1_GL0030242   \n",
       "27      3320656                                O2.UC24-1_GL0155079   \n",
       "28      2811850                                  T2D-63A_GL0089788   \n",
       "29      5500609                                   MH0174_GL0034307   \n",
       "...         ...                                                ...   \n",
       "499970  4238583                                   MH0284_GL0171112   \n",
       "499971  6992342                                   MH0392_GL0212082   \n",
       "499972   186663                                  V1.FI20_GL0146385   \n",
       "499973  4517482                                   DOF003_GL0027603   \n",
       "499974  5233095                                   MH0204_GL0162383   \n",
       "499975  3484802                                   MH0188_GL0112429   \n",
       "499976  3280488                                   MH0028_GL0020655   \n",
       "499977  8177332                            411485.FAEPRAM212_00318   \n",
       "499978  9145946                                   MH0117_GL0078602   \n",
       "499979  7115654                                   DOF004_GL0014088   \n",
       "499980  7451388                                   MH0222_GL0182909   \n",
       "499981  8734258                                470145.BACCOP_04248   \n",
       "499982  7786169                                   MH0157_GL0106576   \n",
       "499983   556756                                SZEY-101A_GL0065406   \n",
       "499984  4147487                                   MH0062_GL0030772   \n",
       "499985  9067856      160400887-stool1_revised_C1382776_1_gene10717   \n",
       "499986  6567495                                   MH0065_GL0023744   \n",
       "499987  6439309                                   MH0158_GL0048733   \n",
       "499988  5254719       159753524-stool2_revised_C538848_1_gene38208   \n",
       "499989  6961318                                   MH0333_GL0111005   \n",
       "499990  6294127                                V1.UC26-4_GL0011854   \n",
       "499991  9631397                                   MH0280_GL0114474   \n",
       "499992  8040629       159611913-stool1_revised_C786324_1_gene54480   \n",
       "499993  7363211                                   MH0260_GL0023264   \n",
       "499994  2951767                                V1.UC26-4_GL0007902   \n",
       "499995   278939  765560005-stool1_revised_scaffold5052_1_gene73628   \n",
       "499996   691723  638754422-stool2_revised_scaffold23317_1_gene1940   \n",
       "499997  4924959                                   MH0161_GL0035075   \n",
       "499998  7655040  764042746-stool2_revised_scaffold22088_1_gene3...   \n",
       "499999  5457503                                V1.UC36-0_GL0134515   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "0               549    Lack both ends           EUR   \n",
       "1               372          Complete           EUR   \n",
       "2               330       Lack 5'-end           EUR   \n",
       "3               177       Lack 5'-end           CHN   \n",
       "4               984          Complete            SP   \n",
       "5               594          Complete           EUR   \n",
       "6              1695    Lack both ends           EUR   \n",
       "7              1740          Complete           EUR   \n",
       "8               261       Lack 3'-end           EUR   \n",
       "9               180       Lack 3'-end           USA   \n",
       "10             1146          Complete           EUR   \n",
       "11              501          Complete           EUR   \n",
       "12              126       Lack 5'-end           EUR   \n",
       "13              486          Complete           EUR   \n",
       "14              234       Lack 3'-end           EUR   \n",
       "15              480          Complete           EUR   \n",
       "16              549    Lack both ends           EUR   \n",
       "17              438          Complete           EUR   \n",
       "18              246       Lack 3'-end           EUR   \n",
       "19              456       Lack 5'-end           EUR   \n",
       "20              162       Lack 5'-end           USA   \n",
       "21              171          Complete           USA   \n",
       "22              681          Complete           EUR   \n",
       "23              753          Complete           EUR   \n",
       "24             1014    Lack both ends           USA   \n",
       "25              330       Lack 3'-end           EUR   \n",
       "26             1329       Lack 5'-end           EUR   \n",
       "27              810    Lack both ends           EUR   \n",
       "28              906          Complete           CHN   \n",
       "29              519    Lack both ends           EUR   \n",
       "...             ...               ...           ...   \n",
       "499970          666          Complete           EUR   \n",
       "499971          354       Lack 5'-end           EUR   \n",
       "499972         2661          Complete           EUR   \n",
       "499973          630          Complete           CHN   \n",
       "499974          546    Lack both ends           EUR   \n",
       "499975          780          Complete           EUR   \n",
       "499976          816          Complete           EUR   \n",
       "499977          240          Complete            SP   \n",
       "499978          159          Complete           EUR   \n",
       "499979          342       Lack 5'-end           CHN   \n",
       "499980          306          Complete           EUR   \n",
       "499981          195          Complete            SP   \n",
       "499982          273          Complete           EUR   \n",
       "499983         1839          Complete           CHN   \n",
       "499984          678          Complete           EUR   \n",
       "499985          168          Complete           USA   \n",
       "499986          399       Lack 5'-end           EUR   \n",
       "499987          414       Lack 5'-end           EUR   \n",
       "499988          546    Lack both ends           USA   \n",
       "499989          357       Lack 3'-end           EUR   \n",
       "499990          432       Lack 5'-end           EUR   \n",
       "499991          120       Lack 5'-end           EUR   \n",
       "499992          252          Complete           USA   \n",
       "499993          315       Lack 3'-end           EUR   \n",
       "499994          879          Complete           EUR   \n",
       "499995         2349       Lack 5'-end           USA   \n",
       "499996         1698          Complete           USA   \n",
       "499997          579          Complete           EUR   \n",
       "499998          288          Complete           USA   \n",
       "499999          525       Lack 5'-end           EUR   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "0                                 unknown                           unknown   \n",
       "1                                 unknown                           unknown   \n",
       "2                                 unknown                           unknown   \n",
       "3                                 unknown                           unknown   \n",
       "4                              Firmicutes                     Coprobacillus   \n",
       "5                                 unknown                           unknown   \n",
       "6                                 unknown                           unknown   \n",
       "7                                 unknown                           unknown   \n",
       "8                                 unknown                           unknown   \n",
       "9                                 unknown                           unknown   \n",
       "10                          Bacteroidetes                           unknown   \n",
       "11                                unknown                           unknown   \n",
       "12                                unknown                           unknown   \n",
       "13                                unknown                           unknown   \n",
       "14                                unknown                           unknown   \n",
       "15                                unknown                           unknown   \n",
       "16                                unknown                           unknown   \n",
       "17                          Bacteroidetes                           unknown   \n",
       "18                                unknown                           unknown   \n",
       "19                             Firmicutes                      Ruminococcus   \n",
       "20                                unknown                           unknown   \n",
       "21                                unknown                           unknown   \n",
       "22                                unknown                           unknown   \n",
       "23                                unknown                           unknown   \n",
       "24                                unknown                           unknown   \n",
       "25                                unknown                           unknown   \n",
       "26                                unknown                           unknown   \n",
       "27                                unknown                           unknown   \n",
       "28                                unknown                           unknown   \n",
       "29                                unknown                           unknown   \n",
       "...                                   ...                               ...   \n",
       "499970                            unknown                           unknown   \n",
       "499971                      Bacteroidetes                       Bacteroides   \n",
       "499972                            unknown                           unknown   \n",
       "499973                      Bacteroidetes                       Bacteroides   \n",
       "499974                            unknown                           unknown   \n",
       "499975                            unknown                           unknown   \n",
       "499976                         Firmicutes                       Eubacterium   \n",
       "499977                         Firmicutes                  Faecalibacterium   \n",
       "499978                            unknown                           unknown   \n",
       "499979                            unknown                           unknown   \n",
       "499980                            unknown                           unknown   \n",
       "499981                      Bacteroidetes                       Bacteroides   \n",
       "499982                            unknown                           unknown   \n",
       "499983                            unknown                           unknown   \n",
       "499984                         Firmicutes                           unknown   \n",
       "499985                            unknown                           unknown   \n",
       "499986                            unknown                           unknown   \n",
       "499987                            unknown                           unknown   \n",
       "499988                            unknown                           unknown   \n",
       "499989                            unknown                           unknown   \n",
       "499990                            unknown                           unknown   \n",
       "499991                            unknown                           unknown   \n",
       "499992                         Firmicutes                  Faecalibacterium   \n",
       "499993                            unknown                           unknown   \n",
       "499994                            unknown                           unknown   \n",
       "499995                            unknown                           unknown   \n",
       "499996                            unknown                           unknown   \n",
       "499997                            unknown                           unknown   \n",
       "499998                            unknown                           unknown   \n",
       "499999                            unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "0              unknown           COG4932                    0.008682   \n",
       "1               K00335           COG3411                    0.187056   \n",
       "2              unknown           unknown                    0.015785   \n",
       "3              unknown         NOG125034                    0.008682   \n",
       "4              unknown           unknown                    0.112865   \n",
       "5              unknown           unknown                    0.008682   \n",
       "6              unknown          NOG12793                    0.004736   \n",
       "7              unknown           unknown                    0.003946   \n",
       "8              unknown           unknown                    0.014996   \n",
       "9              unknown           unknown                    0.000789   \n",
       "10             unknown           COG1073                    0.010260   \n",
       "11              K07095           COG0622                    0.096290   \n",
       "12             unknown           unknown                    0.335438   \n",
       "13             unknown          NOG77285                    0.032360   \n",
       "14             unknown         NOG291728                    0.048145   \n",
       "15              K01520           COG0756                    0.000789   \n",
       "16             unknown           unknown                    0.086819   \n",
       "17              K01493           COG2131                    0.007103   \n",
       "18             unknown         NOG256796                    0.005525   \n",
       "19             unknown           unknown                    0.284136   \n",
       "20             unknown           unknown                    0.003946   \n",
       "21             unknown           unknown                    0.000789   \n",
       "22              K06206           COG1489                    0.046567   \n",
       "23              K02049           COG1116                    0.326756   \n",
       "24             unknown         NOG277117                    0.085241   \n",
       "25              K01897           COG0318                    0.033938   \n",
       "26              K01972           COG0272                    0.024467   \n",
       "27             unknown          NOG69716                    0.019732   \n",
       "28             unknown          NOG40611                    0.008682   \n",
       "29             unknown           unknown                    0.003157   \n",
       "...                ...               ...                         ...   \n",
       "499970         unknown           unknown                    0.152328   \n",
       "499971         unknown           unknown                    0.001579   \n",
       "499972          K02355   COG3688;COG0480                    0.727703   \n",
       "499973         unknown         NOG275003                    0.233623   \n",
       "499974         unknown           unknown                    0.020521   \n",
       "499975         unknown           unknown                    0.003157   \n",
       "499976          K09690           COG1682                    0.361484   \n",
       "499977         unknown           unknown                    0.783741   \n",
       "499978         unknown           unknown                    0.014996   \n",
       "499979         unknown           unknown                    0.015785   \n",
       "499980         unknown           unknown                    0.002368   \n",
       "499981         unknown           unknown                    0.015785   \n",
       "499982         unknown           unknown                    0.000789   \n",
       "499983         unknown           unknown                    0.000789   \n",
       "499984          K07658           COG0745                    0.229676   \n",
       "499985         unknown           unknown                    0.003946   \n",
       "499986         unknown           unknown                    0.007103   \n",
       "499987          K03666           COG1923                    0.440410   \n",
       "499988          K03531           COG0206                    0.004736   \n",
       "499989         unknown           unknown                    0.007893   \n",
       "499990          K07024           COG0561                    0.002368   \n",
       "499991         unknown           unknown                    0.037096   \n",
       "499992         unknown           unknown                    0.018942   \n",
       "499993         unknown           unknown                    0.003946   \n",
       "499994         unknown          NOG47185                    0.004736   \n",
       "499995          K00571           COG1002                    0.153907   \n",
       "499996         unknown           unknown                    0.001579   \n",
       "499997         unknown           unknown                    0.054459   \n",
       "499998         unknown           unknown                    0.003946   \n",
       "499999         unknown           unknown                    0.008682   \n",
       "\n",
       "        Individual Occurence Frequency        KEGG Functional Categories  \\\n",
       "0                             0.009346                           unknown   \n",
       "1                             0.195327                 Energy Metabolism   \n",
       "2                             0.018692                           unknown   \n",
       "3                             0.008411                           unknown   \n",
       "4                             0.113084                           unknown   \n",
       "5                             0.009346                           unknown   \n",
       "6                             0.004673                           unknown   \n",
       "7                             0.004673                           unknown   \n",
       "8                             0.008411                           unknown   \n",
       "9                             0.000935                           unknown   \n",
       "10                            0.011215                           unknown   \n",
       "11                            0.100000              Poorly Characterized   \n",
       "12                            0.330841                           unknown   \n",
       "13                            0.030841                           unknown   \n",
       "14                            0.051402                           unknown   \n",
       "15                            0.000935             Nucleotide Metabolism   \n",
       "16                            0.090654                           unknown   \n",
       "17                            0.004673             Nucleotide Metabolism   \n",
       "18                            0.004673                           unknown   \n",
       "19                            0.285047                           unknown   \n",
       "20                            0.002804                           unknown   \n",
       "21                            0.000935                           unknown   \n",
       "22                            0.034579                        Metabolism   \n",
       "23                            0.332710                Membrane Transport   \n",
       "24                            0.079439                           unknown   \n",
       "25                            0.037383                  Lipid Metabolism   \n",
       "26                            0.021495            Replication and Repair   \n",
       "27                            0.018692                           unknown   \n",
       "28                            0.009346                           unknown   \n",
       "29                            0.003738                           unknown   \n",
       "...                                ...                               ...   \n",
       "499970                        0.147664                           unknown   \n",
       "499971                        0.000935                           unknown   \n",
       "499972                        0.721495                       Translation   \n",
       "499973                        0.229907                           unknown   \n",
       "499974                        0.021495                           unknown   \n",
       "499975                        0.003738                           unknown   \n",
       "499976                        0.362617                Membrane Transport   \n",
       "499977                        0.778505                           unknown   \n",
       "499978                        0.015888                           unknown   \n",
       "499979                        0.017757                           unknown   \n",
       "499980                        0.002804                           unknown   \n",
       "499981                        0.016822                           unknown   \n",
       "499982                        0.000935                           unknown   \n",
       "499983                        0.000935                           unknown   \n",
       "499984                        0.244860               Signal Transduction   \n",
       "499985                        0.004673                           unknown   \n",
       "499986                        0.008411                           unknown   \n",
       "499987                        0.445794  Folding, Sorting and Degradation   \n",
       "499988                        0.002804            Replication and Repair   \n",
       "499989                        0.007477                           unknown   \n",
       "499990                        0.001869              Poorly Characterized   \n",
       "499991                        0.033645                           unknown   \n",
       "499992                        0.017757                           unknown   \n",
       "499993                        0.004673                           unknown   \n",
       "499994                        0.003738                           unknown   \n",
       "499995                        0.152336    Genetic Information Processing   \n",
       "499996                        0.000935                           unknown   \n",
       "499997                        0.064486                           unknown   \n",
       "499998                        0.002804                           unknown   \n",
       "499999                        0.007477                           unknown   \n",
       "\n",
       "                             eggNOG Functional Categories Cohort Assembled  \n",
       "0                  Cell wall/membrane/envelope biogenesis              EUR  \n",
       "1                        Energy production and conversion      EUR;CHN;USA  \n",
       "2                                                 unknown              EUR  \n",
       "3                                        Function unknown              CHN  \n",
       "4                                                 unknown      EUR;CHN;USA  \n",
       "5                                                 unknown              EUR  \n",
       "6                                        Function unknown              EUR  \n",
       "7                                                 unknown          EUR;USA  \n",
       "8                                                 unknown              EUR  \n",
       "9                                                 unknown              USA  \n",
       "10                       General function prediction only          EUR;CHN  \n",
       "11                       General function prediction only              EUR  \n",
       "12                                                unknown              EUR  \n",
       "13                                       Function unknown              EUR  \n",
       "14                                                unknown              EUR  \n",
       "15                    Nucleotide transport and metabolism              EUR  \n",
       "16                                                unknown              EUR  \n",
       "17                    Nucleotide transport and metabolism              EUR  \n",
       "18                                       Function unknown              EUR  \n",
       "19                                                unknown              EUR  \n",
       "20                                                unknown              USA  \n",
       "21                                                unknown              USA  \n",
       "22                       General function prediction only              EUR  \n",
       "23                 Inorganic ion transport and metabolism      EUR;CHN;USA  \n",
       "24                                                unknown              USA  \n",
       "25      Lipid transport and metabolism;Secondary metab...              EUR  \n",
       "26                  Replication, recombination and repair              EUR  \n",
       "27                                       Function unknown              EUR  \n",
       "28                                       Function unknown              CHN  \n",
       "29                                                unknown              EUR  \n",
       "...                                                   ...              ...  \n",
       "499970                                            unknown      EUR;CHN;USA  \n",
       "499971                                            unknown              EUR  \n",
       "499972  Translation, ribosomal structure and biogenesi...              EUR  \n",
       "499973                                            unknown              CHN  \n",
       "499974                                            unknown              EUR  \n",
       "499975                                            unknown              EUR  \n",
       "499976  Carbohydrate transport and metabolism;Cell wal...      EUR;CHN;USA  \n",
       "499977                                            unknown              EUR  \n",
       "499978                                            unknown              EUR  \n",
       "499979                                            unknown              CHN  \n",
       "499980                                            unknown              EUR  \n",
       "499981                                            unknown              NaN  \n",
       "499982                                            unknown              EUR  \n",
       "499983                                            unknown              CHN  \n",
       "499984       Transcription;Signal transduction mechanisms      EUR;CHN;USA  \n",
       "499985                                            unknown              USA  \n",
       "499986                                            unknown              EUR  \n",
       "499987                   General function prediction only      EUR;CHN;USA  \n",
       "499988  Cell cycle control, cell division, chromosome ...              USA  \n",
       "499989                                            unknown              EUR  \n",
       "499990                   General function prediction only              EUR  \n",
       "499991                                            unknown              EUR  \n",
       "499992                                            unknown              USA  \n",
       "499993                                            unknown              EUR  \n",
       "499994                                   Function unknown              EUR  \n",
       "499995                                 Defense mechanisms          EUR;USA  \n",
       "499996                                            unknown              USA  \n",
       "499997                                            unknown          EUR;CHN  \n",
       "499998                                            unknown              USA  \n",
       "499999                                            unknown              EUR  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ordenar por filas(axis 0), Ascendente\n",
    "df.sort_index(axis=0, ascending=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>499999</th>\n",
       "      <td>5457503</td>\n",
       "      <td>V1.UC36-0_GL0134515</td>\n",
       "      <td>525</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499998</th>\n",
       "      <td>7655040</td>\n",
       "      <td>764042746-stool2_revised_scaffold22088_1_gene3...</td>\n",
       "      <td>288</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499997</th>\n",
       "      <td>4924959</td>\n",
       "      <td>MH0161_GL0035075</td>\n",
       "      <td>579</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.054459</td>\n",
       "      <td>0.064486</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499996</th>\n",
       "      <td>691723</td>\n",
       "      <td>638754422-stool2_revised_scaffold23317_1_gene1940</td>\n",
       "      <td>1698</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499995</th>\n",
       "      <td>278939</td>\n",
       "      <td>765560005-stool1_revised_scaffold5052_1_gene73628</td>\n",
       "      <td>2349</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00571</td>\n",
       "      <td>COG1002</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>0.152336</td>\n",
       "      <td>Genetic Information Processing</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499994</th>\n",
       "      <td>2951767</td>\n",
       "      <td>V1.UC26-4_GL0007902</td>\n",
       "      <td>879</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG47185</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499993</th>\n",
       "      <td>7363211</td>\n",
       "      <td>MH0260_GL0023264</td>\n",
       "      <td>315</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499992</th>\n",
       "      <td>8040629</td>\n",
       "      <td>159611913-stool1_revised_C786324_1_gene54480</td>\n",
       "      <td>252</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.018942</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499991</th>\n",
       "      <td>9631397</td>\n",
       "      <td>MH0280_GL0114474</td>\n",
       "      <td>120</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.037096</td>\n",
       "      <td>0.033645</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499990</th>\n",
       "      <td>6294127</td>\n",
       "      <td>V1.UC26-4_GL0011854</td>\n",
       "      <td>432</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07024</td>\n",
       "      <td>COG0561</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499989</th>\n",
       "      <td>6961318</td>\n",
       "      <td>MH0333_GL0111005</td>\n",
       "      <td>357</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007893</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499988</th>\n",
       "      <td>5254719</td>\n",
       "      <td>159753524-stool2_revised_C538848_1_gene38208</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03531</td>\n",
       "      <td>COG0206</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499987</th>\n",
       "      <td>6439309</td>\n",
       "      <td>MH0158_GL0048733</td>\n",
       "      <td>414</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03666</td>\n",
       "      <td>COG1923</td>\n",
       "      <td>0.440410</td>\n",
       "      <td>0.445794</td>\n",
       "      <td>Folding, Sorting and Degradation</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499986</th>\n",
       "      <td>6567495</td>\n",
       "      <td>MH0065_GL0023744</td>\n",
       "      <td>399</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499985</th>\n",
       "      <td>9067856</td>\n",
       "      <td>160400887-stool1_revised_C1382776_1_gene10717</td>\n",
       "      <td>168</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499984</th>\n",
       "      <td>4147487</td>\n",
       "      <td>MH0062_GL0030772</td>\n",
       "      <td>678</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07658</td>\n",
       "      <td>COG0745</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>0.244860</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>Transcription;Signal transduction mechanisms</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499983</th>\n",
       "      <td>556756</td>\n",
       "      <td>SZEY-101A_GL0065406</td>\n",
       "      <td>1839</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499982</th>\n",
       "      <td>7786169</td>\n",
       "      <td>MH0157_GL0106576</td>\n",
       "      <td>273</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499981</th>\n",
       "      <td>8734258</td>\n",
       "      <td>470145.BACCOP_04248</td>\n",
       "      <td>195</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.016822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499980</th>\n",
       "      <td>7451388</td>\n",
       "      <td>MH0222_GL0182909</td>\n",
       "      <td>306</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499979</th>\n",
       "      <td>7115654</td>\n",
       "      <td>DOF004_GL0014088</td>\n",
       "      <td>342</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499978</th>\n",
       "      <td>9145946</td>\n",
       "      <td>MH0117_GL0078602</td>\n",
       "      <td>159</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499977</th>\n",
       "      <td>8177332</td>\n",
       "      <td>411485.FAEPRAM212_00318</td>\n",
       "      <td>240</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.783741</td>\n",
       "      <td>0.778505</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499976</th>\n",
       "      <td>3280488</td>\n",
       "      <td>MH0028_GL0020655</td>\n",
       "      <td>816</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>K09690</td>\n",
       "      <td>COG1682</td>\n",
       "      <td>0.361484</td>\n",
       "      <td>0.362617</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Carbohydrate transport and metabolism;Cell wal...</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499975</th>\n",
       "      <td>3484802</td>\n",
       "      <td>MH0188_GL0112429</td>\n",
       "      <td>780</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499974</th>\n",
       "      <td>5233095</td>\n",
       "      <td>MH0204_GL0162383</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499973</th>\n",
       "      <td>4517482</td>\n",
       "      <td>DOF003_GL0027603</td>\n",
       "      <td>630</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG275003</td>\n",
       "      <td>0.233623</td>\n",
       "      <td>0.229907</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499972</th>\n",
       "      <td>186663</td>\n",
       "      <td>V1.FI20_GL0146385</td>\n",
       "      <td>2661</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02355</td>\n",
       "      <td>COG3688;COG0480</td>\n",
       "      <td>0.727703</td>\n",
       "      <td>0.721495</td>\n",
       "      <td>Translation</td>\n",
       "      <td>Translation, ribosomal structure and biogenesi...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499971</th>\n",
       "      <td>6992342</td>\n",
       "      <td>MH0392_GL0212082</td>\n",
       "      <td>354</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499970</th>\n",
       "      <td>4238583</td>\n",
       "      <td>MH0284_GL0171112</td>\n",
       "      <td>666</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.152328</td>\n",
       "      <td>0.147664</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5500609</td>\n",
       "      <td>MH0174_GL0034307</td>\n",
       "      <td>519</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2811850</td>\n",
       "      <td>T2D-63A_GL0089788</td>\n",
       "      <td>906</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG40611</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3320656</td>\n",
       "      <td>O2.UC24-1_GL0155079</td>\n",
       "      <td>810</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG69716</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1293228</td>\n",
       "      <td>V1.UC50-1_GL0030242</td>\n",
       "      <td>1329</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01972</td>\n",
       "      <td>COG0272</td>\n",
       "      <td>0.024467</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Replication, recombination and repair</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>7219798</td>\n",
       "      <td>MH0355_GL0176135</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01897</td>\n",
       "      <td>COG0318</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.037383</td>\n",
       "      <td>Lipid Metabolism</td>\n",
       "      <td>Lipid transport and metabolism;Secondary metab...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2312660</td>\n",
       "      <td>158256496-stool1_revised_C920453_1_gene161597</td>\n",
       "      <td>1014</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG277117</td>\n",
       "      <td>0.085241</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3650803</td>\n",
       "      <td>MH0012_GL0124464</td>\n",
       "      <td>753</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02049</td>\n",
       "      <td>COG1116</td>\n",
       "      <td>0.326756</td>\n",
       "      <td>0.332710</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Inorganic ion transport and metabolism</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4137075</td>\n",
       "      <td>O2.UC48-0_GL0276256</td>\n",
       "      <td>681</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K06206</td>\n",
       "      <td>COG1489</td>\n",
       "      <td>0.046567</td>\n",
       "      <td>0.034579</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9031743</td>\n",
       "      <td>160765029-stool1_revised_C1108515_1_gene40264</td>\n",
       "      <td>171</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9139876</td>\n",
       "      <td>159753524-stool2_revised_C530648_1_gene94409</td>\n",
       "      <td>162</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6068154</td>\n",
       "      <td>MH0040_GL0048712</td>\n",
       "      <td>456</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Ruminococcus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.284136</td>\n",
       "      <td>0.285047</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>8094620</td>\n",
       "      <td>O2.UC47-2_GL0030825</td>\n",
       "      <td>246</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG256796</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6226899</td>\n",
       "      <td>MH0192_GL0144006</td>\n",
       "      <td>438</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01493</td>\n",
       "      <td>COG2131</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5203442</td>\n",
       "      <td>MH0188_GL0079716</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.086819</td>\n",
       "      <td>0.090654</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5870802</td>\n",
       "      <td>MH0096_GL0115416</td>\n",
       "      <td>480</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01520</td>\n",
       "      <td>COG0756</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>8222582</td>\n",
       "      <td>MH0309_GL0024067</td>\n",
       "      <td>234</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG291728</td>\n",
       "      <td>0.048145</td>\n",
       "      <td>0.051402</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5831026</td>\n",
       "      <td>MH0419_GL0063398</td>\n",
       "      <td>486</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG77285</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.030841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9553442</td>\n",
       "      <td>MH0149_GL0074986</td>\n",
       "      <td>126</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.335438</td>\n",
       "      <td>0.330841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5707244</td>\n",
       "      <td>V1.FI15_GL0072437</td>\n",
       "      <td>501</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07095</td>\n",
       "      <td>COG0622</td>\n",
       "      <td>0.096290</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1821827</td>\n",
       "      <td>MH0333_GL0003962</td>\n",
       "      <td>1146</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1073</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>0.011215</td>\n",
       "      <td>unknown</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8920818</td>\n",
       "      <td>764588959-stool1_revised_C775588_1_gene178391</td>\n",
       "      <td>180</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7928769</td>\n",
       "      <td>O2.UC48-1_GL0076618</td>\n",
       "      <td>261</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>647891</td>\n",
       "      <td>MH0433_GL0005093</td>\n",
       "      <td>1740</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>693534</td>\n",
       "      <td>MH0448_GL0045910</td>\n",
       "      <td>1695</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4807882</td>\n",
       "      <td>V1.FI14_GL0101784</td>\n",
       "      <td>594</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2441669</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "      <td>984</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Coprobacillus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.112865</td>\n",
       "      <td>0.113084</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8950791</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>177</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG125034</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7221353</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6811315</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>372</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00335</td>\n",
       "      <td>COG3411</td>\n",
       "      <td>0.187056</td>\n",
       "      <td>0.195327</td>\n",
       "      <td>Energy Metabolism</td>\n",
       "      <td>Energy production and conversion</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5209933</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4932</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "499999  5457503                                V1.UC36-0_GL0134515   \n",
       "499998  7655040  764042746-stool2_revised_scaffold22088_1_gene3...   \n",
       "499997  4924959                                   MH0161_GL0035075   \n",
       "499996   691723  638754422-stool2_revised_scaffold23317_1_gene1940   \n",
       "499995   278939  765560005-stool1_revised_scaffold5052_1_gene73628   \n",
       "499994  2951767                                V1.UC26-4_GL0007902   \n",
       "499993  7363211                                   MH0260_GL0023264   \n",
       "499992  8040629       159611913-stool1_revised_C786324_1_gene54480   \n",
       "499991  9631397                                   MH0280_GL0114474   \n",
       "499990  6294127                                V1.UC26-4_GL0011854   \n",
       "499989  6961318                                   MH0333_GL0111005   \n",
       "499988  5254719       159753524-stool2_revised_C538848_1_gene38208   \n",
       "499987  6439309                                   MH0158_GL0048733   \n",
       "499986  6567495                                   MH0065_GL0023744   \n",
       "499985  9067856      160400887-stool1_revised_C1382776_1_gene10717   \n",
       "499984  4147487                                   MH0062_GL0030772   \n",
       "499983   556756                                SZEY-101A_GL0065406   \n",
       "499982  7786169                                   MH0157_GL0106576   \n",
       "499981  8734258                                470145.BACCOP_04248   \n",
       "499980  7451388                                   MH0222_GL0182909   \n",
       "499979  7115654                                   DOF004_GL0014088   \n",
       "499978  9145946                                   MH0117_GL0078602   \n",
       "499977  8177332                            411485.FAEPRAM212_00318   \n",
       "499976  3280488                                   MH0028_GL0020655   \n",
       "499975  3484802                                   MH0188_GL0112429   \n",
       "499974  5233095                                   MH0204_GL0162383   \n",
       "499973  4517482                                   DOF003_GL0027603   \n",
       "499972   186663                                  V1.FI20_GL0146385   \n",
       "499971  6992342                                   MH0392_GL0212082   \n",
       "499970  4238583                                   MH0284_GL0171112   \n",
       "...         ...                                                ...   \n",
       "29      5500609                                   MH0174_GL0034307   \n",
       "28      2811850                                  T2D-63A_GL0089788   \n",
       "27      3320656                                O2.UC24-1_GL0155079   \n",
       "26      1293228                                V1.UC50-1_GL0030242   \n",
       "25      7219798                                   MH0355_GL0176135   \n",
       "24      2312660      158256496-stool1_revised_C920453_1_gene161597   \n",
       "23      3650803                                   MH0012_GL0124464   \n",
       "22      4137075                                O2.UC48-0_GL0276256   \n",
       "21      9031743      160765029-stool1_revised_C1108515_1_gene40264   \n",
       "20      9139876       159753524-stool2_revised_C530648_1_gene94409   \n",
       "19      6068154                                   MH0040_GL0048712   \n",
       "18      8094620                                O2.UC47-2_GL0030825   \n",
       "17      6226899                                   MH0192_GL0144006   \n",
       "16      5203442                                   MH0188_GL0079716   \n",
       "15      5870802                                   MH0096_GL0115416   \n",
       "14      8222582                                   MH0309_GL0024067   \n",
       "13      5831026                                   MH0419_GL0063398   \n",
       "12      9553442                                   MH0149_GL0074986   \n",
       "11      5707244                                  V1.FI15_GL0072437   \n",
       "10      1821827                                   MH0333_GL0003962   \n",
       "9       8920818      764588959-stool1_revised_C775588_1_gene178391   \n",
       "8       7928769                                O2.UC48-1_GL0076618   \n",
       "7        647891                                   MH0433_GL0005093   \n",
       "6        693534                                   MH0448_GL0045910   \n",
       "5       4807882                                  V1.FI14_GL0101784   \n",
       "4       2441669                            469596.HMPREF9488_01493   \n",
       "3       8950791                                   DOM015_GL0050638   \n",
       "2       7221353                                   MH0389_GL0170585   \n",
       "1       6811315                                   MH0012_GL0174453   \n",
       "0       5209933                                   MH0396_GL0114156   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "499999          525       Lack 5'-end           EUR   \n",
       "499998          288          Complete           USA   \n",
       "499997          579          Complete           EUR   \n",
       "499996         1698          Complete           USA   \n",
       "499995         2349       Lack 5'-end           USA   \n",
       "499994          879          Complete           EUR   \n",
       "499993          315       Lack 3'-end           EUR   \n",
       "499992          252          Complete           USA   \n",
       "499991          120       Lack 5'-end           EUR   \n",
       "499990          432       Lack 5'-end           EUR   \n",
       "499989          357       Lack 3'-end           EUR   \n",
       "499988          546    Lack both ends           USA   \n",
       "499987          414       Lack 5'-end           EUR   \n",
       "499986          399       Lack 5'-end           EUR   \n",
       "499985          168          Complete           USA   \n",
       "499984          678          Complete           EUR   \n",
       "499983         1839          Complete           CHN   \n",
       "499982          273          Complete           EUR   \n",
       "499981          195          Complete            SP   \n",
       "499980          306          Complete           EUR   \n",
       "499979          342       Lack 5'-end           CHN   \n",
       "499978          159          Complete           EUR   \n",
       "499977          240          Complete            SP   \n",
       "499976          816          Complete           EUR   \n",
       "499975          780          Complete           EUR   \n",
       "499974          546    Lack both ends           EUR   \n",
       "499973          630          Complete           CHN   \n",
       "499972         2661          Complete           EUR   \n",
       "499971          354       Lack 5'-end           EUR   \n",
       "499970          666          Complete           EUR   \n",
       "...             ...               ...           ...   \n",
       "29              519    Lack both ends           EUR   \n",
       "28              906          Complete           CHN   \n",
       "27              810    Lack both ends           EUR   \n",
       "26             1329       Lack 5'-end           EUR   \n",
       "25              330       Lack 3'-end           EUR   \n",
       "24             1014    Lack both ends           USA   \n",
       "23              753          Complete           EUR   \n",
       "22              681          Complete           EUR   \n",
       "21              171          Complete           USA   \n",
       "20              162       Lack 5'-end           USA   \n",
       "19              456       Lack 5'-end           EUR   \n",
       "18              246       Lack 3'-end           EUR   \n",
       "17              438          Complete           EUR   \n",
       "16              549    Lack both ends           EUR   \n",
       "15              480          Complete           EUR   \n",
       "14              234       Lack 3'-end           EUR   \n",
       "13              486          Complete           EUR   \n",
       "12              126       Lack 5'-end           EUR   \n",
       "11              501          Complete           EUR   \n",
       "10             1146          Complete           EUR   \n",
       "9               180       Lack 3'-end           USA   \n",
       "8               261       Lack 3'-end           EUR   \n",
       "7              1740          Complete           EUR   \n",
       "6              1695    Lack both ends           EUR   \n",
       "5               594          Complete           EUR   \n",
       "4               984          Complete            SP   \n",
       "3               177       Lack 5'-end           CHN   \n",
       "2               330       Lack 5'-end           EUR   \n",
       "1               372          Complete           EUR   \n",
       "0               549    Lack both ends           EUR   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "499999                            unknown                           unknown   \n",
       "499998                            unknown                           unknown   \n",
       "499997                            unknown                           unknown   \n",
       "499996                            unknown                           unknown   \n",
       "499995                            unknown                           unknown   \n",
       "499994                            unknown                           unknown   \n",
       "499993                            unknown                           unknown   \n",
       "499992                         Firmicutes                  Faecalibacterium   \n",
       "499991                            unknown                           unknown   \n",
       "499990                            unknown                           unknown   \n",
       "499989                            unknown                           unknown   \n",
       "499988                            unknown                           unknown   \n",
       "499987                            unknown                           unknown   \n",
       "499986                            unknown                           unknown   \n",
       "499985                            unknown                           unknown   \n",
       "499984                         Firmicutes                           unknown   \n",
       "499983                            unknown                           unknown   \n",
       "499982                            unknown                           unknown   \n",
       "499981                      Bacteroidetes                       Bacteroides   \n",
       "499980                            unknown                           unknown   \n",
       "499979                            unknown                           unknown   \n",
       "499978                            unknown                           unknown   \n",
       "499977                         Firmicutes                  Faecalibacterium   \n",
       "499976                         Firmicutes                       Eubacterium   \n",
       "499975                            unknown                           unknown   \n",
       "499974                            unknown                           unknown   \n",
       "499973                      Bacteroidetes                       Bacteroides   \n",
       "499972                            unknown                           unknown   \n",
       "499971                      Bacteroidetes                       Bacteroides   \n",
       "499970                            unknown                           unknown   \n",
       "...                                   ...                               ...   \n",
       "29                                unknown                           unknown   \n",
       "28                                unknown                           unknown   \n",
       "27                                unknown                           unknown   \n",
       "26                                unknown                           unknown   \n",
       "25                                unknown                           unknown   \n",
       "24                                unknown                           unknown   \n",
       "23                                unknown                           unknown   \n",
       "22                                unknown                           unknown   \n",
       "21                                unknown                           unknown   \n",
       "20                                unknown                           unknown   \n",
       "19                             Firmicutes                      Ruminococcus   \n",
       "18                                unknown                           unknown   \n",
       "17                          Bacteroidetes                           unknown   \n",
       "16                                unknown                           unknown   \n",
       "15                                unknown                           unknown   \n",
       "14                                unknown                           unknown   \n",
       "13                                unknown                           unknown   \n",
       "12                                unknown                           unknown   \n",
       "11                                unknown                           unknown   \n",
       "10                          Bacteroidetes                           unknown   \n",
       "9                                 unknown                           unknown   \n",
       "8                                 unknown                           unknown   \n",
       "7                                 unknown                           unknown   \n",
       "6                                 unknown                           unknown   \n",
       "5                                 unknown                           unknown   \n",
       "4                              Firmicutes                     Coprobacillus   \n",
       "3                                 unknown                           unknown   \n",
       "2                                 unknown                           unknown   \n",
       "1                                 unknown                           unknown   \n",
       "0                                 unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "499999         unknown           unknown                    0.008682   \n",
       "499998         unknown           unknown                    0.003946   \n",
       "499997         unknown           unknown                    0.054459   \n",
       "499996         unknown           unknown                    0.001579   \n",
       "499995          K00571           COG1002                    0.153907   \n",
       "499994         unknown          NOG47185                    0.004736   \n",
       "499993         unknown           unknown                    0.003946   \n",
       "499992         unknown           unknown                    0.018942   \n",
       "499991         unknown           unknown                    0.037096   \n",
       "499990          K07024           COG0561                    0.002368   \n",
       "499989         unknown           unknown                    0.007893   \n",
       "499988          K03531           COG0206                    0.004736   \n",
       "499987          K03666           COG1923                    0.440410   \n",
       "499986         unknown           unknown                    0.007103   \n",
       "499985         unknown           unknown                    0.003946   \n",
       "499984          K07658           COG0745                    0.229676   \n",
       "499983         unknown           unknown                    0.000789   \n",
       "499982         unknown           unknown                    0.000789   \n",
       "499981         unknown           unknown                    0.015785   \n",
       "499980         unknown           unknown                    0.002368   \n",
       "499979         unknown           unknown                    0.015785   \n",
       "499978         unknown           unknown                    0.014996   \n",
       "499977         unknown           unknown                    0.783741   \n",
       "499976          K09690           COG1682                    0.361484   \n",
       "499975         unknown           unknown                    0.003157   \n",
       "499974         unknown           unknown                    0.020521   \n",
       "499973         unknown         NOG275003                    0.233623   \n",
       "499972          K02355   COG3688;COG0480                    0.727703   \n",
       "499971         unknown           unknown                    0.001579   \n",
       "499970         unknown           unknown                    0.152328   \n",
       "...                ...               ...                         ...   \n",
       "29             unknown           unknown                    0.003157   \n",
       "28             unknown          NOG40611                    0.008682   \n",
       "27             unknown          NOG69716                    0.019732   \n",
       "26              K01972           COG0272                    0.024467   \n",
       "25              K01897           COG0318                    0.033938   \n",
       "24             unknown         NOG277117                    0.085241   \n",
       "23              K02049           COG1116                    0.326756   \n",
       "22              K06206           COG1489                    0.046567   \n",
       "21             unknown           unknown                    0.000789   \n",
       "20             unknown           unknown                    0.003946   \n",
       "19             unknown           unknown                    0.284136   \n",
       "18             unknown         NOG256796                    0.005525   \n",
       "17              K01493           COG2131                    0.007103   \n",
       "16             unknown           unknown                    0.086819   \n",
       "15              K01520           COG0756                    0.000789   \n",
       "14             unknown         NOG291728                    0.048145   \n",
       "13             unknown          NOG77285                    0.032360   \n",
       "12             unknown           unknown                    0.335438   \n",
       "11              K07095           COG0622                    0.096290   \n",
       "10             unknown           COG1073                    0.010260   \n",
       "9              unknown           unknown                    0.000789   \n",
       "8              unknown           unknown                    0.014996   \n",
       "7              unknown           unknown                    0.003946   \n",
       "6              unknown          NOG12793                    0.004736   \n",
       "5              unknown           unknown                    0.008682   \n",
       "4              unknown           unknown                    0.112865   \n",
       "3              unknown         NOG125034                    0.008682   \n",
       "2              unknown           unknown                    0.015785   \n",
       "1               K00335           COG3411                    0.187056   \n",
       "0              unknown           COG4932                    0.008682   \n",
       "\n",
       "        Individual Occurence Frequency        KEGG Functional Categories  \\\n",
       "499999                        0.007477                           unknown   \n",
       "499998                        0.002804                           unknown   \n",
       "499997                        0.064486                           unknown   \n",
       "499996                        0.000935                           unknown   \n",
       "499995                        0.152336    Genetic Information Processing   \n",
       "499994                        0.003738                           unknown   \n",
       "499993                        0.004673                           unknown   \n",
       "499992                        0.017757                           unknown   \n",
       "499991                        0.033645                           unknown   \n",
       "499990                        0.001869              Poorly Characterized   \n",
       "499989                        0.007477                           unknown   \n",
       "499988                        0.002804            Replication and Repair   \n",
       "499987                        0.445794  Folding, Sorting and Degradation   \n",
       "499986                        0.008411                           unknown   \n",
       "499985                        0.004673                           unknown   \n",
       "499984                        0.244860               Signal Transduction   \n",
       "499983                        0.000935                           unknown   \n",
       "499982                        0.000935                           unknown   \n",
       "499981                        0.016822                           unknown   \n",
       "499980                        0.002804                           unknown   \n",
       "499979                        0.017757                           unknown   \n",
       "499978                        0.015888                           unknown   \n",
       "499977                        0.778505                           unknown   \n",
       "499976                        0.362617                Membrane Transport   \n",
       "499975                        0.003738                           unknown   \n",
       "499974                        0.021495                           unknown   \n",
       "499973                        0.229907                           unknown   \n",
       "499972                        0.721495                       Translation   \n",
       "499971                        0.000935                           unknown   \n",
       "499970                        0.147664                           unknown   \n",
       "...                                ...                               ...   \n",
       "29                            0.003738                           unknown   \n",
       "28                            0.009346                           unknown   \n",
       "27                            0.018692                           unknown   \n",
       "26                            0.021495            Replication and Repair   \n",
       "25                            0.037383                  Lipid Metabolism   \n",
       "24                            0.079439                           unknown   \n",
       "23                            0.332710                Membrane Transport   \n",
       "22                            0.034579                        Metabolism   \n",
       "21                            0.000935                           unknown   \n",
       "20                            0.002804                           unknown   \n",
       "19                            0.285047                           unknown   \n",
       "18                            0.004673                           unknown   \n",
       "17                            0.004673             Nucleotide Metabolism   \n",
       "16                            0.090654                           unknown   \n",
       "15                            0.000935             Nucleotide Metabolism   \n",
       "14                            0.051402                           unknown   \n",
       "13                            0.030841                           unknown   \n",
       "12                            0.330841                           unknown   \n",
       "11                            0.100000              Poorly Characterized   \n",
       "10                            0.011215                           unknown   \n",
       "9                             0.000935                           unknown   \n",
       "8                             0.008411                           unknown   \n",
       "7                             0.004673                           unknown   \n",
       "6                             0.004673                           unknown   \n",
       "5                             0.009346                           unknown   \n",
       "4                             0.113084                           unknown   \n",
       "3                             0.008411                           unknown   \n",
       "2                             0.018692                           unknown   \n",
       "1                             0.195327                 Energy Metabolism   \n",
       "0                             0.009346                           unknown   \n",
       "\n",
       "                             eggNOG Functional Categories Cohort Assembled  \n",
       "499999                                            unknown              EUR  \n",
       "499998                                            unknown              USA  \n",
       "499997                                            unknown          EUR;CHN  \n",
       "499996                                            unknown              USA  \n",
       "499995                                 Defense mechanisms          EUR;USA  \n",
       "499994                                   Function unknown              EUR  \n",
       "499993                                            unknown              EUR  \n",
       "499992                                            unknown              USA  \n",
       "499991                                            unknown              EUR  \n",
       "499990                   General function prediction only              EUR  \n",
       "499989                                            unknown              EUR  \n",
       "499988  Cell cycle control, cell division, chromosome ...              USA  \n",
       "499987                   General function prediction only      EUR;CHN;USA  \n",
       "499986                                            unknown              EUR  \n",
       "499985                                            unknown              USA  \n",
       "499984       Transcription;Signal transduction mechanisms      EUR;CHN;USA  \n",
       "499983                                            unknown              CHN  \n",
       "499982                                            unknown              EUR  \n",
       "499981                                            unknown              NaN  \n",
       "499980                                            unknown              EUR  \n",
       "499979                                            unknown              CHN  \n",
       "499978                                            unknown              EUR  \n",
       "499977                                            unknown              EUR  \n",
       "499976  Carbohydrate transport and metabolism;Cell wal...      EUR;CHN;USA  \n",
       "499975                                            unknown              EUR  \n",
       "499974                                            unknown              EUR  \n",
       "499973                                            unknown              CHN  \n",
       "499972  Translation, ribosomal structure and biogenesi...              EUR  \n",
       "499971                                            unknown              EUR  \n",
       "499970                                            unknown      EUR;CHN;USA  \n",
       "...                                                   ...              ...  \n",
       "29                                                unknown              EUR  \n",
       "28                                       Function unknown              CHN  \n",
       "27                                       Function unknown              EUR  \n",
       "26                  Replication, recombination and repair              EUR  \n",
       "25      Lipid transport and metabolism;Secondary metab...              EUR  \n",
       "24                                                unknown              USA  \n",
       "23                 Inorganic ion transport and metabolism      EUR;CHN;USA  \n",
       "22                       General function prediction only              EUR  \n",
       "21                                                unknown              USA  \n",
       "20                                                unknown              USA  \n",
       "19                                                unknown              EUR  \n",
       "18                                       Function unknown              EUR  \n",
       "17                    Nucleotide transport and metabolism              EUR  \n",
       "16                                                unknown              EUR  \n",
       "15                    Nucleotide transport and metabolism              EUR  \n",
       "14                                                unknown              EUR  \n",
       "13                                       Function unknown              EUR  \n",
       "12                                                unknown              EUR  \n",
       "11                       General function prediction only              EUR  \n",
       "10                       General function prediction only          EUR;CHN  \n",
       "9                                                 unknown              USA  \n",
       "8                                                 unknown              EUR  \n",
       "7                                                 unknown          EUR;USA  \n",
       "6                                        Function unknown              EUR  \n",
       "5                                                 unknown              EUR  \n",
       "4                                                 unknown      EUR;CHN;USA  \n",
       "3                                        Function unknown              CHN  \n",
       "2                                                 unknown              EUR  \n",
       "1                        Energy production and conversion      EUR;CHN;USA  \n",
       "0                  Cell wall/membrane/envelope biogenesis              EUR  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ordenar por filas(axis 0), descendente\n",
    "df.sort_index(axis=0, ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cohort Assembled</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>5209933</td>\n",
       "      <td>549</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4932</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6811315</td>\n",
       "      <td>372</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>0.195327</td>\n",
       "      <td>K00335</td>\n",
       "      <td>Energy Metabolism</td>\n",
       "      <td>0.187056</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG3411</td>\n",
       "      <td>Energy production and conversion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>7221353</td>\n",
       "      <td>330</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CHN</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>8950791</td>\n",
       "      <td>177</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG125034</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>SP</td>\n",
       "      <td>Complete</td>\n",
       "      <td>2441669</td>\n",
       "      <td>984</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "      <td>0.113084</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.112865</td>\n",
       "      <td>Coprobacillus</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4807882</td>\n",
       "      <td>594</td>\n",
       "      <td>V1.FI14_GL0101784</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>693534</td>\n",
       "      <td>1695</td>\n",
       "      <td>MH0448_GL0045910</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>EUR;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>647891</td>\n",
       "      <td>1740</td>\n",
       "      <td>MH0433_GL0005093</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>7928769</td>\n",
       "      <td>261</td>\n",
       "      <td>O2.UC48-1_GL0076618</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>8920818</td>\n",
       "      <td>180</td>\n",
       "      <td>764588959-stool1_revised_C775588_1_gene178391</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>EUR;CHN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>1821827</td>\n",
       "      <td>1146</td>\n",
       "      <td>MH0333_GL0003962</td>\n",
       "      <td>0.011215</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>COG1073</td>\n",
       "      <td>General function prediction only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>5707244</td>\n",
       "      <td>501</td>\n",
       "      <td>V1.FI15_GL0072437</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>K07095</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>0.096290</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0622</td>\n",
       "      <td>General function prediction only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>9553442</td>\n",
       "      <td>126</td>\n",
       "      <td>MH0149_GL0074986</td>\n",
       "      <td>0.330841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.335438</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>5831026</td>\n",
       "      <td>486</td>\n",
       "      <td>MH0419_GL0063398</td>\n",
       "      <td>0.030841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG77285</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>8222582</td>\n",
       "      <td>234</td>\n",
       "      <td>MH0309_GL0024067</td>\n",
       "      <td>0.051402</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.048145</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG291728</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>5870802</td>\n",
       "      <td>480</td>\n",
       "      <td>MH0096_GL0115416</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>K01520</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0756</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>5203442</td>\n",
       "      <td>549</td>\n",
       "      <td>MH0188_GL0079716</td>\n",
       "      <td>0.090654</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.086819</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6226899</td>\n",
       "      <td>438</td>\n",
       "      <td>MH0192_GL0144006</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>K01493</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>COG2131</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>8094620</td>\n",
       "      <td>246</td>\n",
       "      <td>O2.UC47-2_GL0030825</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG256796</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>6068154</td>\n",
       "      <td>456</td>\n",
       "      <td>MH0040_GL0048712</td>\n",
       "      <td>0.285047</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.284136</td>\n",
       "      <td>Ruminococcus</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>9139876</td>\n",
       "      <td>162</td>\n",
       "      <td>159753524-stool2_revised_C530648_1_gene94409</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Complete</td>\n",
       "      <td>9031743</td>\n",
       "      <td>171</td>\n",
       "      <td>160765029-stool1_revised_C1108515_1_gene40264</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4137075</td>\n",
       "      <td>681</td>\n",
       "      <td>O2.UC48-0_GL0276256</td>\n",
       "      <td>0.034579</td>\n",
       "      <td>K06206</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>0.046567</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1489</td>\n",
       "      <td>General function prediction only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>3650803</td>\n",
       "      <td>753</td>\n",
       "      <td>MH0012_GL0124464</td>\n",
       "      <td>0.332710</td>\n",
       "      <td>K02049</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>0.326756</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1116</td>\n",
       "      <td>Inorganic ion transport and metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>2312660</td>\n",
       "      <td>1014</td>\n",
       "      <td>158256496-stool1_revised_C920453_1_gene161597</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.085241</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG277117</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>7219798</td>\n",
       "      <td>330</td>\n",
       "      <td>MH0355_GL0176135</td>\n",
       "      <td>0.037383</td>\n",
       "      <td>K01897</td>\n",
       "      <td>Lipid Metabolism</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0318</td>\n",
       "      <td>Lipid transport and metabolism;Secondary metab...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>1293228</td>\n",
       "      <td>1329</td>\n",
       "      <td>V1.UC50-1_GL0030242</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>K01972</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>0.024467</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0272</td>\n",
       "      <td>Replication, recombination and repair</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>3320656</td>\n",
       "      <td>810</td>\n",
       "      <td>O2.UC24-1_GL0155079</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG69716</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>CHN</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Complete</td>\n",
       "      <td>2811850</td>\n",
       "      <td>906</td>\n",
       "      <td>T2D-63A_GL0089788</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG40611</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>5500609</td>\n",
       "      <td>519</td>\n",
       "      <td>MH0174_GL0034307</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499970</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4238583</td>\n",
       "      <td>666</td>\n",
       "      <td>MH0284_GL0171112</td>\n",
       "      <td>0.147664</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.152328</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499971</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>6992342</td>\n",
       "      <td>354</td>\n",
       "      <td>MH0392_GL0212082</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499972</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>186663</td>\n",
       "      <td>2661</td>\n",
       "      <td>V1.FI20_GL0146385</td>\n",
       "      <td>0.721495</td>\n",
       "      <td>K02355</td>\n",
       "      <td>Translation</td>\n",
       "      <td>0.727703</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG3688;COG0480</td>\n",
       "      <td>Translation, ribosomal structure and biogenesi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499973</th>\n",
       "      <td>CHN</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4517482</td>\n",
       "      <td>630</td>\n",
       "      <td>DOF003_GL0027603</td>\n",
       "      <td>0.229907</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.233623</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>NOG275003</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499974</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>5233095</td>\n",
       "      <td>546</td>\n",
       "      <td>MH0204_GL0162383</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499975</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>3484802</td>\n",
       "      <td>780</td>\n",
       "      <td>MH0188_GL0112429</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499976</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>3280488</td>\n",
       "      <td>816</td>\n",
       "      <td>MH0028_GL0020655</td>\n",
       "      <td>0.362617</td>\n",
       "      <td>K09690</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>0.361484</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>COG1682</td>\n",
       "      <td>Carbohydrate transport and metabolism;Cell wal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499977</th>\n",
       "      <td>EUR</td>\n",
       "      <td>SP</td>\n",
       "      <td>Complete</td>\n",
       "      <td>8177332</td>\n",
       "      <td>240</td>\n",
       "      <td>411485.FAEPRAM212_00318</td>\n",
       "      <td>0.778505</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.783741</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499978</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>9145946</td>\n",
       "      <td>159</td>\n",
       "      <td>MH0117_GL0078602</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499979</th>\n",
       "      <td>CHN</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>7115654</td>\n",
       "      <td>342</td>\n",
       "      <td>DOF004_GL0014088</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499980</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>7451388</td>\n",
       "      <td>306</td>\n",
       "      <td>MH0222_GL0182909</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499981</th>\n",
       "      <td>NaN</td>\n",
       "      <td>SP</td>\n",
       "      <td>Complete</td>\n",
       "      <td>8734258</td>\n",
       "      <td>195</td>\n",
       "      <td>470145.BACCOP_04248</td>\n",
       "      <td>0.016822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499982</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>7786169</td>\n",
       "      <td>273</td>\n",
       "      <td>MH0157_GL0106576</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499983</th>\n",
       "      <td>CHN</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Complete</td>\n",
       "      <td>556756</td>\n",
       "      <td>1839</td>\n",
       "      <td>SZEY-101A_GL0065406</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499984</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4147487</td>\n",
       "      <td>678</td>\n",
       "      <td>MH0062_GL0030772</td>\n",
       "      <td>0.244860</td>\n",
       "      <td>K07658</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>COG0745</td>\n",
       "      <td>Transcription;Signal transduction mechanisms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499985</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Complete</td>\n",
       "      <td>9067856</td>\n",
       "      <td>168</td>\n",
       "      <td>160400887-stool1_revised_C1382776_1_gene10717</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499986</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>6567495</td>\n",
       "      <td>399</td>\n",
       "      <td>MH0065_GL0023744</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499987</th>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>6439309</td>\n",
       "      <td>414</td>\n",
       "      <td>MH0158_GL0048733</td>\n",
       "      <td>0.445794</td>\n",
       "      <td>K03666</td>\n",
       "      <td>Folding, Sorting and Degradation</td>\n",
       "      <td>0.440410</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1923</td>\n",
       "      <td>General function prediction only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499988</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>5254719</td>\n",
       "      <td>546</td>\n",
       "      <td>159753524-stool2_revised_C538848_1_gene38208</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>K03531</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0206</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499989</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>6961318</td>\n",
       "      <td>357</td>\n",
       "      <td>MH0333_GL0111005</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007893</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499990</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>6294127</td>\n",
       "      <td>432</td>\n",
       "      <td>V1.UC26-4_GL0011854</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>K07024</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG0561</td>\n",
       "      <td>General function prediction only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499991</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>9631397</td>\n",
       "      <td>120</td>\n",
       "      <td>MH0280_GL0114474</td>\n",
       "      <td>0.033645</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.037096</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499992</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Complete</td>\n",
       "      <td>8040629</td>\n",
       "      <td>252</td>\n",
       "      <td>159611913-stool1_revised_C786324_1_gene54480</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.018942</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499993</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>7363211</td>\n",
       "      <td>315</td>\n",
       "      <td>MH0260_GL0023264</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499994</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>2951767</td>\n",
       "      <td>879</td>\n",
       "      <td>V1.UC26-4_GL0007902</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG47185</td>\n",
       "      <td>Function unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499995</th>\n",
       "      <td>EUR;USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>278939</td>\n",
       "      <td>2349</td>\n",
       "      <td>765560005-stool1_revised_scaffold5052_1_gene73628</td>\n",
       "      <td>0.152336</td>\n",
       "      <td>K00571</td>\n",
       "      <td>Genetic Information Processing</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1002</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499996</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Complete</td>\n",
       "      <td>691723</td>\n",
       "      <td>1698</td>\n",
       "      <td>638754422-stool2_revised_scaffold23317_1_gene1940</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499997</th>\n",
       "      <td>EUR;CHN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Complete</td>\n",
       "      <td>4924959</td>\n",
       "      <td>579</td>\n",
       "      <td>MH0161_GL0035075</td>\n",
       "      <td>0.064486</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.054459</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499998</th>\n",
       "      <td>USA</td>\n",
       "      <td>USA</td>\n",
       "      <td>Complete</td>\n",
       "      <td>7655040</td>\n",
       "      <td>288</td>\n",
       "      <td>764042746-stool2_revised_scaffold22088_1_gene3...</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499999</th>\n",
       "      <td>EUR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>5457503</td>\n",
       "      <td>525</td>\n",
       "      <td>V1.UC36-0_GL0134515</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cohort Assembled Cohort Origin Gene Completeness  Gene ID  Gene Length  \\\n",
       "0                   EUR           EUR    Lack both ends  5209933          549   \n",
       "1           EUR;CHN;USA           EUR          Complete  6811315          372   \n",
       "2                   EUR           EUR       Lack 5'-end  7221353          330   \n",
       "3                   CHN           CHN       Lack 5'-end  8950791          177   \n",
       "4           EUR;CHN;USA            SP          Complete  2441669          984   \n",
       "5                   EUR           EUR          Complete  4807882          594   \n",
       "6                   EUR           EUR    Lack both ends   693534         1695   \n",
       "7               EUR;USA           EUR          Complete   647891         1740   \n",
       "8                   EUR           EUR       Lack 3'-end  7928769          261   \n",
       "9                   USA           USA       Lack 3'-end  8920818          180   \n",
       "10              EUR;CHN           EUR          Complete  1821827         1146   \n",
       "11                  EUR           EUR          Complete  5707244          501   \n",
       "12                  EUR           EUR       Lack 5'-end  9553442          126   \n",
       "13                  EUR           EUR          Complete  5831026          486   \n",
       "14                  EUR           EUR       Lack 3'-end  8222582          234   \n",
       "15                  EUR           EUR          Complete  5870802          480   \n",
       "16                  EUR           EUR    Lack both ends  5203442          549   \n",
       "17                  EUR           EUR          Complete  6226899          438   \n",
       "18                  EUR           EUR       Lack 3'-end  8094620          246   \n",
       "19                  EUR           EUR       Lack 5'-end  6068154          456   \n",
       "20                  USA           USA       Lack 5'-end  9139876          162   \n",
       "21                  USA           USA          Complete  9031743          171   \n",
       "22                  EUR           EUR          Complete  4137075          681   \n",
       "23          EUR;CHN;USA           EUR          Complete  3650803          753   \n",
       "24                  USA           USA    Lack both ends  2312660         1014   \n",
       "25                  EUR           EUR       Lack 3'-end  7219798          330   \n",
       "26                  EUR           EUR       Lack 5'-end  1293228         1329   \n",
       "27                  EUR           EUR    Lack both ends  3320656          810   \n",
       "28                  CHN           CHN          Complete  2811850          906   \n",
       "29                  EUR           EUR    Lack both ends  5500609          519   \n",
       "...                 ...           ...               ...      ...          ...   \n",
       "499970      EUR;CHN;USA           EUR          Complete  4238583          666   \n",
       "499971              EUR           EUR       Lack 5'-end  6992342          354   \n",
       "499972              EUR           EUR          Complete   186663         2661   \n",
       "499973              CHN           CHN          Complete  4517482          630   \n",
       "499974              EUR           EUR    Lack both ends  5233095          546   \n",
       "499975              EUR           EUR          Complete  3484802          780   \n",
       "499976      EUR;CHN;USA           EUR          Complete  3280488          816   \n",
       "499977              EUR            SP          Complete  8177332          240   \n",
       "499978              EUR           EUR          Complete  9145946          159   \n",
       "499979              CHN           CHN       Lack 5'-end  7115654          342   \n",
       "499980              EUR           EUR          Complete  7451388          306   \n",
       "499981              NaN            SP          Complete  8734258          195   \n",
       "499982              EUR           EUR          Complete  7786169          273   \n",
       "499983              CHN           CHN          Complete   556756         1839   \n",
       "499984      EUR;CHN;USA           EUR          Complete  4147487          678   \n",
       "499985              USA           USA          Complete  9067856          168   \n",
       "499986              EUR           EUR       Lack 5'-end  6567495          399   \n",
       "499987      EUR;CHN;USA           EUR       Lack 5'-end  6439309          414   \n",
       "499988              USA           USA    Lack both ends  5254719          546   \n",
       "499989              EUR           EUR       Lack 3'-end  6961318          357   \n",
       "499990              EUR           EUR       Lack 5'-end  6294127          432   \n",
       "499991              EUR           EUR       Lack 5'-end  9631397          120   \n",
       "499992              USA           USA          Complete  8040629          252   \n",
       "499993              EUR           EUR       Lack 3'-end  7363211          315   \n",
       "499994              EUR           EUR          Complete  2951767          879   \n",
       "499995          EUR;USA           USA       Lack 5'-end   278939         2349   \n",
       "499996              USA           USA          Complete   691723         1698   \n",
       "499997          EUR;CHN           EUR          Complete  4924959          579   \n",
       "499998              USA           USA          Complete  7655040          288   \n",
       "499999              EUR           EUR       Lack 5'-end  5457503          525   \n",
       "\n",
       "                                                Gene Name  \\\n",
       "0                                        MH0396_GL0114156   \n",
       "1                                        MH0012_GL0174453   \n",
       "2                                        MH0389_GL0170585   \n",
       "3                                        DOM015_GL0050638   \n",
       "4                                 469596.HMPREF9488_01493   \n",
       "5                                       V1.FI14_GL0101784   \n",
       "6                                        MH0448_GL0045910   \n",
       "7                                        MH0433_GL0005093   \n",
       "8                                     O2.UC48-1_GL0076618   \n",
       "9           764588959-stool1_revised_C775588_1_gene178391   \n",
       "10                                       MH0333_GL0003962   \n",
       "11                                      V1.FI15_GL0072437   \n",
       "12                                       MH0149_GL0074986   \n",
       "13                                       MH0419_GL0063398   \n",
       "14                                       MH0309_GL0024067   \n",
       "15                                       MH0096_GL0115416   \n",
       "16                                       MH0188_GL0079716   \n",
       "17                                       MH0192_GL0144006   \n",
       "18                                    O2.UC47-2_GL0030825   \n",
       "19                                       MH0040_GL0048712   \n",
       "20           159753524-stool2_revised_C530648_1_gene94409   \n",
       "21          160765029-stool1_revised_C1108515_1_gene40264   \n",
       "22                                    O2.UC48-0_GL0276256   \n",
       "23                                       MH0012_GL0124464   \n",
       "24          158256496-stool1_revised_C920453_1_gene161597   \n",
       "25                                       MH0355_GL0176135   \n",
       "26                                    V1.UC50-1_GL0030242   \n",
       "27                                    O2.UC24-1_GL0155079   \n",
       "28                                      T2D-63A_GL0089788   \n",
       "29                                       MH0174_GL0034307   \n",
       "...                                                   ...   \n",
       "499970                                   MH0284_GL0171112   \n",
       "499971                                   MH0392_GL0212082   \n",
       "499972                                  V1.FI20_GL0146385   \n",
       "499973                                   DOF003_GL0027603   \n",
       "499974                                   MH0204_GL0162383   \n",
       "499975                                   MH0188_GL0112429   \n",
       "499976                                   MH0028_GL0020655   \n",
       "499977                            411485.FAEPRAM212_00318   \n",
       "499978                                   MH0117_GL0078602   \n",
       "499979                                   DOF004_GL0014088   \n",
       "499980                                   MH0222_GL0182909   \n",
       "499981                                470145.BACCOP_04248   \n",
       "499982                                   MH0157_GL0106576   \n",
       "499983                                SZEY-101A_GL0065406   \n",
       "499984                                   MH0062_GL0030772   \n",
       "499985      160400887-stool1_revised_C1382776_1_gene10717   \n",
       "499986                                   MH0065_GL0023744   \n",
       "499987                                   MH0158_GL0048733   \n",
       "499988       159753524-stool2_revised_C538848_1_gene38208   \n",
       "499989                                   MH0333_GL0111005   \n",
       "499990                                V1.UC26-4_GL0011854   \n",
       "499991                                   MH0280_GL0114474   \n",
       "499992       159611913-stool1_revised_C786324_1_gene54480   \n",
       "499993                                   MH0260_GL0023264   \n",
       "499994                                V1.UC26-4_GL0007902   \n",
       "499995  765560005-stool1_revised_scaffold5052_1_gene73628   \n",
       "499996  638754422-stool2_revised_scaffold23317_1_gene1940   \n",
       "499997                                   MH0161_GL0035075   \n",
       "499998  764042746-stool2_revised_scaffold22088_1_gene3...   \n",
       "499999                                V1.UC36-0_GL0134515   \n",
       "\n",
       "        Individual Occurence Frequency KEGG Annotation  \\\n",
       "0                             0.009346         unknown   \n",
       "1                             0.195327          K00335   \n",
       "2                             0.018692         unknown   \n",
       "3                             0.008411         unknown   \n",
       "4                             0.113084         unknown   \n",
       "5                             0.009346         unknown   \n",
       "6                             0.004673         unknown   \n",
       "7                             0.004673         unknown   \n",
       "8                             0.008411         unknown   \n",
       "9                             0.000935         unknown   \n",
       "10                            0.011215         unknown   \n",
       "11                            0.100000          K07095   \n",
       "12                            0.330841         unknown   \n",
       "13                            0.030841         unknown   \n",
       "14                            0.051402         unknown   \n",
       "15                            0.000935          K01520   \n",
       "16                            0.090654         unknown   \n",
       "17                            0.004673          K01493   \n",
       "18                            0.004673         unknown   \n",
       "19                            0.285047         unknown   \n",
       "20                            0.002804         unknown   \n",
       "21                            0.000935         unknown   \n",
       "22                            0.034579          K06206   \n",
       "23                            0.332710          K02049   \n",
       "24                            0.079439         unknown   \n",
       "25                            0.037383          K01897   \n",
       "26                            0.021495          K01972   \n",
       "27                            0.018692         unknown   \n",
       "28                            0.009346         unknown   \n",
       "29                            0.003738         unknown   \n",
       "...                                ...             ...   \n",
       "499970                        0.147664         unknown   \n",
       "499971                        0.000935         unknown   \n",
       "499972                        0.721495          K02355   \n",
       "499973                        0.229907         unknown   \n",
       "499974                        0.021495         unknown   \n",
       "499975                        0.003738         unknown   \n",
       "499976                        0.362617          K09690   \n",
       "499977                        0.778505         unknown   \n",
       "499978                        0.015888         unknown   \n",
       "499979                        0.017757         unknown   \n",
       "499980                        0.002804         unknown   \n",
       "499981                        0.016822         unknown   \n",
       "499982                        0.000935         unknown   \n",
       "499983                        0.000935         unknown   \n",
       "499984                        0.244860          K07658   \n",
       "499985                        0.004673         unknown   \n",
       "499986                        0.008411         unknown   \n",
       "499987                        0.445794          K03666   \n",
       "499988                        0.002804          K03531   \n",
       "499989                        0.007477         unknown   \n",
       "499990                        0.001869          K07024   \n",
       "499991                        0.033645         unknown   \n",
       "499992                        0.017757         unknown   \n",
       "499993                        0.004673         unknown   \n",
       "499994                        0.003738         unknown   \n",
       "499995                        0.152336          K00571   \n",
       "499996                        0.000935         unknown   \n",
       "499997                        0.064486         unknown   \n",
       "499998                        0.002804         unknown   \n",
       "499999                        0.007477         unknown   \n",
       "\n",
       "              KEGG Functional Categories  Sample Occurence Frequency  \\\n",
       "0                                unknown                    0.008682   \n",
       "1                      Energy Metabolism                    0.187056   \n",
       "2                                unknown                    0.015785   \n",
       "3                                unknown                    0.008682   \n",
       "4                                unknown                    0.112865   \n",
       "5                                unknown                    0.008682   \n",
       "6                                unknown                    0.004736   \n",
       "7                                unknown                    0.003946   \n",
       "8                                unknown                    0.014996   \n",
       "9                                unknown                    0.000789   \n",
       "10                               unknown                    0.010260   \n",
       "11                  Poorly Characterized                    0.096290   \n",
       "12                               unknown                    0.335438   \n",
       "13                               unknown                    0.032360   \n",
       "14                               unknown                    0.048145   \n",
       "15                 Nucleotide Metabolism                    0.000789   \n",
       "16                               unknown                    0.086819   \n",
       "17                 Nucleotide Metabolism                    0.007103   \n",
       "18                               unknown                    0.005525   \n",
       "19                               unknown                    0.284136   \n",
       "20                               unknown                    0.003946   \n",
       "21                               unknown                    0.000789   \n",
       "22                            Metabolism                    0.046567   \n",
       "23                    Membrane Transport                    0.326756   \n",
       "24                               unknown                    0.085241   \n",
       "25                      Lipid Metabolism                    0.033938   \n",
       "26                Replication and Repair                    0.024467   \n",
       "27                               unknown                    0.019732   \n",
       "28                               unknown                    0.008682   \n",
       "29                               unknown                    0.003157   \n",
       "...                                  ...                         ...   \n",
       "499970                           unknown                    0.152328   \n",
       "499971                           unknown                    0.001579   \n",
       "499972                       Translation                    0.727703   \n",
       "499973                           unknown                    0.233623   \n",
       "499974                           unknown                    0.020521   \n",
       "499975                           unknown                    0.003157   \n",
       "499976                Membrane Transport                    0.361484   \n",
       "499977                           unknown                    0.783741   \n",
       "499978                           unknown                    0.014996   \n",
       "499979                           unknown                    0.015785   \n",
       "499980                           unknown                    0.002368   \n",
       "499981                           unknown                    0.015785   \n",
       "499982                           unknown                    0.000789   \n",
       "499983                           unknown                    0.000789   \n",
       "499984               Signal Transduction                    0.229676   \n",
       "499985                           unknown                    0.003946   \n",
       "499986                           unknown                    0.007103   \n",
       "499987  Folding, Sorting and Degradation                    0.440410   \n",
       "499988            Replication and Repair                    0.004736   \n",
       "499989                           unknown                    0.007893   \n",
       "499990              Poorly Characterized                    0.002368   \n",
       "499991                           unknown                    0.037096   \n",
       "499992                           unknown                    0.018942   \n",
       "499993                           unknown                    0.003946   \n",
       "499994                           unknown                    0.004736   \n",
       "499995    Genetic Information Processing                    0.153907   \n",
       "499996                           unknown                    0.001579   \n",
       "499997                           unknown                    0.054459   \n",
       "499998                           unknown                    0.003946   \n",
       "499999                           unknown                    0.008682   \n",
       "\n",
       "       Taxonomic Annotation(Genus Level) Taxonomic Annotation(Phylum Level)  \\\n",
       "0                                unknown                            unknown   \n",
       "1                                unknown                            unknown   \n",
       "2                                unknown                            unknown   \n",
       "3                                unknown                            unknown   \n",
       "4                          Coprobacillus                         Firmicutes   \n",
       "5                                unknown                            unknown   \n",
       "6                                unknown                            unknown   \n",
       "7                                unknown                            unknown   \n",
       "8                                unknown                            unknown   \n",
       "9                                unknown                            unknown   \n",
       "10                               unknown                      Bacteroidetes   \n",
       "11                               unknown                            unknown   \n",
       "12                               unknown                            unknown   \n",
       "13                               unknown                            unknown   \n",
       "14                               unknown                            unknown   \n",
       "15                               unknown                            unknown   \n",
       "16                               unknown                            unknown   \n",
       "17                               unknown                      Bacteroidetes   \n",
       "18                               unknown                            unknown   \n",
       "19                          Ruminococcus                         Firmicutes   \n",
       "20                               unknown                            unknown   \n",
       "21                               unknown                            unknown   \n",
       "22                               unknown                            unknown   \n",
       "23                               unknown                            unknown   \n",
       "24                               unknown                            unknown   \n",
       "25                               unknown                            unknown   \n",
       "26                               unknown                            unknown   \n",
       "27                               unknown                            unknown   \n",
       "28                               unknown                            unknown   \n",
       "29                               unknown                            unknown   \n",
       "...                                  ...                                ...   \n",
       "499970                           unknown                            unknown   \n",
       "499971                       Bacteroides                      Bacteroidetes   \n",
       "499972                           unknown                            unknown   \n",
       "499973                       Bacteroides                      Bacteroidetes   \n",
       "499974                           unknown                            unknown   \n",
       "499975                           unknown                            unknown   \n",
       "499976                       Eubacterium                         Firmicutes   \n",
       "499977                  Faecalibacterium                         Firmicutes   \n",
       "499978                           unknown                            unknown   \n",
       "499979                           unknown                            unknown   \n",
       "499980                           unknown                            unknown   \n",
       "499981                       Bacteroides                      Bacteroidetes   \n",
       "499982                           unknown                            unknown   \n",
       "499983                           unknown                            unknown   \n",
       "499984                           unknown                         Firmicutes   \n",
       "499985                           unknown                            unknown   \n",
       "499986                           unknown                            unknown   \n",
       "499987                           unknown                            unknown   \n",
       "499988                           unknown                            unknown   \n",
       "499989                           unknown                            unknown   \n",
       "499990                           unknown                            unknown   \n",
       "499991                           unknown                            unknown   \n",
       "499992                  Faecalibacterium                         Firmicutes   \n",
       "499993                           unknown                            unknown   \n",
       "499994                           unknown                            unknown   \n",
       "499995                           unknown                            unknown   \n",
       "499996                           unknown                            unknown   \n",
       "499997                           unknown                            unknown   \n",
       "499998                           unknown                            unknown   \n",
       "499999                           unknown                            unknown   \n",
       "\n",
       "       eggNOG Annotation                       eggNOG Functional Categories  \n",
       "0                COG4932             Cell wall/membrane/envelope biogenesis  \n",
       "1                COG3411                   Energy production and conversion  \n",
       "2                unknown                                            unknown  \n",
       "3              NOG125034                                   Function unknown  \n",
       "4                unknown                                            unknown  \n",
       "5                unknown                                            unknown  \n",
       "6               NOG12793                                   Function unknown  \n",
       "7                unknown                                            unknown  \n",
       "8                unknown                                            unknown  \n",
       "9                unknown                                            unknown  \n",
       "10               COG1073                   General function prediction only  \n",
       "11               COG0622                   General function prediction only  \n",
       "12               unknown                                            unknown  \n",
       "13              NOG77285                                   Function unknown  \n",
       "14             NOG291728                                            unknown  \n",
       "15               COG0756                Nucleotide transport and metabolism  \n",
       "16               unknown                                            unknown  \n",
       "17               COG2131                Nucleotide transport and metabolism  \n",
       "18             NOG256796                                   Function unknown  \n",
       "19               unknown                                            unknown  \n",
       "20               unknown                                            unknown  \n",
       "21               unknown                                            unknown  \n",
       "22               COG1489                   General function prediction only  \n",
       "23               COG1116             Inorganic ion transport and metabolism  \n",
       "24             NOG277117                                            unknown  \n",
       "25               COG0318  Lipid transport and metabolism;Secondary metab...  \n",
       "26               COG0272              Replication, recombination and repair  \n",
       "27              NOG69716                                   Function unknown  \n",
       "28              NOG40611                                   Function unknown  \n",
       "29               unknown                                            unknown  \n",
       "...                  ...                                                ...  \n",
       "499970           unknown                                            unknown  \n",
       "499971           unknown                                            unknown  \n",
       "499972   COG3688;COG0480  Translation, ribosomal structure and biogenesi...  \n",
       "499973         NOG275003                                            unknown  \n",
       "499974           unknown                                            unknown  \n",
       "499975           unknown                                            unknown  \n",
       "499976           COG1682  Carbohydrate transport and metabolism;Cell wal...  \n",
       "499977           unknown                                            unknown  \n",
       "499978           unknown                                            unknown  \n",
       "499979           unknown                                            unknown  \n",
       "499980           unknown                                            unknown  \n",
       "499981           unknown                                            unknown  \n",
       "499982           unknown                                            unknown  \n",
       "499983           unknown                                            unknown  \n",
       "499984           COG0745       Transcription;Signal transduction mechanisms  \n",
       "499985           unknown                                            unknown  \n",
       "499986           unknown                                            unknown  \n",
       "499987           COG1923                   General function prediction only  \n",
       "499988           COG0206  Cell cycle control, cell division, chromosome ...  \n",
       "499989           unknown                                            unknown  \n",
       "499990           COG0561                   General function prediction only  \n",
       "499991           unknown                                            unknown  \n",
       "499992           unknown                                            unknown  \n",
       "499993           unknown                                            unknown  \n",
       "499994          NOG47185                                   Function unknown  \n",
       "499995           COG1002                                 Defense mechanisms  \n",
       "499996           unknown                                            unknown  \n",
       "499997           unknown                                            unknown  \n",
       "499998           unknown                                            unknown  \n",
       "499999           unknown                                            unknown  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ordenar por columnas(axis 1), Ascendente\n",
    "df.sort_index(axis=1, ascending=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>499999</th>\n",
       "      <td>5457503</td>\n",
       "      <td>V1.UC36-0_GL0134515</td>\n",
       "      <td>525</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499998</th>\n",
       "      <td>7655040</td>\n",
       "      <td>764042746-stool2_revised_scaffold22088_1_gene3...</td>\n",
       "      <td>288</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499997</th>\n",
       "      <td>4924959</td>\n",
       "      <td>MH0161_GL0035075</td>\n",
       "      <td>579</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.054459</td>\n",
       "      <td>0.064486</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499996</th>\n",
       "      <td>691723</td>\n",
       "      <td>638754422-stool2_revised_scaffold23317_1_gene1940</td>\n",
       "      <td>1698</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499995</th>\n",
       "      <td>278939</td>\n",
       "      <td>765560005-stool1_revised_scaffold5052_1_gene73628</td>\n",
       "      <td>2349</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00571</td>\n",
       "      <td>COG1002</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>0.152336</td>\n",
       "      <td>Genetic Information Processing</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499994</th>\n",
       "      <td>2951767</td>\n",
       "      <td>V1.UC26-4_GL0007902</td>\n",
       "      <td>879</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG47185</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499993</th>\n",
       "      <td>7363211</td>\n",
       "      <td>MH0260_GL0023264</td>\n",
       "      <td>315</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499992</th>\n",
       "      <td>8040629</td>\n",
       "      <td>159611913-stool1_revised_C786324_1_gene54480</td>\n",
       "      <td>252</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.018942</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499991</th>\n",
       "      <td>9631397</td>\n",
       "      <td>MH0280_GL0114474</td>\n",
       "      <td>120</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.037096</td>\n",
       "      <td>0.033645</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499990</th>\n",
       "      <td>6294127</td>\n",
       "      <td>V1.UC26-4_GL0011854</td>\n",
       "      <td>432</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07024</td>\n",
       "      <td>COG0561</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499989</th>\n",
       "      <td>6961318</td>\n",
       "      <td>MH0333_GL0111005</td>\n",
       "      <td>357</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007893</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499988</th>\n",
       "      <td>5254719</td>\n",
       "      <td>159753524-stool2_revised_C538848_1_gene38208</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03531</td>\n",
       "      <td>COG0206</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499987</th>\n",
       "      <td>6439309</td>\n",
       "      <td>MH0158_GL0048733</td>\n",
       "      <td>414</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K03666</td>\n",
       "      <td>COG1923</td>\n",
       "      <td>0.440410</td>\n",
       "      <td>0.445794</td>\n",
       "      <td>Folding, Sorting and Degradation</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499986</th>\n",
       "      <td>6567495</td>\n",
       "      <td>MH0065_GL0023744</td>\n",
       "      <td>399</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499985</th>\n",
       "      <td>9067856</td>\n",
       "      <td>160400887-stool1_revised_C1382776_1_gene10717</td>\n",
       "      <td>168</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499984</th>\n",
       "      <td>4147487</td>\n",
       "      <td>MH0062_GL0030772</td>\n",
       "      <td>678</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07658</td>\n",
       "      <td>COG0745</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>0.244860</td>\n",
       "      <td>Signal Transduction</td>\n",
       "      <td>Transcription;Signal transduction mechanisms</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499983</th>\n",
       "      <td>556756</td>\n",
       "      <td>SZEY-101A_GL0065406</td>\n",
       "      <td>1839</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499982</th>\n",
       "      <td>7786169</td>\n",
       "      <td>MH0157_GL0106576</td>\n",
       "      <td>273</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499981</th>\n",
       "      <td>8734258</td>\n",
       "      <td>470145.BACCOP_04248</td>\n",
       "      <td>195</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.016822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499980</th>\n",
       "      <td>7451388</td>\n",
       "      <td>MH0222_GL0182909</td>\n",
       "      <td>306</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499979</th>\n",
       "      <td>7115654</td>\n",
       "      <td>DOF004_GL0014088</td>\n",
       "      <td>342</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.017757</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499978</th>\n",
       "      <td>9145946</td>\n",
       "      <td>MH0117_GL0078602</td>\n",
       "      <td>159</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499977</th>\n",
       "      <td>8177332</td>\n",
       "      <td>411485.FAEPRAM212_00318</td>\n",
       "      <td>240</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.783741</td>\n",
       "      <td>0.778505</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499976</th>\n",
       "      <td>3280488</td>\n",
       "      <td>MH0028_GL0020655</td>\n",
       "      <td>816</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>K09690</td>\n",
       "      <td>COG1682</td>\n",
       "      <td>0.361484</td>\n",
       "      <td>0.362617</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Carbohydrate transport and metabolism;Cell wal...</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499975</th>\n",
       "      <td>3484802</td>\n",
       "      <td>MH0188_GL0112429</td>\n",
       "      <td>780</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499974</th>\n",
       "      <td>5233095</td>\n",
       "      <td>MH0204_GL0162383</td>\n",
       "      <td>546</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499973</th>\n",
       "      <td>4517482</td>\n",
       "      <td>DOF003_GL0027603</td>\n",
       "      <td>630</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG275003</td>\n",
       "      <td>0.233623</td>\n",
       "      <td>0.229907</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499972</th>\n",
       "      <td>186663</td>\n",
       "      <td>V1.FI20_GL0146385</td>\n",
       "      <td>2661</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02355</td>\n",
       "      <td>COG3688;COG0480</td>\n",
       "      <td>0.727703</td>\n",
       "      <td>0.721495</td>\n",
       "      <td>Translation</td>\n",
       "      <td>Translation, ribosomal structure and biogenesi...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499971</th>\n",
       "      <td>6992342</td>\n",
       "      <td>MH0392_GL0212082</td>\n",
       "      <td>354</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499970</th>\n",
       "      <td>4238583</td>\n",
       "      <td>MH0284_GL0171112</td>\n",
       "      <td>666</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.152328</td>\n",
       "      <td>0.147664</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5500609</td>\n",
       "      <td>MH0174_GL0034307</td>\n",
       "      <td>519</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2811850</td>\n",
       "      <td>T2D-63A_GL0089788</td>\n",
       "      <td>906</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG40611</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3320656</td>\n",
       "      <td>O2.UC24-1_GL0155079</td>\n",
       "      <td>810</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG69716</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1293228</td>\n",
       "      <td>V1.UC50-1_GL0030242</td>\n",
       "      <td>1329</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01972</td>\n",
       "      <td>COG0272</td>\n",
       "      <td>0.024467</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>Replication and Repair</td>\n",
       "      <td>Replication, recombination and repair</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>7219798</td>\n",
       "      <td>MH0355_GL0176135</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01897</td>\n",
       "      <td>COG0318</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.037383</td>\n",
       "      <td>Lipid Metabolism</td>\n",
       "      <td>Lipid transport and metabolism;Secondary metab...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2312660</td>\n",
       "      <td>158256496-stool1_revised_C920453_1_gene161597</td>\n",
       "      <td>1014</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG277117</td>\n",
       "      <td>0.085241</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3650803</td>\n",
       "      <td>MH0012_GL0124464</td>\n",
       "      <td>753</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K02049</td>\n",
       "      <td>COG1116</td>\n",
       "      <td>0.326756</td>\n",
       "      <td>0.332710</td>\n",
       "      <td>Membrane Transport</td>\n",
       "      <td>Inorganic ion transport and metabolism</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4137075</td>\n",
       "      <td>O2.UC48-0_GL0276256</td>\n",
       "      <td>681</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K06206</td>\n",
       "      <td>COG1489</td>\n",
       "      <td>0.046567</td>\n",
       "      <td>0.034579</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9031743</td>\n",
       "      <td>160765029-stool1_revised_C1108515_1_gene40264</td>\n",
       "      <td>171</td>\n",
       "      <td>Complete</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9139876</td>\n",
       "      <td>159753524-stool2_revised_C530648_1_gene94409</td>\n",
       "      <td>162</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6068154</td>\n",
       "      <td>MH0040_GL0048712</td>\n",
       "      <td>456</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Ruminococcus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.284136</td>\n",
       "      <td>0.285047</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>8094620</td>\n",
       "      <td>O2.UC47-2_GL0030825</td>\n",
       "      <td>246</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG256796</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6226899</td>\n",
       "      <td>MH0192_GL0144006</td>\n",
       "      <td>438</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01493</td>\n",
       "      <td>COG2131</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5203442</td>\n",
       "      <td>MH0188_GL0079716</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.086819</td>\n",
       "      <td>0.090654</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5870802</td>\n",
       "      <td>MH0096_GL0115416</td>\n",
       "      <td>480</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K01520</td>\n",
       "      <td>COG0756</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>Nucleotide Metabolism</td>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>8222582</td>\n",
       "      <td>MH0309_GL0024067</td>\n",
       "      <td>234</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG291728</td>\n",
       "      <td>0.048145</td>\n",
       "      <td>0.051402</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5831026</td>\n",
       "      <td>MH0419_GL0063398</td>\n",
       "      <td>486</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG77285</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.030841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9553442</td>\n",
       "      <td>MH0149_GL0074986</td>\n",
       "      <td>126</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.335438</td>\n",
       "      <td>0.330841</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5707244</td>\n",
       "      <td>V1.FI15_GL0072437</td>\n",
       "      <td>501</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07095</td>\n",
       "      <td>COG0622</td>\n",
       "      <td>0.096290</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>Poorly Characterized</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1821827</td>\n",
       "      <td>MH0333_GL0003962</td>\n",
       "      <td>1146</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG1073</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>0.011215</td>\n",
       "      <td>unknown</td>\n",
       "      <td>General function prediction only</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8920818</td>\n",
       "      <td>764588959-stool1_revised_C775588_1_gene178391</td>\n",
       "      <td>180</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7928769</td>\n",
       "      <td>O2.UC48-1_GL0076618</td>\n",
       "      <td>261</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014996</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>647891</td>\n",
       "      <td>MH0433_GL0005093</td>\n",
       "      <td>1740</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>693534</td>\n",
       "      <td>MH0448_GL0045910</td>\n",
       "      <td>1695</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4807882</td>\n",
       "      <td>V1.FI14_GL0101784</td>\n",
       "      <td>594</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2441669</td>\n",
       "      <td>469596.HMPREF9488_01493</td>\n",
       "      <td>984</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Coprobacillus</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.112865</td>\n",
       "      <td>0.113084</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8950791</td>\n",
       "      <td>DOM015_GL0050638</td>\n",
       "      <td>177</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG125034</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7221353</td>\n",
       "      <td>MH0389_GL0170585</td>\n",
       "      <td>330</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6811315</td>\n",
       "      <td>MH0012_GL0174453</td>\n",
       "      <td>372</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00335</td>\n",
       "      <td>COG3411</td>\n",
       "      <td>0.187056</td>\n",
       "      <td>0.195327</td>\n",
       "      <td>Energy Metabolism</td>\n",
       "      <td>Energy production and conversion</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5209933</td>\n",
       "      <td>MH0396_GL0114156</td>\n",
       "      <td>549</td>\n",
       "      <td>Lack both ends</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4932</td>\n",
       "      <td>0.008682</td>\n",
       "      <td>0.009346</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "499999  5457503                                V1.UC36-0_GL0134515   \n",
       "499998  7655040  764042746-stool2_revised_scaffold22088_1_gene3...   \n",
       "499997  4924959                                   MH0161_GL0035075   \n",
       "499996   691723  638754422-stool2_revised_scaffold23317_1_gene1940   \n",
       "499995   278939  765560005-stool1_revised_scaffold5052_1_gene73628   \n",
       "499994  2951767                                V1.UC26-4_GL0007902   \n",
       "499993  7363211                                   MH0260_GL0023264   \n",
       "499992  8040629       159611913-stool1_revised_C786324_1_gene54480   \n",
       "499991  9631397                                   MH0280_GL0114474   \n",
       "499990  6294127                                V1.UC26-4_GL0011854   \n",
       "499989  6961318                                   MH0333_GL0111005   \n",
       "499988  5254719       159753524-stool2_revised_C538848_1_gene38208   \n",
       "499987  6439309                                   MH0158_GL0048733   \n",
       "499986  6567495                                   MH0065_GL0023744   \n",
       "499985  9067856      160400887-stool1_revised_C1382776_1_gene10717   \n",
       "499984  4147487                                   MH0062_GL0030772   \n",
       "499983   556756                                SZEY-101A_GL0065406   \n",
       "499982  7786169                                   MH0157_GL0106576   \n",
       "499981  8734258                                470145.BACCOP_04248   \n",
       "499980  7451388                                   MH0222_GL0182909   \n",
       "499979  7115654                                   DOF004_GL0014088   \n",
       "499978  9145946                                   MH0117_GL0078602   \n",
       "499977  8177332                            411485.FAEPRAM212_00318   \n",
       "499976  3280488                                   MH0028_GL0020655   \n",
       "499975  3484802                                   MH0188_GL0112429   \n",
       "499974  5233095                                   MH0204_GL0162383   \n",
       "499973  4517482                                   DOF003_GL0027603   \n",
       "499972   186663                                  V1.FI20_GL0146385   \n",
       "499971  6992342                                   MH0392_GL0212082   \n",
       "499970  4238583                                   MH0284_GL0171112   \n",
       "...         ...                                                ...   \n",
       "29      5500609                                   MH0174_GL0034307   \n",
       "28      2811850                                  T2D-63A_GL0089788   \n",
       "27      3320656                                O2.UC24-1_GL0155079   \n",
       "26      1293228                                V1.UC50-1_GL0030242   \n",
       "25      7219798                                   MH0355_GL0176135   \n",
       "24      2312660      158256496-stool1_revised_C920453_1_gene161597   \n",
       "23      3650803                                   MH0012_GL0124464   \n",
       "22      4137075                                O2.UC48-0_GL0276256   \n",
       "21      9031743      160765029-stool1_revised_C1108515_1_gene40264   \n",
       "20      9139876       159753524-stool2_revised_C530648_1_gene94409   \n",
       "19      6068154                                   MH0040_GL0048712   \n",
       "18      8094620                                O2.UC47-2_GL0030825   \n",
       "17      6226899                                   MH0192_GL0144006   \n",
       "16      5203442                                   MH0188_GL0079716   \n",
       "15      5870802                                   MH0096_GL0115416   \n",
       "14      8222582                                   MH0309_GL0024067   \n",
       "13      5831026                                   MH0419_GL0063398   \n",
       "12      9553442                                   MH0149_GL0074986   \n",
       "11      5707244                                  V1.FI15_GL0072437   \n",
       "10      1821827                                   MH0333_GL0003962   \n",
       "9       8920818      764588959-stool1_revised_C775588_1_gene178391   \n",
       "8       7928769                                O2.UC48-1_GL0076618   \n",
       "7        647891                                   MH0433_GL0005093   \n",
       "6        693534                                   MH0448_GL0045910   \n",
       "5       4807882                                  V1.FI14_GL0101784   \n",
       "4       2441669                            469596.HMPREF9488_01493   \n",
       "3       8950791                                   DOM015_GL0050638   \n",
       "2       7221353                                   MH0389_GL0170585   \n",
       "1       6811315                                   MH0012_GL0174453   \n",
       "0       5209933                                   MH0396_GL0114156   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "499999          525       Lack 5'-end           EUR   \n",
       "499998          288          Complete           USA   \n",
       "499997          579          Complete           EUR   \n",
       "499996         1698          Complete           USA   \n",
       "499995         2349       Lack 5'-end           USA   \n",
       "499994          879          Complete           EUR   \n",
       "499993          315       Lack 3'-end           EUR   \n",
       "499992          252          Complete           USA   \n",
       "499991          120       Lack 5'-end           EUR   \n",
       "499990          432       Lack 5'-end           EUR   \n",
       "499989          357       Lack 3'-end           EUR   \n",
       "499988          546    Lack both ends           USA   \n",
       "499987          414       Lack 5'-end           EUR   \n",
       "499986          399       Lack 5'-end           EUR   \n",
       "499985          168          Complete           USA   \n",
       "499984          678          Complete           EUR   \n",
       "499983         1839          Complete           CHN   \n",
       "499982          273          Complete           EUR   \n",
       "499981          195          Complete            SP   \n",
       "499980          306          Complete           EUR   \n",
       "499979          342       Lack 5'-end           CHN   \n",
       "499978          159          Complete           EUR   \n",
       "499977          240          Complete            SP   \n",
       "499976          816          Complete           EUR   \n",
       "499975          780          Complete           EUR   \n",
       "499974          546    Lack both ends           EUR   \n",
       "499973          630          Complete           CHN   \n",
       "499972         2661          Complete           EUR   \n",
       "499971          354       Lack 5'-end           EUR   \n",
       "499970          666          Complete           EUR   \n",
       "...             ...               ...           ...   \n",
       "29              519    Lack both ends           EUR   \n",
       "28              906          Complete           CHN   \n",
       "27              810    Lack both ends           EUR   \n",
       "26             1329       Lack 5'-end           EUR   \n",
       "25              330       Lack 3'-end           EUR   \n",
       "24             1014    Lack both ends           USA   \n",
       "23              753          Complete           EUR   \n",
       "22              681          Complete           EUR   \n",
       "21              171          Complete           USA   \n",
       "20              162       Lack 5'-end           USA   \n",
       "19              456       Lack 5'-end           EUR   \n",
       "18              246       Lack 3'-end           EUR   \n",
       "17              438          Complete           EUR   \n",
       "16              549    Lack both ends           EUR   \n",
       "15              480          Complete           EUR   \n",
       "14              234       Lack 3'-end           EUR   \n",
       "13              486          Complete           EUR   \n",
       "12              126       Lack 5'-end           EUR   \n",
       "11              501          Complete           EUR   \n",
       "10             1146          Complete           EUR   \n",
       "9               180       Lack 3'-end           USA   \n",
       "8               261       Lack 3'-end           EUR   \n",
       "7              1740          Complete           EUR   \n",
       "6              1695    Lack both ends           EUR   \n",
       "5               594          Complete           EUR   \n",
       "4               984          Complete            SP   \n",
       "3               177       Lack 5'-end           CHN   \n",
       "2               330       Lack 5'-end           EUR   \n",
       "1               372          Complete           EUR   \n",
       "0               549    Lack both ends           EUR   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "499999                            unknown                           unknown   \n",
       "499998                            unknown                           unknown   \n",
       "499997                            unknown                           unknown   \n",
       "499996                            unknown                           unknown   \n",
       "499995                            unknown                           unknown   \n",
       "499994                            unknown                           unknown   \n",
       "499993                            unknown                           unknown   \n",
       "499992                         Firmicutes                  Faecalibacterium   \n",
       "499991                            unknown                           unknown   \n",
       "499990                            unknown                           unknown   \n",
       "499989                            unknown                           unknown   \n",
       "499988                            unknown                           unknown   \n",
       "499987                            unknown                           unknown   \n",
       "499986                            unknown                           unknown   \n",
       "499985                            unknown                           unknown   \n",
       "499984                         Firmicutes                           unknown   \n",
       "499983                            unknown                           unknown   \n",
       "499982                            unknown                           unknown   \n",
       "499981                      Bacteroidetes                       Bacteroides   \n",
       "499980                            unknown                           unknown   \n",
       "499979                            unknown                           unknown   \n",
       "499978                            unknown                           unknown   \n",
       "499977                         Firmicutes                  Faecalibacterium   \n",
       "499976                         Firmicutes                       Eubacterium   \n",
       "499975                            unknown                           unknown   \n",
       "499974                            unknown                           unknown   \n",
       "499973                      Bacteroidetes                       Bacteroides   \n",
       "499972                            unknown                           unknown   \n",
       "499971                      Bacteroidetes                       Bacteroides   \n",
       "499970                            unknown                           unknown   \n",
       "...                                   ...                               ...   \n",
       "29                                unknown                           unknown   \n",
       "28                                unknown                           unknown   \n",
       "27                                unknown                           unknown   \n",
       "26                                unknown                           unknown   \n",
       "25                                unknown                           unknown   \n",
       "24                                unknown                           unknown   \n",
       "23                                unknown                           unknown   \n",
       "22                                unknown                           unknown   \n",
       "21                                unknown                           unknown   \n",
       "20                                unknown                           unknown   \n",
       "19                             Firmicutes                      Ruminococcus   \n",
       "18                                unknown                           unknown   \n",
       "17                          Bacteroidetes                           unknown   \n",
       "16                                unknown                           unknown   \n",
       "15                                unknown                           unknown   \n",
       "14                                unknown                           unknown   \n",
       "13                                unknown                           unknown   \n",
       "12                                unknown                           unknown   \n",
       "11                                unknown                           unknown   \n",
       "10                          Bacteroidetes                           unknown   \n",
       "9                                 unknown                           unknown   \n",
       "8                                 unknown                           unknown   \n",
       "7                                 unknown                           unknown   \n",
       "6                                 unknown                           unknown   \n",
       "5                                 unknown                           unknown   \n",
       "4                              Firmicutes                     Coprobacillus   \n",
       "3                                 unknown                           unknown   \n",
       "2                                 unknown                           unknown   \n",
       "1                                 unknown                           unknown   \n",
       "0                                 unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "499999         unknown           unknown                    0.008682   \n",
       "499998         unknown           unknown                    0.003946   \n",
       "499997         unknown           unknown                    0.054459   \n",
       "499996         unknown           unknown                    0.001579   \n",
       "499995          K00571           COG1002                    0.153907   \n",
       "499994         unknown          NOG47185                    0.004736   \n",
       "499993         unknown           unknown                    0.003946   \n",
       "499992         unknown           unknown                    0.018942   \n",
       "499991         unknown           unknown                    0.037096   \n",
       "499990          K07024           COG0561                    0.002368   \n",
       "499989         unknown           unknown                    0.007893   \n",
       "499988          K03531           COG0206                    0.004736   \n",
       "499987          K03666           COG1923                    0.440410   \n",
       "499986         unknown           unknown                    0.007103   \n",
       "499985         unknown           unknown                    0.003946   \n",
       "499984          K07658           COG0745                    0.229676   \n",
       "499983         unknown           unknown                    0.000789   \n",
       "499982         unknown           unknown                    0.000789   \n",
       "499981         unknown           unknown                    0.015785   \n",
       "499980         unknown           unknown                    0.002368   \n",
       "499979         unknown           unknown                    0.015785   \n",
       "499978         unknown           unknown                    0.014996   \n",
       "499977         unknown           unknown                    0.783741   \n",
       "499976          K09690           COG1682                    0.361484   \n",
       "499975         unknown           unknown                    0.003157   \n",
       "499974         unknown           unknown                    0.020521   \n",
       "499973         unknown         NOG275003                    0.233623   \n",
       "499972          K02355   COG3688;COG0480                    0.727703   \n",
       "499971         unknown           unknown                    0.001579   \n",
       "499970         unknown           unknown                    0.152328   \n",
       "...                ...               ...                         ...   \n",
       "29             unknown           unknown                    0.003157   \n",
       "28             unknown          NOG40611                    0.008682   \n",
       "27             unknown          NOG69716                    0.019732   \n",
       "26              K01972           COG0272                    0.024467   \n",
       "25              K01897           COG0318                    0.033938   \n",
       "24             unknown         NOG277117                    0.085241   \n",
       "23              K02049           COG1116                    0.326756   \n",
       "22              K06206           COG1489                    0.046567   \n",
       "21             unknown           unknown                    0.000789   \n",
       "20             unknown           unknown                    0.003946   \n",
       "19             unknown           unknown                    0.284136   \n",
       "18             unknown         NOG256796                    0.005525   \n",
       "17              K01493           COG2131                    0.007103   \n",
       "16             unknown           unknown                    0.086819   \n",
       "15              K01520           COG0756                    0.000789   \n",
       "14             unknown         NOG291728                    0.048145   \n",
       "13             unknown          NOG77285                    0.032360   \n",
       "12             unknown           unknown                    0.335438   \n",
       "11              K07095           COG0622                    0.096290   \n",
       "10             unknown           COG1073                    0.010260   \n",
       "9              unknown           unknown                    0.000789   \n",
       "8              unknown           unknown                    0.014996   \n",
       "7              unknown           unknown                    0.003946   \n",
       "6              unknown          NOG12793                    0.004736   \n",
       "5              unknown           unknown                    0.008682   \n",
       "4              unknown           unknown                    0.112865   \n",
       "3              unknown         NOG125034                    0.008682   \n",
       "2              unknown           unknown                    0.015785   \n",
       "1               K00335           COG3411                    0.187056   \n",
       "0              unknown           COG4932                    0.008682   \n",
       "\n",
       "        Individual Occurence Frequency        KEGG Functional Categories  \\\n",
       "499999                        0.007477                           unknown   \n",
       "499998                        0.002804                           unknown   \n",
       "499997                        0.064486                           unknown   \n",
       "499996                        0.000935                           unknown   \n",
       "499995                        0.152336    Genetic Information Processing   \n",
       "499994                        0.003738                           unknown   \n",
       "499993                        0.004673                           unknown   \n",
       "499992                        0.017757                           unknown   \n",
       "499991                        0.033645                           unknown   \n",
       "499990                        0.001869              Poorly Characterized   \n",
       "499989                        0.007477                           unknown   \n",
       "499988                        0.002804            Replication and Repair   \n",
       "499987                        0.445794  Folding, Sorting and Degradation   \n",
       "499986                        0.008411                           unknown   \n",
       "499985                        0.004673                           unknown   \n",
       "499984                        0.244860               Signal Transduction   \n",
       "499983                        0.000935                           unknown   \n",
       "499982                        0.000935                           unknown   \n",
       "499981                        0.016822                           unknown   \n",
       "499980                        0.002804                           unknown   \n",
       "499979                        0.017757                           unknown   \n",
       "499978                        0.015888                           unknown   \n",
       "499977                        0.778505                           unknown   \n",
       "499976                        0.362617                Membrane Transport   \n",
       "499975                        0.003738                           unknown   \n",
       "499974                        0.021495                           unknown   \n",
       "499973                        0.229907                           unknown   \n",
       "499972                        0.721495                       Translation   \n",
       "499971                        0.000935                           unknown   \n",
       "499970                        0.147664                           unknown   \n",
       "...                                ...                               ...   \n",
       "29                            0.003738                           unknown   \n",
       "28                            0.009346                           unknown   \n",
       "27                            0.018692                           unknown   \n",
       "26                            0.021495            Replication and Repair   \n",
       "25                            0.037383                  Lipid Metabolism   \n",
       "24                            0.079439                           unknown   \n",
       "23                            0.332710                Membrane Transport   \n",
       "22                            0.034579                        Metabolism   \n",
       "21                            0.000935                           unknown   \n",
       "20                            0.002804                           unknown   \n",
       "19                            0.285047                           unknown   \n",
       "18                            0.004673                           unknown   \n",
       "17                            0.004673             Nucleotide Metabolism   \n",
       "16                            0.090654                           unknown   \n",
       "15                            0.000935             Nucleotide Metabolism   \n",
       "14                            0.051402                           unknown   \n",
       "13                            0.030841                           unknown   \n",
       "12                            0.330841                           unknown   \n",
       "11                            0.100000              Poorly Characterized   \n",
       "10                            0.011215                           unknown   \n",
       "9                             0.000935                           unknown   \n",
       "8                             0.008411                           unknown   \n",
       "7                             0.004673                           unknown   \n",
       "6                             0.004673                           unknown   \n",
       "5                             0.009346                           unknown   \n",
       "4                             0.113084                           unknown   \n",
       "3                             0.008411                           unknown   \n",
       "2                             0.018692                           unknown   \n",
       "1                             0.195327                 Energy Metabolism   \n",
       "0                             0.009346                           unknown   \n",
       "\n",
       "                             eggNOG Functional Categories Cohort Assembled  \n",
       "499999                                            unknown              EUR  \n",
       "499998                                            unknown              USA  \n",
       "499997                                            unknown          EUR;CHN  \n",
       "499996                                            unknown              USA  \n",
       "499995                                 Defense mechanisms          EUR;USA  \n",
       "499994                                   Function unknown              EUR  \n",
       "499993                                            unknown              EUR  \n",
       "499992                                            unknown              USA  \n",
       "499991                                            unknown              EUR  \n",
       "499990                   General function prediction only              EUR  \n",
       "499989                                            unknown              EUR  \n",
       "499988  Cell cycle control, cell division, chromosome ...              USA  \n",
       "499987                   General function prediction only      EUR;CHN;USA  \n",
       "499986                                            unknown              EUR  \n",
       "499985                                            unknown              USA  \n",
       "499984       Transcription;Signal transduction mechanisms      EUR;CHN;USA  \n",
       "499983                                            unknown              CHN  \n",
       "499982                                            unknown              EUR  \n",
       "499981                                            unknown              NaN  \n",
       "499980                                            unknown              EUR  \n",
       "499979                                            unknown              CHN  \n",
       "499978                                            unknown              EUR  \n",
       "499977                                            unknown              EUR  \n",
       "499976  Carbohydrate transport and metabolism;Cell wal...      EUR;CHN;USA  \n",
       "499975                                            unknown              EUR  \n",
       "499974                                            unknown              EUR  \n",
       "499973                                            unknown              CHN  \n",
       "499972  Translation, ribosomal structure and biogenesi...              EUR  \n",
       "499971                                            unknown              EUR  \n",
       "499970                                            unknown      EUR;CHN;USA  \n",
       "...                                                   ...              ...  \n",
       "29                                                unknown              EUR  \n",
       "28                                       Function unknown              CHN  \n",
       "27                                       Function unknown              EUR  \n",
       "26                  Replication, recombination and repair              EUR  \n",
       "25      Lipid transport and metabolism;Secondary metab...              EUR  \n",
       "24                                                unknown              USA  \n",
       "23                 Inorganic ion transport and metabolism      EUR;CHN;USA  \n",
       "22                       General function prediction only              EUR  \n",
       "21                                                unknown              USA  \n",
       "20                                                unknown              USA  \n",
       "19                                                unknown              EUR  \n",
       "18                                       Function unknown              EUR  \n",
       "17                    Nucleotide transport and metabolism              EUR  \n",
       "16                                                unknown              EUR  \n",
       "15                    Nucleotide transport and metabolism              EUR  \n",
       "14                                                unknown              EUR  \n",
       "13                                       Function unknown              EUR  \n",
       "12                                                unknown              EUR  \n",
       "11                       General function prediction only              EUR  \n",
       "10                       General function prediction only          EUR;CHN  \n",
       "9                                                 unknown              USA  \n",
       "8                                                 unknown              EUR  \n",
       "7                                                 unknown          EUR;USA  \n",
       "6                                        Function unknown              EUR  \n",
       "5                                                 unknown              EUR  \n",
       "4                                                 unknown      EUR;CHN;USA  \n",
       "3                                        Function unknown              CHN  \n",
       "2                                                 unknown              EUR  \n",
       "1                        Energy production and conversion      EUR;CHN;USA  \n",
       "0                  Cell wall/membrane/envelope biogenesis              EUR  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ordenar por columnas(axis 1), descendente\n",
    "df.sort_index(axis=0, ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88836</th>\n",
       "      <td>9865705</td>\n",
       "      <td>V1.CD6-0-PT_GL0096973</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476498</th>\n",
       "      <td>9862460</td>\n",
       "      <td>O2.UC4-2_GL0053162</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.021310</td>\n",
       "      <td>0.020561</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151345</th>\n",
       "      <td>9862491</td>\n",
       "      <td>O2.UC43-0_GL0033611</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.014953</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34192</th>\n",
       "      <td>9877190</td>\n",
       "      <td>159551223-stool2_revised_C916401_1_gene84247</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014207</td>\n",
       "      <td>0.012150</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331083</th>\n",
       "      <td>9860951</td>\n",
       "      <td>O2.UC21-1_GL0136539</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.059195</td>\n",
       "      <td>0.061682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280181</th>\n",
       "      <td>9857596</td>\n",
       "      <td>MH0415_GL0056639</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483179</th>\n",
       "      <td>9878379</td>\n",
       "      <td>763577454-stool2_revised_scaffold23171_1_gene1...</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.169692</td>\n",
       "      <td>0.157944</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370156</th>\n",
       "      <td>9853996</td>\n",
       "      <td>MH0335_GL0009364</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409047</th>\n",
       "      <td>9860917</td>\n",
       "      <td>O2.UC2-0_GL0121235</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34135</th>\n",
       "      <td>9872403</td>\n",
       "      <td>N003A_GL0007810</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408969</th>\n",
       "      <td>9869351</td>\n",
       "      <td>V1.UC52-0_GL0023463</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81900</th>\n",
       "      <td>9855675</td>\n",
       "      <td>MH0373_GL0182676</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.022889</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15105</th>\n",
       "      <td>9848328</td>\n",
       "      <td>MH0143_GL0088163</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53716</th>\n",
       "      <td>9877737</td>\n",
       "      <td>160502038-stool1_revised_C976100_1_gene185746</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4930</th>\n",
       "      <td>9873886</td>\n",
       "      <td>T2D-113A_GL0111098</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.115233</td>\n",
       "      <td>0.116822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225835</th>\n",
       "      <td>9847115</td>\n",
       "      <td>MH0100_GL0041020</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.047356</td>\n",
       "      <td>0.043925</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409075</th>\n",
       "      <td>9872514</td>\n",
       "      <td>N017A_GL0080793</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.072612</td>\n",
       "      <td>0.065421</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151252</th>\n",
       "      <td>9853377</td>\n",
       "      <td>MH0318_GL0022747</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53803</th>\n",
       "      <td>9856149</td>\n",
       "      <td>MH0384_GL0070245</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151236</th>\n",
       "      <td>9855061</td>\n",
       "      <td>MH0362_GL0041997</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.089187</td>\n",
       "      <td>0.091589</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197036</th>\n",
       "      <td>9846712</td>\n",
       "      <td>MH0086_GL0082324</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.140489</td>\n",
       "      <td>0.115888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34125</th>\n",
       "      <td>9858623</td>\n",
       "      <td>MH0435_GL0149202</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.089976</td>\n",
       "      <td>0.088785</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197042</th>\n",
       "      <td>9859108</td>\n",
       "      <td>MH0445_GL0066112</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.033149</td>\n",
       "      <td>0.036449</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116515</th>\n",
       "      <td>9879793</td>\n",
       "      <td>411459.RUMOBE_01786</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Blautia</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369973</th>\n",
       "      <td>9854153</td>\n",
       "      <td>MH0339_GL0008776</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.101815</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369662</th>\n",
       "      <td>9849875</td>\n",
       "      <td>MH0196_GL0163961</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.013418</td>\n",
       "      <td>0.014019</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369663</th>\n",
       "      <td>9852450</td>\n",
       "      <td>MH0276_GL0235765</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320790</th>\n",
       "      <td>9845699</td>\n",
       "      <td>MH0022_GL0010478</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4686</th>\n",
       "      <td>9874722</td>\n",
       "      <td>T2D-29A_GL0000548</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.160221</td>\n",
       "      <td>0.163551</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296144</th>\n",
       "      <td>9849405</td>\n",
       "      <td>MH0185_GL0064951</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.228887</td>\n",
       "      <td>0.232710</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245450</th>\n",
       "      <td>412</td>\n",
       "      <td>V1.FI27_GL0164628</td>\n",
       "      <td>15222</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.053670</td>\n",
       "      <td>0.055140</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285369</th>\n",
       "      <td>399</td>\n",
       "      <td>MH0150_GL0033854</td>\n",
       "      <td>15306</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13734</td>\n",
       "      <td>bactNOG74050</td>\n",
       "      <td>0.823204</td>\n",
       "      <td>0.835514</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101624</th>\n",
       "      <td>392</td>\n",
       "      <td>MH0264_GL0016567</td>\n",
       "      <td>15447</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.009471</td>\n",
       "      <td>0.010280</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183807</th>\n",
       "      <td>377</td>\n",
       "      <td>MH0404_GL0080882</td>\n",
       "      <td>15591</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00558</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.013084</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59169</th>\n",
       "      <td>349</td>\n",
       "      <td>159571453-stool2_revised_scaffold27522_1_gene3...</td>\n",
       "      <td>15945</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07279</td>\n",
       "      <td>COG3468</td>\n",
       "      <td>0.048934</td>\n",
       "      <td>0.050467</td>\n",
       "      <td>Cellular Processes and Signaling</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis;Intrace...</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267886</th>\n",
       "      <td>340</td>\n",
       "      <td>V1.FI25_GL0154021</td>\n",
       "      <td>16074</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K15125</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.013418</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355168</th>\n",
       "      <td>331</td>\n",
       "      <td>V1.FI28_GL0063861</td>\n",
       "      <td>16161</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00599</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.028037</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215409</th>\n",
       "      <td>321</td>\n",
       "      <td>V1.FI14_GL0255454</td>\n",
       "      <td>16278</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.022099</td>\n",
       "      <td>0.026168</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470957</th>\n",
       "      <td>318</td>\n",
       "      <td>O2.UC5-0_GL0058609</td>\n",
       "      <td>16311</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189070</th>\n",
       "      <td>317</td>\n",
       "      <td>O2.UC40-1_GL0020660</td>\n",
       "      <td>16344</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.047356</td>\n",
       "      <td>0.042056</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164915</th>\n",
       "      <td>312</td>\n",
       "      <td>V1.UC58-0_GL0107276</td>\n",
       "      <td>16419</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00558</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.011839</td>\n",
       "      <td>0.013084</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425189</th>\n",
       "      <td>309</td>\n",
       "      <td>V1.UC21-4_GL0157994</td>\n",
       "      <td>16491</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00599</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56580</th>\n",
       "      <td>301</td>\n",
       "      <td>MH0313_GL0120834</td>\n",
       "      <td>16626</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG34255</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>0.012150</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366267</th>\n",
       "      <td>290</td>\n",
       "      <td>O2.CD2-0_GL0055034</td>\n",
       "      <td>16704</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259192</th>\n",
       "      <td>281</td>\n",
       "      <td>MH0230_GL0027875</td>\n",
       "      <td>16845</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.044988</td>\n",
       "      <td>0.053271</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363401</th>\n",
       "      <td>271</td>\n",
       "      <td>T2D-122A_GL0027378</td>\n",
       "      <td>16941</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217952</th>\n",
       "      <td>266</td>\n",
       "      <td>MH0448_GL0184927</td>\n",
       "      <td>17049</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399905</th>\n",
       "      <td>249</td>\n",
       "      <td>MH0250_GL0010325</td>\n",
       "      <td>17202</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.074980</td>\n",
       "      <td>0.076636</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458845</th>\n",
       "      <td>228</td>\n",
       "      <td>O2.UC22-1_GL0043947</td>\n",
       "      <td>17361</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.155485</td>\n",
       "      <td>0.157944</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339944</th>\n",
       "      <td>205</td>\n",
       "      <td>V1.UC63-0_GL0053529</td>\n",
       "      <td>17601</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13734</td>\n",
       "      <td>NOG244570</td>\n",
       "      <td>0.060773</td>\n",
       "      <td>0.070093</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103410</th>\n",
       "      <td>170</td>\n",
       "      <td>V1.CD20-0_GL0037007</td>\n",
       "      <td>18738</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.421468</td>\n",
       "      <td>0.402804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498872</th>\n",
       "      <td>124</td>\n",
       "      <td>V1.FI17_GL0005370</td>\n",
       "      <td>19764</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.095501</td>\n",
       "      <td>0.095327</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412707</th>\n",
       "      <td>114</td>\n",
       "      <td>O2.UC50-0_GL0310928</td>\n",
       "      <td>19995</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.093923</td>\n",
       "      <td>0.092523</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126993</th>\n",
       "      <td>108</td>\n",
       "      <td>MH0328_GL0176549</td>\n",
       "      <td>20100</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.029203</td>\n",
       "      <td>0.032710</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64334</th>\n",
       "      <td>92</td>\n",
       "      <td>O2.UC43-0_GL0090208</td>\n",
       "      <td>20445</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.029203</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18027</th>\n",
       "      <td>81</td>\n",
       "      <td>MH0276_GL0250660</td>\n",
       "      <td>20691</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.006314</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83989</th>\n",
       "      <td>76</td>\n",
       "      <td>SZEY-48A_GL0052647</td>\n",
       "      <td>20778</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>K01317</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.203631</td>\n",
       "      <td>0.212150</td>\n",
       "      <td>Enzyme Families</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53975</th>\n",
       "      <td>74</td>\n",
       "      <td>MH0375_GL0055361</td>\n",
       "      <td>20790</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.036306</td>\n",
       "      <td>0.038318</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4901</th>\n",
       "      <td>42</td>\n",
       "      <td>ED16A_GL0163742</td>\n",
       "      <td>23358</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K14194</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.204420</td>\n",
       "      <td>0.207477</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336287</th>\n",
       "      <td>6</td>\n",
       "      <td>SZEY-41A_GL0088289</td>\n",
       "      <td>41277</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13735</td>\n",
       "      <td>COG5184</td>\n",
       "      <td>0.864246</td>\n",
       "      <td>0.873832</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "88836   9865705                              V1.CD6-0-PT_GL0096973   \n",
       "476498  9862460                                 O2.UC4-2_GL0053162   \n",
       "151345  9862491                                O2.UC43-0_GL0033611   \n",
       "34192   9877190       159551223-stool2_revised_C916401_1_gene84247   \n",
       "331083  9860951                                O2.UC21-1_GL0136539   \n",
       "280181  9857596                                   MH0415_GL0056639   \n",
       "483179  9878379  763577454-stool2_revised_scaffold23171_1_gene1...   \n",
       "370156  9853996                                   MH0335_GL0009364   \n",
       "409047  9860917                                 O2.UC2-0_GL0121235   \n",
       "34135   9872403                                    N003A_GL0007810   \n",
       "408969  9869351                                V1.UC52-0_GL0023463   \n",
       "81900   9855675                                   MH0373_GL0182676   \n",
       "15105   9848328                                   MH0143_GL0088163   \n",
       "53716   9877737      160502038-stool1_revised_C976100_1_gene185746   \n",
       "4930    9873886                                 T2D-113A_GL0111098   \n",
       "225835  9847115                                   MH0100_GL0041020   \n",
       "409075  9872514                                    N017A_GL0080793   \n",
       "151252  9853377                                   MH0318_GL0022747   \n",
       "53803   9856149                                   MH0384_GL0070245   \n",
       "151236  9855061                                   MH0362_GL0041997   \n",
       "197036  9846712                                   MH0086_GL0082324   \n",
       "34125   9858623                                   MH0435_GL0149202   \n",
       "197042  9859108                                   MH0445_GL0066112   \n",
       "116515  9879793                                411459.RUMOBE_01786   \n",
       "369973  9854153                                   MH0339_GL0008776   \n",
       "369662  9849875                                   MH0196_GL0163961   \n",
       "369663  9852450                                   MH0276_GL0235765   \n",
       "320790  9845699                                   MH0022_GL0010478   \n",
       "4686    9874722                                  T2D-29A_GL0000548   \n",
       "296144  9849405                                   MH0185_GL0064951   \n",
       "...         ...                                                ...   \n",
       "245450      412                                  V1.FI27_GL0164628   \n",
       "285369      399                                   MH0150_GL0033854   \n",
       "101624      392                                   MH0264_GL0016567   \n",
       "183807      377                                   MH0404_GL0080882   \n",
       "59169       349  159571453-stool2_revised_scaffold27522_1_gene3...   \n",
       "267886      340                                  V1.FI25_GL0154021   \n",
       "355168      331                                  V1.FI28_GL0063861   \n",
       "215409      321                                  V1.FI14_GL0255454   \n",
       "470957      318                                 O2.UC5-0_GL0058609   \n",
       "189070      317                                O2.UC40-1_GL0020660   \n",
       "164915      312                                V1.UC58-0_GL0107276   \n",
       "425189      309                                V1.UC21-4_GL0157994   \n",
       "56580       301                                   MH0313_GL0120834   \n",
       "366267      290                                 O2.CD2-0_GL0055034   \n",
       "259192      281                                   MH0230_GL0027875   \n",
       "363401      271                                 T2D-122A_GL0027378   \n",
       "217952      266                                   MH0448_GL0184927   \n",
       "399905      249                                   MH0250_GL0010325   \n",
       "458845      228                                O2.UC22-1_GL0043947   \n",
       "339944      205                                V1.UC63-0_GL0053529   \n",
       "103410      170                                V1.CD20-0_GL0037007   \n",
       "498872      124                                  V1.FI17_GL0005370   \n",
       "412707      114                                O2.UC50-0_GL0310928   \n",
       "126993      108                                   MH0328_GL0176549   \n",
       "64334        92                                O2.UC43-0_GL0090208   \n",
       "18027        81                                   MH0276_GL0250660   \n",
       "83989        76                                 SZEY-48A_GL0052647   \n",
       "53975        74                                   MH0375_GL0055361   \n",
       "4901         42                                    ED16A_GL0163742   \n",
       "336287        6                                 SZEY-41A_GL0088289   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "88836           102       Lack 5'-end           EUR   \n",
       "476498          102       Lack 5'-end           EUR   \n",
       "151345          102       Lack 3'-end           EUR   \n",
       "34192           102       Lack 3'-end           USA   \n",
       "331083          102       Lack 3'-end           EUR   \n",
       "280181          102       Lack 3'-end           EUR   \n",
       "483179          102       Lack 5'-end           USA   \n",
       "370156          102          Complete           EUR   \n",
       "409047          102       Lack 5'-end           EUR   \n",
       "34135           102       Lack 5'-end           CHN   \n",
       "408969          102       Lack 5'-end           EUR   \n",
       "81900           102       Lack 5'-end           EUR   \n",
       "15105           102          Complete           EUR   \n",
       "53716           102       Lack 5'-end           USA   \n",
       "4930            102       Lack 5'-end           CHN   \n",
       "225835          102       Lack 5'-end           EUR   \n",
       "409075          102       Lack 5'-end           CHN   \n",
       "151252          102          Complete           EUR   \n",
       "53803           102          Complete           EUR   \n",
       "151236          102       Lack 3'-end           EUR   \n",
       "197036          102       Lack 3'-end           EUR   \n",
       "34125           102       Lack 5'-end           EUR   \n",
       "197042          102       Lack 5'-end           EUR   \n",
       "116515          102          Complete            SP   \n",
       "369973          102       Lack 5'-end           EUR   \n",
       "369662          102       Lack 3'-end           EUR   \n",
       "369663          102          Complete           EUR   \n",
       "320790          102       Lack 5'-end           EUR   \n",
       "4686            102       Lack 5'-end           CHN   \n",
       "296144          102       Lack 3'-end           EUR   \n",
       "...             ...               ...           ...   \n",
       "245450        15222       Lack 3'-end           EUR   \n",
       "285369        15306          Complete           EUR   \n",
       "101624        15447          Complete           EUR   \n",
       "183807        15591          Complete           EUR   \n",
       "59169         15945       Lack 3'-end           USA   \n",
       "267886        16074       Lack 3'-end           EUR   \n",
       "355168        16161          Complete           EUR   \n",
       "215409        16278       Lack 3'-end           EUR   \n",
       "470957        16311          Complete           EUR   \n",
       "189070        16344          Complete           EUR   \n",
       "164915        16419          Complete           EUR   \n",
       "425189        16491          Complete           EUR   \n",
       "56580         16626          Complete           EUR   \n",
       "366267        16704          Complete           EUR   \n",
       "259192        16845          Complete           EUR   \n",
       "363401        16941          Complete           CHN   \n",
       "217952        17049          Complete           EUR   \n",
       "399905        17202          Complete           EUR   \n",
       "458845        17361          Complete           EUR   \n",
       "339944        17601       Lack 3'-end           EUR   \n",
       "103410        18738          Complete           EUR   \n",
       "498872        19764          Complete           EUR   \n",
       "412707        19995          Complete           EUR   \n",
       "126993        20100          Complete           EUR   \n",
       "64334         20445          Complete           EUR   \n",
       "18027         20691          Complete           EUR   \n",
       "83989         20778          Complete           CHN   \n",
       "53975         20790          Complete           EUR   \n",
       "4901          23358       Lack 3'-end           CHN   \n",
       "336287        41277       Lack 5'-end           CHN   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "88836                             unknown                           unknown   \n",
       "476498                            unknown                           unknown   \n",
       "151345                            unknown                           unknown   \n",
       "34192                          Firmicutes                       Eubacterium   \n",
       "331083                            unknown                           unknown   \n",
       "280181                            unknown                           unknown   \n",
       "483179                     Proteobacteria                           unknown   \n",
       "370156                            unknown                           unknown   \n",
       "409047                            unknown                           unknown   \n",
       "34135                             unknown                           unknown   \n",
       "408969                            unknown                           unknown   \n",
       "81900                             unknown                           unknown   \n",
       "15105                             unknown                           unknown   \n",
       "53716                             unknown                           unknown   \n",
       "4930                              unknown                           unknown   \n",
       "225835                            unknown                           unknown   \n",
       "409075                            unknown                           unknown   \n",
       "151252                            unknown                           unknown   \n",
       "53803                             unknown                           unknown   \n",
       "151236                            unknown                           unknown   \n",
       "197036                            unknown                           unknown   \n",
       "34125                          Firmicutes                  Faecalibacterium   \n",
       "197042                            unknown                           unknown   \n",
       "116515                         Firmicutes                           Blautia   \n",
       "369973                            unknown                           unknown   \n",
       "369662                            unknown                           unknown   \n",
       "369663                            unknown                           unknown   \n",
       "320790                            unknown                           unknown   \n",
       "4686                              unknown                           unknown   \n",
       "296144                            unknown                           unknown   \n",
       "...                                   ...                               ...   \n",
       "245450                            unknown                           unknown   \n",
       "285369                            unknown                           unknown   \n",
       "101624                            unknown                           unknown   \n",
       "183807                            unknown                           unknown   \n",
       "59169                             unknown                           unknown   \n",
       "267886                            unknown                           unknown   \n",
       "355168                            unknown                           unknown   \n",
       "215409                            unknown                           unknown   \n",
       "470957                            unknown                           unknown   \n",
       "189070                            unknown                           unknown   \n",
       "164915                            unknown                           unknown   \n",
       "425189                            unknown                           unknown   \n",
       "56580                             unknown                           unknown   \n",
       "366267                            unknown                           unknown   \n",
       "259192                            unknown                           unknown   \n",
       "363401                            unknown                           unknown   \n",
       "217952                            unknown                           unknown   \n",
       "399905                            unknown                           unknown   \n",
       "458845                      Bacteroidetes                       Bacteroides   \n",
       "339944                            unknown                           unknown   \n",
       "103410                     Proteobacteria                           unknown   \n",
       "498872                            unknown                           unknown   \n",
       "412707                            unknown                           unknown   \n",
       "126993                            unknown                           unknown   \n",
       "64334                             unknown                           unknown   \n",
       "18027                             unknown                           unknown   \n",
       "83989                      Proteobacteria                       Escherichia   \n",
       "53975                             unknown                           unknown   \n",
       "4901                              unknown                           unknown   \n",
       "336287                            unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "88836          unknown           unknown                    0.005525   \n",
       "476498         unknown           unknown                    0.021310   \n",
       "151345         unknown           unknown                    0.015785   \n",
       "34192          unknown           unknown                    0.014207   \n",
       "331083         unknown           unknown                    0.059195   \n",
       "280181         unknown           unknown                    0.020521   \n",
       "483179         unknown           unknown                    0.169692   \n",
       "370156         unknown           unknown                    0.004736   \n",
       "409047         unknown           unknown                    0.011050   \n",
       "34135          unknown           unknown                    0.004736   \n",
       "408969         unknown           unknown                    0.002368   \n",
       "81900          unknown           unknown                    0.022889   \n",
       "15105          unknown           unknown                    0.003946   \n",
       "53716          unknown           unknown                    0.000000   \n",
       "4930           unknown           unknown                    0.115233   \n",
       "225835         unknown           unknown                    0.047356   \n",
       "409075         unknown           unknown                    0.072612   \n",
       "151252         unknown           unknown                    0.000789   \n",
       "53803          unknown           unknown                    0.001579   \n",
       "151236         unknown           unknown                    0.089187   \n",
       "197036         unknown           unknown                    0.140489   \n",
       "34125          unknown           unknown                    0.089976   \n",
       "197042         unknown           unknown                    0.033149   \n",
       "116515         unknown           unknown                    0.004736   \n",
       "369973         unknown           unknown                    0.101815   \n",
       "369662         unknown           unknown                    0.013418   \n",
       "369663         unknown           unknown                    0.004736   \n",
       "320790         unknown           unknown                    0.005525   \n",
       "4686           unknown           unknown                    0.160221   \n",
       "296144         unknown           unknown                    0.228887   \n",
       "...                ...               ...                         ...   \n",
       "245450         unknown      bactNOG98109                    0.053670   \n",
       "285369          K13734      bactNOG74050                    0.823204   \n",
       "101624         unknown      bactNOG23571                    0.009471   \n",
       "183807          K00558      bactNOG23571                    0.011050   \n",
       "59169           K07279           COG3468                    0.048934   \n",
       "267886          K15125          NOG12793                    0.013418   \n",
       "355168          K00599   COG4646;COG0827                    0.032360   \n",
       "215409         unknown          NOG12793                    0.022099   \n",
       "470957         unknown          NOG12793                    0.032360   \n",
       "189070         unknown          NOG12793                    0.047356   \n",
       "164915          K00558      bactNOG23571                    0.011839   \n",
       "425189          K00599   COG4646;COG0827                    0.004736   \n",
       "56580          unknown      bactNOG34255                    0.010260   \n",
       "366267         unknown   COG4646;COG0827                    0.011050   \n",
       "259192         unknown      bactNOG98109                    0.044988   \n",
       "363401         unknown      bactNOG98109                    0.004736   \n",
       "217952         unknown           unknown                    0.002368   \n",
       "399905         unknown           unknown                    0.074980   \n",
       "458845         unknown   COG4646;COG0827                    0.155485   \n",
       "339944          K13734         NOG244570                    0.060773   \n",
       "103410         unknown          NOG12793                    0.421468   \n",
       "498872         unknown           unknown                    0.095501   \n",
       "412707         unknown           unknown                    0.093923   \n",
       "126993         unknown          NOG12793                    0.029203   \n",
       "64334          unknown   COG4646;COG0827                    0.029203   \n",
       "18027          unknown           unknown                    0.006314   \n",
       "83989           K01317          NOG12793                    0.203631   \n",
       "53975          unknown          NOG12793                    0.036306   \n",
       "4901            K14194          NOG12793                    0.204420   \n",
       "336287          K13735           COG5184                    0.864246   \n",
       "\n",
       "        Individual Occurence Frequency        KEGG Functional Categories  \\\n",
       "88836                         0.005607                           unknown   \n",
       "476498                        0.020561                           unknown   \n",
       "151345                        0.014953                           unknown   \n",
       "34192                         0.012150                           unknown   \n",
       "331083                        0.061682                           unknown   \n",
       "280181                        0.023364                           unknown   \n",
       "483179                        0.157944                           unknown   \n",
       "370156                        0.005607                           unknown   \n",
       "409047                        0.007477                           unknown   \n",
       "34135                         0.005607                           unknown   \n",
       "408969                        0.001869                           unknown   \n",
       "81900                         0.023364                           unknown   \n",
       "15105                         0.004673                           unknown   \n",
       "53716                         0.000000                           unknown   \n",
       "4930                          0.116822                           unknown   \n",
       "225835                        0.043925                           unknown   \n",
       "409075                        0.065421                           unknown   \n",
       "151252                        0.000935                           unknown   \n",
       "53803                         0.001869                           unknown   \n",
       "151236                        0.091589                           unknown   \n",
       "197036                        0.115888                           unknown   \n",
       "34125                         0.088785                           unknown   \n",
       "197042                        0.036449                           unknown   \n",
       "116515                        0.003738                           unknown   \n",
       "369973                        0.079439                           unknown   \n",
       "369662                        0.014019                           unknown   \n",
       "369663                        0.005607                           unknown   \n",
       "320790                        0.005607                           unknown   \n",
       "4686                          0.163551                           unknown   \n",
       "296144                        0.232710                           unknown   \n",
       "...                                ...                               ...   \n",
       "245450                        0.055140                           unknown   \n",
       "285369                        0.835514               Infectious Diseases   \n",
       "101624                        0.010280                           unknown   \n",
       "183807                        0.013084             Amino Acid Metabolism   \n",
       "59169                         0.050467  Cellular Processes and Signaling   \n",
       "267886                        0.015888               Infectious Diseases   \n",
       "355168                        0.028037             Amino Acid Metabolism   \n",
       "215409                        0.026168                           unknown   \n",
       "470957                        0.023364                           unknown   \n",
       "189070                        0.042056                           unknown   \n",
       "164915                        0.013084             Amino Acid Metabolism   \n",
       "425189                        0.003738             Amino Acid Metabolism   \n",
       "56580                         0.012150                           unknown   \n",
       "366267                        0.007477                           unknown   \n",
       "259192                        0.053271                           unknown   \n",
       "363401                        0.005607                           unknown   \n",
       "217952                        0.002804                           unknown   \n",
       "399905                        0.076636                           unknown   \n",
       "458845                        0.157944                           unknown   \n",
       "339944                        0.070093               Infectious Diseases   \n",
       "103410                        0.402804                           unknown   \n",
       "498872                        0.095327                           unknown   \n",
       "412707                        0.092523                           unknown   \n",
       "126993                        0.032710                           unknown   \n",
       "64334                         0.021495                           unknown   \n",
       "18027                         0.007477                           unknown   \n",
       "83989                         0.212150                   Enzyme Families   \n",
       "53975                         0.038318                           unknown   \n",
       "4901                          0.207477               Infectious Diseases   \n",
       "336287                        0.873832               Infectious Diseases   \n",
       "\n",
       "                             eggNOG Functional Categories Cohort Assembled  \n",
       "88836                                             unknown              EUR  \n",
       "476498                                            unknown              EUR  \n",
       "151345                                            unknown              EUR  \n",
       "34192                                             unknown              USA  \n",
       "331083                                            unknown              EUR  \n",
       "280181                                            unknown              EUR  \n",
       "483179                                            unknown              USA  \n",
       "370156                                            unknown              EUR  \n",
       "409047                                            unknown              EUR  \n",
       "34135                                             unknown              CHN  \n",
       "408969                                            unknown              EUR  \n",
       "81900                                             unknown              EUR  \n",
       "15105                                             unknown              EUR  \n",
       "53716                                             unknown              USA  \n",
       "4930                                              unknown              CHN  \n",
       "225835                                            unknown              EUR  \n",
       "409075                                            unknown              CHN  \n",
       "151252                                            unknown              EUR  \n",
       "53803                                             unknown              EUR  \n",
       "151236                                            unknown              EUR  \n",
       "197036                                            unknown              EUR  \n",
       "34125                                             unknown              EUR  \n",
       "197042                                            unknown              EUR  \n",
       "116515                                            unknown              NaN  \n",
       "369973                                            unknown              EUR  \n",
       "369662                                            unknown              EUR  \n",
       "369663                                            unknown              EUR  \n",
       "320790                                            unknown              EUR  \n",
       "4686                                              unknown              CHN  \n",
       "296144                                            unknown              EUR  \n",
       "...                                                   ...              ...  \n",
       "245450                                   Function unknown              EUR  \n",
       "285369             Cell wall/membrane/envelope biogenesis      EUR;CHN;USA  \n",
       "101624                                   Function unknown              EUR  \n",
       "183807                                   Function unknown              EUR  \n",
       "59169   Cell wall/membrane/envelope biogenesis;Intrace...              USA  \n",
       "267886                                   Function unknown              EUR  \n",
       "355168  Transcription;Replication, recombination and r...          EUR;CHN  \n",
       "215409                                   Function unknown              EUR  \n",
       "470957                                   Function unknown              EUR  \n",
       "189070                                   Function unknown              EUR  \n",
       "164915                                   Function unknown              EUR  \n",
       "425189  Transcription;Replication, recombination and r...          EUR;CHN  \n",
       "56580                                    Function unknown              EUR  \n",
       "366267  Transcription;Replication, recombination and r...          EUR;USA  \n",
       "259192                                   Function unknown          EUR;CHN  \n",
       "363401                                   Function unknown              CHN  \n",
       "217952                                            unknown              EUR  \n",
       "399905                                            unknown              EUR  \n",
       "458845  Transcription;Replication, recombination and r...              EUR  \n",
       "339944                                   Function unknown          EUR;CHN  \n",
       "103410                                   Function unknown      EUR;CHN;USA  \n",
       "498872                                            unknown          EUR;USA  \n",
       "412707                                            unknown              EUR  \n",
       "126993                                   Function unknown              EUR  \n",
       "64334   Transcription;Replication, recombination and r...          EUR;USA  \n",
       "18027                                             unknown              EUR  \n",
       "83989                                    Function unknown          EUR;CHN  \n",
       "53975                                    Function unknown              EUR  \n",
       "4901                                     Function unknown          EUR;CHN  \n",
       "336287  Cell cycle control, cell division, chromosome ...      EUR;CHN;USA  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by='Gene Length')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on DataFrame in module pandas.core.frame object:\n",
      "\n",
      "class DataFrame(pandas.core.generic.NDFrame)\n",
      " |  Two-dimensional size-mutable, potentially heterogeneous tabular data\n",
      " |  structure with labeled axes (rows and columns). Arithmetic operations\n",
      " |  align on both row and column labels. Can be thought of as a dict-like\n",
      " |  container for Series objects. The primary pandas data structure\n",
      " |  \n",
      " |  Parameters\n",
      " |  ----------\n",
      " |  data : numpy ndarray (structured or homogeneous), dict, or DataFrame\n",
      " |      Dict can contain Series, arrays, constants, or list-like objects\n",
      " |  index : Index or array-like\n",
      " |      Index to use for resulting frame. Will default to np.arange(n) if\n",
      " |      no indexing information part of input data and no index provided\n",
      " |  columns : Index or array-like\n",
      " |      Column labels to use for resulting frame. Will default to\n",
      " |      np.arange(n) if no column labels are provided\n",
      " |  dtype : dtype, default None\n",
      " |      Data type to force, otherwise infer\n",
      " |  copy : boolean, default False\n",
      " |      Copy data from inputs. Only affects DataFrame / 2d ndarray input\n",
      " |  \n",
      " |  Examples\n",
      " |  --------\n",
      " |  >>> d = {'col1': ts1, 'col2': ts2}\n",
      " |  >>> df = DataFrame(data=d, index=index)\n",
      " |  >>> df2 = DataFrame(np.random.randn(10, 5))\n",
      " |  >>> df3 = DataFrame(np.random.randn(10, 5),\n",
      " |  ...                 columns=['a', 'b', 'c', 'd', 'e'])\n",
      " |  \n",
      " |  See also\n",
      " |  --------\n",
      " |  DataFrame.from_records : constructor from tuples, also record arrays\n",
      " |  DataFrame.from_dict : from dicts of Series, arrays, or dicts\n",
      " |  DataFrame.from_items : from sequence of (key, value) pairs\n",
      " |  pandas.read_csv, pandas.read_table, pandas.read_clipboard\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      DataFrame\n",
      " |      pandas.core.generic.NDFrame\n",
      " |      pandas.core.base.PandasObject\n",
      " |      pandas.core.base.StringMixin\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __add__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __add__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __and__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __and__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __div__ = __truediv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |  \n",
      " |  __eq__(self, other)\n",
      " |      Wrapper for comparison method __eq__\n",
      " |  \n",
      " |  __floordiv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __floordiv__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __ge__(self, other)\n",
      " |      Wrapper for comparison method __ge__\n",
      " |  \n",
      " |  __getitem__(self, key)\n",
      " |  \n",
      " |  __gt__(self, other)\n",
      " |      Wrapper for comparison method __gt__\n",
      " |  \n",
      " |  __iadd__ = f(self, other)\n",
      " |  \n",
      " |  __imul__ = f(self, other)\n",
      " |  \n",
      " |  __init__(self, data=None, index=None, columns=None, dtype=None, copy=False)\n",
      " |      Initialize self.  See help(type(self)) for accurate signature.\n",
      " |  \n",
      " |  __ipow__ = f(self, other)\n",
      " |  \n",
      " |  __isub__ = f(self, other)\n",
      " |  \n",
      " |  __itruediv__ = f(self, other)\n",
      " |  \n",
      " |  __le__(self, other)\n",
      " |      Wrapper for comparison method __le__\n",
      " |  \n",
      " |  __len__(self)\n",
      " |      Returns length of info axis, but here we use the index\n",
      " |  \n",
      " |  __lt__(self, other)\n",
      " |      Wrapper for comparison method __lt__\n",
      " |  \n",
      " |  __mod__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __mod__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __mul__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __mul__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __ne__(self, other)\n",
      " |      Wrapper for comparison method __ne__\n",
      " |  \n",
      " |  __or__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __or__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __pow__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __pow__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __radd__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __radd__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rand__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __rand__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rdiv__ = __rtruediv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |  \n",
      " |  __rfloordiv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rfloordiv__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rmod__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rmod__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rmul__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rmul__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __ror__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __ror__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rpow__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rpow__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rsub__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rsub__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rtruediv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __rtruediv__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __rxor__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __rxor__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __setitem__(self, key, value)\n",
      " |  \n",
      " |  __sub__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __sub__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __truediv__(self, other, axis=None, level=None, fill_value=None)\n",
      " |      Binary operator __truediv__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  __unicode__(self)\n",
      " |      Return a string representation for a particular DataFrame\n",
      " |      \n",
      " |      Invoked by unicode(df) in py2 only. Yields a Unicode String in both\n",
      " |      py2/py3.\n",
      " |  \n",
      " |  __xor__(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Binary operator __xor__ with support to substitute a fill_value for missing data in\n",
      " |      one of the inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame locations are\n",
      " |          missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  add(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Addition of dataframe and other, element-wise (binary operator `add`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe + other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.radd\n",
      " |  \n",
      " |  align(self, other, join='outer', axis=None, level=None, copy=True, fill_value=None, method=None, limit=None, fill_axis=0, broadcast_axis=None)\n",
      " |      Align two object on their axes with the\n",
      " |      specified join method for each axis Index\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame or Series\n",
      " |      join : {'outer', 'inner', 'left', 'right'}, default 'outer'\n",
      " |      axis : allowed axis of the other object, default None\n",
      " |          Align on index (0), columns (1), or both (None)\n",
      " |      level : int or level name, default None\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      copy : boolean, default True\n",
      " |          Always returns new objects. If copy=False and no reindexing is\n",
      " |          required then original objects are returned.\n",
      " |      fill_value : scalar, default np.NaN\n",
      " |          Value to use for missing values. Defaults to NaN, but can be any\n",
      " |          \"compatible\" value\n",
      " |      method : str, default None\n",
      " |      limit : int, default None\n",
      " |      fill_axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          Filling axis, method and limit\n",
      " |      broadcast_axis : {0 or 'index', 1 or 'columns'}, default None\n",
      " |          Broadcast values along this axis, if aligning two objects of\n",
      " |          different dimensions\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      (left, right) : (DataFrame, type of other)\n",
      " |          Aligned objects\n",
      " |  \n",
      " |  all(self, axis=None, bool_only=None, skipna=None, level=None, **kwargs)\n",
      " |      Return whether all elements are True over requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      bool_only : boolean, default None\n",
      " |          Include only boolean columns. If None, will attempt to use everything,\n",
      " |          then use only boolean data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      all : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  any(self, axis=None, bool_only=None, skipna=None, level=None, **kwargs)\n",
      " |      Return whether any element is True over requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      bool_only : boolean, default None\n",
      " |          Include only boolean columns. If None, will attempt to use everything,\n",
      " |          then use only boolean data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      any : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  append(self, other, ignore_index=False, verify_integrity=False)\n",
      " |      Append rows of `other` to the end of this frame, returning a new\n",
      " |      object. Columns not in this frame are added as new columns.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame or Series/dict-like object, or list of these\n",
      " |          The data to append.\n",
      " |      ignore_index : boolean, default False\n",
      " |          If True, do not use the index labels.\n",
      " |      verify_integrity : boolean, default False\n",
      " |          If True, raise ValueError on creating index with duplicates.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      appended : DataFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      If a list of dict/series is passed and the keys are all contained in\n",
      " |      the DataFrame's index, the order of the columns in the resulting\n",
      " |      DataFrame will be unchanged.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      pandas.concat : General function to concatenate DataFrame, Series\n",
      " |          or Panel objects\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'))\n",
      " |      >>> df\n",
      " |         A  B\n",
      " |      0  1  2\n",
      " |      1  3  4\n",
      " |      >>> df2 = pd.DataFrame([[5, 6], [7, 8]], columns=list('AB'))\n",
      " |      >>> df.append(df2)\n",
      " |         A  B\n",
      " |      0  1  2\n",
      " |      1  3  4\n",
      " |      0  5  6\n",
      " |      1  7  8\n",
      " |      \n",
      " |      With `ignore_index` set to True:\n",
      " |      \n",
      " |      >>> df.append(df2, ignore_index=True)\n",
      " |         A  B\n",
      " |      0  1  2\n",
      " |      1  3  4\n",
      " |      2  5  6\n",
      " |      3  7  8\n",
      " |  \n",
      " |  apply(self, func, axis=0, broadcast=False, raw=False, reduce=None, args=(), **kwds)\n",
      " |      Applies function along input axis of DataFrame.\n",
      " |      \n",
      " |      Objects passed to functions are Series objects having index\n",
      " |      either the DataFrame's index (axis=0) or the columns (axis=1).\n",
      " |      Return type depends on whether passed function aggregates, or the\n",
      " |      reduce argument if the DataFrame is empty.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      func : function\n",
      " |          Function to apply to each column/row\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          * 0 or 'index': apply function to each column\n",
      " |          * 1 or 'columns': apply function to each row\n",
      " |      broadcast : boolean, default False\n",
      " |          For aggregation functions, return object of same size with values\n",
      " |          propagated\n",
      " |      raw : boolean, default False\n",
      " |          If False, convert each row or column into a Series. If raw=True the\n",
      " |          passed function will receive ndarray objects instead. If you are\n",
      " |          just applying a NumPy reduction function this will achieve much\n",
      " |          better performance\n",
      " |      reduce : boolean or None, default None\n",
      " |          Try to apply reduction procedures. If the DataFrame is empty,\n",
      " |          apply will use reduce to determine whether the result should be a\n",
      " |          Series or a DataFrame. If reduce is None (the default), apply's\n",
      " |          return value will be guessed by calling func an empty Series (note:\n",
      " |          while guessing, exceptions raised by func will be ignored). If\n",
      " |          reduce is True a Series will always be returned, and if False a\n",
      " |          DataFrame will always be returned.\n",
      " |      args : tuple\n",
      " |          Positional arguments to pass to function in addition to the\n",
      " |          array/series\n",
      " |      Additional keyword arguments will be passed as keywords to the function\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      In the current implementation apply calls func twice on the\n",
      " |      first column/row to decide whether it can take a fast or slow\n",
      " |      code path. This can lead to unexpected behavior if func has\n",
      " |      side-effects, as they will take effect twice for the first\n",
      " |      column/row.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df.apply(numpy.sqrt) # returns DataFrame\n",
      " |      >>> df.apply(numpy.sum, axis=0) # equiv to df.sum(0)\n",
      " |      >>> df.apply(numpy.sum, axis=1) # equiv to df.sum(1)\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.applymap: For elementwise operations\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      applied : Series or DataFrame\n",
      " |  \n",
      " |  applymap(self, func)\n",
      " |      Apply a function to a DataFrame that is intended to operate\n",
      " |      elementwise, i.e. like doing map(func, series) for each series in the\n",
      " |      DataFrame\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      func : function\n",
      " |          Python function, returns a single value from a single value\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = pd.DataFrame(np.random.randn(3, 3))\n",
      " |      >>> df\n",
      " |          0         1          2\n",
      " |      0  -0.029638  1.081563   1.280300\n",
      " |      1   0.647747  0.831136  -1.549481\n",
      " |      2   0.513416 -0.884417   0.195343\n",
      " |      >>> df = df.applymap(lambda x: '%.2f' % x)\n",
      " |      >>> df\n",
      " |          0         1          2\n",
      " |      0  -0.03      1.08       1.28\n",
      " |      1   0.65      0.83      -1.55\n",
      " |      2   0.51     -0.88       0.20\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      applied : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.apply : For operations on rows/columns\n",
      " |  \n",
      " |  assign(self, **kwargs)\n",
      " |      Assign new columns to a DataFrame, returning a new object\n",
      " |      (a copy) with all the original columns in addition to the new ones.\n",
      " |      \n",
      " |      .. versionadded:: 0.16.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      kwargs : keyword, value pairs\n",
      " |          keywords are the column names. If the values are\n",
      " |          callable, they are computed on the DataFrame and\n",
      " |          assigned to the new columns. The callable must not\n",
      " |          change input DataFrame (though pandas doesn't check it).\n",
      " |          If the values are not callable, (e.g. a Series, scalar, or array),\n",
      " |          they are simply assigned.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      df : DataFrame\n",
      " |          A new DataFrame with the new columns in addition to\n",
      " |          all the existing columns.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Since ``kwargs`` is a dictionary, the order of your\n",
      " |      arguments may not be preserved. The make things predicatable,\n",
      " |      the columns are inserted in alphabetical order, at the end of\n",
      " |      your DataFrame. Assigning multiple columns within the same\n",
      " |      ``assign`` is possible, but you cannot reference other columns\n",
      " |      created within the same ``assign`` call.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = DataFrame({'A': range(1, 11), 'B': np.random.randn(10)})\n",
      " |      \n",
      " |      Where the value is a callable, evaluated on `df`:\n",
      " |      \n",
      " |      >>> df.assign(ln_A = lambda x: np.log(x.A))\n",
      " |          A         B      ln_A\n",
      " |      0   1  0.426905  0.000000\n",
      " |      1   2 -0.780949  0.693147\n",
      " |      2   3 -0.418711  1.098612\n",
      " |      3   4 -0.269708  1.386294\n",
      " |      4   5 -0.274002  1.609438\n",
      " |      5   6 -0.500792  1.791759\n",
      " |      6   7  1.649697  1.945910\n",
      " |      7   8 -1.495604  2.079442\n",
      " |      8   9  0.549296  2.197225\n",
      " |      9  10 -0.758542  2.302585\n",
      " |      \n",
      " |      Where the value already exists and is inserted:\n",
      " |      \n",
      " |      >>> newcol = np.log(df['A'])\n",
      " |      >>> df.assign(ln_A=newcol)\n",
      " |          A         B      ln_A\n",
      " |      0   1  0.426905  0.000000\n",
      " |      1   2 -0.780949  0.693147\n",
      " |      2   3 -0.418711  1.098612\n",
      " |      3   4 -0.269708  1.386294\n",
      " |      4   5 -0.274002  1.609438\n",
      " |      5   6 -0.500792  1.791759\n",
      " |      6   7  1.649697  1.945910\n",
      " |      7   8 -1.495604  2.079442\n",
      " |      8   9  0.549296  2.197225\n",
      " |      9  10 -0.758542  2.302585\n",
      " |  \n",
      " |  boxplot(self, column=None, by=None, ax=None, fontsize=None, rot=0, grid=True, figsize=None, layout=None, return_type=None, **kwds)\n",
      " |      Make a box plot from DataFrame column optionally grouped by some columns or\n",
      " |      other inputs\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      data : the pandas object holding the data\n",
      " |      column : column name or list of names, or vector\n",
      " |          Can be any valid input to groupby\n",
      " |      by : string or sequence\n",
      " |          Column in the DataFrame to group by\n",
      " |      ax : Matplotlib axes object, optional\n",
      " |      fontsize : int or string\n",
      " |      rot : label rotation angle\n",
      " |      figsize : A tuple (width, height) in inches\n",
      " |      grid : Setting this to True will show the grid\n",
      " |      layout : tuple (optional)\n",
      " |          (rows, columns) for the layout of the plot\n",
      " |      return_type : {None, 'axes', 'dict', 'both'}, default None\n",
      " |          The kind of object to return. The default is ``axes``\n",
      " |          'axes' returns the matplotlib axes the boxplot is drawn on;\n",
      " |          'dict' returns a dictionary  whose values are the matplotlib\n",
      " |          Lines of the boxplot;\n",
      " |          'both' returns a namedtuple with the axes and dict.\n",
      " |      \n",
      " |          When grouping with ``by``, a Series mapping columns to ``return_type``\n",
      " |          is returned, unless ``return_type`` is None, in which case a NumPy\n",
      " |          array of axes is returned with the same shape as ``layout``.\n",
      " |          See the prose documentation for more.\n",
      " |      \n",
      " |      kwds : other plotting keyword arguments to be passed to matplotlib boxplot\n",
      " |             function\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      lines : dict\n",
      " |      ax : matplotlib Axes\n",
      " |      (ax, lines): namedtuple\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Use ``return_type='dict'`` when you want to tweak the appearance\n",
      " |      of the lines after plotting. In this case a dict containing the Lines\n",
      " |      making up the boxes, caps, fliers, medians, and whiskers is returned.\n",
      " |  \n",
      " |  combine(self, other, func, fill_value=None, overwrite=True)\n",
      " |      Add two DataFrame objects and do not propagate NaN values, so if for a\n",
      " |      (column, time) one frame is missing a value, it will default to the\n",
      " |      other frame's value (which might be NaN as well)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame\n",
      " |      func : function\n",
      " |      fill_value : scalar value\n",
      " |      overwrite : boolean, default True\n",
      " |          If True then overwrite values for common keys in the calling frame\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |  \n",
      " |  combineAdd(self, other)\n",
      " |      DEPRECATED. Use ``DataFrame.add(other, fill_value=0.)`` instead.\n",
      " |      \n",
      " |      Add two DataFrame objects and do not propagate\n",
      " |      NaN values, so if for a (column, time) one frame is missing a\n",
      " |      value, it will default to the other frame's value (which might\n",
      " |      be NaN as well)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.add\n",
      " |  \n",
      " |  combineMult(self, other)\n",
      " |      DEPRECATED. Use ``DataFrame.mul(other, fill_value=1.)`` instead.\n",
      " |      \n",
      " |      Multiply two DataFrame objects and do not propagate NaN values, so if\n",
      " |      for a (column, time) one frame is missing a value, it will default to\n",
      " |      the other frame's value (which might be NaN as well)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.mul\n",
      " |  \n",
      " |  combine_first(self, other)\n",
      " |      Combine two DataFrame objects and default to non-null values in frame\n",
      " |      calling the method. Result index columns will be the union of the\n",
      " |      respective indexes and columns\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      a's values prioritized, use values from b to fill holes:\n",
      " |      \n",
      " |      >>> a.combine_first(b)\n",
      " |      \n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      combined : DataFrame\n",
      " |  \n",
      " |  compound(self, axis=None, skipna=None, level=None)\n",
      " |      Return the compound percentage of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      compounded : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  corr(self, method='pearson', min_periods=1)\n",
      " |      Compute pairwise correlation of columns, excluding NA/null values\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      method : {'pearson', 'kendall', 'spearman'}\n",
      " |          * pearson : standard correlation coefficient\n",
      " |          * kendall : Kendall Tau correlation coefficient\n",
      " |          * spearman : Spearman rank correlation\n",
      " |      min_periods : int, optional\n",
      " |          Minimum number of observations required per pair of columns\n",
      " |          to have a valid result. Currently only available for pearson\n",
      " |          and spearman correlation\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : DataFrame\n",
      " |  \n",
      " |  corrwith(self, other, axis=0, drop=False)\n",
      " |      Compute pairwise correlation between rows or columns of two DataFrame\n",
      " |      objects.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          0 or 'index' to compute column-wise, 1 or 'columns' for row-wise\n",
      " |      drop : boolean, default False\n",
      " |          Drop missing indices from result, default returns union of all\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      correls : Series\n",
      " |  \n",
      " |  count(self, axis=0, level=None, numeric_only=False)\n",
      " |      Return Series with number of non-NA/null observations over requested\n",
      " |      axis. Works with non-floating point data as well (detects NaN and None)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          0 or 'index' for row-wise, 1 or 'columns' for column-wise\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a DataFrame\n",
      " |      numeric_only : boolean, default False\n",
      " |          Include only float, int, boolean data\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      count : Series (or DataFrame if level specified)\n",
      " |  \n",
      " |  cov(self, min_periods=None)\n",
      " |      Compute pairwise covariance of columns, excluding NA/null values\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      min_periods : int, optional\n",
      " |          Minimum number of observations required per pair of columns\n",
      " |          to have a valid result.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : DataFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      `y` contains the covariance matrix of the DataFrame's time series.\n",
      " |      The covariance is normalized by N-1 (unbiased estimator).\n",
      " |  \n",
      " |  cummax(self, axis=None, skipna=True, *args, **kwargs)\n",
      " |      Return cumulative max over requested axis.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      cummax : Series\n",
      " |  \n",
      " |  cummin(self, axis=None, skipna=True, *args, **kwargs)\n",
      " |      Return cumulative minimum over requested axis.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      cummin : Series\n",
      " |  \n",
      " |  cumprod(self, axis=None, skipna=True, *args, **kwargs)\n",
      " |      Return cumulative product over requested axis.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      cumprod : Series\n",
      " |  \n",
      " |  cumsum(self, axis=None, skipna=True, *args, **kwargs)\n",
      " |      Return cumulative sum over requested axis.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      cumsum : Series\n",
      " |  \n",
      " |  diff(self, periods=1, axis=0)\n",
      " |      1st discrete difference of object\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      periods : int, default 1\n",
      " |          Periods to shift for forming difference\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          Take difference over rows (0) or columns (1).\n",
      " |      \n",
      " |          .. versionadded: 0.16.1\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      diffed : DataFrame\n",
      " |  \n",
      " |  div = truediv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |  \n",
      " |  divide = truediv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |  \n",
      " |  dot(self, other)\n",
      " |      Matrix multiplication with DataFrame or Series objects\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame or Series\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      dot_product : DataFrame or Series\n",
      " |  \n",
      " |  drop_duplicates(self, subset=None, keep='first', inplace=False)\n",
      " |      Return DataFrame with duplicate rows removed, optionally only\n",
      " |      considering certain columns\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      subset : column label or sequence of labels, optional\n",
      " |          Only consider certain columns for identifying duplicates, by\n",
      " |          default use all of the columns\n",
      " |      keep : {'first', 'last', False}, default 'first'\n",
      " |          - ``first`` : Drop duplicates except for the first occurrence.\n",
      " |          - ``last`` : Drop duplicates except for the last occurrence.\n",
      " |          - False : Drop all duplicates.\n",
      " |      take_last : deprecated\n",
      " |      inplace : boolean, default False\n",
      " |          Whether to drop duplicates in place or to return a copy\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      deduplicated : DataFrame\n",
      " |  \n",
      " |  dropna(self, axis=0, how='any', thresh=None, subset=None, inplace=False)\n",
      " |      Return object with labels on given axis omitted where alternately any\n",
      " |      or all of the data are missing\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, or tuple/list thereof\n",
      " |          Pass tuple or list to drop on multiple axes\n",
      " |      how : {'any', 'all'}\n",
      " |          * any : if any NA values are present, drop that label\n",
      " |          * all : if all values are NA, drop that label\n",
      " |      thresh : int, default None\n",
      " |          int value : require that many non-NA values\n",
      " |      subset : array-like\n",
      " |          Labels along other axis to consider, e.g. if you are dropping rows\n",
      " |          these would be a list of columns to include\n",
      " |      inplace : boolean, default False\n",
      " |          If True, do operation inplace and return None.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      dropped : DataFrame\n",
      " |  \n",
      " |  duplicated(self, subset=None, keep='first')\n",
      " |      Return boolean Series denoting duplicate rows, optionally only\n",
      " |      considering certain columns\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      subset : column label or sequence of labels, optional\n",
      " |          Only consider certain columns for identifying duplicates, by\n",
      " |          default use all of the columns\n",
      " |      keep : {'first', 'last', False}, default 'first'\n",
      " |          - ``first`` : Mark duplicates as ``True`` except for the\n",
      " |            first occurrence.\n",
      " |          - ``last`` : Mark duplicates as ``True`` except for the\n",
      " |            last occurrence.\n",
      " |          - False : Mark all duplicates as ``True``.\n",
      " |      take_last : deprecated\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      duplicated : Series\n",
      " |  \n",
      " |  eq(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods eq\n",
      " |  \n",
      " |  eval(self, expr, inplace=None, **kwargs)\n",
      " |      Evaluate an expression in the context of the calling DataFrame\n",
      " |      instance.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      expr : string\n",
      " |          The expression string to evaluate.\n",
      " |      inplace : bool\n",
      " |          If the expression contains an assignment, whether to return a new\n",
      " |          DataFrame or mutate the existing.\n",
      " |      \n",
      " |          WARNING: inplace=None currently falls back to to True, but\n",
      " |          in a future version, will default to False.  Use inplace=True\n",
      " |          explicitly rather than relying on the default.\n",
      " |      \n",
      " |          .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      kwargs : dict\n",
      " |          See the documentation for :func:`~pandas.eval` for complete details\n",
      " |          on the keyword arguments accepted by\n",
      " |          :meth:`~pandas.DataFrame.query`.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      ret : ndarray, scalar, or pandas object\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.DataFrame.query\n",
      " |      pandas.DataFrame.assign\n",
      " |      pandas.eval\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      For more details see the API documentation for :func:`~pandas.eval`.\n",
      " |      For detailed examples see :ref:`enhancing performance with eval\n",
      " |      <enhancingperf.eval>`.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> from numpy.random import randn\n",
      " |      >>> from pandas import DataFrame\n",
      " |      >>> df = DataFrame(randn(10, 2), columns=list('ab'))\n",
      " |      >>> df.eval('a + b')\n",
      " |      >>> df.eval('c = a + b')\n",
      " |  \n",
      " |  ewm(self, com=None, span=None, halflife=None, alpha=None, min_periods=0, freq=None, adjust=True, ignore_na=False, axis=0)\n",
      " |      Provides exponential weighted functions\n",
      " |      \n",
      " |      .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      com : float, optional\n",
      " |          Specify decay in terms of center of mass,\n",
      " |          :math:`\\alpha = 1 / (1 + com),\\text{ for } com \\geq 0`\n",
      " |      span : float, optional\n",
      " |          Specify decay in terms of span,\n",
      " |          :math:`\\alpha = 2 / (span + 1),\\text{ for } span \\geq 1`\n",
      " |      halflife : float, optional\n",
      " |          Specify decay in terms of half-life,\n",
      " |          :math:`\\alpha = 1 - exp(log(0.5) / halflife),\\text{ for } halflife > 0`\n",
      " |      alpha : float, optional\n",
      " |          Specify smoothing factor :math:`\\alpha` directly,\n",
      " |          :math:`0 < \\alpha \\leq 1`\n",
      " |      \n",
      " |          .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      min_periods : int, default 0\n",
      " |          Minimum number of observations in window required to have a value\n",
      " |          (otherwise result is NA).\n",
      " |      freq : None or string alias / date offset object, default=None (DEPRECATED)\n",
      " |          Frequency to conform to before computing statistic\n",
      " |      adjust : boolean, default True\n",
      " |          Divide by decaying adjustment factor in beginning periods to account\n",
      " |          for imbalance in relative weightings (viewing EWMA as a moving average)\n",
      " |      ignore_na : boolean, default False\n",
      " |          Ignore missing values when calculating weights;\n",
      " |          specify True to reproduce pre-0.15.0 behavior\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      a Window sub-classed for the particular operation\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = DataFrame({'B': [0, 1, 2, np.nan, 4]})\n",
      " |           B\n",
      " |      0  0.0\n",
      " |      1  1.0\n",
      " |      2  2.0\n",
      " |      3  NaN\n",
      " |      4  4.0\n",
      " |      \n",
      " |      >>> df.ewm(com=0.5).mean()\n",
      " |                B\n",
      " |      0  0.000000\n",
      " |      1  0.750000\n",
      " |      2  1.615385\n",
      " |      3  1.615385\n",
      " |      4  3.670213\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Exactly one of center of mass, span, half-life, and alpha must be provided.\n",
      " |      Allowed values and relationship between the parameters are specified in the\n",
      " |      parameter descriptions above; see the link at the end of this section for\n",
      " |      a detailed explanation.\n",
      " |      \n",
      " |      The `freq` keyword is used to conform time series data to a specified\n",
      " |      frequency by resampling the data. This is done with the default parameters\n",
      " |      of :meth:`~pandas.Series.resample` (i.e. using the `mean`).\n",
      " |      \n",
      " |      When adjust is True (default), weighted averages are calculated using\n",
      " |      weights (1-alpha)**(n-1), (1-alpha)**(n-2), ..., 1-alpha, 1.\n",
      " |      \n",
      " |      When adjust is False, weighted averages are calculated recursively as:\n",
      " |         weighted_average[0] = arg[0];\n",
      " |         weighted_average[i] = (1-alpha)*weighted_average[i-1] + alpha*arg[i].\n",
      " |      \n",
      " |      When ignore_na is False (default), weights are based on absolute positions.\n",
      " |      For example, the weights of x and y used in calculating the final weighted\n",
      " |      average of [x, None, y] are (1-alpha)**2 and 1 (if adjust is True), and\n",
      " |      (1-alpha)**2 and alpha (if adjust is False).\n",
      " |      \n",
      " |      When ignore_na is True (reproducing pre-0.15.0 behavior), weights are based\n",
      " |      on relative positions. For example, the weights of x and y used in\n",
      " |      calculating the final weighted average of [x, None, y] are 1-alpha and 1\n",
      " |      (if adjust is True), and 1-alpha and alpha (if adjust is False).\n",
      " |      \n",
      " |      More details can be found at\n",
      " |      http://pandas.pydata.org/pandas-docs/stable/computation.html#exponentially-weighted-windows\n",
      " |  \n",
      " |  expanding(self, min_periods=1, freq=None, center=False, axis=0)\n",
      " |      Provides expanding transformations.\n",
      " |      \n",
      " |      .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      min_periods : int, default None\n",
      " |          Minimum number of observations in window required to have a value\n",
      " |          (otherwise result is NA).\n",
      " |      freq : string or DateOffset object, optional (default None) (DEPRECATED)\n",
      " |          Frequency to conform the data to before computing the statistic.\n",
      " |          Specified as a frequency string or DateOffset object.\n",
      " |      center : boolean, default False\n",
      " |          Set the labels at the center of the window.\n",
      " |      axis : int or string, default 0\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      a Window sub-classed for the particular operation\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = DataFrame({'B': [0, 1, 2, np.nan, 4]})\n",
      " |           B\n",
      " |      0  0.0\n",
      " |      1  1.0\n",
      " |      2  2.0\n",
      " |      3  NaN\n",
      " |      4  4.0\n",
      " |      \n",
      " |      >>> df.expanding(2).sum()\n",
      " |           B\n",
      " |      0  NaN\n",
      " |      1  1.0\n",
      " |      2  3.0\n",
      " |      3  3.0\n",
      " |      4  7.0\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      By default, the result is set to the right edge of the window. This can be\n",
      " |      changed to the center of the window by setting ``center=True``.\n",
      " |      \n",
      " |      The `freq` keyword is used to conform time series data to a specified\n",
      " |      frequency by resampling the data. This is done with the default parameters\n",
      " |      of :meth:`~pandas.Series.resample` (i.e. using the `mean`).\n",
      " |  \n",
      " |  fillna(self, value=None, method=None, axis=None, inplace=False, limit=None, downcast=None, **kwargs)\n",
      " |      Fill NA/NaN values using the specified method\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      value : scalar, dict, Series, or DataFrame\n",
      " |          Value to use to fill holes (e.g. 0), alternately a\n",
      " |          dict/Series/DataFrame of values specifying which value to use for\n",
      " |          each index (for a Series) or column (for a DataFrame). (values not\n",
      " |          in the dict/Series/DataFrame will not be filled). This value cannot\n",
      " |          be a list.\n",
      " |      method : {'backfill', 'bfill', 'pad', 'ffill', None}, default None\n",
      " |          Method to use for filling holes in reindexed Series\n",
      " |          pad / ffill: propagate last valid observation forward to next valid\n",
      " |          backfill / bfill: use NEXT valid observation to fill gap\n",
      " |      axis : {0 or 'index', 1 or 'columns'}\n",
      " |      inplace : boolean, default False\n",
      " |          If True, fill in place. Note: this will modify any\n",
      " |          other views on this object, (e.g. a no-copy slice for a column in a\n",
      " |          DataFrame).\n",
      " |      limit : int, default None\n",
      " |          If method is specified, this is the maximum number of consecutive\n",
      " |          NaN values to forward/backward fill. In other words, if there is\n",
      " |          a gap with more than this number of consecutive NaNs, it will only\n",
      " |          be partially filled. If method is not specified, this is the\n",
      " |          maximum number of entries along the entire axis where NaNs will be\n",
      " |          filled.\n",
      " |      downcast : dict, default is None\n",
      " |          a dict of item->dtype of what to downcast if possible,\n",
      " |          or the string 'infer' which will try to downcast to an appropriate\n",
      " |          equal type (e.g. float64 to int64 if possible)\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      reindex, asfreq\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      filled : DataFrame\n",
      " |  \n",
      " |  first_valid_index(self)\n",
      " |      Return label for first non-NA/null value\n",
      " |  \n",
      " |  floordiv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Integer division of dataframe and other, element-wise (binary operator `floordiv`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe // other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rfloordiv\n",
      " |  \n",
      " |  ge(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods ge\n",
      " |  \n",
      " |  get_value(self, index, col, takeable=False)\n",
      " |      Quickly retrieve single value at passed column and index\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : row label\n",
      " |      col : column label\n",
      " |      takeable : interpret the index/col as indexers, default False\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      value : scalar value\n",
      " |  \n",
      " |  gt(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods gt\n",
      " |  \n",
      " |  hist = hist_frame(data, column=None, by=None, grid=True, xlabelsize=None, xrot=None, ylabelsize=None, yrot=None, ax=None, sharex=False, sharey=False, figsize=None, layout=None, bins=10, **kwds)\n",
      " |      Draw histogram of the DataFrame's series using matplotlib / pylab.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      data : DataFrame\n",
      " |      column : string or sequence\n",
      " |          If passed, will be used to limit data to a subset of columns\n",
      " |      by : object, optional\n",
      " |          If passed, then used to form histograms for separate groups\n",
      " |      grid : boolean, default True\n",
      " |          Whether to show axis grid lines\n",
      " |      xlabelsize : int, default None\n",
      " |          If specified changes the x-axis label size\n",
      " |      xrot : float, default None\n",
      " |          rotation of x axis labels\n",
      " |      ylabelsize : int, default None\n",
      " |          If specified changes the y-axis label size\n",
      " |      yrot : float, default None\n",
      " |          rotation of y axis labels\n",
      " |      ax : matplotlib axes object, default None\n",
      " |      sharex : boolean, default True if ax is None else False\n",
      " |          In case subplots=True, share x axis and set some x axis labels to\n",
      " |          invisible; defaults to True if ax is None otherwise False if an ax\n",
      " |          is passed in; Be aware, that passing in both an ax and sharex=True\n",
      " |          will alter all x axis labels for all subplots in a figure!\n",
      " |      sharey : boolean, default False\n",
      " |          In case subplots=True, share y axis and set some y axis labels to\n",
      " |          invisible\n",
      " |      figsize : tuple\n",
      " |          The size of the figure to create in inches by default\n",
      " |      layout: (optional) a tuple (rows, columns) for the layout of the histograms\n",
      " |      bins: integer, default 10\n",
      " |          Number of histogram bins to be used\n",
      " |      kwds : other plotting keyword arguments\n",
      " |          To be passed to hist function\n",
      " |  \n",
      " |  icol(self, i)\n",
      " |      DEPRECATED. Use ``.iloc[:, i]`` instead\n",
      " |  \n",
      " |  idxmax(self, axis=0, skipna=True)\n",
      " |      Return index of first occurrence of maximum over requested axis.\n",
      " |      NA/null values are excluded.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          0 or 'index' for row-wise, 1 or 'columns' for column-wise\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be first index.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      idxmax : Series\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      This method is the DataFrame version of ``ndarray.argmax``.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      Series.idxmax\n",
      " |  \n",
      " |  idxmin(self, axis=0, skipna=True)\n",
      " |      Return index of first occurrence of minimum over requested axis.\n",
      " |      NA/null values are excluded.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          0 or 'index' for row-wise, 1 or 'columns' for column-wise\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      idxmin : Series\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      This method is the DataFrame version of ``ndarray.argmin``.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      Series.idxmin\n",
      " |  \n",
      " |  iget_value(self, i, j)\n",
      " |      DEPRECATED. Use ``.iat[i, j]`` instead\n",
      " |  \n",
      " |  info(self, verbose=None, buf=None, max_cols=None, memory_usage=None, null_counts=None)\n",
      " |      Concise summary of a DataFrame.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      verbose : {None, True, False}, optional\n",
      " |          Whether to print the full summary.\n",
      " |          None follows the `display.max_info_columns` setting.\n",
      " |          True or False overrides the `display.max_info_columns` setting.\n",
      " |      buf : writable buffer, defaults to sys.stdout\n",
      " |      max_cols : int, default None\n",
      " |          Determines whether full summary or short summary is printed.\n",
      " |          None follows the `display.max_info_columns` setting.\n",
      " |      memory_usage : boolean/string, default None\n",
      " |          Specifies whether total memory usage of the DataFrame\n",
      " |          elements (including index) should be displayed. None follows\n",
      " |          the `display.memory_usage` setting. True or False overrides\n",
      " |          the `display.memory_usage` setting. A value of 'deep' is equivalent\n",
      " |          of True, with deep introspection. Memory usage is shown in\n",
      " |          human-readable units (base-2 representation).\n",
      " |      null_counts : boolean, default None\n",
      " |          Whether to show the non-null counts\n",
      " |      \n",
      " |          - If None, then only show if the frame is smaller than\n",
      " |            max_info_rows and max_info_columns.\n",
      " |          - If True, always show counts.\n",
      " |          - If False, never show counts.\n",
      " |  \n",
      " |  insert(self, loc, column, value, allow_duplicates=False)\n",
      " |      Insert column into DataFrame at specified location.\n",
      " |      \n",
      " |      If `allow_duplicates` is False, raises Exception if column\n",
      " |      is already contained in the DataFrame.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      loc : int\n",
      " |          Must have 0 <= loc <= len(columns)\n",
      " |      column : object\n",
      " |      value : scalar, Series, or array-like\n",
      " |  \n",
      " |  irow(self, i, copy=False)\n",
      " |      DEPRECATED. Use ``.iloc[i]`` instead\n",
      " |  \n",
      " |  isin(self, values)\n",
      " |      Return boolean DataFrame showing whether each element in the\n",
      " |      DataFrame is contained in values.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      values : iterable, Series, DataFrame or dictionary\n",
      " |          The result will only be true at a location if all the\n",
      " |          labels match. If `values` is a Series, that's the index. If\n",
      " |          `values` is a dictionary, the keys must be the column names,\n",
      " |          which must match. If `values` is a DataFrame,\n",
      " |          then both the index and column labels must match.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      \n",
      " |      DataFrame of booleans\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      When ``values`` is a list:\n",
      " |      \n",
      " |      >>> df = DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})\n",
      " |      >>> df.isin([1, 3, 12, 'a'])\n",
      " |             A      B\n",
      " |      0   True   True\n",
      " |      1  False  False\n",
      " |      2   True  False\n",
      " |      \n",
      " |      When ``values`` is a dict:\n",
      " |      \n",
      " |      >>> df = DataFrame({'A': [1, 2, 3], 'B': [1, 4, 7]})\n",
      " |      >>> df.isin({'A': [1, 3], 'B': [4, 7, 12]})\n",
      " |             A      B\n",
      " |      0   True  False  # Note that B didn't match the 1 here.\n",
      " |      1  False   True\n",
      " |      2   True   True\n",
      " |      \n",
      " |      When ``values`` is a Series or DataFrame:\n",
      " |      \n",
      " |      >>> df = DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})\n",
      " |      >>> other = DataFrame({'A': [1, 3, 3, 2], 'B': ['e', 'f', 'f', 'e']})\n",
      " |      >>> df.isin(other)\n",
      " |             A      B\n",
      " |      0   True  False\n",
      " |      1  False  False  # Column A in `other` has a 3, but not at index 1.\n",
      " |      2   True   True\n",
      " |  \n",
      " |  items = iteritems(self)\n",
      " |  \n",
      " |  iteritems(self)\n",
      " |      Iterator over (column name, Series) pairs.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      iterrows : Iterate over DataFrame rows as (index, Series) pairs.\n",
      " |      itertuples : Iterate over DataFrame rows as namedtuples of the values.\n",
      " |  \n",
      " |  iterrows(self)\n",
      " |      Iterate over DataFrame rows as (index, Series) pairs.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      \n",
      " |      1. Because ``iterrows`` returns a Series for each row,\n",
      " |         it does **not** preserve dtypes across the rows (dtypes are\n",
      " |         preserved across columns for DataFrames). For example,\n",
      " |      \n",
      " |         >>> df = pd.DataFrame([[1, 1.5]], columns=['int', 'float'])\n",
      " |         >>> row = next(df.iterrows())[1]\n",
      " |         >>> row\n",
      " |         int      1.0\n",
      " |         float    1.5\n",
      " |         Name: 0, dtype: float64\n",
      " |         >>> print(row['int'].dtype)\n",
      " |         float64\n",
      " |         >>> print(df['int'].dtype)\n",
      " |         int64\n",
      " |      \n",
      " |         To preserve dtypes while iterating over the rows, it is better\n",
      " |         to use :meth:`itertuples` which returns namedtuples of the values\n",
      " |         and which is generally faster than ``iterrows``.\n",
      " |      \n",
      " |      2. You should **never modify** something you are iterating over.\n",
      " |         This is not guaranteed to work in all cases. Depending on the\n",
      " |         data types, the iterator returns a copy and not a view, and writing\n",
      " |         to it will have no effect.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      it : generator\n",
      " |          A generator that iterates over the rows of the frame.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      itertuples : Iterate over DataFrame rows as namedtuples of the values.\n",
      " |      iteritems : Iterate over (column name, Series) pairs.\n",
      " |  \n",
      " |  itertuples(self, index=True, name='Pandas')\n",
      " |      Iterate over DataFrame rows as namedtuples, with index value as first\n",
      " |      element of the tuple.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : boolean, default True\n",
      " |          If True, return the index as the first element of the tuple.\n",
      " |      name : string, default \"Pandas\"\n",
      " |          The name of the returned namedtuples or None to return regular\n",
      " |          tuples.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The column names will be renamed to positional names if they are\n",
      " |      invalid Python identifiers, repeated, or start with an underscore.\n",
      " |      With a large number of columns (>255), regular tuples are returned.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      iterrows : Iterate over DataFrame rows as (index, Series) pairs.\n",
      " |      iteritems : Iterate over (column name, Series) pairs.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'col1': [1, 2], 'col2': [0.1, 0.2]},\n",
      " |                            index=['a', 'b'])\n",
      " |      >>> df\n",
      " |         col1  col2\n",
      " |      a     1   0.1\n",
      " |      b     2   0.2\n",
      " |      >>> for row in df.itertuples():\n",
      " |      ...     print(row)\n",
      " |      ...\n",
      " |      Pandas(Index='a', col1=1, col2=0.10000000000000001)\n",
      " |      Pandas(Index='b', col1=2, col2=0.20000000000000001)\n",
      " |  \n",
      " |  join(self, other, on=None, how='left', lsuffix='', rsuffix='', sort=False)\n",
      " |      Join columns with other DataFrame either on index or on a key\n",
      " |      column. Efficiently Join multiple DataFrame objects by index at once by\n",
      " |      passing a list.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame, Series with name field set, or list of DataFrame\n",
      " |          Index should be similar to one of the columns in this one. If a\n",
      " |          Series is passed, its name attribute must be set, and that will be\n",
      " |          used as the column name in the resulting joined DataFrame\n",
      " |      on : column name, tuple/list of column names, or array-like\n",
      " |          Column(s) in the caller to join on the index in other,\n",
      " |          otherwise joins index-on-index. If multiples\n",
      " |          columns given, the passed DataFrame must have a MultiIndex. Can\n",
      " |          pass an array as the join key if not already contained in the\n",
      " |          calling DataFrame. Like an Excel VLOOKUP operation\n",
      " |      how : {'left', 'right', 'outer', 'inner'}, default: 'left'\n",
      " |          How to handle the operation of the two objects.\n",
      " |      \n",
      " |          * left: use calling frame's index (or column if on is specified)\n",
      " |          * right: use other frame's index\n",
      " |          * outer: form union of calling frame's index (or column if on is\n",
      " |              specified) with other frame's index\n",
      " |          * inner: form intersection of calling frame's index (or column if\n",
      " |              on is specified) with other frame's index\n",
      " |      lsuffix : string\n",
      " |          Suffix to use from left frame's overlapping columns\n",
      " |      rsuffix : string\n",
      " |          Suffix to use from right frame's overlapping columns\n",
      " |      sort : boolean, default False\n",
      " |          Order result DataFrame lexicographically by the join key. If False,\n",
      " |          preserves the index order of the calling (left) DataFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      on, lsuffix, and rsuffix options are not supported when passing a list\n",
      " |      of DataFrame objects\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> caller = pd.DataFrame({'key': ['K0', 'K1', 'K2', 'K3', 'K4', 'K5'],\n",
      " |      ...                        'A': ['A0', 'A1', 'A2', 'A3', 'A4', 'A5']})\n",
      " |      \n",
      " |      >>> caller\n",
      " |          A key\n",
      " |      0  A0  K0\n",
      " |      1  A1  K1\n",
      " |      2  A2  K2\n",
      " |      3  A3  K3\n",
      " |      4  A4  K4\n",
      " |      5  A5  K5\n",
      " |      \n",
      " |      >>> other = pd.DataFrame({'key': ['K0', 'K1', 'K2'],\n",
      " |      ...                       'B': ['B0', 'B1', 'B2']})\n",
      " |      \n",
      " |      >>> other\n",
      " |          B key\n",
      " |      0  B0  K0\n",
      " |      1  B1  K1\n",
      " |      2  B2  K2\n",
      " |      \n",
      " |      Join DataFrames using their indexes.\n",
      " |      \n",
      " |      >>> caller.join(other, lsuffix='_caller', rsuffix='_other')\n",
      " |      \n",
      " |      >>>     A key_caller    B key_other\n",
      " |          0  A0         K0   B0        K0\n",
      " |          1  A1         K1   B1        K1\n",
      " |          2  A2         K2   B2        K2\n",
      " |          3  A3         K3  NaN       NaN\n",
      " |          4  A4         K4  NaN       NaN\n",
      " |          5  A5         K5  NaN       NaN\n",
      " |      \n",
      " |      \n",
      " |      If we want to join using the key columns, we need to set key to be\n",
      " |      the index in both caller and other. The joined DataFrame will have\n",
      " |      key as its index.\n",
      " |      \n",
      " |      >>> caller.set_index('key').join(other.set_index('key'))\n",
      " |      \n",
      " |      >>>      A    B\n",
      " |          key\n",
      " |          K0   A0   B0\n",
      " |          K1   A1   B1\n",
      " |          K2   A2   B2\n",
      " |          K3   A3  NaN\n",
      " |          K4   A4  NaN\n",
      " |          K5   A5  NaN\n",
      " |      \n",
      " |      Another option to join using the key columns is to use the on\n",
      " |      parameter. DataFrame.join always uses other's index but we can use any\n",
      " |      column in the caller. This method preserves the original caller's\n",
      " |      index in the result.\n",
      " |      \n",
      " |      >>> caller.join(other.set_index('key'), on='key')\n",
      " |      \n",
      " |      >>>     A key    B\n",
      " |          0  A0  K0   B0\n",
      " |          1  A1  K1   B1\n",
      " |          2  A2  K2   B2\n",
      " |          3  A3  K3  NaN\n",
      " |          4  A4  K4  NaN\n",
      " |          5  A5  K5  NaN\n",
      " |      \n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.merge : For column(s)-on-columns(s) operations\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      joined : DataFrame\n",
      " |  \n",
      " |  kurt(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return unbiased kurtosis over requested axis using Fisher's definition of\n",
      " |      kurtosis (kurtosis of normal == 0.0). Normalized by N-1\n",
      " |      \n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      kurt : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  kurtosis = kurt(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |  \n",
      " |  last_valid_index(self)\n",
      " |      Return label for last non-NA/null value\n",
      " |  \n",
      " |  le(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods le\n",
      " |  \n",
      " |  lookup(self, row_labels, col_labels)\n",
      " |      Label-based \"fancy indexing\" function for DataFrame.\n",
      " |      Given equal-length arrays of row and column labels, return an\n",
      " |      array of the values corresponding to each (row, col) pair.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      row_labels : sequence\n",
      " |          The row labels to use for lookup\n",
      " |      col_labels : sequence\n",
      " |          The column labels to use for lookup\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Akin to::\n",
      " |      \n",
      " |          result = []\n",
      " |          for row, col in zip(row_labels, col_labels):\n",
      " |              result.append(df.get_value(row, col))\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      values : ndarray\n",
      " |          The found values\n",
      " |  \n",
      " |  lt(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods lt\n",
      " |  \n",
      " |  mad(self, axis=None, skipna=None, level=None)\n",
      " |      Return the mean absolute deviation of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      mad : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  max(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      This method returns the maximum of the values in the object.\n",
      " |                  If you want the *index* of the maximum, use ``idxmax``. This is\n",
      " |                  the equivalent of the ``numpy.ndarray`` method ``argmax``.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      max : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  mean(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return the mean of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      mean : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  median(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return the median of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      median : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  memory_usage(self, index=True, deep=False)\n",
      " |      Memory usage of DataFrame columns.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : bool\n",
      " |          Specifies whether to include memory usage of DataFrame's\n",
      " |          index in returned Series. If `index=True` (default is False)\n",
      " |          the first index of the Series is `Index`.\n",
      " |      deep : bool\n",
      " |          Introspect the data deeply, interrogate\n",
      " |          `object` dtypes for system-level memory consumption\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sizes : Series\n",
      " |          A series with column names as index and memory usage of\n",
      " |          columns with units of bytes.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Memory usage does not include memory consumed by elements that\n",
      " |      are not components of the array if deep=False\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      numpy.ndarray.nbytes\n",
      " |  \n",
      " |  merge(self, right, how='inner', on=None, left_on=None, right_on=None, left_index=False, right_index=False, sort=False, suffixes=('_x', '_y'), copy=True, indicator=False)\n",
      " |      Merge DataFrame objects by performing a database-style join operation by\n",
      " |      columns or indexes.\n",
      " |      \n",
      " |      If joining columns on columns, the DataFrame indexes *will be\n",
      " |      ignored*. Otherwise if joining indexes on indexes or indexes on a column or\n",
      " |      columns, the index will be passed on.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      right : DataFrame\n",
      " |      how : {'left', 'right', 'outer', 'inner'}, default 'inner'\n",
      " |          * left: use only keys from left frame (SQL: left outer join)\n",
      " |          * right: use only keys from right frame (SQL: right outer join)\n",
      " |          * outer: use union of keys from both frames (SQL: full outer join)\n",
      " |          * inner: use intersection of keys from both frames (SQL: inner join)\n",
      " |      on : label or list\n",
      " |          Field names to join on. Must be found in both DataFrames. If on is\n",
      " |          None and not merging on indexes, then it merges on the intersection of\n",
      " |          the columns by default.\n",
      " |      left_on : label or list, or array-like\n",
      " |          Field names to join on in left DataFrame. Can be a vector or list of\n",
      " |          vectors of the length of the DataFrame to use a particular vector as\n",
      " |          the join key instead of columns\n",
      " |      right_on : label or list, or array-like\n",
      " |          Field names to join on in right DataFrame or vector/list of vectors per\n",
      " |          left_on docs\n",
      " |      left_index : boolean, default False\n",
      " |          Use the index from the left DataFrame as the join key(s). If it is a\n",
      " |          MultiIndex, the number of keys in the other DataFrame (either the index\n",
      " |          or a number of columns) must match the number of levels\n",
      " |      right_index : boolean, default False\n",
      " |          Use the index from the right DataFrame as the join key. Same caveats as\n",
      " |          left_index\n",
      " |      sort : boolean, default False\n",
      " |          Sort the join keys lexicographically in the result DataFrame\n",
      " |      suffixes : 2-length sequence (tuple, list, ...)\n",
      " |          Suffix to apply to overlapping column names in the left and right\n",
      " |          side, respectively\n",
      " |      copy : boolean, default True\n",
      " |          If False, do not copy data unnecessarily\n",
      " |      indicator : boolean or string, default False\n",
      " |          If True, adds a column to output DataFrame called \"_merge\" with\n",
      " |          information on the source of each row.\n",
      " |          If string, column with information on source of each row will be added to\n",
      " |          output DataFrame, and column will be named value of string.\n",
      " |          Information column is Categorical-type and takes on a value of \"left_only\"\n",
      " |          for observations whose merge key only appears in 'left' DataFrame,\n",
      " |          \"right_only\" for observations whose merge key only appears in 'right'\n",
      " |          DataFrame, and \"both\" if the observation's merge key is found in both.\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> A              >>> B\n",
      " |          lkey value         rkey value\n",
      " |      0   foo  1         0   foo  5\n",
      " |      1   bar  2         1   bar  6\n",
      " |      2   baz  3         2   qux  7\n",
      " |      3   foo  4         3   bar  8\n",
      " |      \n",
      " |      >>> A.merge(B, left_on='lkey', right_on='rkey', how='outer')\n",
      " |         lkey  value_x  rkey  value_y\n",
      " |      0  foo   1        foo   5\n",
      " |      1  foo   4        foo   5\n",
      " |      2  bar   2        bar   6\n",
      " |      3  bar   2        bar   8\n",
      " |      4  baz   3        NaN   NaN\n",
      " |      5  NaN   NaN      qux   7\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      merged : DataFrame\n",
      " |          The output type will the be same as 'left', if it is a subclass\n",
      " |          of DataFrame.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      merge_ordered\n",
      " |      merge_asof\n",
      " |  \n",
      " |  min(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      This method returns the minimum of the values in the object.\n",
      " |                  If you want the *index* of the minimum, use ``idxmin``. This is\n",
      " |                  the equivalent of the ``numpy.ndarray`` method ``argmin``.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      min : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  mod(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Modulo of dataframe and other, element-wise (binary operator `mod`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe % other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rmod\n",
      " |  \n",
      " |  mode(self, axis=0, numeric_only=False)\n",
      " |      Gets the mode(s) of each element along the axis selected. Empty if\n",
      " |      nothing has 2+ occurrences. Adds a row for each mode per label, fills\n",
      " |      in gaps with nan.\n",
      " |      \n",
      " |      Note that there could be multiple values returned for the selected\n",
      " |      axis (when more than one item share the maximum frequency), which is\n",
      " |      the reason why a dataframe is returned. If you want to impute missing\n",
      " |      values with the mode in a dataframe ``df``, you can just do this:\n",
      " |      ``df.fillna(df.mode().iloc[0])``\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          * 0 or 'index' : get mode of each column\n",
      " |          * 1 or 'columns' : get mode of each row\n",
      " |      numeric_only : boolean, default False\n",
      " |          if True, only apply to numeric columns\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      modes : DataFrame (sorted)\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = pd.DataFrame({'A': [1, 2, 1, 2, 1, 2, 3]})\n",
      " |      >>> df.mode()\n",
      " |         A\n",
      " |      0  1\n",
      " |      1  2\n",
      " |  \n",
      " |  mul(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Multiplication of dataframe and other, element-wise (binary operator `mul`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe * other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rmul\n",
      " |  \n",
      " |  multiply = mul(self, other, axis='columns', level=None, fill_value=None)\n",
      " |  \n",
      " |  ne(self, other, axis='columns', level=None)\n",
      " |      Wrapper for flexible comparison methods ne\n",
      " |  \n",
      " |  nlargest(self, n, columns, keep='first')\n",
      " |      Get the rows of a DataFrame sorted by the `n` largest\n",
      " |      values of `columns`.\n",
      " |      \n",
      " |      .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      n : int\n",
      " |          Number of items to retrieve\n",
      " |      columns : list or str\n",
      " |          Column name or names to order by\n",
      " |      keep : {'first', 'last', False}, default 'first'\n",
      " |          Where there are duplicate values:\n",
      " |          - ``first`` : take the first occurrence.\n",
      " |          - ``last`` : take the last occurrence.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = DataFrame({'a': [1, 10, 8, 11, -1],\n",
      " |      ...                 'b': list('abdce'),\n",
      " |      ...                 'c': [1.0, 2.0, np.nan, 3.0, 4.0]})\n",
      " |      >>> df.nlargest(3, 'a')\n",
      " |          a  b   c\n",
      " |      3  11  c   3\n",
      " |      1  10  b   2\n",
      " |      2   8  d NaN\n",
      " |  \n",
      " |  nsmallest(self, n, columns, keep='first')\n",
      " |      Get the rows of a DataFrame sorted by the `n` smallest\n",
      " |      values of `columns`.\n",
      " |      \n",
      " |      .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      n : int\n",
      " |          Number of items to retrieve\n",
      " |      columns : list or str\n",
      " |          Column name or names to order by\n",
      " |      keep : {'first', 'last', False}, default 'first'\n",
      " |          Where there are duplicate values:\n",
      " |          - ``first`` : take the first occurrence.\n",
      " |          - ``last`` : take the last occurrence.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = DataFrame({'a': [1, 10, 8, 11, -1],\n",
      " |      ...                 'b': list('abdce'),\n",
      " |      ...                 'c': [1.0, 2.0, np.nan, 3.0, 4.0]})\n",
      " |      >>> df.nsmallest(3, 'a')\n",
      " |         a  b   c\n",
      " |      4 -1  e   4\n",
      " |      0  1  a   1\n",
      " |      2  8  d NaN\n",
      " |  \n",
      " |  pivot(self, index=None, columns=None, values=None)\n",
      " |      Reshape data (produce a \"pivot\" table) based on column values. Uses\n",
      " |      unique values from index / columns to form axes of the resulting\n",
      " |      DataFrame.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : string or object, optional\n",
      " |          Column name to use to make new frame's index. If None, uses\n",
      " |          existing index.\n",
      " |      columns : string or object\n",
      " |          Column name to use to make new frame's columns\n",
      " |      values : string or object, optional\n",
      " |          Column name to use for populating new frame's values. If not\n",
      " |          specified, all remaining columns will be used and the result will\n",
      " |          have hierarchically indexed columns\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      pivoted : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.pivot_table : generalization of pivot that can handle\n",
      " |          duplicate values for one index/column pair\n",
      " |      DataFrame.unstack : pivot based on the index values instead of a\n",
      " |          column\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      For finer-tuned control, see hierarchical indexing documentation along\n",
      " |      with the related stack/unstack methods\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'foo': ['one','one','one','two','two','two'],\n",
      " |                             'bar': ['A', 'B', 'C', 'A', 'B', 'C'],\n",
      " |                             'baz': [1, 2, 3, 4, 5, 6]})\n",
      " |      >>> df\n",
      " |          foo   bar  baz\n",
      " |      0   one   A    1\n",
      " |      1   one   B    2\n",
      " |      2   one   C    3\n",
      " |      3   two   A    4\n",
      " |      4   two   B    5\n",
      " |      5   two   C    6\n",
      " |      \n",
      " |      >>> df.pivot(index='foo', columns='bar', values='baz')\n",
      " |           A   B   C\n",
      " |      one  1   2   3\n",
      " |      two  4   5   6\n",
      " |      \n",
      " |      >>> df.pivot(index='foo', columns='bar')['baz']\n",
      " |           A   B   C\n",
      " |      one  1   2   3\n",
      " |      two  4   5   6\n",
      " |  \n",
      " |  pivot_table(data, values=None, index=None, columns=None, aggfunc='mean', fill_value=None, margins=False, dropna=True, margins_name='All')\n",
      " |      Create a spreadsheet-style pivot table as a DataFrame. The levels in the\n",
      " |      pivot table will be stored in MultiIndex objects (hierarchical indexes) on\n",
      " |      the index and columns of the result DataFrame\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      data : DataFrame\n",
      " |      values : column to aggregate, optional\n",
      " |      index : column, Grouper, array, or list of the previous\n",
      " |          If an array is passed, it must be the same length as the data. The list\n",
      " |          can contain any of the other types (except list).\n",
      " |          Keys to group by on the pivot table index.  If an array is passed, it\n",
      " |          is being used as the same manner as column values.\n",
      " |      columns : column, Grouper, array, or list of the previous\n",
      " |          If an array is passed, it must be the same length as the data. The list\n",
      " |          can contain any of the other types (except list).\n",
      " |          Keys to group by on the pivot table column.  If an array is passed, it\n",
      " |          is being used as the same manner as column values.\n",
      " |      aggfunc : function or list of functions, default numpy.mean\n",
      " |          If list of functions passed, the resulting pivot table will have\n",
      " |          hierarchical columns whose top level are the function names (inferred\n",
      " |          from the function objects themselves)\n",
      " |      fill_value : scalar, default None\n",
      " |          Value to replace missing values with\n",
      " |      margins : boolean, default False\n",
      " |          Add all row / columns (e.g. for subtotal / grand totals)\n",
      " |      dropna : boolean, default True\n",
      " |          Do not include columns whose entries are all NaN\n",
      " |      margins_name : string, default 'All'\n",
      " |          Name of the row / column that will contain the totals\n",
      " |          when margins is True.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df\n",
      " |         A   B   C      D\n",
      " |      0  foo one small  1\n",
      " |      1  foo one large  2\n",
      " |      2  foo one large  2\n",
      " |      3  foo two small  3\n",
      " |      4  foo two small  3\n",
      " |      5  bar one large  4\n",
      " |      6  bar one small  5\n",
      " |      7  bar two small  6\n",
      " |      8  bar two large  7\n",
      " |      \n",
      " |      >>> table = pivot_table(df, values='D', index=['A', 'B'],\n",
      " |      ...                     columns=['C'], aggfunc=np.sum)\n",
      " |      >>> table\n",
      " |                small  large\n",
      " |      foo  one  1      4\n",
      " |           two  6      NaN\n",
      " |      bar  one  5      4\n",
      " |           two  6      7\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      table : DataFrame\n",
      " |  \n",
      " |  pow(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Exponential power of dataframe and other, element-wise (binary operator `pow`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe ** other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rpow\n",
      " |  \n",
      " |  prod(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return the product of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      prod : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  product = prod(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |  \n",
      " |  quantile(self, q=0.5, axis=0, numeric_only=True, interpolation='linear')\n",
      " |      Return values at the given quantile over requested axis, a la\n",
      " |      numpy.percentile.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      q : float or array-like, default 0.5 (50% quantile)\n",
      " |          0 <= q <= 1, the quantile(s) to compute\n",
      " |      axis : {0, 1, 'index', 'columns'} (default 0)\n",
      " |          0 or 'index' for row-wise, 1 or 'columns' for column-wise\n",
      " |      interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}\n",
      " |          .. versionadded:: 0.18.0\n",
      " |      \n",
      " |          This optional parameter specifies the interpolation method to use,\n",
      " |          when the desired quantile lies between two data points `i` and `j`:\n",
      " |      \n",
      " |          * linear: `i + (j - i) * fraction`, where `fraction` is the\n",
      " |            fractional part of the index surrounded by `i` and `j`.\n",
      " |          * lower: `i`.\n",
      " |          * higher: `j`.\n",
      " |          * nearest: `i` or `j` whichever is nearest.\n",
      " |          * midpoint: (`i` + `j`) / 2.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      quantiles : Series or DataFrame\n",
      " |      \n",
      " |          - If ``q`` is an array, a DataFrame will be returned where the\n",
      " |            index is ``q``, the columns are the columns of self, and the\n",
      " |            values are the quantiles.\n",
      " |          - If ``q`` is a float, a Series will be returned where the\n",
      " |            index is the columns of self and the values are the quantiles.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = DataFrame(np.array([[1, 1], [2, 10], [3, 100], [4, 100]]),\n",
      " |                         columns=['a', 'b'])\n",
      " |      >>> df.quantile(.1)\n",
      " |      a    1.3\n",
      " |      b    3.7\n",
      " |      dtype: float64\n",
      " |      >>> df.quantile([.1, .5])\n",
      " |             a     b\n",
      " |      0.1  1.3   3.7\n",
      " |      0.5  2.5  55.0\n",
      " |  \n",
      " |  query(self, expr, inplace=False, **kwargs)\n",
      " |      Query the columns of a frame with a boolean expression.\n",
      " |      \n",
      " |      .. versionadded:: 0.13\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      expr : string\n",
      " |          The query string to evaluate.  You can refer to variables\n",
      " |          in the environment by prefixing them with an '@' character like\n",
      " |          ``@a + b``.\n",
      " |      inplace : bool\n",
      " |          Whether the query should modify the data in place or return\n",
      " |          a modified copy\n",
      " |      \n",
      " |          .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      kwargs : dict\n",
      " |          See the documentation for :func:`pandas.eval` for complete details\n",
      " |          on the keyword arguments accepted by :meth:`DataFrame.query`.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      q : DataFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The result of the evaluation of this expression is first passed to\n",
      " |      :attr:`DataFrame.loc` and if that fails because of a\n",
      " |      multidimensional key (e.g., a DataFrame) then the result will be passed\n",
      " |      to :meth:`DataFrame.__getitem__`.\n",
      " |      \n",
      " |      This method uses the top-level :func:`pandas.eval` function to\n",
      " |      evaluate the passed query.\n",
      " |      \n",
      " |      The :meth:`~pandas.DataFrame.query` method uses a slightly\n",
      " |      modified Python syntax by default. For example, the ``&`` and ``|``\n",
      " |      (bitwise) operators have the precedence of their boolean cousins,\n",
      " |      :keyword:`and` and :keyword:`or`. This *is* syntactically valid Python,\n",
      " |      however the semantics are different.\n",
      " |      \n",
      " |      You can change the semantics of the expression by passing the keyword\n",
      " |      argument ``parser='python'``. This enforces the same semantics as\n",
      " |      evaluation in Python space. Likewise, you can pass ``engine='python'``\n",
      " |      to evaluate an expression using Python itself as a backend. This is not\n",
      " |      recommended as it is inefficient compared to using ``numexpr`` as the\n",
      " |      engine.\n",
      " |      \n",
      " |      The :attr:`DataFrame.index` and\n",
      " |      :attr:`DataFrame.columns` attributes of the\n",
      " |      :class:`~pandas.DataFrame` instance are placed in the query namespace\n",
      " |      by default, which allows you to treat both the index and columns of the\n",
      " |      frame as a column in the frame.\n",
      " |      The identifier ``index`` is used for the frame index; you can also\n",
      " |      use the name of the index to identify it in a query.\n",
      " |      \n",
      " |      For further details and examples see the ``query`` documentation in\n",
      " |      :ref:`indexing <indexing.query>`.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.eval\n",
      " |      DataFrame.eval\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> from numpy.random import randn\n",
      " |      >>> from pandas import DataFrame\n",
      " |      >>> df = DataFrame(randn(10, 2), columns=list('ab'))\n",
      " |      >>> df.query('a > b')\n",
      " |      >>> df[df.a > df.b]  # same result as the previous expression\n",
      " |  \n",
      " |  radd(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Addition of dataframe and other, element-wise (binary operator `radd`).\n",
      " |      \n",
      " |      Equivalent to ``other + dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.add\n",
      " |  \n",
      " |  rdiv = rtruediv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |  \n",
      " |  reindex(self, index=None, columns=None, **kwargs)\n",
      " |      Conform DataFrame to new index with optional filling logic, placing\n",
      " |      NA/NaN in locations having no value in the previous index. A new object\n",
      " |      is produced unless the new index is equivalent to the current one and\n",
      " |      copy=False\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index, columns : array-like, optional (can be specified in order, or as\n",
      " |          keywords)\n",
      " |          New labels / index to conform to. Preferably an Index object to\n",
      " |          avoid duplicating data\n",
      " |      method : {None, 'backfill'/'bfill', 'pad'/'ffill', 'nearest'}, optional\n",
      " |          method to use for filling holes in reindexed DataFrame.\n",
      " |          Please note: this is only  applicable to DataFrames/Series with a\n",
      " |          monotonically increasing/decreasing index.\n",
      " |      \n",
      " |          * default: don't fill gaps\n",
      " |          * pad / ffill: propagate last valid observation forward to next\n",
      " |            valid\n",
      " |          * backfill / bfill: use next valid observation to fill gap\n",
      " |          * nearest: use nearest valid observations to fill gap\n",
      " |      \n",
      " |      copy : boolean, default True\n",
      " |          Return a new object, even if the passed indexes are the same\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      fill_value : scalar, default np.NaN\n",
      " |          Value to use for missing values. Defaults to NaN, but can be any\n",
      " |          \"compatible\" value\n",
      " |      limit : int, default None\n",
      " |          Maximum number of consecutive elements to forward or backward fill\n",
      " |      tolerance : optional\n",
      " |          Maximum distance between original and new labels for inexact\n",
      " |          matches. The values of the index at the matching locations most\n",
      " |          satisfy the equation ``abs(index[indexer] - target) <= tolerance``.\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      Create a dataframe with some fictional data.\n",
      " |      \n",
      " |      >>> index = ['Firefox', 'Chrome', 'Safari', 'IE10', 'Konqueror']\n",
      " |      >>> df = pd.DataFrame({\n",
      " |      ...      'http_status': [200,200,404,404,301],\n",
      " |      ...      'response_time': [0.04, 0.02, 0.07, 0.08, 1.0]},\n",
      " |      ...       index=index)\n",
      " |      >>> df\n",
      " |                  http_status  response_time\n",
      " |      Firefox            200           0.04\n",
      " |      Chrome             200           0.02\n",
      " |      Safari             404           0.07\n",
      " |      IE10               404           0.08\n",
      " |      Konqueror          301           1.00\n",
      " |      \n",
      " |      Create a new index and reindex the dataframe. By default\n",
      " |      values in the new index that do not have corresponding\n",
      " |      records in the dataframe are assigned ``NaN``.\n",
      " |      \n",
      " |      >>> new_index= ['Safari', 'Iceweasel', 'Comodo Dragon', 'IE10',\n",
      " |      ...             'Chrome']\n",
      " |      >>> df.reindex(new_index)\n",
      " |                     http_status  response_time\n",
      " |      Safari                 404           0.07\n",
      " |      Iceweasel              NaN            NaN\n",
      " |      Comodo Dragon          NaN            NaN\n",
      " |      IE10                   404           0.08\n",
      " |      Chrome                 200           0.02\n",
      " |      \n",
      " |      We can fill in the missing values by passing a value to\n",
      " |      the keyword ``fill_value``. Because the index is not monotonically\n",
      " |      increasing or decreasing, we cannot use arguments to the keyword\n",
      " |      ``method`` to fill the ``NaN`` values.\n",
      " |      \n",
      " |      >>> df.reindex(new_index, fill_value=0)\n",
      " |                     http_status  response_time\n",
      " |      Safari                 404           0.07\n",
      " |      Iceweasel                0           0.00\n",
      " |      Comodo Dragon            0           0.00\n",
      " |      IE10                   404           0.08\n",
      " |      Chrome                 200           0.02\n",
      " |      \n",
      " |      >>> df.reindex(new_index, fill_value='missing')\n",
      " |                    http_status response_time\n",
      " |      Safari                404          0.07\n",
      " |      Iceweasel         missing       missing\n",
      " |      Comodo Dragon     missing       missing\n",
      " |      IE10                  404          0.08\n",
      " |      Chrome                200          0.02\n",
      " |      \n",
      " |      To further illustrate the filling functionality in\n",
      " |      ``reindex``, we will create a dataframe with a\n",
      " |      monotonically increasing index (for example, a sequence\n",
      " |      of dates).\n",
      " |      \n",
      " |      >>> date_index = pd.date_range('1/1/2010', periods=6, freq='D')\n",
      " |      >>> df2 = pd.DataFrame({\"prices\": [100, 101, np.nan, 100, 89, 88]},\n",
      " |      ...                    index=date_index)\n",
      " |      >>> df2\n",
      " |                  prices\n",
      " |      2010-01-01     100\n",
      " |      2010-01-02     101\n",
      " |      2010-01-03     NaN\n",
      " |      2010-01-04     100\n",
      " |      2010-01-05      89\n",
      " |      2010-01-06      88\n",
      " |      \n",
      " |      Suppose we decide to expand the dataframe to cover a wider\n",
      " |      date range.\n",
      " |      \n",
      " |      >>> date_index2 = pd.date_range('12/29/2009', periods=10, freq='D')\n",
      " |      >>> df2.reindex(date_index2)\n",
      " |                  prices\n",
      " |      2009-12-29     NaN\n",
      " |      2009-12-30     NaN\n",
      " |      2009-12-31     NaN\n",
      " |      2010-01-01     100\n",
      " |      2010-01-02     101\n",
      " |      2010-01-03     NaN\n",
      " |      2010-01-04     100\n",
      " |      2010-01-05      89\n",
      " |      2010-01-06      88\n",
      " |      2010-01-07     NaN\n",
      " |      \n",
      " |      The index entries that did not have a value in the original data frame\n",
      " |      (for example, '2009-12-29') are by default filled with ``NaN``.\n",
      " |      If desired, we can fill in the missing values using one of several\n",
      " |      options.\n",
      " |      \n",
      " |      For example, to backpropagate the last valid value to fill the ``NaN``\n",
      " |      values, pass ``bfill`` as an argument to the ``method`` keyword.\n",
      " |      \n",
      " |      >>> df2.reindex(date_index2, method='bfill')\n",
      " |                  prices\n",
      " |      2009-12-29     100\n",
      " |      2009-12-30     100\n",
      " |      2009-12-31     100\n",
      " |      2010-01-01     100\n",
      " |      2010-01-02     101\n",
      " |      2010-01-03     NaN\n",
      " |      2010-01-04     100\n",
      " |      2010-01-05      89\n",
      " |      2010-01-06      88\n",
      " |      2010-01-07     NaN\n",
      " |      \n",
      " |      Please note that the ``NaN`` value present in the original dataframe\n",
      " |      (at index value 2010-01-03) will not be filled by any of the\n",
      " |      value propagation schemes. This is because filling while reindexing\n",
      " |      does not look at dataframe values, but only compares the original and\n",
      " |      desired indexes. If you do want to fill in the ``NaN`` values present\n",
      " |      in the original dataframe, use the ``fillna()`` method.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      reindexed : DataFrame\n",
      " |  \n",
      " |  reindex_axis(self, labels, axis=0, method=None, level=None, copy=True, limit=None, fill_value=nan)\n",
      " |      Conform input object to new index with optional\n",
      " |      filling logic, placing NA/NaN in locations having no value in the\n",
      " |      previous index. A new object is produced unless the new index is\n",
      " |      equivalent to the current one and copy=False\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      labels : array-like\n",
      " |          New labels / index to conform to. Preferably an Index object to\n",
      " |          avoid duplicating data\n",
      " |      axis : {0 or 'index', 1 or 'columns'}\n",
      " |      method : {None, 'backfill'/'bfill', 'pad'/'ffill', 'nearest'}, optional\n",
      " |          Method to use for filling holes in reindexed DataFrame:\n",
      " |      \n",
      " |          * default: don't fill gaps\n",
      " |          * pad / ffill: propagate last valid observation forward to next\n",
      " |            valid\n",
      " |          * backfill / bfill: use next valid observation to fill gap\n",
      " |          * nearest: use nearest valid observations to fill gap\n",
      " |      \n",
      " |      copy : boolean, default True\n",
      " |          Return a new object, even if the passed indexes are the same\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      limit : int, default None\n",
      " |          Maximum number of consecutive elements to forward or backward fill\n",
      " |      tolerance : optional\n",
      " |          Maximum distance between original and new labels for inexact\n",
      " |          matches. The values of the index at the matching locations most\n",
      " |          satisfy the equation ``abs(index[indexer] - target) <= tolerance``.\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df.reindex_axis(['A', 'B', 'C'], axis=1)\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      reindex, reindex_like\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      reindexed : DataFrame\n",
      " |  \n",
      " |  rename(self, index=None, columns=None, **kwargs)\n",
      " |      Alter axes input function or functions. Function / dict values must be\n",
      " |      unique (1-to-1). Labels not contained in a dict / Series will be left\n",
      " |      as-is. Extra labels listed don't throw an error. Alternatively, change\n",
      " |      ``Series.name`` with a scalar value (Series only).\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index, columns : scalar, list-like, dict-like or function, optional\n",
      " |          Scalar or list-like will alter the ``Series.name`` attribute,\n",
      " |          and raise on DataFrame or Panel.\n",
      " |          dict-like or functions are transformations to apply to\n",
      " |          that axis' values\n",
      " |      copy : boolean, default True\n",
      " |          Also copy underlying data\n",
      " |      inplace : boolean, default False\n",
      " |          Whether to return a new DataFrame. If True then value of copy is\n",
      " |          ignored.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      renamed : DataFrame (new object)\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.NDFrame.rename_axis\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> s = pd.Series([1, 2, 3])\n",
      " |      >>> s\n",
      " |      0    1\n",
      " |      1    2\n",
      " |      2    3\n",
      " |      dtype: int64\n",
      " |      >>> s.rename(\"my_name\") # scalar, changes Series.name\n",
      " |      0    1\n",
      " |      1    2\n",
      " |      2    3\n",
      " |      Name: my_name, dtype: int64\n",
      " |      >>> s.rename(lambda x: x ** 2)  # function, changes labels\n",
      " |      0    1\n",
      " |      1    2\n",
      " |      4    3\n",
      " |      dtype: int64\n",
      " |      >>> s.rename({1: 3, 2: 5})  # mapping, changes labels\n",
      " |      0    1\n",
      " |      3    2\n",
      " |      5    3\n",
      " |      dtype: int64\n",
      " |      >>> df = pd.DataFrame({\"A\": [1, 2, 3], \"B\": [4, 5, 6]})\n",
      " |      >>> df.rename(2)\n",
      " |      ...\n",
      " |      TypeError: 'int' object is not callable\n",
      " |      >>> df.rename(index=str, columns={\"A\": \"a\", \"B\": \"c\"})\n",
      " |         a  c\n",
      " |      0  1  4\n",
      " |      1  2  5\n",
      " |      2  3  6\n",
      " |      >>> df.rename(index=str, columns={\"A\": \"a\", \"C\": \"c\"})\n",
      " |         a  B\n",
      " |      0  1  4\n",
      " |      1  2  5\n",
      " |      2  3  6\n",
      " |  \n",
      " |  reorder_levels(self, order, axis=0)\n",
      " |      Rearrange index levels using input order.\n",
      " |      May not drop or duplicate levels\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      order : list of int or list of str\n",
      " |          List representing new level order. Reference level by number\n",
      " |          (position) or by key (label).\n",
      " |      axis : int\n",
      " |          Where to reorder levels.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      type of caller (new object)\n",
      " |  \n",
      " |  reset_index(self, level=None, drop=False, inplace=False, col_level=0, col_fill='')\n",
      " |      For DataFrame with multi-level index, return new DataFrame with\n",
      " |      labeling information in the columns under the index names, defaulting\n",
      " |      to 'level_0', 'level_1', etc. if any are None. For a standard index,\n",
      " |      the index name will be used (if set), otherwise a default 'index' or\n",
      " |      'level_0' (if 'index' is already taken) will be used.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      level : int, str, tuple, or list, default None\n",
      " |          Only remove the given levels from the index. Removes all levels by\n",
      " |          default\n",
      " |      drop : boolean, default False\n",
      " |          Do not try to insert index into dataframe columns. This resets\n",
      " |          the index to the default integer index.\n",
      " |      inplace : boolean, default False\n",
      " |          Modify the DataFrame in place (do not create a new object)\n",
      " |      col_level : int or str, default 0\n",
      " |          If the columns have multiple levels, determines which level the\n",
      " |          labels are inserted into. By default it is inserted into the first\n",
      " |          level.\n",
      " |      col_fill : object, default ''\n",
      " |          If the columns have multiple levels, determines how the other\n",
      " |          levels are named. If None then the index name is repeated.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      resetted : DataFrame\n",
      " |  \n",
      " |  rfloordiv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Integer division of dataframe and other, element-wise (binary operator `rfloordiv`).\n",
      " |      \n",
      " |      Equivalent to ``other // dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.floordiv\n",
      " |  \n",
      " |  rmod(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Modulo of dataframe and other, element-wise (binary operator `rmod`).\n",
      " |      \n",
      " |      Equivalent to ``other % dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.mod\n",
      " |  \n",
      " |  rmul(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Multiplication of dataframe and other, element-wise (binary operator `rmul`).\n",
      " |      \n",
      " |      Equivalent to ``other * dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.mul\n",
      " |  \n",
      " |  rolling(self, window, min_periods=None, freq=None, center=False, win_type=None, on=None, axis=0)\n",
      " |      Provides rolling window calculcations.\n",
      " |      \n",
      " |      .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      window : int, or offset\n",
      " |          Size of the moving window. This is the number of observations used for\n",
      " |          calculating the statistic. Each window will be a fixed size.\n",
      " |      \n",
      " |          If its an offset then this will be the time period of each window. Each\n",
      " |          window will be a variable sized based on the observations included in\n",
      " |          the time-period. This is only valid for datetimelike indexes. This is\n",
      " |          new in 0.19.0\n",
      " |      min_periods : int, default None\n",
      " |          Minimum number of observations in window required to have a value\n",
      " |          (otherwise result is NA). For a window that is specified by an offset,\n",
      " |          this will default to 1.\n",
      " |      freq : string or DateOffset object, optional (default None) (DEPRECATED)\n",
      " |          Frequency to conform the data to before computing the statistic.\n",
      " |          Specified as a frequency string or DateOffset object.\n",
      " |      center : boolean, default False\n",
      " |          Set the labels at the center of the window.\n",
      " |      win_type : string, default None\n",
      " |          Provide a window type. See the notes below.\n",
      " |      on : string, optional\n",
      " |          For a DataFrame, column on which to calculate\n",
      " |          the rolling window, rather than the index\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      axis : int or string, default 0\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      a Window or Rolling sub-classed for the particular operation\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'B': [0, 1, 2, np.nan, 4]})\n",
      " |      >>> df\n",
      " |           B\n",
      " |      0  0.0\n",
      " |      1  1.0\n",
      " |      2  2.0\n",
      " |      3  NaN\n",
      " |      4  4.0\n",
      " |      \n",
      " |      Rolling sum with a window length of 2, using the 'triang'\n",
      " |      window type.\n",
      " |      \n",
      " |      >>> df.rolling(2, win_type='triang').sum()\n",
      " |           B\n",
      " |      0  NaN\n",
      " |      1  1.0\n",
      " |      2  2.5\n",
      " |      3  NaN\n",
      " |      4  NaN\n",
      " |      \n",
      " |      Rolling sum with a window length of 2, min_periods defaults\n",
      " |      to the window length.\n",
      " |      \n",
      " |      >>> df.rolling(2).sum()\n",
      " |           B\n",
      " |      0  NaN\n",
      " |      1  1.0\n",
      " |      2  3.0\n",
      " |      3  NaN\n",
      " |      4  NaN\n",
      " |      \n",
      " |      Same as above, but explicity set the min_periods\n",
      " |      \n",
      " |      >>> df.rolling(2, min_periods=1).sum()\n",
      " |           B\n",
      " |      0  0.0\n",
      " |      1  1.0\n",
      " |      2  3.0\n",
      " |      3  2.0\n",
      " |      4  4.0\n",
      " |      \n",
      " |      A ragged (meaning not-a-regular frequency), time-indexed DataFrame\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'B': [0, 1, 2, np.nan, 4]},\n",
      " |      ....:                 index = [pd.Timestamp('20130101 09:00:00'),\n",
      " |      ....:                          pd.Timestamp('20130101 09:00:02'),\n",
      " |      ....:                          pd.Timestamp('20130101 09:00:03'),\n",
      " |      ....:                          pd.Timestamp('20130101 09:00:05'),\n",
      " |      ....:                          pd.Timestamp('20130101 09:00:06')])\n",
      " |      \n",
      " |      >>> df\n",
      " |                             B\n",
      " |      2013-01-01 09:00:00  0.0\n",
      " |      2013-01-01 09:00:02  1.0\n",
      " |      2013-01-01 09:00:03  2.0\n",
      " |      2013-01-01 09:00:05  NaN\n",
      " |      2013-01-01 09:00:06  4.0\n",
      " |      \n",
      " |      \n",
      " |      Contrasting to an integer rolling window, this will roll a variable\n",
      " |      length window corresponding to the time period.\n",
      " |      The default for min_periods is 1.\n",
      " |      \n",
      " |      >>> df.rolling('2s').sum()\n",
      " |                             B\n",
      " |      2013-01-01 09:00:00  0.0\n",
      " |      2013-01-01 09:00:02  1.0\n",
      " |      2013-01-01 09:00:03  3.0\n",
      " |      2013-01-01 09:00:05  NaN\n",
      " |      2013-01-01 09:00:06  4.0\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      By default, the result is set to the right edge of the window. This can be\n",
      " |      changed to the center of the window by setting ``center=True``.\n",
      " |      \n",
      " |      The `freq` keyword is used to conform time series data to a specified\n",
      " |      frequency by resampling the data. This is done with the default parameters\n",
      " |      of :meth:`~pandas.Series.resample` (i.e. using the `mean`).\n",
      " |      \n",
      " |      To learn more about the offsets & frequency strings, please see `this link\n",
      " |      <http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases>`__.\n",
      " |      \n",
      " |      The recognized win_types are:\n",
      " |      \n",
      " |      * ``boxcar``\n",
      " |      * ``triang``\n",
      " |      * ``blackman``\n",
      " |      * ``hamming``\n",
      " |      * ``bartlett``\n",
      " |      * ``parzen``\n",
      " |      * ``bohman``\n",
      " |      * ``blackmanharris``\n",
      " |      * ``nuttall``\n",
      " |      * ``barthann``\n",
      " |      * ``kaiser`` (needs beta)\n",
      " |      * ``gaussian`` (needs std)\n",
      " |      * ``general_gaussian`` (needs power, width)\n",
      " |      * ``slepian`` (needs width).\n",
      " |  \n",
      " |  round(self, decimals=0, *args, **kwargs)\n",
      " |      Round a DataFrame to a variable number of decimal places.\n",
      " |      \n",
      " |      .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      decimals : int, dict, Series\n",
      " |          Number of decimal places to round each column to. If an int is\n",
      " |          given, round each column to the same number of places.\n",
      " |          Otherwise dict and Series round to variable numbers of places.\n",
      " |          Column names should be in the keys if `decimals` is a\n",
      " |          dict-like, or in the index if `decimals` is a Series. Any\n",
      " |          columns not included in `decimals` will be left as is. Elements\n",
      " |          of `decimals` which are not columns of the input will be\n",
      " |          ignored.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = pd.DataFrame(np.random.random([3, 3]),\n",
      " |      ...     columns=['A', 'B', 'C'], index=['first', 'second', 'third'])\n",
      " |      >>> df\n",
      " |                     A         B         C\n",
      " |      first   0.028208  0.992815  0.173891\n",
      " |      second  0.038683  0.645646  0.577595\n",
      " |      third   0.877076  0.149370  0.491027\n",
      " |      >>> df.round(2)\n",
      " |                 A     B     C\n",
      " |      first   0.03  0.99  0.17\n",
      " |      second  0.04  0.65  0.58\n",
      " |      third   0.88  0.15  0.49\n",
      " |      >>> df.round({'A': 1, 'C': 2})\n",
      " |                A         B     C\n",
      " |      first   0.0  0.992815  0.17\n",
      " |      second  0.0  0.645646  0.58\n",
      " |      third   0.9  0.149370  0.49\n",
      " |      >>> decimals = pd.Series([1, 0, 2], index=['A', 'B', 'C'])\n",
      " |      >>> df.round(decimals)\n",
      " |                A  B     C\n",
      " |      first   0.0  1  0.17\n",
      " |      second  0.0  1  0.58\n",
      " |      third   0.9  0  0.49\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame object\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      numpy.around\n",
      " |      Series.round\n",
      " |  \n",
      " |  rpow(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Exponential power of dataframe and other, element-wise (binary operator `rpow`).\n",
      " |      \n",
      " |      Equivalent to ``other ** dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.pow\n",
      " |  \n",
      " |  rsub(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Subtraction of dataframe and other, element-wise (binary operator `rsub`).\n",
      " |      \n",
      " |      Equivalent to ``other - dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.sub\n",
      " |  \n",
      " |  rtruediv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Floating division of dataframe and other, element-wise (binary operator `rtruediv`).\n",
      " |      \n",
      " |      Equivalent to ``other / dataframe``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.truediv\n",
      " |  \n",
      " |  select_dtypes(self, include=None, exclude=None)\n",
      " |      Return a subset of a DataFrame including/excluding columns based on\n",
      " |      their ``dtype``.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      include, exclude : list-like\n",
      " |          A list of dtypes or strings to be included/excluded. You must pass\n",
      " |          in a non-empty sequence for at least one of these.\n",
      " |      \n",
      " |      Raises\n",
      " |      ------\n",
      " |      ValueError\n",
      " |          * If both of ``include`` and ``exclude`` are empty\n",
      " |          * If ``include`` and ``exclude`` have overlapping elements\n",
      " |          * If any kind of string dtype is passed in.\n",
      " |      TypeError\n",
      " |          * If either of ``include`` or ``exclude`` is not a sequence\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      subset : DataFrame\n",
      " |          The subset of the frame including the dtypes in ``include`` and\n",
      " |          excluding the dtypes in ``exclude``.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      * To select all *numeric* types use the numpy dtype ``numpy.number``\n",
      " |      * To select strings you must use the ``object`` dtype, but note that\n",
      " |        this will return *all* object dtype columns\n",
      " |      * See the `numpy dtype hierarchy\n",
      " |        <http://docs.scipy.org/doc/numpy/reference/arrays.scalars.html>`__\n",
      " |      * To select Pandas categorical dtypes, use 'category'\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = pd.DataFrame({'a': np.random.randn(6).astype('f4'),\n",
      " |      ...                    'b': [True, False] * 3,\n",
      " |      ...                    'c': [1.0, 2.0] * 3})\n",
      " |      >>> df\n",
      " |              a      b  c\n",
      " |      0  0.3962   True  1\n",
      " |      1  0.1459  False  2\n",
      " |      2  0.2623   True  1\n",
      " |      3  0.0764  False  2\n",
      " |      4 -0.9703   True  1\n",
      " |      5 -1.2094  False  2\n",
      " |      >>> df.select_dtypes(include=['float64'])\n",
      " |         c\n",
      " |      0  1\n",
      " |      1  2\n",
      " |      2  1\n",
      " |      3  2\n",
      " |      4  1\n",
      " |      5  2\n",
      " |      >>> df.select_dtypes(exclude=['floating'])\n",
      " |             b\n",
      " |      0   True\n",
      " |      1  False\n",
      " |      2   True\n",
      " |      3  False\n",
      " |      4   True\n",
      " |      5  False\n",
      " |  \n",
      " |  sem(self, axis=None, skipna=None, level=None, ddof=1, numeric_only=None, **kwargs)\n",
      " |      Return unbiased standard error of the mean over requested axis.\n",
      " |      \n",
      " |      Normalized by N-1 by default. This can be changed using the ddof argument\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      ddof : int, default 1\n",
      " |          degrees of freedom\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sem : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  set_index(self, keys, drop=True, append=False, inplace=False, verify_integrity=False)\n",
      " |      Set the DataFrame index (row labels) using one or more existing\n",
      " |      columns. By default yields a new object.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      keys : column label or list of column labels / arrays\n",
      " |      drop : boolean, default True\n",
      " |          Delete columns to be used as the new index\n",
      " |      append : boolean, default False\n",
      " |          Whether to append columns to existing index\n",
      " |      inplace : boolean, default False\n",
      " |          Modify the DataFrame in place (do not create a new object)\n",
      " |      verify_integrity : boolean, default False\n",
      " |          Check the new index for duplicates. Otherwise defer the check until\n",
      " |          necessary. Setting to False will improve the performance of this\n",
      " |          method\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> indexed_df = df.set_index(['A', 'B'])\n",
      " |      >>> indexed_df2 = df.set_index(['A', [0, 1, 2, 0, 1, 2]])\n",
      " |      >>> indexed_df3 = df.set_index([[0, 1, 2, 0, 1, 2]])\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      dataframe : DataFrame\n",
      " |  \n",
      " |  set_value(self, index, col, value, takeable=False)\n",
      " |      Put single value at passed column and index\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : row label\n",
      " |      col : column label\n",
      " |      value : scalar value\n",
      " |      takeable : interpret the index/col as indexers, default False\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      frame : DataFrame\n",
      " |          If label pair is contained, will be reference to calling DataFrame,\n",
      " |          otherwise a new object\n",
      " |  \n",
      " |  shift(self, periods=1, freq=None, axis=0)\n",
      " |      Shift index by desired number of periods with an optional time freq\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      periods : int\n",
      " |          Number of periods to move, can be positive or negative\n",
      " |      freq : DateOffset, timedelta, or time rule string, optional\n",
      " |          Increment to use from the tseries module or time rule (e.g. 'EOM').\n",
      " |          See Notes.\n",
      " |      axis : {0 or 'index', 1 or 'columns'}\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      If freq is specified then the index values are shifted but the data\n",
      " |      is not realigned. That is, use freq if you would like to extend the\n",
      " |      index when shifting and preserve the original data.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      shifted : DataFrame\n",
      " |  \n",
      " |  skew(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return unbiased skew over requested axis\n",
      " |      Normalized by N-1\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      skew : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  sort(self, columns=None, axis=0, ascending=True, inplace=False, kind='quicksort', na_position='last', **kwargs)\n",
      " |      DEPRECATED: use :meth:`DataFrame.sort_values`\n",
      " |      \n",
      " |      Sort DataFrame either by labels (along either axis) or by the values in\n",
      " |      column(s)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      columns : object\n",
      " |          Column name(s) in frame. Accepts a column name or a list\n",
      " |          for a nested sort. A tuple will be interpreted as the\n",
      " |          levels of a multi-index.\n",
      " |      ascending : boolean or list, default True\n",
      " |          Sort ascending vs. descending. Specify list for multiple sort\n",
      " |          orders\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          Sort index/rows versus columns\n",
      " |      inplace : boolean, default False\n",
      " |          Sort the DataFrame without creating a new instance\n",
      " |      kind : {'quicksort', 'mergesort', 'heapsort'}, optional\n",
      " |          This option is only applied when sorting on a single column or\n",
      " |          label.\n",
      " |      na_position : {'first', 'last'} (optional, default='last')\n",
      " |          'first' puts NaNs at the beginning\n",
      " |          'last' puts NaNs at the end\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> result = df.sort(['A', 'B'], ascending=[1, 0])\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sorted : DataFrame\n",
      " |  \n",
      " |  sort_index(self, axis=0, level=None, ascending=True, inplace=False, kind='quicksort', na_position='last', sort_remaining=True, by=None)\n",
      " |      Sort object by labels (along an axis)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : index, columns to direct sorting\n",
      " |      level : int or level name or list of ints or list of level names\n",
      " |          if not None, sort on values in specified index level(s)\n",
      " |      ascending : boolean, default True\n",
      " |          Sort ascending vs. descending\n",
      " |      inplace : bool, default False\n",
      " |          if True, perform operation in-place\n",
      " |      kind : {'quicksort', 'mergesort', 'heapsort'}, default 'quicksort'\n",
      " |           Choice of sorting algorithm. See also ndarray.np.sort for more\n",
      " |           information.  `mergesort` is the only stable algorithm. For\n",
      " |           DataFrames, this option is only applied when sorting on a single\n",
      " |           column or label.\n",
      " |      na_position : {'first', 'last'}, default 'last'\n",
      " |           `first` puts NaNs at the beginning, `last` puts NaNs at the end\n",
      " |      sort_remaining : bool, default True\n",
      " |          if true and sorting by level and index is multilevel, sort by other\n",
      " |          levels too (in order) after sorting by specified level\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sorted_obj : DataFrame\n",
      " |  \n",
      " |  sort_values(self, by, axis=0, ascending=True, inplace=False, kind='quicksort', na_position='last')\n",
      " |      Sort by the values along either axis\n",
      " |      \n",
      " |      .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      by : str or list of str\n",
      " |          Name or list of names which refer to the axis items.\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          Axis to direct sorting\n",
      " |      ascending : bool or list of bool, default True\n",
      " |           Sort ascending vs. descending. Specify list for multiple sort\n",
      " |           orders.  If this is a list of bools, must match the length of\n",
      " |           the by.\n",
      " |      inplace : bool, default False\n",
      " |           if True, perform operation in-place\n",
      " |      kind : {'quicksort', 'mergesort', 'heapsort'}, default 'quicksort'\n",
      " |           Choice of sorting algorithm. See also ndarray.np.sort for more\n",
      " |           information.  `mergesort` is the only stable algorithm. For\n",
      " |           DataFrames, this option is only applied when sorting on a single\n",
      " |           column or label.\n",
      " |      na_position : {'first', 'last'}, default 'last'\n",
      " |           `first` puts NaNs at the beginning, `last` puts NaNs at the end\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sorted_obj : DataFrame\n",
      " |  \n",
      " |  sortlevel(self, level=0, axis=0, ascending=True, inplace=False, sort_remaining=True)\n",
      " |      Sort multilevel index by chosen axis and primary level. Data will be\n",
      " |      lexicographically sorted by the chosen level followed by the other\n",
      " |      levels (in order)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      level : int\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |      ascending : boolean, default True\n",
      " |      inplace : boolean, default False\n",
      " |          Sort the DataFrame without creating a new instance\n",
      " |      sort_remaining : boolean, default True\n",
      " |          Sort by the other levels too.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sorted : DataFrame\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      DataFrame.sort_index(level=...)\n",
      " |  \n",
      " |  stack(self, level=-1, dropna=True)\n",
      " |      Pivot a level of the (possibly hierarchical) column labels, returning a\n",
      " |      DataFrame (or Series in the case of an object with a single level of\n",
      " |      column labels) having a hierarchical index with a new inner-most level\n",
      " |      of row labels.\n",
      " |      The level involved will automatically get sorted.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      level : int, string, or list of these, default last level\n",
      " |          Level(s) to stack, can pass level name\n",
      " |      dropna : boolean, default True\n",
      " |          Whether to drop rows in the resulting Frame/Series with no valid\n",
      " |          values\n",
      " |      \n",
      " |      Examples\n",
      " |      ----------\n",
      " |      >>> s\n",
      " |           a   b\n",
      " |      one  1.  2.\n",
      " |      two  3.  4.\n",
      " |      \n",
      " |      >>> s.stack()\n",
      " |      one a    1\n",
      " |          b    2\n",
      " |      two a    3\n",
      " |          b    4\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      stacked : DataFrame or Series\n",
      " |  \n",
      " |  std(self, axis=None, skipna=None, level=None, ddof=1, numeric_only=None, **kwargs)\n",
      " |      Return sample standard deviation over requested axis.\n",
      " |      \n",
      " |      Normalized by N-1 by default. This can be changed using the ddof argument\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      ddof : int, default 1\n",
      " |          degrees of freedom\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      std : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  sub(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Subtraction of dataframe and other, element-wise (binary operator `sub`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe - other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rsub\n",
      " |  \n",
      " |  subtract = sub(self, other, axis='columns', level=None, fill_value=None)\n",
      " |  \n",
      " |  sum(self, axis=None, skipna=None, level=None, numeric_only=None, **kwargs)\n",
      " |      Return the sum of the values for the requested axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      sum : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  swaplevel(self, i=-2, j=-1, axis=0)\n",
      " |      Swap levels i and j in a MultiIndex on a particular axis\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      i, j : int, string (can be mixed)\n",
      " |          Level of index to be swapped. Can pass level name as string.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      swapped : type of caller (new object)\n",
      " |      \n",
      " |      .. versionchanged:: 0.18.1\n",
      " |      \n",
      " |         The indexes ``i`` and ``j`` are now optional, and default to\n",
      " |         the two innermost levels of the index.\n",
      " |  \n",
      " |  to_csv(self, path_or_buf=None, sep=',', na_rep='', float_format=None, columns=None, header=True, index=True, index_label=None, mode='w', encoding=None, compression=None, quoting=None, quotechar='\"', line_terminator='\\n', chunksize=None, tupleize_cols=False, date_format=None, doublequote=True, escapechar=None, decimal='.')\n",
      " |      Write DataFrame to a comma-separated values (csv) file\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path_or_buf : string or file handle, default None\n",
      " |          File path or object, if None is provided the result is returned as\n",
      " |          a string.\n",
      " |      sep : character, default ','\n",
      " |          Field delimiter for the output file.\n",
      " |      na_rep : string, default ''\n",
      " |          Missing data representation\n",
      " |      float_format : string, default None\n",
      " |          Format string for floating point numbers\n",
      " |      columns : sequence, optional\n",
      " |          Columns to write\n",
      " |      header : boolean or list of string, default True\n",
      " |          Write out column names. If a list of string is given it is assumed\n",
      " |          to be aliases for the column names\n",
      " |      index : boolean, default True\n",
      " |          Write row names (index)\n",
      " |      index_label : string or sequence, or False, default None\n",
      " |          Column label for index column(s) if desired. If None is given, and\n",
      " |          `header` and `index` are True, then the index names are used. A\n",
      " |          sequence should be given if the DataFrame uses MultiIndex.  If\n",
      " |          False do not print fields for index names. Use index_label=False\n",
      " |          for easier importing in R\n",
      " |      mode : str\n",
      " |          Python write mode, default 'w'\n",
      " |      encoding : string, optional\n",
      " |          A string representing the encoding to use in the output file,\n",
      " |          defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.\n",
      " |      compression : string, optional\n",
      " |          a string representing the compression to use in the output file,\n",
      " |          allowed values are 'gzip', 'bz2', 'xz',\n",
      " |          only used when the first argument is a filename\n",
      " |      line_terminator : string, default ``'\\n'``\n",
      " |          The newline character or character sequence to use in the output\n",
      " |          file\n",
      " |      quoting : optional constant from csv module\n",
      " |          defaults to csv.QUOTE_MINIMAL. If you have set a `float_format`\n",
      " |          then floats are comverted to strings and thus csv.QUOTE_NONNUMERIC\n",
      " |          will treat them as non-numeric\n",
      " |      quotechar : string (length 1), default '\\\"'\n",
      " |          character used to quote fields\n",
      " |      doublequote : boolean, default True\n",
      " |          Control quoting of `quotechar` inside a field\n",
      " |      escapechar : string (length 1), default None\n",
      " |          character used to escape `sep` and `quotechar` when appropriate\n",
      " |      chunksize : int or None\n",
      " |          rows to write at a time\n",
      " |      tupleize_cols : boolean, default False\n",
      " |          write multi_index columns as a list of tuples (if True)\n",
      " |          or new (expanded format) if False)\n",
      " |      date_format : string, default None\n",
      " |          Format string for datetime objects\n",
      " |      decimal: string, default '.'\n",
      " |          Character recognized as decimal separator. E.g. use ',' for\n",
      " |          European data\n",
      " |      \n",
      " |          .. versionadded:: 0.16.0\n",
      " |  \n",
      " |  to_dict(self, orient='dict')\n",
      " |      Convert DataFrame to dictionary.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      orient : str {'dict', 'list', 'series', 'split', 'records', 'index'}\n",
      " |          Determines the type of the values of the dictionary.\n",
      " |      \n",
      " |          - dict (default) : dict like {column -> {index -> value}}\n",
      " |          - list : dict like {column -> [values]}\n",
      " |          - series : dict like {column -> Series(values)}\n",
      " |          - split : dict like\n",
      " |            {index -> [index], columns -> [columns], data -> [values]}\n",
      " |          - records : list like\n",
      " |            [{column -> value}, ... , {column -> value}]\n",
      " |          - index : dict like {index -> {column -> value}}\n",
      " |      \n",
      " |            .. versionadded:: 0.17.0\n",
      " |      \n",
      " |          Abbreviations are allowed. `s` indicates `series` and `sp`\n",
      " |          indicates `split`.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : dict like {column -> {index -> value}}\n",
      " |  \n",
      " |  to_excel(self, excel_writer, sheet_name='Sheet1', na_rep='', float_format=None, columns=None, header=True, index=True, index_label=None, startrow=0, startcol=0, engine=None, merge_cells=True, encoding=None, inf_rep='inf', verbose=True)\n",
      " |      Write DataFrame to a excel sheet\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      excel_writer : string or ExcelWriter object\n",
      " |          File path or existing ExcelWriter\n",
      " |      sheet_name : string, default 'Sheet1'\n",
      " |          Name of sheet which will contain DataFrame\n",
      " |      na_rep : string, default ''\n",
      " |          Missing data representation\n",
      " |      float_format : string, default None\n",
      " |          Format string for floating point numbers\n",
      " |      columns : sequence, optional\n",
      " |          Columns to write\n",
      " |      header : boolean or list of string, default True\n",
      " |          Write out column names. If a list of string is given it is\n",
      " |          assumed to be aliases for the column names\n",
      " |      index : boolean, default True\n",
      " |          Write row names (index)\n",
      " |      index_label : string or sequence, default None\n",
      " |          Column label for index column(s) if desired. If None is given, and\n",
      " |          `header` and `index` are True, then the index names are used. A\n",
      " |          sequence should be given if the DataFrame uses MultiIndex.\n",
      " |      startrow :\n",
      " |          upper left cell row to dump data frame\n",
      " |      startcol :\n",
      " |          upper left cell column to dump data frame\n",
      " |      engine : string, default None\n",
      " |          write engine to use - you can also set this via the options\n",
      " |          ``io.excel.xlsx.writer``, ``io.excel.xls.writer``, and\n",
      " |          ``io.excel.xlsm.writer``.\n",
      " |      merge_cells : boolean, default True\n",
      " |          Write MultiIndex and Hierarchical Rows as merged cells.\n",
      " |      encoding: string, default None\n",
      " |          encoding of the resulting excel file. Only necessary for xlwt,\n",
      " |          other writers support unicode natively.\n",
      " |      inf_rep : string, default 'inf'\n",
      " |          Representation for infinity (there is no native representation for\n",
      " |          infinity in Excel)\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      If passing an existing ExcelWriter object, then the sheet will be added\n",
      " |      to the existing workbook.  This can be used to save different\n",
      " |      DataFrames to one workbook:\n",
      " |      \n",
      " |      >>> writer = ExcelWriter('output.xlsx')\n",
      " |      >>> df1.to_excel(writer,'Sheet1')\n",
      " |      >>> df2.to_excel(writer,'Sheet2')\n",
      " |      >>> writer.save()\n",
      " |      \n",
      " |      For compatibility with to_csv, to_excel serializes lists and dicts to\n",
      " |      strings before writing.\n",
      " |  \n",
      " |  to_gbq(self, destination_table, project_id, chunksize=10000, verbose=True, reauth=False, if_exists='fail', private_key=None)\n",
      " |      Write a DataFrame to a Google BigQuery table.\n",
      " |      \n",
      " |      THIS IS AN EXPERIMENTAL LIBRARY\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      dataframe : DataFrame\n",
      " |          DataFrame to be written\n",
      " |      destination_table : string\n",
      " |          Name of table to be written, in the form 'dataset.tablename'\n",
      " |      project_id : str\n",
      " |          Google BigQuery Account project ID.\n",
      " |      chunksize : int (default 10000)\n",
      " |          Number of rows to be inserted in each chunk from the dataframe.\n",
      " |      verbose : boolean (default True)\n",
      " |          Show percentage complete\n",
      " |      reauth : boolean (default False)\n",
      " |          Force Google BigQuery to reauthenticate the user. This is useful\n",
      " |          if multiple accounts are used.\n",
      " |      if_exists : {'fail', 'replace', 'append'}, default 'fail'\n",
      " |          'fail': If table exists, do nothing.\n",
      " |          'replace': If table exists, drop it, recreate it, and insert data.\n",
      " |          'append': If table exists, insert data. Create if does not exist.\n",
      " |      private_key : str (optional)\n",
      " |          Service account private key in JSON format. Can be file path\n",
      " |          or string contents. This is useful for remote server\n",
      " |          authentication (eg. jupyter iPython notebook on remote host)\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |  \n",
      " |  to_html(self, buf=None, columns=None, col_space=None, header=True, index=True, na_rep='NaN', formatters=None, float_format=None, sparsify=None, index_names=True, justify=None, bold_rows=True, classes=None, escape=True, max_rows=None, max_cols=None, show_dimensions=False, notebook=False, decimal='.', border=None)\n",
      " |      Render a DataFrame as an HTML table.\n",
      " |      \n",
      " |      `to_html`-specific options:\n",
      " |      \n",
      " |      bold_rows : boolean, default True\n",
      " |          Make the row labels bold in the output\n",
      " |      classes : str or list or tuple, default None\n",
      " |          CSS class(es) to apply to the resulting html table\n",
      " |      escape : boolean, default True\n",
      " |          Convert the characters <, >, and & to HTML-safe sequences.=\n",
      " |      max_rows : int, optional\n",
      " |          Maximum number of rows to show before truncating. If None, show\n",
      " |          all.\n",
      " |      max_cols : int, optional\n",
      " |          Maximum number of columns to show before truncating. If None, show\n",
      " |          all.\n",
      " |      decimal : string, default '.'\n",
      " |          Character recognized as decimal separator, e.g. ',' in Europe\n",
      " |      \n",
      " |          .. versionadded:: 0.18.0\n",
      " |      border : int\n",
      " |          A ``border=border`` attribute is included in the opening\n",
      " |          `<table>` tag. Default ``pd.options.html.border``.\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      buf : StringIO-like, optional\n",
      " |          buffer to write to\n",
      " |      columns : sequence, optional\n",
      " |          the subset of columns to write; default None writes all columns\n",
      " |      col_space : int, optional\n",
      " |          the minimum width of each column\n",
      " |      header : bool, optional\n",
      " |          whether to print column labels, default True\n",
      " |      index : bool, optional\n",
      " |          whether to print index (row) labels, default True\n",
      " |      na_rep : string, optional\n",
      " |          string representation of NAN to use, default 'NaN'\n",
      " |      formatters : list or dict of one-parameter functions, optional\n",
      " |          formatter functions to apply to columns' elements by position or name,\n",
      " |          default None. The result of each function must be a unicode string.\n",
      " |          List must be of length equal to the number of columns.\n",
      " |      float_format : one-parameter function, optional\n",
      " |          formatter function to apply to columns' elements if they are floats,\n",
      " |          default None. The result of this function must be a unicode string.\n",
      " |      sparsify : bool, optional\n",
      " |          Set to False for a DataFrame with a hierarchical index to print every\n",
      " |          multiindex key at each row, default True\n",
      " |      index_names : bool, optional\n",
      " |          Prints the names of the indexes, default True\n",
      " |      line_width : int, optional\n",
      " |          Width to wrap a line in characters, default no wrap\n",
      " |      justify : {'left', 'right'}, default None\n",
      " |          Left or right-justify the column labels. If None uses the option from\n",
      " |          the print configuration (controlled by set_option), 'right' out\n",
      " |          of the box.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      formatted : string (or unicode, depending on data and options)\n",
      " |  \n",
      " |  to_latex(self, buf=None, columns=None, col_space=None, header=True, index=True, na_rep='NaN', formatters=None, float_format=None, sparsify=None, index_names=True, bold_rows=True, column_format=None, longtable=None, escape=None, encoding=None, decimal='.')\n",
      " |      Render a DataFrame to a tabular environment table. You can splice\n",
      " |      this into a LaTeX document. Requires \\usepackage{booktabs}.\n",
      " |      \n",
      " |      `to_latex`-specific options:\n",
      " |      \n",
      " |      bold_rows : boolean, default True\n",
      " |          Make the row labels bold in the output\n",
      " |      column_format : str, default None\n",
      " |          The columns format as specified in `LaTeX table format\n",
      " |          <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g 'rcl' for 3\n",
      " |          columns\n",
      " |      longtable : boolean, default will be read from the pandas config module\n",
      " |          default: False\n",
      " |          Use a longtable environment instead of tabular. Requires adding\n",
      " |          a \\usepackage{longtable} to your LaTeX preamble.\n",
      " |      escape : boolean, default will be read from the pandas config module\n",
      " |          default: True\n",
      " |          When set to False prevents from escaping latex special\n",
      " |          characters in column names.\n",
      " |      encoding : str, default None\n",
      " |          A string representing the encoding to use in the output file,\n",
      " |          defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.\n",
      " |      decimal : string, default '.'\n",
      " |          Character recognized as decimal separator, e.g. ',' in Europe\n",
      " |      \n",
      " |          .. versionadded:: 0.18.0\n",
      " |      \n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      buf : StringIO-like, optional\n",
      " |          buffer to write to\n",
      " |      columns : sequence, optional\n",
      " |          the subset of columns to write; default None writes all columns\n",
      " |      col_space : int, optional\n",
      " |          the minimum width of each column\n",
      " |      header : bool, optional\n",
      " |          whether to print column labels, default True\n",
      " |      index : bool, optional\n",
      " |          whether to print index (row) labels, default True\n",
      " |      na_rep : string, optional\n",
      " |          string representation of NAN to use, default 'NaN'\n",
      " |      formatters : list or dict of one-parameter functions, optional\n",
      " |          formatter functions to apply to columns' elements by position or name,\n",
      " |          default None. The result of each function must be a unicode string.\n",
      " |          List must be of length equal to the number of columns.\n",
      " |      float_format : one-parameter function, optional\n",
      " |          formatter function to apply to columns' elements if they are floats,\n",
      " |          default None. The result of this function must be a unicode string.\n",
      " |      sparsify : bool, optional\n",
      " |          Set to False for a DataFrame with a hierarchical index to print every\n",
      " |          multiindex key at each row, default True\n",
      " |      index_names : bool, optional\n",
      " |          Prints the names of the indexes, default True\n",
      " |      line_width : int, optional\n",
      " |          Width to wrap a line in characters, default no wrap\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      formatted : string (or unicode, depending on data and options)\n",
      " |  \n",
      " |  to_panel(self)\n",
      " |      Transform long (stacked) format (DataFrame) into wide (3D, Panel)\n",
      " |      format.\n",
      " |      \n",
      " |      Currently the index of the DataFrame must be a 2-level MultiIndex. This\n",
      " |      may be generalized later\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      panel : Panel\n",
      " |  \n",
      " |  to_period(self, freq=None, axis=0, copy=True)\n",
      " |      Convert DataFrame from DatetimeIndex to PeriodIndex with desired\n",
      " |      frequency (inferred from index if not passed)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      freq : string, default\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          The axis to convert (the index by default)\n",
      " |      copy : boolean, default True\n",
      " |          If False then underlying input data is not copied\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      ts : TimeSeries with PeriodIndex\n",
      " |  \n",
      " |  to_records(self, index=True, convert_datetime64=True)\n",
      " |      Convert DataFrame to record array. Index will be put in the\n",
      " |      'index' field of the record array if requested\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      index : boolean, default True\n",
      " |          Include index in resulting record array, stored in 'index' field\n",
      " |      convert_datetime64 : boolean, default True\n",
      " |          Whether to convert the index to datetime.datetime if it is a\n",
      " |          DatetimeIndex\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : recarray\n",
      " |  \n",
      " |  to_sparse(self, fill_value=None, kind='block')\n",
      " |      Convert to SparseDataFrame\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      fill_value : float, default NaN\n",
      " |      kind : {'block', 'integer'}\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : SparseDataFrame\n",
      " |  \n",
      " |  to_stata(self, fname, convert_dates=None, write_index=True, encoding='latin-1', byteorder=None, time_stamp=None, data_label=None, variable_labels=None)\n",
      " |      A class for writing Stata binary dta files from array-like objects\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      fname : str or buffer\n",
      " |          String path of file-like object\n",
      " |      convert_dates : dict\n",
      " |          Dictionary mapping columns containing datetime types to stata\n",
      " |          internal format to use when wirting the dates. Options are 'tc',\n",
      " |          'td', 'tm', 'tw', 'th', 'tq', 'ty'. Column can be either an integer\n",
      " |          or a name. Datetime columns that do not have a conversion type\n",
      " |          specified will be converted to 'tc'. Raises NotImplementedError if\n",
      " |          a datetime column has timezone information\n",
      " |      write_index : bool\n",
      " |          Write the index to Stata dataset.\n",
      " |      encoding : str\n",
      " |          Default is latin-1. Unicode is not supported\n",
      " |      byteorder : str\n",
      " |          Can be \">\", \"<\", \"little\", or \"big\". default is `sys.byteorder`\n",
      " |      time_stamp : datetime\n",
      " |          A datetime to use as file creation date.  Default is the current\n",
      " |          time.\n",
      " |      dataset_label : str\n",
      " |          A label for the data set.  Must be 80 characters or smaller.\n",
      " |      variable_labels : dict\n",
      " |          Dictionary containing columns as keys and variable labels as\n",
      " |          values. Each label must be 80 characters or smaller.\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      Raises\n",
      " |      ------\n",
      " |      NotImplementedError\n",
      " |          * If datetimes contain timezone information\n",
      " |          * Column dtype is not representable in Stata\n",
      " |      ValueError\n",
      " |          * Columns listed in convert_dates are noth either datetime64[ns]\n",
      " |            or datetime.datetime\n",
      " |          * Column listed in convert_dates is not in DataFrame\n",
      " |          * Categorical label contains more than 32,000 characters\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> writer = StataWriter('./data_file.dta', data)\n",
      " |      >>> writer.write_file()\n",
      " |      \n",
      " |      Or with dates\n",
      " |      \n",
      " |      >>> writer = StataWriter('./date_data_file.dta', data, {2 : 'tw'})\n",
      " |      >>> writer.write_file()\n",
      " |  \n",
      " |  to_string(self, buf=None, columns=None, col_space=None, header=True, index=True, na_rep='NaN', formatters=None, float_format=None, sparsify=None, index_names=True, justify=None, line_width=None, max_rows=None, max_cols=None, show_dimensions=False)\n",
      " |      Render a DataFrame to a console-friendly tabular output.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      buf : StringIO-like, optional\n",
      " |          buffer to write to\n",
      " |      columns : sequence, optional\n",
      " |          the subset of columns to write; default None writes all columns\n",
      " |      col_space : int, optional\n",
      " |          the minimum width of each column\n",
      " |      header : bool, optional\n",
      " |          whether to print column labels, default True\n",
      " |      index : bool, optional\n",
      " |          whether to print index (row) labels, default True\n",
      " |      na_rep : string, optional\n",
      " |          string representation of NAN to use, default 'NaN'\n",
      " |      formatters : list or dict of one-parameter functions, optional\n",
      " |          formatter functions to apply to columns' elements by position or name,\n",
      " |          default None. The result of each function must be a unicode string.\n",
      " |          List must be of length equal to the number of columns.\n",
      " |      float_format : one-parameter function, optional\n",
      " |          formatter function to apply to columns' elements if they are floats,\n",
      " |          default None. The result of this function must be a unicode string.\n",
      " |      sparsify : bool, optional\n",
      " |          Set to False for a DataFrame with a hierarchical index to print every\n",
      " |          multiindex key at each row, default True\n",
      " |      index_names : bool, optional\n",
      " |          Prints the names of the indexes, default True\n",
      " |      line_width : int, optional\n",
      " |          Width to wrap a line in characters, default no wrap\n",
      " |      justify : {'left', 'right'}, default None\n",
      " |          Left or right-justify the column labels. If None uses the option from\n",
      " |          the print configuration (controlled by set_option), 'right' out\n",
      " |          of the box.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      formatted : string (or unicode, depending on data and options)\n",
      " |  \n",
      " |  to_timestamp(self, freq=None, how='start', axis=0, copy=True)\n",
      " |      Cast to DatetimeIndex of timestamps, at *beginning* of period\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      freq : string, default frequency of PeriodIndex\n",
      " |          Desired frequency\n",
      " |      how : {'s', 'e', 'start', 'end'}\n",
      " |          Convention for converting period to timestamp; start of period\n",
      " |          vs. end\n",
      " |      axis : {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          The axis to convert (the index by default)\n",
      " |      copy : boolean, default True\n",
      " |          If false then underlying input data is not copied\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      df : DataFrame with DatetimeIndex\n",
      " |  \n",
      " |  transpose(self, *args, **kwargs)\n",
      " |      Transpose index and columns\n",
      " |  \n",
      " |  truediv(self, other, axis='columns', level=None, fill_value=None)\n",
      " |      Floating division of dataframe and other, element-wise (binary operator `truediv`).\n",
      " |      \n",
      " |      Equivalent to ``dataframe / other``, but with support to substitute a fill_value for\n",
      " |      missing data in one of the inputs.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Series, DataFrame, or constant\n",
      " |      axis : {0, 1, 'index', 'columns'}\n",
      " |          For Series input, axis to match Series index on\n",
      " |      fill_value : None or float value, default None\n",
      " |          Fill missing (NaN) values with this value. If both DataFrame\n",
      " |          locations are missing, the result will be missing\n",
      " |      level : int or name\n",
      " |          Broadcast across a level, matching Index values on the\n",
      " |          passed MultiIndex level\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Mismatched indices will be unioned together\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      result : DataFrame\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.rtruediv\n",
      " |  \n",
      " |  unstack(self, level=-1, fill_value=None)\n",
      " |      Pivot a level of the (necessarily hierarchical) index labels, returning\n",
      " |      a DataFrame having a new level of column labels whose inner-most level\n",
      " |      consists of the pivoted index labels. If the index is not a MultiIndex,\n",
      " |      the output will be a Series (the analogue of stack when the columns are\n",
      " |      not a MultiIndex).\n",
      " |      The level involved will automatically get sorted.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      level : int, string, or list of these, default -1 (last level)\n",
      " |          Level(s) of index to unstack, can pass level name\n",
      " |      fill_value : replace NaN with this value if the unstack produces\n",
      " |          missing values\n",
      " |      \n",
      " |          .. versionadded: 0.18.0\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      DataFrame.pivot : Pivot a table based on column values.\n",
      " |      DataFrame.stack : Pivot a level of the column labels (inverse operation\n",
      " |          from `unstack`).\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> index = pd.MultiIndex.from_tuples([('one', 'a'), ('one', 'b'),\n",
      " |      ...                                    ('two', 'a'), ('two', 'b')])\n",
      " |      >>> s = pd.Series(np.arange(1.0, 5.0), index=index)\n",
      " |      >>> s\n",
      " |      one  a   1.0\n",
      " |           b   2.0\n",
      " |      two  a   3.0\n",
      " |           b   4.0\n",
      " |      dtype: float64\n",
      " |      \n",
      " |      >>> s.unstack(level=-1)\n",
      " |           a   b\n",
      " |      one  1.0  2.0\n",
      " |      two  3.0  4.0\n",
      " |      \n",
      " |      >>> s.unstack(level=0)\n",
      " |         one  two\n",
      " |      a  1.0   3.0\n",
      " |      b  2.0   4.0\n",
      " |      \n",
      " |      >>> df = s.unstack(level=0)\n",
      " |      >>> df.unstack()\n",
      " |      one  a  1.0\n",
      " |           b  2.0\n",
      " |      two  a  3.0\n",
      " |           b  4.0\n",
      " |      dtype: float64\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      unstacked : DataFrame or Series\n",
      " |  \n",
      " |  update(self, other, join='left', overwrite=True, filter_func=None, raise_conflict=False)\n",
      " |      Modify DataFrame in place using non-NA values from passed\n",
      " |      DataFrame. Aligns on indices\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : DataFrame, or object coercible into a DataFrame\n",
      " |      join : {'left'}, default 'left'\n",
      " |      overwrite : boolean, default True\n",
      " |          If True then overwrite values for common keys in the calling frame\n",
      " |      filter_func : callable(1d-array) -> 1d-array<boolean>, default None\n",
      " |          Can choose to replace values other than NA. Return True for values\n",
      " |          that should be updated\n",
      " |      raise_conflict : boolean\n",
      " |          If True, will raise an error if the DataFrame and other both\n",
      " |          contain data in the same place.\n",
      " |  \n",
      " |  var(self, axis=None, skipna=None, level=None, ddof=1, numeric_only=None, **kwargs)\n",
      " |      Return unbiased variance over requested axis.\n",
      " |      \n",
      " |      Normalized by N-1 by default. This can be changed using the ddof argument\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis : {index (0), columns (1)}\n",
      " |      skipna : boolean, default True\n",
      " |          Exclude NA/null values. If an entire row/column is NA, the result\n",
      " |          will be NA\n",
      " |      level : int or level name, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), count along a\n",
      " |          particular level, collapsing into a Series\n",
      " |      ddof : int, default 1\n",
      " |          degrees of freedom\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean columns. If None, will attempt to use\n",
      " |          everything, then use only numeric data. Not implemented for Series.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      var : Series or DataFrame (if level specified)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods defined here:\n",
      " |  \n",
      " |  from_csv(path, header=0, sep=',', index_col=0, parse_dates=True, encoding=None, tupleize_cols=False, infer_datetime_format=False) from builtins.type\n",
      " |      Read CSV file (DISCOURAGED, please use :func:`pandas.read_csv`\n",
      " |      instead).\n",
      " |      \n",
      " |      It is preferable to use the more powerful :func:`pandas.read_csv`\n",
      " |      for most general purposes, but ``from_csv`` makes for an easy\n",
      " |      roundtrip to and from a file (the exact counterpart of\n",
      " |      ``to_csv``), especially with a DataFrame of time series data.\n",
      " |      \n",
      " |      This method only differs from the preferred :func:`pandas.read_csv`\n",
      " |      in some defaults:\n",
      " |      \n",
      " |      - `index_col` is ``0`` instead of ``None`` (take first column as index\n",
      " |        by default)\n",
      " |      - `parse_dates` is ``True`` instead of ``False`` (try parsing the index\n",
      " |        as datetime by default)\n",
      " |      \n",
      " |      So a ``pd.DataFrame.from_csv(path)`` can be replaced by\n",
      " |      ``pd.read_csv(path, index_col=0, parse_dates=True)``.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path : string file path or file handle / StringIO\n",
      " |      header : int, default 0\n",
      " |          Row to use as header (skip prior rows)\n",
      " |      sep : string, default ','\n",
      " |          Field delimiter\n",
      " |      index_col : int or sequence, default 0\n",
      " |          Column to use for index. If a sequence is given, a MultiIndex\n",
      " |          is used. Different default from read_table\n",
      " |      parse_dates : boolean, default True\n",
      " |          Parse dates. Different default from read_table\n",
      " |      tupleize_cols : boolean, default False\n",
      " |          write multi_index columns as a list of tuples (if True)\n",
      " |          or new (expanded format) if False)\n",
      " |      infer_datetime_format: boolean, default False\n",
      " |          If True and `parse_dates` is True for a column, try to infer the\n",
      " |          datetime format based on the first datetime string. If the format\n",
      " |          can be inferred, there often will be a large parsing speed-up.\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      pandas.read_csv\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : DataFrame\n",
      " |  \n",
      " |  from_dict(data, orient='columns', dtype=None) from builtins.type\n",
      " |      Construct DataFrame from dict of array-like or dicts\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      data : dict\n",
      " |          {field : array-like} or {field : dict}\n",
      " |      orient : {'columns', 'index'}, default 'columns'\n",
      " |          The \"orientation\" of the data. If the keys of the passed dict\n",
      " |          should be the columns of the resulting DataFrame, pass 'columns'\n",
      " |          (default). Otherwise if the keys should be rows, pass 'index'.\n",
      " |      dtype : dtype, default None\n",
      " |          Data type to force, otherwise infer\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      DataFrame\n",
      " |  \n",
      " |  from_items(items, columns=None, orient='columns') from builtins.type\n",
      " |      Convert (key, value) pairs to DataFrame. The keys will be the axis\n",
      " |      index (usually the columns, but depends on the specified\n",
      " |      orientation). The values should be arrays or Series.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      items : sequence of (key, value) pairs\n",
      " |          Values should be arrays or Series.\n",
      " |      columns : sequence of column labels, optional\n",
      " |          Must be passed if orient='index'.\n",
      " |      orient : {'columns', 'index'}, default 'columns'\n",
      " |          The \"orientation\" of the data. If the keys of the\n",
      " |          input correspond to column labels, pass 'columns'\n",
      " |          (default). Otherwise if the keys correspond to the index,\n",
      " |          pass 'index'.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      frame : DataFrame\n",
      " |  \n",
      " |  from_records(data, index=None, exclude=None, columns=None, coerce_float=False, nrows=None) from builtins.type\n",
      " |      Convert structured or record ndarray to DataFrame\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      data : ndarray (structured dtype), list of tuples, dict, or DataFrame\n",
      " |      index : string, list of fields, array-like\n",
      " |          Field of array to use as the index, alternately a specific set of\n",
      " |          input labels to use\n",
      " |      exclude : sequence, default None\n",
      " |          Columns or fields to exclude\n",
      " |      columns : sequence, default None\n",
      " |          Column names to use. If the passed data do not have names\n",
      " |          associated with them, this argument provides names for the\n",
      " |          columns. Otherwise this argument indicates the order of the columns\n",
      " |          in the result (any names not found in the data will become all-NA\n",
      " |          columns)\n",
      " |      coerce_float : boolean, default False\n",
      " |          Attempt to convert values to non-string, non-numeric objects (like\n",
      " |          decimal.Decimal) to floating point, useful for SQL result sets\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      df : DataFrame\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  T\n",
      " |      Transpose index and columns\n",
      " |  \n",
      " |  axes\n",
      " |      Return a list with the row axis labels and column axis labels as the\n",
      " |      only members. They are returned in that order.\n",
      " |  \n",
      " |  columns\n",
      " |  \n",
      " |  index\n",
      " |  \n",
      " |  shape\n",
      " |      Return a tuple representing the dimensionality of the DataFrame.\n",
      " |  \n",
      " |  style\n",
      " |      Property returning a Styler object containing methods for\n",
      " |      building a styled HTML representation fo the DataFrame.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.formats.style.Styler\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  plot = <class 'pandas.tools.plotting.FramePlotMethods'>\n",
      " |      DataFrame plotting accessor and method\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df.plot.line()\n",
      " |      >>> df.plot.scatter('x', 'y')\n",
      " |      >>> df.plot.hexbin()\n",
      " |      \n",
      " |      These plotting methods can also be accessed by calling the accessor as a\n",
      " |      method with the ``kind`` argument:\n",
      " |      ``df.plot(kind='line')`` is equivalent to ``df.plot.line()``\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from pandas.core.generic.NDFrame:\n",
      " |  \n",
      " |  __abs__(self)\n",
      " |  \n",
      " |  __array__(self, dtype=None)\n",
      " |  \n",
      " |  __array_wrap__(self, result, context=None)\n",
      " |  \n",
      " |  __bool__ = __nonzero__(self)\n",
      " |  \n",
      " |  __contains__(self, key)\n",
      " |      True if the key is in the info axis\n",
      " |  \n",
      " |  __delitem__(self, key)\n",
      " |      Delete item\n",
      " |  \n",
      " |  __finalize__(self, other, method=None, **kwargs)\n",
      " |      Propagate metadata from other to self.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : the object from which to get the attributes that we are going\n",
      " |          to propagate\n",
      " |      method : optional, a passed method name ; possibly to take different\n",
      " |          types of propagation actions based on this\n",
      " |  \n",
      " |  __getattr__(self, name)\n",
      " |      After regular attribute access, try looking up the name\n",
      " |      This allows simpler access to columns for interactive use.\n",
      " |  \n",
      " |  __getstate__(self)\n",
      " |  \n",
      " |  __hash__(self)\n",
      " |      Return hash(self).\n",
      " |  \n",
      " |  __invert__(self)\n",
      " |  \n",
      " |  __iter__(self)\n",
      " |      Iterate over infor axis\n",
      " |  \n",
      " |  __neg__(self)\n",
      " |  \n",
      " |  __nonzero__(self)\n",
      " |  \n",
      " |  __round__(self, decimals=0)\n",
      " |  \n",
      " |  __setattr__(self, name, value)\n",
      " |      After regular attribute access, try setting the name\n",
      " |      This allows simpler access to columns for interactive use.\n",
      " |  \n",
      " |  __setstate__(self, state)\n",
      " |  \n",
      " |  abs(self)\n",
      " |      Return an object with absolute value taken--only applicable to objects\n",
      " |      that are all numeric.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      abs: type of caller\n",
      " |  \n",
      " |  add_prefix(self, prefix)\n",
      " |      Concatenate prefix string with panel items names.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      prefix : string\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      with_prefix : type of caller\n",
      " |  \n",
      " |  add_suffix(self, suffix)\n",
      " |      Concatenate suffix string with panel items names.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      suffix : string\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      with_suffix : type of caller\n",
      " |  \n",
      " |  as_blocks(self, copy=True)\n",
      " |      Convert the frame to a dict of dtype -> Constructor Types that each has\n",
      " |      a homogeneous dtype.\n",
      " |      \n",
      " |      NOTE: the dtypes of the blocks WILL BE PRESERVED HERE (unlike in\n",
      " |            as_matrix)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      copy : boolean, default True\n",
      " |      \n",
      " |             .. versionadded: 0.16.1\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      values : a dict of dtype -> Constructor Types\n",
      " |  \n",
      " |  as_matrix(self, columns=None)\n",
      " |      Convert the frame to its Numpy-array representation.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      columns: list, optional, default:None\n",
      " |          If None, return all columns, otherwise, returns specified columns.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      values : ndarray\n",
      " |          If the caller is heterogeneous and contains booleans or objects,\n",
      " |          the result will be of dtype=object. See Notes.\n",
      " |      \n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Return is NOT a Numpy-matrix, rather, a Numpy-array.\n",
      " |      \n",
      " |      The dtype will be a lower-common-denominator dtype (implicit\n",
      " |      upcasting); that is to say if the dtypes (even of numeric types)\n",
      " |      are mixed, the one that accommodates all will be chosen. Use this\n",
      " |      with care if you are not dealing with the blocks.\n",
      " |      \n",
      " |      e.g. If the dtypes are float16 and float32, dtype will be upcast to\n",
      " |      float32.  If dtypes are int32 and uint8, dtype will be upcase to\n",
      " |      int32. By numpy.find_common_type convention, mixing int64 and uint64\n",
      " |      will result in a flot64 dtype.\n",
      " |      \n",
      " |      This method is provided for backwards compatibility. Generally,\n",
      " |      it is recommended to use '.values'.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.DataFrame.values\n",
      " |  \n",
      " |  asfreq(self, freq, method=None, how=None, normalize=False)\n",
      " |      Convert TimeSeries to specified frequency.\n",
      " |      \n",
      " |      Optionally provide filling method to pad/backfill missing values.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      freq : DateOffset object, or string\n",
      " |      method : {'backfill'/'bfill', 'pad'/'ffill'}, default None\n",
      " |          Method to use for filling holes in reindexed Series (note this\n",
      " |          does not fill NaNs that already were present):\n",
      " |      \n",
      " |          * 'pad' / 'ffill': propagate last valid observation forward to next\n",
      " |            valid\n",
      " |          * 'backfill' / 'bfill': use NEXT valid observation to fill\n",
      " |      how : {'start', 'end'}, default end\n",
      " |          For PeriodIndex only, see PeriodIndex.asfreq\n",
      " |      normalize : bool, default False\n",
      " |          Whether to reset output index to midnight\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      converted : type of caller\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      To learn more about the frequency strings, please see `this link\n",
      " |      <http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases>`__.\n",
      " |  \n",
      " |  asof(self, where, subset=None)\n",
      " |      The last row without any NaN is taken (or the last row without\n",
      " |      NaN considering only the subset of columns in the case of a DataFrame)\n",
      " |      \n",
      " |      .. versionadded:: 0.19.0 For DataFrame\n",
      " |      \n",
      " |      If there is no good value, NaN is returned.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      where : date or array of dates\n",
      " |      subset : string or list of strings, default None\n",
      " |         if not None use these columns for NaN propagation\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Dates are assumed to be sorted\n",
      " |      Raises if this is not the case\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      where is scalar\n",
      " |      \n",
      " |        - value or NaN if input is Series\n",
      " |        - Series if input is DataFrame\n",
      " |      \n",
      " |      where is Index: same shape object as input\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      merge_asof\n",
      " |  \n",
      " |  astype(self, dtype, copy=True, raise_on_error=True, **kwargs)\n",
      " |      Cast object to input numpy.dtype\n",
      " |      Return a copy when copy = True (be really careful with this!)\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      dtype : data type, or dict of column name -> data type\n",
      " |          Use a numpy.dtype or Python type to cast entire pandas object to\n",
      " |          the same type. Alternatively, use {col: dtype, ...}, where col is a\n",
      " |          column label and dtype is a numpy.dtype or Python type to cast one\n",
      " |          or more of the DataFrame's columns to column-specific types.\n",
      " |      raise_on_error : raise on invalid input\n",
      " |      kwargs : keyword arguments to pass on to the constructor\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      casted : type of caller\n",
      " |  \n",
      " |  at_time(self, time, asof=False)\n",
      " |      Select values at particular time of day (e.g. 9:30AM).\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      time : datetime.time or string\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      values_at_time : type of caller\n",
      " |  \n",
      " |  between_time(self, start_time, end_time, include_start=True, include_end=True)\n",
      " |      Select values between particular times of the day (e.g., 9:00-9:30 AM).\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      start_time : datetime.time or string\n",
      " |      end_time : datetime.time or string\n",
      " |      include_start : boolean, default True\n",
      " |      include_end : boolean, default True\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      values_between_time : type of caller\n",
      " |  \n",
      " |  bfill(self, axis=None, inplace=False, limit=None, downcast=None)\n",
      " |      Synonym for NDFrame.fillna(method='bfill')\n",
      " |  \n",
      " |  bool(self)\n",
      " |      Return the bool of a single element PandasObject.\n",
      " |      \n",
      " |      This must be a boolean scalar value, either True or False.  Raise a\n",
      " |      ValueError if the PandasObject does not have exactly 1 element, or that\n",
      " |      element is not boolean\n",
      " |  \n",
      " |  clip(self, lower=None, upper=None, axis=None, *args, **kwargs)\n",
      " |      Trim values at input threshold(s).\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      lower : float or array_like, default None\n",
      " |      upper : float or array_like, default None\n",
      " |      axis : int or string axis name, optional\n",
      " |          Align object with lower and upper along the given axis.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      clipped : Series\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df\n",
      " |        0         1\n",
      " |      0  0.335232 -1.256177\n",
      " |      1 -1.367855  0.746646\n",
      " |      2  0.027753 -1.176076\n",
      " |      3  0.230930 -0.679613\n",
      " |      4  1.261967  0.570967\n",
      " |      >>> df.clip(-1.0, 0.5)\n",
      " |                0         1\n",
      " |      0  0.335232 -1.000000\n",
      " |      1 -1.000000  0.500000\n",
      " |      2  0.027753 -1.000000\n",
      " |      3  0.230930 -0.679613\n",
      " |      4  0.500000  0.500000\n",
      " |      >>> t\n",
      " |      0   -0.3\n",
      " |      1   -0.2\n",
      " |      2   -0.1\n",
      " |      3    0.0\n",
      " |      4    0.1\n",
      " |      dtype: float64\n",
      " |      >>> df.clip(t, t + 1, axis=0)\n",
      " |                0         1\n",
      " |      0  0.335232 -0.300000\n",
      " |      1 -0.200000  0.746646\n",
      " |      2  0.027753 -0.100000\n",
      " |      3  0.230930  0.000000\n",
      " |      4  1.100000  0.570967\n",
      " |  \n",
      " |  clip_lower(self, threshold, axis=None)\n",
      " |      Return copy of the input with values below given value(s) truncated.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      threshold : float or array_like\n",
      " |      axis : int or string axis name, optional\n",
      " |          Align object with threshold along the given axis.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      clip\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      clipped : same type as input\n",
      " |  \n",
      " |  clip_upper(self, threshold, axis=None)\n",
      " |      Return copy of input with values above given value(s) truncated.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      threshold : float or array_like\n",
      " |      axis : int or string axis name, optional\n",
      " |          Align object with threshold along the given axis.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      clip\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      clipped : same type as input\n",
      " |  \n",
      " |  consolidate(self, inplace=False)\n",
      " |      Compute NDFrame with \"consolidated\" internals (data of each dtype\n",
      " |      grouped together in a single ndarray). Mainly an internal API function,\n",
      " |      but available here to the savvy user\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      inplace : boolean, default False\n",
      " |          If False return new object, otherwise modify existing object\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      consolidated : type of caller\n",
      " |  \n",
      " |  convert_objects(self, convert_dates=True, convert_numeric=False, convert_timedeltas=True, copy=True)\n",
      " |      Deprecated.\n",
      " |      \n",
      " |      Attempt to infer better dtype for object columns\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      convert_dates : boolean, default True\n",
      " |          If True, convert to date where possible. If 'coerce', force\n",
      " |          conversion, with unconvertible values becoming NaT.\n",
      " |      convert_numeric : boolean, default False\n",
      " |          If True, attempt to coerce to numbers (including strings), with\n",
      " |          unconvertible values becoming NaN.\n",
      " |      convert_timedeltas : boolean, default True\n",
      " |          If True, convert to timedelta where possible. If 'coerce', force\n",
      " |          conversion, with unconvertible values becoming NaT.\n",
      " |      copy : boolean, default True\n",
      " |          If True, return a copy even if no copy is necessary (e.g. no\n",
      " |          conversion was done). Note: This is meant for internal use, and\n",
      " |          should not be confused with inplace.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.to_datetime : Convert argument to datetime.\n",
      " |      pandas.to_timedelta : Convert argument to timedelta.\n",
      " |      pandas.to_numeric : Return a fixed frequency timedelta index,\n",
      " |          with day as the default.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      converted : same as input object\n",
      " |  \n",
      " |  copy(self, deep=True)\n",
      " |      Make a copy of this objects data.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      deep : boolean or string, default True\n",
      " |          Make a deep copy, including a copy of the data and the indices.\n",
      " |          With ``deep=False`` neither the indices or the data are copied.\n",
      " |      \n",
      " |          Note that when ``deep=True`` data is copied, actual python objects\n",
      " |          will not be copied recursively, only the reference to the object.\n",
      " |          This is in contrast to ``copy.deepcopy`` in the Standard Library,\n",
      " |          which recursively copies object data.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      copy : type of caller\n",
      " |  \n",
      " |  describe(self, percentiles=None, include=None, exclude=None)\n",
      " |      Generate various summary statistics, excluding NaN values.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      percentiles : array-like, optional\n",
      " |          The percentiles to include in the output. Should all\n",
      " |          be in the interval [0, 1]. By default `percentiles` is\n",
      " |          [.25, .5, .75], returning the 25th, 50th, and 75th percentiles.\n",
      " |      include, exclude : list-like, 'all', or None (default)\n",
      " |          Specify the form of the returned result. Either:\n",
      " |      \n",
      " |          - None to both (default). The result will include only\n",
      " |            numeric-typed columns or, if none are, only categorical columns.\n",
      " |          - A list of dtypes or strings to be included/excluded.\n",
      " |            To select all numeric types use numpy numpy.number. To select\n",
      " |            categorical objects use type object. See also the select_dtypes\n",
      " |            documentation. eg. df.describe(include=['O'])\n",
      " |          - If include is the string 'all', the output column-set will\n",
      " |            match the input one.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      summary: NDFrame of summary statistics\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The output DataFrame index depends on the requested dtypes:\n",
      " |      \n",
      " |      For numeric dtypes, it will include: count, mean, std, min,\n",
      " |      max, and lower, 50, and upper percentiles.\n",
      " |      \n",
      " |      For object dtypes (e.g. timestamps or strings), the index\n",
      " |      will include the count, unique, most common, and frequency of the\n",
      " |      most common. Timestamps also include the first and last items.\n",
      " |      \n",
      " |      For mixed dtypes, the index will be the union of the corresponding\n",
      " |      output types. Non-applicable entries will be filled with NaN.\n",
      " |      Note that mixed-dtype outputs can only be returned from mixed-dtype\n",
      " |      inputs and appropriate use of the include/exclude arguments.\n",
      " |      \n",
      " |      If multiple values have the highest count, then the\n",
      " |      `count` and `most common` pair will be arbitrarily chosen from\n",
      " |      among those with the highest count.\n",
      " |      \n",
      " |      The include, exclude arguments are ignored for Series.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      DataFrame.select_dtypes\n",
      " |  \n",
      " |  drop(self, labels, axis=0, level=None, inplace=False, errors='raise')\n",
      " |      Return new object with labels in requested axis removed.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      labels : single label or list-like\n",
      " |      axis : int or axis name\n",
      " |      level : int or level name, default None\n",
      " |          For MultiIndex\n",
      " |      inplace : bool, default False\n",
      " |          If True, do operation inplace and return None.\n",
      " |      errors : {'ignore', 'raise'}, default 'raise'\n",
      " |          If 'ignore', suppress error and existing labels are dropped.\n",
      " |      \n",
      " |          .. versionadded:: 0.16.1\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      dropped : type of caller\n",
      " |  \n",
      " |  equals(self, other)\n",
      " |      Determines if two NDFrame objects contain the same elements. NaNs in\n",
      " |      the same location are considered equal.\n",
      " |  \n",
      " |  ffill(self, axis=None, inplace=False, limit=None, downcast=None)\n",
      " |      Synonym for NDFrame.fillna(method='ffill')\n",
      " |  \n",
      " |  filter(self, items=None, like=None, regex=None, axis=None)\n",
      " |      Subset rows or columns of dataframe according to labels in\n",
      " |      the specified index.\n",
      " |      \n",
      " |      Note that this routine does not filter a dataframe on its\n",
      " |      contents. The filter is applied to the labels of the index.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      items : list-like\n",
      " |          List of info axis to restrict to (must not all be present)\n",
      " |      like : string\n",
      " |          Keep info axis where \"arg in col == True\"\n",
      " |      regex : string (regular expression)\n",
      " |          Keep info axis with re.search(regex, col) == True\n",
      " |      axis : int or string axis name\n",
      " |          The axis to filter on.  By default this is the info axis,\n",
      " |          'index' for Series, 'columns' for DataFrame\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      same type as input object\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df\n",
      " |      one  two  three\n",
      " |      mouse     1    2      3\n",
      " |      rabbit    4    5      6\n",
      " |      \n",
      " |      >>> # select columns by name\n",
      " |      >>> df.filter(items=['one', 'three'])\n",
      " |      one  three\n",
      " |      mouse     1      3\n",
      " |      rabbit    4      6\n",
      " |      \n",
      " |      >>> # select columns by regular expression\n",
      " |      >>> df.filter(regex='e$', axis=1)\n",
      " |      one  three\n",
      " |      mouse     1      3\n",
      " |      rabbit    4      6\n",
      " |      \n",
      " |      >>> # select rows containing 'bbi'\n",
      " |      >>> df.filter(like='bbi', axis=0)\n",
      " |      one  two  three\n",
      " |      rabbit    4    5      6\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.DataFrame.select\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The ``items``, ``like``, and ``regex`` parameters are\n",
      " |      enforced to be mutually exclusive.\n",
      " |      \n",
      " |      ``axis`` defaults to the info axis that is used when indexing\n",
      " |      with ``[]``.\n",
      " |  \n",
      " |  first(self, offset)\n",
      " |      Convenience method for subsetting initial periods of time series data\n",
      " |      based on a date offset.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      offset : string, DateOffset, dateutil.relativedelta\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      ts.first('10D') -> First 10 days\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      subset : type of caller\n",
      " |  \n",
      " |  get(self, key, default=None)\n",
      " |      Get item from object for given key (DataFrame column, Panel slice,\n",
      " |      etc.). Returns default value if not found.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      key : object\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      value : type of items contained in object\n",
      " |  \n",
      " |  get_dtype_counts(self)\n",
      " |      Return the counts of dtypes in this object.\n",
      " |  \n",
      " |  get_ftype_counts(self)\n",
      " |      Return the counts of ftypes in this object.\n",
      " |  \n",
      " |  get_values(self)\n",
      " |      same as values (but handles sparseness conversions)\n",
      " |  \n",
      " |  groupby(self, by=None, axis=0, level=None, as_index=True, sort=True, group_keys=True, squeeze=False, **kwargs)\n",
      " |      Group series using mapper (dict or key function, apply given function\n",
      " |      to group, return result as series) or by a series of columns.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      by : mapping function / list of functions, dict, Series, or tuple /\n",
      " |          list of column names.\n",
      " |          Called on each element of the object index to determine the groups.\n",
      " |          If a dict or Series is passed, the Series or dict VALUES will be\n",
      " |          used to determine the groups\n",
      " |      axis : int, default 0\n",
      " |      level : int, level name, or sequence of such, default None\n",
      " |          If the axis is a MultiIndex (hierarchical), group by a particular\n",
      " |          level or levels\n",
      " |      as_index : boolean, default True\n",
      " |          For aggregated output, return object with group labels as the\n",
      " |          index. Only relevant for DataFrame input. as_index=False is\n",
      " |          effectively \"SQL-style\" grouped output\n",
      " |      sort : boolean, default True\n",
      " |          Sort group keys. Get better performance by turning this off.\n",
      " |          Note this does not influence the order of observations within each\n",
      " |          group.  groupby preserves the order of rows within each group.\n",
      " |      group_keys : boolean, default True\n",
      " |          When calling apply, add group keys to index to identify pieces\n",
      " |      squeeze : boolean, default False\n",
      " |          reduce the dimensionality of the return type if possible,\n",
      " |          otherwise return a consistent type\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      DataFrame results\n",
      " |      \n",
      " |      >>> data.groupby(func, axis=0).mean()\n",
      " |      >>> data.groupby(['col1', 'col2'])['col3'].mean()\n",
      " |      \n",
      " |      DataFrame with hierarchical index\n",
      " |      \n",
      " |      >>> data.groupby(['col1', 'col2']).mean()\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      GroupBy object\n",
      " |  \n",
      " |  head(self, n=5)\n",
      " |      Returns first n rows\n",
      " |  \n",
      " |  interpolate(self, method='linear', axis=0, limit=None, inplace=False, limit_direction='forward', downcast=None, **kwargs)\n",
      " |      Interpolate values according to different methods.\n",
      " |      \n",
      " |      Please note that only ``method='linear'`` is supported for\n",
      " |      DataFrames/Series with a MultiIndex.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      method : {'linear', 'time', 'index', 'values', 'nearest', 'zero',\n",
      " |                'slinear', 'quadratic', 'cubic', 'barycentric', 'krogh',\n",
      " |                'polynomial', 'spline', 'piecewise_polynomial',\n",
      " |                'from_derivatives', 'pchip', 'akima'}\n",
      " |      \n",
      " |          * 'linear': ignore the index and treat the values as equally\n",
      " |            spaced. This is the only method supported on MultiIndexes.\n",
      " |            default\n",
      " |          * 'time': interpolation works on daily and higher resolution\n",
      " |            data to interpolate given length of interval\n",
      " |          * 'index', 'values': use the actual numerical values of the index\n",
      " |          * 'nearest', 'zero', 'slinear', 'quadratic', 'cubic',\n",
      " |            'barycentric', 'polynomial' is passed to\n",
      " |            ``scipy.interpolate.interp1d``. Both 'polynomial' and 'spline'\n",
      " |            require that you also specify an `order` (int),\n",
      " |            e.g. df.interpolate(method='polynomial', order=4).\n",
      " |            These use the actual numerical values of the index.\n",
      " |          * 'krogh', 'piecewise_polynomial', 'spline', 'pchip' and 'akima' are all\n",
      " |            wrappers around the scipy interpolation methods of similar\n",
      " |            names. These use the actual numerical values of the index. See\n",
      " |            the scipy documentation for more on their behavior\n",
      " |            `here <http://docs.scipy.org/doc/scipy/reference/interpolate.html#univariate-interpolation>`__  # noqa\n",
      " |            `and here <http://docs.scipy.org/doc/scipy/reference/tutorial/interpolate.html>`__  # noqa\n",
      " |          * 'from_derivatives' refers to BPoly.from_derivatives which\n",
      " |            replaces 'piecewise_polynomial' interpolation method in scipy 0.18\n",
      " |      \n",
      " |          .. versionadded:: 0.18.1\n",
      " |      \n",
      " |             Added support for the 'akima' method\n",
      " |             Added interpolate method 'from_derivatives' which replaces\n",
      " |             'piecewise_polynomial' in scipy 0.18; backwards-compatible with\n",
      " |             scipy < 0.18\n",
      " |      \n",
      " |      axis : {0, 1}, default 0\n",
      " |          * 0: fill column-by-column\n",
      " |          * 1: fill row-by-row\n",
      " |      limit : int, default None.\n",
      " |          Maximum number of consecutive NaNs to fill.\n",
      " |      limit_direction : {'forward', 'backward', 'both'}, defaults to 'forward'\n",
      " |          If limit is specified, consecutive NaNs will be filled in this\n",
      " |          direction.\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      inplace : bool, default False\n",
      " |          Update the NDFrame in place if possible.\n",
      " |      downcast : optional, 'infer' or None, defaults to None\n",
      " |          Downcast dtypes if possible.\n",
      " |      kwargs : keyword arguments to pass on to the interpolating function.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      Series or DataFrame of same shape interpolated at the NaNs\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      reindex, replace, fillna\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      Filling in NaNs\n",
      " |      \n",
      " |      >>> s = pd.Series([0, 1, np.nan, 3])\n",
      " |      >>> s.interpolate()\n",
      " |      0    0\n",
      " |      1    1\n",
      " |      2    2\n",
      " |      3    3\n",
      " |      dtype: float64\n",
      " |  \n",
      " |  isnull(self)\n",
      " |      Return a boolean same-sized object indicating if the values are null.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      notnull : boolean inverse of isnull\n",
      " |  \n",
      " |  iterkv(self, *args, **kwargs)\n",
      " |      iteritems alias used to get around 2to3. Deprecated\n",
      " |  \n",
      " |  keys(self)\n",
      " |      Get the 'info axis' (see Indexing for more)\n",
      " |      \n",
      " |      This is index for Series, columns for DataFrame and major_axis for\n",
      " |      Panel.\n",
      " |  \n",
      " |  last(self, offset)\n",
      " |      Convenience method for subsetting final periods of time series data\n",
      " |      based on a date offset.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      offset : string, DateOffset, dateutil.relativedelta\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      ts.last('5M') -> Last 5 months\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      subset : type of caller\n",
      " |  \n",
      " |  mask(self, cond, other=nan, inplace=False, axis=None, level=None, try_cast=False, raise_on_error=True)\n",
      " |      Return an object of same shape as self and whose corresponding\n",
      " |      entries are from self where cond is False and otherwise are from\n",
      " |      other.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      cond : boolean NDFrame, array or callable\n",
      " |          If cond is callable, it is computed on the NDFrame and\n",
      " |          should return boolean NDFrame or array.\n",
      " |          The callable must not change input NDFrame\n",
      " |          (though pandas doesn't check it).\n",
      " |      \n",
      " |          .. versionadded:: 0.18.1\n",
      " |      \n",
      " |          A callable can be used as cond.\n",
      " |      \n",
      " |      other : scalar, NDFrame, or callable\n",
      " |          If other is callable, it is computed on the NDFrame and\n",
      " |          should return scalar or NDFrame.\n",
      " |          The callable must not change input NDFrame\n",
      " |          (though pandas doesn't check it).\n",
      " |      \n",
      " |          .. versionadded:: 0.18.1\n",
      " |      \n",
      " |          A callable can be used as other.\n",
      " |      \n",
      " |      inplace : boolean, default False\n",
      " |          Whether to perform the operation in place on the data\n",
      " |      axis : alignment axis if needed, default None\n",
      " |      level : alignment level if needed, default None\n",
      " |      try_cast : boolean, default False\n",
      " |          try to cast the result back to the input type (if possible),\n",
      " |      raise_on_error : boolean, default True\n",
      " |          Whether to raise on invalid data types (e.g. trying to where on\n",
      " |          strings)\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      wh : same type as caller\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The mask method is an application of the if-then idiom. For each\n",
      " |      element in the calling DataFrame, if ``cond`` is ``False`` the\n",
      " |      element is used; otherwise the corresponding element from the DataFrame\n",
      " |      ``other`` is used.\n",
      " |      \n",
      " |      The signature for :func:`DataFrame.where` differs from\n",
      " |      :func:`numpy.where`. Roughly ``df1.where(m, df2)`` is equivalent to\n",
      " |      ``np.where(m, df1, df2)``.\n",
      " |      \n",
      " |      For further details and examples see the ``mask`` documentation in\n",
      " |      :ref:`indexing <indexing.where_mask>`.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> s = pd.Series(range(5))\n",
      " |      >>> s.where(s > 0)\n",
      " |      0    NaN\n",
      " |      1    1.0\n",
      " |      2    2.0\n",
      " |      3    3.0\n",
      " |      4    4.0\n",
      " |      \n",
      " |      >>> df = pd.DataFrame(np.arange(10).reshape(-1, 2), columns=['A', 'B'])\n",
      " |      >>> m = df % 3 == 0\n",
      " |      >>> df.where(m, -df)\n",
      " |         A  B\n",
      " |      0  0 -1\n",
      " |      1 -2  3\n",
      " |      2 -4 -5\n",
      " |      3  6 -7\n",
      " |      4 -8  9\n",
      " |      >>> df.where(m, -df) == np.where(m, df, -df)\n",
      " |            A     B\n",
      " |      0  True  True\n",
      " |      1  True  True\n",
      " |      2  True  True\n",
      " |      3  True  True\n",
      " |      4  True  True\n",
      " |      >>> df.where(m, -df) == df.mask(~m, -df)\n",
      " |            A     B\n",
      " |      0  True  True\n",
      " |      1  True  True\n",
      " |      2  True  True\n",
      " |      3  True  True\n",
      " |      4  True  True\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      :func:`DataFrame.where`\n",
      " |  \n",
      " |  notnull(self)\n",
      " |      Return a boolean same-sized object indicating if the values are\n",
      " |      not null.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      isnull : boolean inverse of notnull\n",
      " |  \n",
      " |  pct_change(self, periods=1, fill_method='pad', limit=None, freq=None, **kwargs)\n",
      " |      Percent change over given number of periods.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      periods : int, default 1\n",
      " |          Periods to shift for forming percent change\n",
      " |      fill_method : str, default 'pad'\n",
      " |          How to handle NAs before computing percent changes\n",
      " |      limit : int, default None\n",
      " |          The number of consecutive NAs to fill before stopping\n",
      " |      freq : DateOffset, timedelta, or offset alias string, optional\n",
      " |          Increment to use from time series API (e.g. 'M' or BDay())\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      chg : NDFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      \n",
      " |      By default, the percentage change is calculated along the stat\n",
      " |      axis: 0, or ``Index``, for ``DataFrame`` and 1, or ``minor`` for\n",
      " |      ``Panel``. You can change this with the ``axis`` keyword argument.\n",
      " |  \n",
      " |  pipe(self, func, *args, **kwargs)\n",
      " |      Apply func(self, \\*args, \\*\\*kwargs)\n",
      " |      \n",
      " |      .. versionadded:: 0.16.2\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      func : function\n",
      " |          function to apply to the NDFrame.\n",
      " |          ``args``, and ``kwargs`` are passed into ``func``.\n",
      " |          Alternatively a ``(callable, data_keyword)`` tuple where\n",
      " |          ``data_keyword`` is a string indicating the keyword of\n",
      " |          ``callable`` that expects the NDFrame.\n",
      " |      args : positional arguments passed into ``func``.\n",
      " |      kwargs : a dictionary of keyword arguments passed into ``func``.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      object : the return type of ``func``.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      \n",
      " |      Use ``.pipe`` when chaining together functions that expect\n",
      " |      on Series or DataFrames. Instead of writing\n",
      " |      \n",
      " |      >>> f(g(h(df), arg1=a), arg2=b, arg3=c)\n",
      " |      \n",
      " |      You can write\n",
      " |      \n",
      " |      >>> (df.pipe(h)\n",
      " |      ...    .pipe(g, arg1=a)\n",
      " |      ...    .pipe(f, arg2=b, arg3=c)\n",
      " |      ... )\n",
      " |      \n",
      " |      If you have a function that takes the data as (say) the second\n",
      " |      argument, pass a tuple indicating which keyword expects the\n",
      " |      data. For example, suppose ``f`` takes its data as ``arg2``:\n",
      " |      \n",
      " |      >>> (df.pipe(h)\n",
      " |      ...    .pipe(g, arg1=a)\n",
      " |      ...    .pipe((f, 'arg2'), arg1=a, arg3=c)\n",
      " |      ...  )\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.DataFrame.apply\n",
      " |      pandas.DataFrame.applymap\n",
      " |      pandas.Series.map\n",
      " |  \n",
      " |  pop(self, item)\n",
      " |      Return item and drop from frame. Raise KeyError if not found.\n",
      " |  \n",
      " |  rank(self, axis=0, method='average', numeric_only=None, na_option='keep', ascending=True, pct=False)\n",
      " |      Compute numerical data ranks (1 through n) along axis. Equal values are\n",
      " |      assigned a rank that is the average of the ranks of those values\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      axis: {0 or 'index', 1 or 'columns'}, default 0\n",
      " |          index to direct ranking\n",
      " |      method : {'average', 'min', 'max', 'first', 'dense'}\n",
      " |          * average: average rank of group\n",
      " |          * min: lowest rank in group\n",
      " |          * max: highest rank in group\n",
      " |          * first: ranks assigned in order they appear in the array\n",
      " |          * dense: like 'min', but rank always increases by 1 between groups\n",
      " |      numeric_only : boolean, default None\n",
      " |          Include only float, int, boolean data. Valid only for DataFrame or\n",
      " |          Panel objects\n",
      " |      na_option : {'keep', 'top', 'bottom'}\n",
      " |          * keep: leave NA values where they are\n",
      " |          * top: smallest rank if ascending\n",
      " |          * bottom: smallest rank if descending\n",
      " |      ascending : boolean, default True\n",
      " |          False for ranks by high (1) to low (N)\n",
      " |      pct : boolean, default False\n",
      " |          Computes percentage rank of data\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      ranks : same type as caller\n",
      " |  \n",
      " |  reindex_like(self, other, method=None, copy=True, limit=None, tolerance=None)\n",
      " |      Return an object with matching indices to myself.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      other : Object\n",
      " |      method : string or None\n",
      " |      copy : boolean, default True\n",
      " |      limit : int, default None\n",
      " |          Maximum number of consecutive labels to fill for inexact matches.\n",
      " |      tolerance : optional\n",
      " |          Maximum distance between labels of the other object and this\n",
      " |          object for inexact matches.\n",
      " |      \n",
      " |          .. versionadded:: 0.17.0\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Like calling s.reindex(index=other.index, columns=other.columns,\n",
      " |                             method=...)\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      reindexed : same as input\n",
      " |  \n",
      " |  rename_axis(self, mapper, axis=0, copy=True, inplace=False)\n",
      " |      Alter index and / or columns using input function or functions.\n",
      " |      A scaler or list-like for ``mapper`` will alter the ``Index.name``\n",
      " |      or ``MultiIndex.names`` attribute.\n",
      " |      A function or dict for ``mapper`` will alter the labels.\n",
      " |      Function / dict values must be unique (1-to-1). Labels not contained in\n",
      " |      a dict / Series will be left as-is.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      mapper : scalar, list-like, dict-like or function, optional\n",
      " |      axis : int or string, default 0\n",
      " |      copy : boolean, default True\n",
      " |          Also copy underlying data\n",
      " |      inplace : boolean, default False\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      renamed : type of caller\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      pandas.NDFrame.rename\n",
      " |      pandas.Index.rename\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = pd.DataFrame({\"A\": [1, 2, 3], \"B\": [4, 5, 6]})\n",
      " |      >>> df.rename_axis(\"foo\")  # scalar, alters df.index.name\n",
      " |           A  B\n",
      " |      foo\n",
      " |      0    1  4\n",
      " |      1    2  5\n",
      " |      2    3  6\n",
      " |      >>> df.rename_axis(lambda x: 2 * x)  # function: alters labels\n",
      " |         A  B\n",
      " |      0  1  4\n",
      " |      2  2  5\n",
      " |      4  3  6\n",
      " |      >>> df.rename_axis({\"A\": \"ehh\", \"C\": \"see\"}, axis=\"columns\")  # mapping\n",
      " |         ehh  B\n",
      " |      0    1  4\n",
      " |      1    2  5\n",
      " |      2    3  6\n",
      " |  \n",
      " |  replace(self, to_replace=None, value=None, inplace=False, limit=None, regex=False, method='pad', axis=None)\n",
      " |      Replace values given in 'to_replace' with 'value'.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      to_replace : str, regex, list, dict, Series, numeric, or None\n",
      " |      \n",
      " |          * str or regex:\n",
      " |      \n",
      " |              - str: string exactly matching `to_replace` will be replaced\n",
      " |                with `value`\n",
      " |              - regex: regexs matching `to_replace` will be replaced with\n",
      " |                `value`\n",
      " |      \n",
      " |          * list of str, regex, or numeric:\n",
      " |      \n",
      " |              - First, if `to_replace` and `value` are both lists, they\n",
      " |                **must** be the same length.\n",
      " |              - Second, if ``regex=True`` then all of the strings in **both**\n",
      " |                lists will be interpreted as regexs otherwise they will match\n",
      " |                directly. This doesn't matter much for `value` since there\n",
      " |                are only a few possible substitution regexes you can use.\n",
      " |              - str and regex rules apply as above.\n",
      " |      \n",
      " |          * dict:\n",
      " |      \n",
      " |              - Nested dictionaries, e.g., {'a': {'b': nan}}, are read as\n",
      " |                follows: look in column 'a' for the value 'b' and replace it\n",
      " |                with nan. You can nest regular expressions as well. Note that\n",
      " |                column names (the top-level dictionary keys in a nested\n",
      " |                dictionary) **cannot** be regular expressions.\n",
      " |              - Keys map to column names and values map to substitution\n",
      " |                values. You can treat this as a special case of passing two\n",
      " |                lists except that you are specifying the column to search in.\n",
      " |      \n",
      " |          * None:\n",
      " |      \n",
      " |              - This means that the ``regex`` argument must be a string,\n",
      " |                compiled regular expression, or list, dict, ndarray or Series\n",
      " |                of such elements. If `value` is also ``None`` then this\n",
      " |                **must** be a nested dictionary or ``Series``.\n",
      " |      \n",
      " |          See the examples section for examples of each of these.\n",
      " |      value : scalar, dict, list, str, regex, default None\n",
      " |          Value to use to fill holes (e.g. 0), alternately a dict of values\n",
      " |          specifying which value to use for each column (columns not in the\n",
      " |          dict will not be filled). Regular expressions, strings and lists or\n",
      " |          dicts of such objects are also allowed.\n",
      " |      inplace : boolean, default False\n",
      " |          If True, in place. Note: this will modify any\n",
      " |          other views on this object (e.g. a column form a DataFrame).\n",
      " |          Returns the caller if this is True.\n",
      " |      limit : int, default None\n",
      " |          Maximum size gap to forward or backward fill\n",
      " |      regex : bool or same types as `to_replace`, default False\n",
      " |          Whether to interpret `to_replace` and/or `value` as regular\n",
      " |          expressions. If this is ``True`` then `to_replace` *must* be a\n",
      " |          string. Otherwise, `to_replace` must be ``None`` because this\n",
      " |          parameter will be interpreted as a regular expression or a list,\n",
      " |          dict, or array of regular expressions.\n",
      " |      method : string, optional, {'pad', 'ffill', 'bfill'}\n",
      " |          The method to use when for replacement, when ``to_replace`` is a\n",
      " |          ``list``.\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      NDFrame.reindex\n",
      " |      NDFrame.asfreq\n",
      " |      NDFrame.fillna\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      filled : NDFrame\n",
      " |      \n",
      " |      Raises\n",
      " |      ------\n",
      " |      AssertionError\n",
      " |          * If `regex` is not a ``bool`` and `to_replace` is not ``None``.\n",
      " |      TypeError\n",
      " |          * If `to_replace` is a ``dict`` and `value` is not a ``list``,\n",
      " |            ``dict``, ``ndarray``, or ``Series``\n",
      " |          * If `to_replace` is ``None`` and `regex` is not compilable into a\n",
      " |            regular expression or is a list, dict, ndarray, or Series.\n",
      " |      ValueError\n",
      " |          * If `to_replace` and `value` are ``list`` s or ``ndarray`` s, but\n",
      " |            they are not the same length.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      * Regex substitution is performed under the hood with ``re.sub``. The\n",
      " |        rules for substitution for ``re.sub`` are the same.\n",
      " |      * Regular expressions will only substitute on strings, meaning you\n",
      " |        cannot provide, for example, a regular expression matching floating\n",
      " |        point numbers and expect the columns in your frame that have a\n",
      " |        numeric dtype to be matched. However, if those floating point numbers\n",
      " |        *are* strings, then you can do this.\n",
      " |      * This method has *a lot* of options. You are encouraged to experiment\n",
      " |        and play with this method to gain intuition about how it works.\n",
      " |  \n",
      " |  resample(self, rule, how=None, axis=0, fill_method=None, closed=None, label=None, convention='start', kind=None, loffset=None, limit=None, base=0, on=None, level=None)\n",
      " |      Convenience method for frequency conversion and resampling of time\n",
      " |      series.  Object must have a datetime-like index (DatetimeIndex,\n",
      " |      PeriodIndex, or TimedeltaIndex), or pass datetime-like values\n",
      " |      to the on or level keyword.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      rule : string\n",
      " |          the offset string or object representing target conversion\n",
      " |      axis : int, optional, default 0\n",
      " |      closed : {'right', 'left'}\n",
      " |          Which side of bin interval is closed\n",
      " |      label : {'right', 'left'}\n",
      " |          Which bin edge label to label bucket with\n",
      " |      convention : {'start', 'end', 's', 'e'}\n",
      " |      loffset : timedelta\n",
      " |          Adjust the resampled time labels\n",
      " |      base : int, default 0\n",
      " |          For frequencies that evenly subdivide 1 day, the \"origin\" of the\n",
      " |          aggregated intervals. For example, for '5min' frequency, base could\n",
      " |          range from 0 through 4. Defaults to 0\n",
      " |      on : string, optional\n",
      " |          For a DataFrame, column to use instead of index for resampling.\n",
      " |          Column must be datetime-like.\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      level : string or int, optional\n",
      " |          For a MultiIndex, level (name or number) to use for\n",
      " |          resampling.  Level must be datetime-like.\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      To learn more about the offset strings, please see `this link\n",
      " |      <http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases>`__.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      Start by creating a series with 9 one minute timestamps.\n",
      " |      \n",
      " |      >>> index = pd.date_range('1/1/2000', periods=9, freq='T')\n",
      " |      >>> series = pd.Series(range(9), index=index)\n",
      " |      >>> series\n",
      " |      2000-01-01 00:00:00    0\n",
      " |      2000-01-01 00:01:00    1\n",
      " |      2000-01-01 00:02:00    2\n",
      " |      2000-01-01 00:03:00    3\n",
      " |      2000-01-01 00:04:00    4\n",
      " |      2000-01-01 00:05:00    5\n",
      " |      2000-01-01 00:06:00    6\n",
      " |      2000-01-01 00:07:00    7\n",
      " |      2000-01-01 00:08:00    8\n",
      " |      Freq: T, dtype: int64\n",
      " |      \n",
      " |      Downsample the series into 3 minute bins and sum the values\n",
      " |      of the timestamps falling into a bin.\n",
      " |      \n",
      " |      >>> series.resample('3T').sum()\n",
      " |      2000-01-01 00:00:00     3\n",
      " |      2000-01-01 00:03:00    12\n",
      " |      2000-01-01 00:06:00    21\n",
      " |      Freq: 3T, dtype: int64\n",
      " |      \n",
      " |      Downsample the series into 3 minute bins as above, but label each\n",
      " |      bin using the right edge instead of the left. Please note that the\n",
      " |      value in the bucket used as the label is not included in the bucket,\n",
      " |      which it labels. For example, in the original series the\n",
      " |      bucket ``2000-01-01 00:03:00`` contains the value 3, but the summed\n",
      " |      value in the resampled bucket with the label``2000-01-01 00:03:00``\n",
      " |      does not include 3 (if it did, the summed value would be 6, not 3).\n",
      " |      To include this value close the right side of the bin interval as\n",
      " |      illustrated in the example below this one.\n",
      " |      \n",
      " |      >>> series.resample('3T', label='right').sum()\n",
      " |      2000-01-01 00:03:00     3\n",
      " |      2000-01-01 00:06:00    12\n",
      " |      2000-01-01 00:09:00    21\n",
      " |      Freq: 3T, dtype: int64\n",
      " |      \n",
      " |      Downsample the series into 3 minute bins as above, but close the right\n",
      " |      side of the bin interval.\n",
      " |      \n",
      " |      >>> series.resample('3T', label='right', closed='right').sum()\n",
      " |      2000-01-01 00:00:00     0\n",
      " |      2000-01-01 00:03:00     6\n",
      " |      2000-01-01 00:06:00    15\n",
      " |      2000-01-01 00:09:00    15\n",
      " |      Freq: 3T, dtype: int64\n",
      " |      \n",
      " |      Upsample the series into 30 second bins.\n",
      " |      \n",
      " |      >>> series.resample('30S').asfreq()[0:5] #select first 5 rows\n",
      " |      2000-01-01 00:00:00     0\n",
      " |      2000-01-01 00:00:30   NaN\n",
      " |      2000-01-01 00:01:00     1\n",
      " |      2000-01-01 00:01:30   NaN\n",
      " |      2000-01-01 00:02:00     2\n",
      " |      Freq: 30S, dtype: float64\n",
      " |      \n",
      " |      Upsample the series into 30 second bins and fill the ``NaN``\n",
      " |      values using the ``pad`` method.\n",
      " |      \n",
      " |      >>> series.resample('30S').pad()[0:5]\n",
      " |      2000-01-01 00:00:00    0\n",
      " |      2000-01-01 00:00:30    0\n",
      " |      2000-01-01 00:01:00    1\n",
      " |      2000-01-01 00:01:30    1\n",
      " |      2000-01-01 00:02:00    2\n",
      " |      Freq: 30S, dtype: int64\n",
      " |      \n",
      " |      Upsample the series into 30 second bins and fill the\n",
      " |      ``NaN`` values using the ``bfill`` method.\n",
      " |      \n",
      " |      >>> series.resample('30S').bfill()[0:5]\n",
      " |      2000-01-01 00:00:00    0\n",
      " |      2000-01-01 00:00:30    1\n",
      " |      2000-01-01 00:01:00    1\n",
      " |      2000-01-01 00:01:30    2\n",
      " |      2000-01-01 00:02:00    2\n",
      " |      Freq: 30S, dtype: int64\n",
      " |      \n",
      " |      Pass a custom function via ``apply``\n",
      " |      \n",
      " |      >>> def custom_resampler(array_like):\n",
      " |      ...     return np.sum(array_like)+5\n",
      " |      \n",
      " |      >>> series.resample('3T').apply(custom_resampler)\n",
      " |      2000-01-01 00:00:00     8\n",
      " |      2000-01-01 00:03:00    17\n",
      " |      2000-01-01 00:06:00    26\n",
      " |      Freq: 3T, dtype: int64\n",
      " |  \n",
      " |  sample(self, n=None, frac=None, replace=False, weights=None, random_state=None, axis=None)\n",
      " |      Returns a random sample of items from an axis of object.\n",
      " |      \n",
      " |      .. versionadded:: 0.16.1\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      n : int, optional\n",
      " |          Number of items from axis to return. Cannot be used with `frac`.\n",
      " |          Default = 1 if `frac` = None.\n",
      " |      frac : float, optional\n",
      " |          Fraction of axis items to return. Cannot be used with `n`.\n",
      " |      replace : boolean, optional\n",
      " |          Sample with or without replacement. Default = False.\n",
      " |      weights : str or ndarray-like, optional\n",
      " |          Default 'None' results in equal probability weighting.\n",
      " |          If passed a Series, will align with target object on index. Index\n",
      " |          values in weights not found in sampled object will be ignored and\n",
      " |          index values in sampled object not in weights will be assigned\n",
      " |          weights of zero.\n",
      " |          If called on a DataFrame, will accept the name of a column\n",
      " |          when axis = 0.\n",
      " |          Unless weights are a Series, weights must be same length as axis\n",
      " |          being sampled.\n",
      " |          If weights do not sum to 1, they will be normalized to sum to 1.\n",
      " |          Missing values in the weights column will be treated as zero.\n",
      " |          inf and -inf values not allowed.\n",
      " |      random_state : int or numpy.random.RandomState, optional\n",
      " |          Seed for the random number generator (if int), or numpy RandomState\n",
      " |          object.\n",
      " |      axis : int or string, optional\n",
      " |          Axis to sample. Accepts axis number or name. Default is stat axis\n",
      " |          for given data type (0 for Series and DataFrames, 1 for Panels).\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      A new object of same type as caller.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      \n",
      " |      Generate an example ``Series`` and ``DataFrame``:\n",
      " |      \n",
      " |      >>> s = pd.Series(np.random.randn(50))\n",
      " |      >>> s.head()\n",
      " |      0   -0.038497\n",
      " |      1    1.820773\n",
      " |      2   -0.972766\n",
      " |      3   -1.598270\n",
      " |      4   -1.095526\n",
      " |      dtype: float64\n",
      " |      >>> df = pd.DataFrame(np.random.randn(50, 4), columns=list('ABCD'))\n",
      " |      >>> df.head()\n",
      " |                A         B         C         D\n",
      " |      0  0.016443 -2.318952 -0.566372 -1.028078\n",
      " |      1 -1.051921  0.438836  0.658280 -0.175797\n",
      " |      2 -1.243569 -0.364626 -0.215065  0.057736\n",
      " |      3  1.768216  0.404512 -0.385604 -1.457834\n",
      " |      4  1.072446 -1.137172  0.314194 -0.046661\n",
      " |      \n",
      " |      Next extract a random sample from both of these objects...\n",
      " |      \n",
      " |      3 random elements from the ``Series``:\n",
      " |      \n",
      " |      >>> s.sample(n=3)\n",
      " |      27   -0.994689\n",
      " |      55   -1.049016\n",
      " |      67   -0.224565\n",
      " |      dtype: float64\n",
      " |      \n",
      " |      And a random 10% of the ``DataFrame`` with replacement:\n",
      " |      \n",
      " |      >>> df.sample(frac=0.1, replace=True)\n",
      " |                 A         B         C         D\n",
      " |      35  1.981780  0.142106  1.817165 -0.290805\n",
      " |      49 -1.336199 -0.448634 -0.789640  0.217116\n",
      " |      40  0.823173 -0.078816  1.009536  1.015108\n",
      " |      15  1.421154 -0.055301 -1.922594 -0.019696\n",
      " |      6  -0.148339  0.832938  1.787600 -1.383767\n",
      " |  \n",
      " |  select(self, crit, axis=0)\n",
      " |      Return data corresponding to axis labels matching criteria\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      crit : function\n",
      " |          To be called on each index (label). Should return True or False\n",
      " |      axis : int\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      selection : type of caller\n",
      " |  \n",
      " |  set_axis(self, axis, labels)\n",
      " |      public verson of axis assignment\n",
      " |  \n",
      " |  slice_shift(self, periods=1, axis=0)\n",
      " |      Equivalent to `shift` without copying data. The shifted data will\n",
      " |      not include the dropped periods and the shifted axis will be smaller\n",
      " |      than the original.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      periods : int\n",
      " |          Number of periods to move, can be positive or negative\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      While the `slice_shift` is faster than `shift`, you may pay for it\n",
      " |      later during alignment.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      shifted : same type as caller\n",
      " |  \n",
      " |  squeeze(self, **kwargs)\n",
      " |      Squeeze length 1 dimensions.\n",
      " |  \n",
      " |  swapaxes(self, axis1, axis2, copy=True)\n",
      " |      Interchange axes and swap values axes appropriately\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      y : same as input\n",
      " |  \n",
      " |  tail(self, n=5)\n",
      " |      Returns last n rows\n",
      " |  \n",
      " |  take(self, indices, axis=0, convert=True, is_copy=True, **kwargs)\n",
      " |      Analogous to ndarray.take\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      indices : list / array of ints\n",
      " |      axis : int, default 0\n",
      " |      convert : translate neg to pos indices (default)\n",
      " |      is_copy : mark the returned frame as a copy\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      taken : type of caller\n",
      " |  \n",
      " |  to_clipboard(self, excel=None, sep=None, **kwargs)\n",
      " |      Attempt to write text representation of object to the system clipboard\n",
      " |      This can be pasted into Excel, for example.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      excel : boolean, defaults to True\n",
      " |              if True, use the provided separator, writing in a csv\n",
      " |              format for allowing easy pasting into excel.\n",
      " |              if False, write a string representation of the object\n",
      " |              to the clipboard\n",
      " |      sep : optional, defaults to tab\n",
      " |      other keywords are passed to to_csv\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      Requirements for your platform\n",
      " |        - Linux: xclip, or xsel (with gtk or PyQt4 modules)\n",
      " |        - Windows: none\n",
      " |        - OS X: none\n",
      " |  \n",
      " |  to_dense(self)\n",
      " |      Return dense representation of NDFrame (as opposed to sparse)\n",
      " |  \n",
      " |  to_hdf(self, path_or_buf, key, **kwargs)\n",
      " |      Write the contained data to an HDF5 file using HDFStore.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path_or_buf : the path (string) or HDFStore object\n",
      " |      key : string\n",
      " |          indentifier for the group in the store\n",
      " |      mode : optional, {'a', 'w', 'r+'}, default 'a'\n",
      " |      \n",
      " |        ``'w'``\n",
      " |            Write; a new file is created (an existing file with the same\n",
      " |            name would be deleted).\n",
      " |        ``'a'``\n",
      " |            Append; an existing file is opened for reading and writing,\n",
      " |            and if the file does not exist it is created.\n",
      " |        ``'r+'``\n",
      " |            It is similar to ``'a'``, but the file must already exist.\n",
      " |      format : 'fixed(f)|table(t)', default is 'fixed'\n",
      " |          fixed(f) : Fixed format\n",
      " |                     Fast writing/reading. Not-appendable, nor searchable\n",
      " |          table(t) : Table format\n",
      " |                     Write as a PyTables Table structure which may perform\n",
      " |                     worse but allow more flexible operations like searching\n",
      " |                     / selecting subsets of the data\n",
      " |      append : boolean, default False\n",
      " |          For Table formats, append the input data to the existing\n",
      " |      data_columns :  list of columns, or True, default None\n",
      " |          List of columns to create as indexed data columns for on-disk\n",
      " |          queries, or True to use all columns. By default only the axes\n",
      " |          of the object are indexed. See `here\n",
      " |          <http://pandas.pydata.org/pandas-docs/stable/io.html#query-via-data-columns>`__.\n",
      " |      \n",
      " |          Applicable only to format='table'.\n",
      " |      complevel : int, 1-9, default 0\n",
      " |          If a complib is specified compression will be applied\n",
      " |          where possible\n",
      " |      complib : {'zlib', 'bzip2', 'lzo', 'blosc', None}, default None\n",
      " |          If complevel is > 0 apply compression to objects written\n",
      " |          in the store wherever possible\n",
      " |      fletcher32 : bool, default False\n",
      " |          If applying compression use the fletcher32 checksum\n",
      " |      dropna : boolean, default False.\n",
      " |          If true, ALL nan rows will not be written to store.\n",
      " |  \n",
      " |  to_json(self, path_or_buf=None, orient=None, date_format='epoch', double_precision=10, force_ascii=True, date_unit='ms', default_handler=None, lines=False)\n",
      " |      Convert the object to a JSON string.\n",
      " |      \n",
      " |      Note NaN's and None will be converted to null and datetime objects\n",
      " |      will be converted to UNIX timestamps.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path_or_buf : the path or buffer to write the result string\n",
      " |          if this is None, return a StringIO of the converted string\n",
      " |      orient : string\n",
      " |      \n",
      " |          * Series\n",
      " |      \n",
      " |            - default is 'index'\n",
      " |            - allowed values are: {'split','records','index'}\n",
      " |      \n",
      " |          * DataFrame\n",
      " |      \n",
      " |            - default is 'columns'\n",
      " |            - allowed values are:\n",
      " |              {'split','records','index','columns','values'}\n",
      " |      \n",
      " |          * The format of the JSON string\n",
      " |      \n",
      " |            - split : dict like\n",
      " |              {index -> [index], columns -> [columns], data -> [values]}\n",
      " |            - records : list like\n",
      " |              [{column -> value}, ... , {column -> value}]\n",
      " |            - index : dict like {index -> {column -> value}}\n",
      " |            - columns : dict like {column -> {index -> value}}\n",
      " |            - values : just the values array\n",
      " |      \n",
      " |      date_format : {'epoch', 'iso'}\n",
      " |          Type of date conversion. `epoch` = epoch milliseconds,\n",
      " |          `iso`` = ISO8601, default is epoch.\n",
      " |      double_precision : The number of decimal places to use when encoding\n",
      " |          floating point values, default 10.\n",
      " |      force_ascii : force encoded string to be ASCII, default True.\n",
      " |      date_unit : string, default 'ms' (milliseconds)\n",
      " |          The time unit to encode to, governs timestamp and ISO8601\n",
      " |          precision.  One of 's', 'ms', 'us', 'ns' for second, millisecond,\n",
      " |          microsecond, and nanosecond respectively.\n",
      " |      default_handler : callable, default None\n",
      " |          Handler to call if object cannot otherwise be converted to a\n",
      " |          suitable format for JSON. Should receive a single argument which is\n",
      " |          the object to convert and return a serialisable object.\n",
      " |      lines : boolean, defalut False\n",
      " |          If 'orient' is 'records' write out line delimited json format. Will\n",
      " |          throw ValueError if incorrect 'orient' since others are not list\n",
      " |          like.\n",
      " |      \n",
      " |          .. versionadded:: 0.19.0\n",
      " |      \n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      same type as input object with filtered info axis\n",
      " |  \n",
      " |  to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs)\n",
      " |      msgpack (serialize) object to input file path\n",
      " |      \n",
      " |      THIS IS AN EXPERIMENTAL LIBRARY and the storage format\n",
      " |      may not be stable until a future release.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path : string File path, buffer-like, or None\n",
      " |          if None, return generated string\n",
      " |      append : boolean whether to append to an existing msgpack\n",
      " |          (default is False)\n",
      " |      compress : type of compressor (zlib or blosc), default to None (no\n",
      " |          compression)\n",
      " |  \n",
      " |  to_pickle(self, path)\n",
      " |      Pickle (serialize) object to input file path.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      path : string\n",
      " |          File path\n",
      " |  \n",
      " |  to_sql(self, name, con, flavor=None, schema=None, if_exists='fail', index=True, index_label=None, chunksize=None, dtype=None)\n",
      " |      Write records stored in a DataFrame to a SQL database.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      name : string\n",
      " |          Name of SQL table\n",
      " |      con : SQLAlchemy engine or DBAPI2 connection (legacy mode)\n",
      " |          Using SQLAlchemy makes it possible to use any DB supported by that\n",
      " |          library. If a DBAPI2 object, only sqlite3 is supported.\n",
      " |      flavor : 'sqlite', default None\n",
      " |          DEPRECATED: this parameter will be removed in a future version,\n",
      " |          as 'sqlite' is the only supported option if SQLAlchemy is not\n",
      " |          installed.\n",
      " |      schema : string, default None\n",
      " |          Specify the schema (if database flavor supports this). If None, use\n",
      " |          default schema.\n",
      " |      if_exists : {'fail', 'replace', 'append'}, default 'fail'\n",
      " |          - fail: If table exists, do nothing.\n",
      " |          - replace: If table exists, drop it, recreate it, and insert data.\n",
      " |          - append: If table exists, insert data. Create if does not exist.\n",
      " |      index : boolean, default True\n",
      " |          Write DataFrame index as a column.\n",
      " |      index_label : string or sequence, default None\n",
      " |          Column label for index column(s). If None is given (default) and\n",
      " |          `index` is True, then the index names are used.\n",
      " |          A sequence should be given if the DataFrame uses MultiIndex.\n",
      " |      chunksize : int, default None\n",
      " |          If not None, then rows will be written in batches of this size at a\n",
      " |          time.  If None, all rows will be written at once.\n",
      " |      dtype : dict of column name to SQL type, default None\n",
      " |          Optional specifying the datatype for columns. The SQL type should\n",
      " |          be a SQLAlchemy type, or a string for sqlite3 fallback connection.\n",
      " |  \n",
      " |  to_xarray(self)\n",
      " |      Return an xarray object from the pandas object.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      a DataArray for a Series\n",
      " |      a Dataset for a DataFrame\n",
      " |      a DataArray for higher dims\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df = pd.DataFrame({'A' : [1, 1, 2],\n",
      " |                             'B' : ['foo', 'bar', 'foo'],\n",
      " |                             'C' : np.arange(4.,7)})\n",
      " |      >>> df\n",
      " |         A    B    C\n",
      " |      0  1  foo  4.0\n",
      " |      1  1  bar  5.0\n",
      " |      2  2  foo  6.0\n",
      " |      \n",
      " |      >>> df.to_xarray()\n",
      " |      <xarray.Dataset>\n",
      " |      Dimensions:  (index: 3)\n",
      " |      Coordinates:\n",
      " |        * index    (index) int64 0 1 2\n",
      " |      Data variables:\n",
      " |          A        (index) int64 1 1 2\n",
      " |          B        (index) object 'foo' 'bar' 'foo'\n",
      " |          C        (index) float64 4.0 5.0 6.0\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'A' : [1, 1, 2],\n",
      " |                             'B' : ['foo', 'bar', 'foo'],\n",
      " |                             'C' : np.arange(4.,7)}\n",
      " |                           ).set_index(['B','A'])\n",
      " |      >>> df\n",
      " |               C\n",
      " |      B   A\n",
      " |      foo 1  4.0\n",
      " |      bar 1  5.0\n",
      " |      foo 2  6.0\n",
      " |      \n",
      " |      >>> df.to_xarray()\n",
      " |      <xarray.Dataset>\n",
      " |      Dimensions:  (A: 2, B: 2)\n",
      " |      Coordinates:\n",
      " |        * B        (B) object 'bar' 'foo'\n",
      " |        * A        (A) int64 1 2\n",
      " |      Data variables:\n",
      " |          C        (B, A) float64 5.0 nan 4.0 6.0\n",
      " |      \n",
      " |      >>> p = pd.Panel(np.arange(24).reshape(4,3,2),\n",
      " |                       items=list('ABCD'),\n",
      " |                       major_axis=pd.date_range('20130101', periods=3),\n",
      " |                       minor_axis=['first', 'second'])\n",
      " |      >>> p\n",
      " |      <class 'pandas.core.panel.Panel'>\n",
      " |      Dimensions: 4 (items) x 3 (major_axis) x 2 (minor_axis)\n",
      " |      Items axis: A to D\n",
      " |      Major_axis axis: 2013-01-01 00:00:00 to 2013-01-03 00:00:00\n",
      " |      Minor_axis axis: first to second\n",
      " |      \n",
      " |      >>> p.to_xarray()\n",
      " |      <xarray.DataArray (items: 4, major_axis: 3, minor_axis: 2)>\n",
      " |      array([[[ 0,  1],\n",
      " |              [ 2,  3],\n",
      " |              [ 4,  5]],\n",
      " |             [[ 6,  7],\n",
      " |              [ 8,  9],\n",
      " |              [10, 11]],\n",
      " |             [[12, 13],\n",
      " |              [14, 15],\n",
      " |              [16, 17]],\n",
      " |             [[18, 19],\n",
      " |              [20, 21],\n",
      " |              [22, 23]]])\n",
      " |      Coordinates:\n",
      " |        * items       (items) object 'A' 'B' 'C' 'D'\n",
      " |        * major_axis  (major_axis) datetime64[ns] 2013-01-01 2013-01-02 2013-01-03  # noqa\n",
      " |        * minor_axis  (minor_axis) object 'first' 'second'\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      See the `xarray docs <http://xarray.pydata.org/en/stable/>`__\n",
      " |  \n",
      " |  truncate(self, before=None, after=None, axis=None, copy=True)\n",
      " |      Truncates a sorted NDFrame before and/or after some particular\n",
      " |      index value. If the axis contains only datetime values, before/after\n",
      " |      parameters are converted to datetime values.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      before : date\n",
      " |          Truncate before index value\n",
      " |      after : date\n",
      " |          Truncate after index value\n",
      " |      axis : the truncation axis, defaults to the stat axis\n",
      " |      copy : boolean, default is True,\n",
      " |          return a copy of the truncated section\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      truncated : type of caller\n",
      " |  \n",
      " |  tshift(self, periods=1, freq=None, axis=0)\n",
      " |      Shift the time index, using the index's frequency if available.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      periods : int\n",
      " |          Number of periods to move, can be positive or negative\n",
      " |      freq : DateOffset, timedelta, or time rule string, default None\n",
      " |          Increment to use from the tseries module or time rule (e.g. 'EOM')\n",
      " |      axis : int or basestring\n",
      " |          Corresponds to the axis that contains the Index\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      If freq is not specified then tries to use the freq or inferred_freq\n",
      " |      attributes of the index. If neither of those attributes exist, a\n",
      " |      ValueError is thrown\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      shifted : NDFrame\n",
      " |  \n",
      " |  tz_convert(self, tz, axis=0, level=None, copy=True)\n",
      " |      Convert tz-aware axis to target time zone.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      tz : string or pytz.timezone object\n",
      " |      axis : the axis to convert\n",
      " |      level : int, str, default None\n",
      " |          If axis ia a MultiIndex, convert a specific level. Otherwise\n",
      " |          must be None\n",
      " |      copy : boolean, default True\n",
      " |          Also make a copy of the underlying data\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      \n",
      " |      Raises\n",
      " |      ------\n",
      " |      TypeError\n",
      " |          If the axis is tz-naive.\n",
      " |  \n",
      " |  tz_localize(self, tz, axis=0, level=None, copy=True, ambiguous='raise')\n",
      " |      Localize tz-naive TimeSeries to target time zone.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      tz : string or pytz.timezone object\n",
      " |      axis : the axis to localize\n",
      " |      level : int, str, default None\n",
      " |          If axis ia a MultiIndex, localize a specific level. Otherwise\n",
      " |          must be None\n",
      " |      copy : boolean, default True\n",
      " |          Also make a copy of the underlying data\n",
      " |      ambiguous : 'infer', bool-ndarray, 'NaT', default 'raise'\n",
      " |          - 'infer' will attempt to infer fall dst-transition hours based on\n",
      " |            order\n",
      " |          - bool-ndarray where True signifies a DST time, False designates\n",
      " |            a non-DST time (note that this flag is only applicable for\n",
      " |            ambiguous times)\n",
      " |          - 'NaT' will return NaT where there are ambiguous times\n",
      " |          - 'raise' will raise an AmbiguousTimeError if there are ambiguous\n",
      " |            times\n",
      " |      infer_dst : boolean, default False (DEPRECATED)\n",
      " |          Attempt to infer fall dst-transition hours based on order\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      \n",
      " |      Raises\n",
      " |      ------\n",
      " |      TypeError\n",
      " |          If the TimeSeries is tz-aware and tz is not None.\n",
      " |  \n",
      " |  where(self, cond, other=nan, inplace=False, axis=None, level=None, try_cast=False, raise_on_error=True)\n",
      " |      Return an object of same shape as self and whose corresponding\n",
      " |      entries are from self where cond is True and otherwise are from\n",
      " |      other.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      cond : boolean NDFrame, array or callable\n",
      " |          If cond is callable, it is computed on the NDFrame and\n",
      " |          should return boolean NDFrame or array.\n",
      " |          The callable must not change input NDFrame\n",
      " |          (though pandas doesn't check it).\n",
      " |      \n",
      " |          .. versionadded:: 0.18.1\n",
      " |      \n",
      " |          A callable can be used as cond.\n",
      " |      \n",
      " |      other : scalar, NDFrame, or callable\n",
      " |          If other is callable, it is computed on the NDFrame and\n",
      " |          should return scalar or NDFrame.\n",
      " |          The callable must not change input NDFrame\n",
      " |          (though pandas doesn't check it).\n",
      " |      \n",
      " |          .. versionadded:: 0.18.1\n",
      " |      \n",
      " |          A callable can be used as other.\n",
      " |      \n",
      " |      inplace : boolean, default False\n",
      " |          Whether to perform the operation in place on the data\n",
      " |      axis : alignment axis if needed, default None\n",
      " |      level : alignment level if needed, default None\n",
      " |      try_cast : boolean, default False\n",
      " |          try to cast the result back to the input type (if possible),\n",
      " |      raise_on_error : boolean, default True\n",
      " |          Whether to raise on invalid data types (e.g. trying to where on\n",
      " |          strings)\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      wh : same type as caller\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The where method is an application of the if-then idiom. For each\n",
      " |      element in the calling DataFrame, if ``cond`` is ``True`` the\n",
      " |      element is used; otherwise the corresponding element from the DataFrame\n",
      " |      ``other`` is used.\n",
      " |      \n",
      " |      The signature for :func:`DataFrame.where` differs from\n",
      " |      :func:`numpy.where`. Roughly ``df1.where(m, df2)`` is equivalent to\n",
      " |      ``np.where(m, df1, df2)``.\n",
      " |      \n",
      " |      For further details and examples see the ``where`` documentation in\n",
      " |      :ref:`indexing <indexing.where_mask>`.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> s = pd.Series(range(5))\n",
      " |      >>> s.where(s > 0)\n",
      " |      0    NaN\n",
      " |      1    1.0\n",
      " |      2    2.0\n",
      " |      3    3.0\n",
      " |      4    4.0\n",
      " |      \n",
      " |      >>> df = pd.DataFrame(np.arange(10).reshape(-1, 2), columns=['A', 'B'])\n",
      " |      >>> m = df % 3 == 0\n",
      " |      >>> df.where(m, -df)\n",
      " |         A  B\n",
      " |      0  0 -1\n",
      " |      1 -2  3\n",
      " |      2 -4 -5\n",
      " |      3  6 -7\n",
      " |      4 -8  9\n",
      " |      >>> df.where(m, -df) == np.where(m, df, -df)\n",
      " |            A     B\n",
      " |      0  True  True\n",
      " |      1  True  True\n",
      " |      2  True  True\n",
      " |      3  True  True\n",
      " |      4  True  True\n",
      " |      >>> df.where(m, -df) == df.mask(~m, -df)\n",
      " |            A     B\n",
      " |      0  True  True\n",
      " |      1  True  True\n",
      " |      2  True  True\n",
      " |      3  True  True\n",
      " |      4  True  True\n",
      " |      \n",
      " |      See Also\n",
      " |      --------\n",
      " |      :func:`DataFrame.mask`\n",
      " |  \n",
      " |  xs(self, key, axis=0, level=None, drop_level=True)\n",
      " |      Returns a cross-section (row(s) or column(s)) from the\n",
      " |      Series/DataFrame. Defaults to cross-section on the rows (axis=0).\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      key : object\n",
      " |          Some label contained in the index, or partially in a MultiIndex\n",
      " |      axis : int, default 0\n",
      " |          Axis to retrieve cross-section on\n",
      " |      level : object, defaults to first n levels (n=1 or len(key))\n",
      " |          In case of a key partially contained in a MultiIndex, indicate\n",
      " |          which levels are used. Levels can be referred by label or position.\n",
      " |      drop_level : boolean, default True\n",
      " |          If False, returns object with same levels as self.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      >>> df\n",
      " |         A  B  C\n",
      " |      a  4  5  2\n",
      " |      b  4  0  9\n",
      " |      c  9  7  3\n",
      " |      >>> df.xs('a')\n",
      " |      A    4\n",
      " |      B    5\n",
      " |      C    2\n",
      " |      Name: a\n",
      " |      >>> df.xs('C', axis=1)\n",
      " |      a    2\n",
      " |      b    9\n",
      " |      c    3\n",
      " |      Name: C\n",
      " |      \n",
      " |      >>> df\n",
      " |                          A  B  C  D\n",
      " |      first second third\n",
      " |      bar   one    1      4  1  8  9\n",
      " |            two    1      7  5  5  0\n",
      " |      baz   one    1      6  6  8  0\n",
      " |            three  2      5  3  5  3\n",
      " |      >>> df.xs(('baz', 'three'))\n",
      " |             A  B  C  D\n",
      " |      third\n",
      " |      2      5  3  5  3\n",
      " |      >>> df.xs('one', level=1)\n",
      " |                   A  B  C  D\n",
      " |      first third\n",
      " |      bar   1      4  1  8  9\n",
      " |      baz   1      6  6  8  0\n",
      " |      >>> df.xs(('baz', 2), level=[0, 'third'])\n",
      " |              A  B  C  D\n",
      " |      second\n",
      " |      three   5  3  5  3\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      xs : Series or DataFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      xs is only for getting, not setting values.\n",
      " |      \n",
      " |      MultiIndex Slicers is a generic way to get/set values on any level or\n",
      " |      levels.  It is a superset of xs functionality, see\n",
      " |      :ref:`MultiIndex Slicers <advanced.mi_slicers>`\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from pandas.core.generic.NDFrame:\n",
      " |  \n",
      " |  at\n",
      " |      Fast label-based scalar accessor\n",
      " |      \n",
      " |      Similarly to ``loc``, ``at`` provides **label** based scalar lookups.\n",
      " |      You can also set using these indexers.\n",
      " |  \n",
      " |  blocks\n",
      " |      Internal property, property synonym for as_blocks()\n",
      " |  \n",
      " |  dtypes\n",
      " |      Return the dtypes in this object.\n",
      " |  \n",
      " |  empty\n",
      " |      True if NDFrame is entirely empty [no items], meaning any of the\n",
      " |      axes are of length 0.\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      If NDFrame contains only NaNs, it is still not considered empty. See\n",
      " |      the example below.\n",
      " |      \n",
      " |      Examples\n",
      " |      --------\n",
      " |      An example of an actual empty DataFrame. Notice the index is empty:\n",
      " |      \n",
      " |      >>> df_empty = pd.DataFrame({'A' : []})\n",
      " |      >>> df_empty\n",
      " |      Empty DataFrame\n",
      " |      Columns: [A]\n",
      " |      Index: []\n",
      " |      >>> df_empty.empty\n",
      " |      True\n",
      " |      \n",
      " |      If we only have NaNs in our DataFrame, it is not considered empty! We\n",
      " |      will need to drop the NaNs to make the DataFrame empty:\n",
      " |      \n",
      " |      >>> df = pd.DataFrame({'A' : [np.nan]})\n",
      " |      >>> df\n",
      " |          A\n",
      " |      0 NaN\n",
      " |      >>> df.empty\n",
      " |      False\n",
      " |      >>> df.dropna().empty\n",
      " |      True\n",
      " |      \n",
      " |      See also\n",
      " |      --------\n",
      " |      pandas.Series.dropna\n",
      " |      pandas.DataFrame.dropna\n",
      " |  \n",
      " |  ftypes\n",
      " |      Return the ftypes (indication of sparse/dense and dtype)\n",
      " |      in this object.\n",
      " |  \n",
      " |  iat\n",
      " |      Fast integer location scalar accessor.\n",
      " |      \n",
      " |      Similarly to ``iloc``, ``iat`` provides **integer** based lookups.\n",
      " |      You can also set using these indexers.\n",
      " |  \n",
      " |  iloc\n",
      " |      Purely integer-location based indexing for selection by position.\n",
      " |      \n",
      " |      ``.iloc[]`` is primarily integer position based (from ``0`` to\n",
      " |      ``length-1`` of the axis), but may also be used with a boolean\n",
      " |      array.\n",
      " |      \n",
      " |      Allowed inputs are:\n",
      " |      \n",
      " |      - An integer, e.g. ``5``.\n",
      " |      - A list or array of integers, e.g. ``[4, 3, 0]``.\n",
      " |      - A slice object with ints, e.g. ``1:7``.\n",
      " |      - A boolean array.\n",
      " |      - A ``callable`` function with one argument (the calling Series, DataFrame\n",
      " |        or Panel) and that returns valid output for indexing (one of the above)\n",
      " |      \n",
      " |      ``.iloc`` will raise ``IndexError`` if a requested indexer is\n",
      " |      out-of-bounds, except *slice* indexers which allow out-of-bounds\n",
      " |      indexing (this conforms with python/numpy *slice* semantics).\n",
      " |      \n",
      " |      See more at :ref:`Selection by Position <indexing.integer>`\n",
      " |  \n",
      " |  ix\n",
      " |      A primarily label-location based indexer, with integer position\n",
      " |      fallback.\n",
      " |      \n",
      " |      ``.ix[]`` supports mixed integer and label based access. It is\n",
      " |      primarily label based, but will fall back to integer positional\n",
      " |      access unless the corresponding axis is of integer type.\n",
      " |      \n",
      " |      ``.ix`` is the most general indexer and will support any of the\n",
      " |      inputs in ``.loc`` and ``.iloc``. ``.ix`` also supports floating\n",
      " |      point label schemes. ``.ix`` is exceptionally useful when dealing\n",
      " |      with mixed positional and label based hierachical indexes.\n",
      " |      \n",
      " |      However, when an axis is integer based, ONLY label based access\n",
      " |      and not positional access is supported. Thus, in such cases, it's\n",
      " |      usually better to be explicit and use ``.iloc`` or ``.loc``.\n",
      " |      \n",
      " |      See more at :ref:`Advanced Indexing <advanced>`.\n",
      " |  \n",
      " |  loc\n",
      " |      Purely label-location based indexer for selection by label.\n",
      " |      \n",
      " |      ``.loc[]`` is primarily label based, but may also be used with a\n",
      " |      boolean array.\n",
      " |      \n",
      " |      Allowed inputs are:\n",
      " |      \n",
      " |      - A single label, e.g. ``5`` or ``'a'``, (note that ``5`` is\n",
      " |        interpreted as a *label* of the index, and **never** as an\n",
      " |        integer position along the index).\n",
      " |      - A list or array of labels, e.g. ``['a', 'b', 'c']``.\n",
      " |      - A slice object with labels, e.g. ``'a':'f'`` (note that contrary\n",
      " |        to usual python slices, **both** the start and the stop are included!).\n",
      " |      - A boolean array.\n",
      " |      - A ``callable`` function with one argument (the calling Series, DataFrame\n",
      " |        or Panel) and that returns valid output for indexing (one of the above)\n",
      " |      \n",
      " |      ``.loc`` will raise a ``KeyError`` when the items are not found.\n",
      " |      \n",
      " |      See more at :ref:`Selection by Label <indexing.label>`\n",
      " |  \n",
      " |  ndim\n",
      " |      Number of axes / array dimensions\n",
      " |  \n",
      " |  size\n",
      " |      number of elements in the NDFrame\n",
      " |  \n",
      " |  values\n",
      " |      Numpy representation of NDFrame\n",
      " |      \n",
      " |      Notes\n",
      " |      -----\n",
      " |      The dtype will be a lower-common-denominator dtype (implicit\n",
      " |      upcasting); that is to say if the dtypes (even of numeric types)\n",
      " |      are mixed, the one that accommodates all will be chosen. Use this\n",
      " |      with care if you are not dealing with the blocks.\n",
      " |      \n",
      " |      e.g. If the dtypes are float16 and float32, dtype will be upcast to\n",
      " |      float32.  If dtypes are int32 and uint8, dtype will be upcast to\n",
      " |      int32. By numpy.find_common_type convention, mixing int64 and uint64\n",
      " |      will result in a flot64 dtype.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes inherited from pandas.core.generic.NDFrame:\n",
      " |  \n",
      " |  is_copy = None\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from pandas.core.base.PandasObject:\n",
      " |  \n",
      " |  __dir__(self)\n",
      " |      Provide method name lookup and completion\n",
      " |      Only provide 'public' methods\n",
      " |  \n",
      " |  __sizeof__(self)\n",
      " |      Generates the total memory usage for a object that returns\n",
      " |      either a value or Series of values\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from pandas.core.base.StringMixin:\n",
      " |  \n",
      " |  __bytes__(self)\n",
      " |      Return a string representation for a particular object.\n",
      " |      \n",
      " |      Invoked by bytes(obj) in py3 only.\n",
      " |      Yields a bytestring in both py2/py3.\n",
      " |  \n",
      " |  __repr__(self)\n",
      " |      Return a string representation for a particular object.\n",
      " |      \n",
      " |      Yields Bytestring in Py2, Unicode String in py3.\n",
      " |  \n",
      " |  __str__(self)\n",
      " |      Return a string representation for a particular Object\n",
      " |      \n",
      " |      Invoked by str(df) in both py2/py3.\n",
      " |      Yields Bytestring in Py2, Unicode String in py3.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from pandas.core.base.StringMixin:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>Gene Completeness</th>\n",
       "      <th>Cohort Origin</th>\n",
       "      <th>Taxonomic Annotation(Phylum Level)</th>\n",
       "      <th>Taxonomic Annotation(Genus Level)</th>\n",
       "      <th>KEGG Annotation</th>\n",
       "      <th>eggNOG Annotation</th>\n",
       "      <th>Sample Occurence Frequency</th>\n",
       "      <th>Individual Occurence Frequency</th>\n",
       "      <th>KEGG Functional Categories</th>\n",
       "      <th>eggNOG Functional Categories</th>\n",
       "      <th>Cohort Assembled</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88836</th>\n",
       "      <td>9865705</td>\n",
       "      <td>V1.CD6-0-PT_GL0096973</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476498</th>\n",
       "      <td>9862460</td>\n",
       "      <td>O2.UC4-2_GL0053162</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.021310</td>\n",
       "      <td>0.020561</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151345</th>\n",
       "      <td>9862491</td>\n",
       "      <td>O2.UC43-0_GL0033611</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>0.014953</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34192</th>\n",
       "      <td>9877190</td>\n",
       "      <td>159551223-stool2_revised_C916401_1_gene84247</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Eubacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.014207</td>\n",
       "      <td>0.012150</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331083</th>\n",
       "      <td>9860951</td>\n",
       "      <td>O2.UC21-1_GL0136539</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.059195</td>\n",
       "      <td>0.061682</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280181</th>\n",
       "      <td>9857596</td>\n",
       "      <td>MH0415_GL0056639</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.020521</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483179</th>\n",
       "      <td>9878379</td>\n",
       "      <td>763577454-stool2_revised_scaffold23171_1_gene1...</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.169692</td>\n",
       "      <td>0.157944</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370156</th>\n",
       "      <td>9853996</td>\n",
       "      <td>MH0335_GL0009364</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409047</th>\n",
       "      <td>9860917</td>\n",
       "      <td>O2.UC2-0_GL0121235</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34135</th>\n",
       "      <td>9872403</td>\n",
       "      <td>N003A_GL0007810</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408969</th>\n",
       "      <td>9869351</td>\n",
       "      <td>V1.UC52-0_GL0023463</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81900</th>\n",
       "      <td>9855675</td>\n",
       "      <td>MH0373_GL0182676</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.022889</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15105</th>\n",
       "      <td>9848328</td>\n",
       "      <td>MH0143_GL0088163</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53716</th>\n",
       "      <td>9877737</td>\n",
       "      <td>160502038-stool1_revised_C976100_1_gene185746</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4930</th>\n",
       "      <td>9873886</td>\n",
       "      <td>T2D-113A_GL0111098</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.115233</td>\n",
       "      <td>0.116822</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225835</th>\n",
       "      <td>9847115</td>\n",
       "      <td>MH0100_GL0041020</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.047356</td>\n",
       "      <td>0.043925</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409075</th>\n",
       "      <td>9872514</td>\n",
       "      <td>N017A_GL0080793</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.072612</td>\n",
       "      <td>0.065421</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151252</th>\n",
       "      <td>9853377</td>\n",
       "      <td>MH0318_GL0022747</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.000789</td>\n",
       "      <td>0.000935</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53803</th>\n",
       "      <td>9856149</td>\n",
       "      <td>MH0384_GL0070245</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.001579</td>\n",
       "      <td>0.001869</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151236</th>\n",
       "      <td>9855061</td>\n",
       "      <td>MH0362_GL0041997</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.089187</td>\n",
       "      <td>0.091589</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197036</th>\n",
       "      <td>9846712</td>\n",
       "      <td>MH0086_GL0082324</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.140489</td>\n",
       "      <td>0.115888</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34125</th>\n",
       "      <td>9858623</td>\n",
       "      <td>MH0435_GL0149202</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.089976</td>\n",
       "      <td>0.088785</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197042</th>\n",
       "      <td>9859108</td>\n",
       "      <td>MH0445_GL0066112</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.033149</td>\n",
       "      <td>0.036449</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116515</th>\n",
       "      <td>9879793</td>\n",
       "      <td>411459.RUMOBE_01786</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>SP</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Blautia</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369973</th>\n",
       "      <td>9854153</td>\n",
       "      <td>MH0339_GL0008776</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.101815</td>\n",
       "      <td>0.079439</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369662</th>\n",
       "      <td>9849875</td>\n",
       "      <td>MH0196_GL0163961</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.013418</td>\n",
       "      <td>0.014019</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369663</th>\n",
       "      <td>9852450</td>\n",
       "      <td>MH0276_GL0235765</td>\n",
       "      <td>102</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320790</th>\n",
       "      <td>9845699</td>\n",
       "      <td>MH0022_GL0010478</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.005525</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4686</th>\n",
       "      <td>9874722</td>\n",
       "      <td>T2D-29A_GL0000548</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.160221</td>\n",
       "      <td>0.163551</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296144</th>\n",
       "      <td>9849405</td>\n",
       "      <td>MH0185_GL0064951</td>\n",
       "      <td>102</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.228887</td>\n",
       "      <td>0.232710</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245450</th>\n",
       "      <td>412</td>\n",
       "      <td>V1.FI27_GL0164628</td>\n",
       "      <td>15222</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.053670</td>\n",
       "      <td>0.055140</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285369</th>\n",
       "      <td>399</td>\n",
       "      <td>MH0150_GL0033854</td>\n",
       "      <td>15306</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13734</td>\n",
       "      <td>bactNOG74050</td>\n",
       "      <td>0.823204</td>\n",
       "      <td>0.835514</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101624</th>\n",
       "      <td>392</td>\n",
       "      <td>MH0264_GL0016567</td>\n",
       "      <td>15447</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.009471</td>\n",
       "      <td>0.010280</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183807</th>\n",
       "      <td>377</td>\n",
       "      <td>MH0404_GL0080882</td>\n",
       "      <td>15591</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00558</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.013084</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59169</th>\n",
       "      <td>349</td>\n",
       "      <td>159571453-stool2_revised_scaffold27522_1_gene3...</td>\n",
       "      <td>15945</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>USA</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K07279</td>\n",
       "      <td>COG3468</td>\n",
       "      <td>0.048934</td>\n",
       "      <td>0.050467</td>\n",
       "      <td>Cellular Processes and Signaling</td>\n",
       "      <td>Cell wall/membrane/envelope biogenesis;Intrace...</td>\n",
       "      <td>USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267886</th>\n",
       "      <td>340</td>\n",
       "      <td>V1.FI25_GL0154021</td>\n",
       "      <td>16074</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K15125</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.013418</td>\n",
       "      <td>0.015888</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355168</th>\n",
       "      <td>331</td>\n",
       "      <td>V1.FI28_GL0063861</td>\n",
       "      <td>16161</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00599</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.028037</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215409</th>\n",
       "      <td>321</td>\n",
       "      <td>V1.FI14_GL0255454</td>\n",
       "      <td>16278</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.022099</td>\n",
       "      <td>0.026168</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470957</th>\n",
       "      <td>318</td>\n",
       "      <td>O2.UC5-0_GL0058609</td>\n",
       "      <td>16311</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.032360</td>\n",
       "      <td>0.023364</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189070</th>\n",
       "      <td>317</td>\n",
       "      <td>O2.UC40-1_GL0020660</td>\n",
       "      <td>16344</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.047356</td>\n",
       "      <td>0.042056</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164915</th>\n",
       "      <td>312</td>\n",
       "      <td>V1.UC58-0_GL0107276</td>\n",
       "      <td>16419</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00558</td>\n",
       "      <td>bactNOG23571</td>\n",
       "      <td>0.011839</td>\n",
       "      <td>0.013084</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425189</th>\n",
       "      <td>309</td>\n",
       "      <td>V1.UC21-4_GL0157994</td>\n",
       "      <td>16491</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K00599</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>Amino Acid Metabolism</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56580</th>\n",
       "      <td>301</td>\n",
       "      <td>MH0313_GL0120834</td>\n",
       "      <td>16626</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG34255</td>\n",
       "      <td>0.010260</td>\n",
       "      <td>0.012150</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366267</th>\n",
       "      <td>290</td>\n",
       "      <td>O2.CD2-0_GL0055034</td>\n",
       "      <td>16704</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.011050</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259192</th>\n",
       "      <td>281</td>\n",
       "      <td>MH0230_GL0027875</td>\n",
       "      <td>16845</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.044988</td>\n",
       "      <td>0.053271</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363401</th>\n",
       "      <td>271</td>\n",
       "      <td>T2D-122A_GL0027378</td>\n",
       "      <td>16941</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>bactNOG98109</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.005607</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217952</th>\n",
       "      <td>266</td>\n",
       "      <td>MH0448_GL0184927</td>\n",
       "      <td>17049</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.002368</td>\n",
       "      <td>0.002804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399905</th>\n",
       "      <td>249</td>\n",
       "      <td>MH0250_GL0010325</td>\n",
       "      <td>17202</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.074980</td>\n",
       "      <td>0.076636</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458845</th>\n",
       "      <td>228</td>\n",
       "      <td>O2.UC22-1_GL0043947</td>\n",
       "      <td>17361</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.155485</td>\n",
       "      <td>0.157944</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339944</th>\n",
       "      <td>205</td>\n",
       "      <td>V1.UC63-0_GL0053529</td>\n",
       "      <td>17601</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13734</td>\n",
       "      <td>NOG244570</td>\n",
       "      <td>0.060773</td>\n",
       "      <td>0.070093</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103410</th>\n",
       "      <td>170</td>\n",
       "      <td>V1.CD20-0_GL0037007</td>\n",
       "      <td>18738</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.421468</td>\n",
       "      <td>0.402804</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498872</th>\n",
       "      <td>124</td>\n",
       "      <td>V1.FI17_GL0005370</td>\n",
       "      <td>19764</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.095501</td>\n",
       "      <td>0.095327</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412707</th>\n",
       "      <td>114</td>\n",
       "      <td>O2.UC50-0_GL0310928</td>\n",
       "      <td>19995</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.093923</td>\n",
       "      <td>0.092523</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126993</th>\n",
       "      <td>108</td>\n",
       "      <td>MH0328_GL0176549</td>\n",
       "      <td>20100</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.029203</td>\n",
       "      <td>0.032710</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64334</th>\n",
       "      <td>92</td>\n",
       "      <td>O2.UC43-0_GL0090208</td>\n",
       "      <td>20445</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>COG4646;COG0827</td>\n",
       "      <td>0.029203</td>\n",
       "      <td>0.021495</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transcription;Replication, recombination and r...</td>\n",
       "      <td>EUR;USA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18027</th>\n",
       "      <td>81</td>\n",
       "      <td>MH0276_GL0250660</td>\n",
       "      <td>20691</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0.006314</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83989</th>\n",
       "      <td>76</td>\n",
       "      <td>SZEY-48A_GL0052647</td>\n",
       "      <td>20778</td>\n",
       "      <td>Complete</td>\n",
       "      <td>CHN</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>K01317</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.203631</td>\n",
       "      <td>0.212150</td>\n",
       "      <td>Enzyme Families</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53975</th>\n",
       "      <td>74</td>\n",
       "      <td>MH0375_GL0055361</td>\n",
       "      <td>20790</td>\n",
       "      <td>Complete</td>\n",
       "      <td>EUR</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.036306</td>\n",
       "      <td>0.038318</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4901</th>\n",
       "      <td>42</td>\n",
       "      <td>ED16A_GL0163742</td>\n",
       "      <td>23358</td>\n",
       "      <td>Lack 3'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K14194</td>\n",
       "      <td>NOG12793</td>\n",
       "      <td>0.204420</td>\n",
       "      <td>0.207477</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>EUR;CHN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336287</th>\n",
       "      <td>6</td>\n",
       "      <td>SZEY-41A_GL0088289</td>\n",
       "      <td>41277</td>\n",
       "      <td>Lack 5'-end</td>\n",
       "      <td>CHN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>K13735</td>\n",
       "      <td>COG5184</td>\n",
       "      <td>0.864246</td>\n",
       "      <td>0.873832</td>\n",
       "      <td>Infectious Diseases</td>\n",
       "      <td>Cell cycle control, cell division, chromosome ...</td>\n",
       "      <td>EUR;CHN;USA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500000 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene ID                                          Gene Name  \\\n",
       "88836   9865705                              V1.CD6-0-PT_GL0096973   \n",
       "476498  9862460                                 O2.UC4-2_GL0053162   \n",
       "151345  9862491                                O2.UC43-0_GL0033611   \n",
       "34192   9877190       159551223-stool2_revised_C916401_1_gene84247   \n",
       "331083  9860951                                O2.UC21-1_GL0136539   \n",
       "280181  9857596                                   MH0415_GL0056639   \n",
       "483179  9878379  763577454-stool2_revised_scaffold23171_1_gene1...   \n",
       "370156  9853996                                   MH0335_GL0009364   \n",
       "409047  9860917                                 O2.UC2-0_GL0121235   \n",
       "34135   9872403                                    N003A_GL0007810   \n",
       "408969  9869351                                V1.UC52-0_GL0023463   \n",
       "81900   9855675                                   MH0373_GL0182676   \n",
       "15105   9848328                                   MH0143_GL0088163   \n",
       "53716   9877737      160502038-stool1_revised_C976100_1_gene185746   \n",
       "4930    9873886                                 T2D-113A_GL0111098   \n",
       "225835  9847115                                   MH0100_GL0041020   \n",
       "409075  9872514                                    N017A_GL0080793   \n",
       "151252  9853377                                   MH0318_GL0022747   \n",
       "53803   9856149                                   MH0384_GL0070245   \n",
       "151236  9855061                                   MH0362_GL0041997   \n",
       "197036  9846712                                   MH0086_GL0082324   \n",
       "34125   9858623                                   MH0435_GL0149202   \n",
       "197042  9859108                                   MH0445_GL0066112   \n",
       "116515  9879793                                411459.RUMOBE_01786   \n",
       "369973  9854153                                   MH0339_GL0008776   \n",
       "369662  9849875                                   MH0196_GL0163961   \n",
       "369663  9852450                                   MH0276_GL0235765   \n",
       "320790  9845699                                   MH0022_GL0010478   \n",
       "4686    9874722                                  T2D-29A_GL0000548   \n",
       "296144  9849405                                   MH0185_GL0064951   \n",
       "...         ...                                                ...   \n",
       "245450      412                                  V1.FI27_GL0164628   \n",
       "285369      399                                   MH0150_GL0033854   \n",
       "101624      392                                   MH0264_GL0016567   \n",
       "183807      377                                   MH0404_GL0080882   \n",
       "59169       349  159571453-stool2_revised_scaffold27522_1_gene3...   \n",
       "267886      340                                  V1.FI25_GL0154021   \n",
       "355168      331                                  V1.FI28_GL0063861   \n",
       "215409      321                                  V1.FI14_GL0255454   \n",
       "470957      318                                 O2.UC5-0_GL0058609   \n",
       "189070      317                                O2.UC40-1_GL0020660   \n",
       "164915      312                                V1.UC58-0_GL0107276   \n",
       "425189      309                                V1.UC21-4_GL0157994   \n",
       "56580       301                                   MH0313_GL0120834   \n",
       "366267      290                                 O2.CD2-0_GL0055034   \n",
       "259192      281                                   MH0230_GL0027875   \n",
       "363401      271                                 T2D-122A_GL0027378   \n",
       "217952      266                                   MH0448_GL0184927   \n",
       "399905      249                                   MH0250_GL0010325   \n",
       "458845      228                                O2.UC22-1_GL0043947   \n",
       "339944      205                                V1.UC63-0_GL0053529   \n",
       "103410      170                                V1.CD20-0_GL0037007   \n",
       "498872      124                                  V1.FI17_GL0005370   \n",
       "412707      114                                O2.UC50-0_GL0310928   \n",
       "126993      108                                   MH0328_GL0176549   \n",
       "64334        92                                O2.UC43-0_GL0090208   \n",
       "18027        81                                   MH0276_GL0250660   \n",
       "83989        76                                 SZEY-48A_GL0052647   \n",
       "53975        74                                   MH0375_GL0055361   \n",
       "4901         42                                    ED16A_GL0163742   \n",
       "336287        6                                 SZEY-41A_GL0088289   \n",
       "\n",
       "        Gene Length Gene Completeness Cohort Origin  \\\n",
       "88836           102       Lack 5'-end           EUR   \n",
       "476498          102       Lack 5'-end           EUR   \n",
       "151345          102       Lack 3'-end           EUR   \n",
       "34192           102       Lack 3'-end           USA   \n",
       "331083          102       Lack 3'-end           EUR   \n",
       "280181          102       Lack 3'-end           EUR   \n",
       "483179          102       Lack 5'-end           USA   \n",
       "370156          102          Complete           EUR   \n",
       "409047          102       Lack 5'-end           EUR   \n",
       "34135           102       Lack 5'-end           CHN   \n",
       "408969          102       Lack 5'-end           EUR   \n",
       "81900           102       Lack 5'-end           EUR   \n",
       "15105           102          Complete           EUR   \n",
       "53716           102       Lack 5'-end           USA   \n",
       "4930            102       Lack 5'-end           CHN   \n",
       "225835          102       Lack 5'-end           EUR   \n",
       "409075          102       Lack 5'-end           CHN   \n",
       "151252          102          Complete           EUR   \n",
       "53803           102          Complete           EUR   \n",
       "151236          102       Lack 3'-end           EUR   \n",
       "197036          102       Lack 3'-end           EUR   \n",
       "34125           102       Lack 5'-end           EUR   \n",
       "197042          102       Lack 5'-end           EUR   \n",
       "116515          102          Complete            SP   \n",
       "369973          102       Lack 5'-end           EUR   \n",
       "369662          102       Lack 3'-end           EUR   \n",
       "369663          102          Complete           EUR   \n",
       "320790          102       Lack 5'-end           EUR   \n",
       "4686            102       Lack 5'-end           CHN   \n",
       "296144          102       Lack 3'-end           EUR   \n",
       "...             ...               ...           ...   \n",
       "245450        15222       Lack 3'-end           EUR   \n",
       "285369        15306          Complete           EUR   \n",
       "101624        15447          Complete           EUR   \n",
       "183807        15591          Complete           EUR   \n",
       "59169         15945       Lack 3'-end           USA   \n",
       "267886        16074       Lack 3'-end           EUR   \n",
       "355168        16161          Complete           EUR   \n",
       "215409        16278       Lack 3'-end           EUR   \n",
       "470957        16311          Complete           EUR   \n",
       "189070        16344          Complete           EUR   \n",
       "164915        16419          Complete           EUR   \n",
       "425189        16491          Complete           EUR   \n",
       "56580         16626          Complete           EUR   \n",
       "366267        16704          Complete           EUR   \n",
       "259192        16845          Complete           EUR   \n",
       "363401        16941          Complete           CHN   \n",
       "217952        17049          Complete           EUR   \n",
       "399905        17202          Complete           EUR   \n",
       "458845        17361          Complete           EUR   \n",
       "339944        17601       Lack 3'-end           EUR   \n",
       "103410        18738          Complete           EUR   \n",
       "498872        19764          Complete           EUR   \n",
       "412707        19995          Complete           EUR   \n",
       "126993        20100          Complete           EUR   \n",
       "64334         20445          Complete           EUR   \n",
       "18027         20691          Complete           EUR   \n",
       "83989         20778          Complete           CHN   \n",
       "53975         20790          Complete           EUR   \n",
       "4901          23358       Lack 3'-end           CHN   \n",
       "336287        41277       Lack 5'-end           CHN   \n",
       "\n",
       "       Taxonomic Annotation(Phylum Level) Taxonomic Annotation(Genus Level)  \\\n",
       "88836                             unknown                           unknown   \n",
       "476498                            unknown                           unknown   \n",
       "151345                            unknown                           unknown   \n",
       "34192                          Firmicutes                       Eubacterium   \n",
       "331083                            unknown                           unknown   \n",
       "280181                            unknown                           unknown   \n",
       "483179                     Proteobacteria                           unknown   \n",
       "370156                            unknown                           unknown   \n",
       "409047                            unknown                           unknown   \n",
       "34135                             unknown                           unknown   \n",
       "408969                            unknown                           unknown   \n",
       "81900                             unknown                           unknown   \n",
       "15105                             unknown                           unknown   \n",
       "53716                             unknown                           unknown   \n",
       "4930                              unknown                           unknown   \n",
       "225835                            unknown                           unknown   \n",
       "409075                            unknown                           unknown   \n",
       "151252                            unknown                           unknown   \n",
       "53803                             unknown                           unknown   \n",
       "151236                            unknown                           unknown   \n",
       "197036                            unknown                           unknown   \n",
       "34125                          Firmicutes                  Faecalibacterium   \n",
       "197042                            unknown                           unknown   \n",
       "116515                         Firmicutes                           Blautia   \n",
       "369973                            unknown                           unknown   \n",
       "369662                            unknown                           unknown   \n",
       "369663                            unknown                           unknown   \n",
       "320790                            unknown                           unknown   \n",
       "4686                              unknown                           unknown   \n",
       "296144                            unknown                           unknown   \n",
       "...                                   ...                               ...   \n",
       "245450                            unknown                           unknown   \n",
       "285369                            unknown                           unknown   \n",
       "101624                            unknown                           unknown   \n",
       "183807                            unknown                           unknown   \n",
       "59169                             unknown                           unknown   \n",
       "267886                            unknown                           unknown   \n",
       "355168                            unknown                           unknown   \n",
       "215409                            unknown                           unknown   \n",
       "470957                            unknown                           unknown   \n",
       "189070                            unknown                           unknown   \n",
       "164915                            unknown                           unknown   \n",
       "425189                            unknown                           unknown   \n",
       "56580                             unknown                           unknown   \n",
       "366267                            unknown                           unknown   \n",
       "259192                            unknown                           unknown   \n",
       "363401                            unknown                           unknown   \n",
       "217952                            unknown                           unknown   \n",
       "399905                            unknown                           unknown   \n",
       "458845                      Bacteroidetes                       Bacteroides   \n",
       "339944                            unknown                           unknown   \n",
       "103410                     Proteobacteria                           unknown   \n",
       "498872                            unknown                           unknown   \n",
       "412707                            unknown                           unknown   \n",
       "126993                            unknown                           unknown   \n",
       "64334                             unknown                           unknown   \n",
       "18027                             unknown                           unknown   \n",
       "83989                      Proteobacteria                       Escherichia   \n",
       "53975                             unknown                           unknown   \n",
       "4901                              unknown                           unknown   \n",
       "336287                            unknown                           unknown   \n",
       "\n",
       "       KEGG Annotation eggNOG Annotation  Sample Occurence Frequency  \\\n",
       "88836          unknown           unknown                    0.005525   \n",
       "476498         unknown           unknown                    0.021310   \n",
       "151345         unknown           unknown                    0.015785   \n",
       "34192          unknown           unknown                    0.014207   \n",
       "331083         unknown           unknown                    0.059195   \n",
       "280181         unknown           unknown                    0.020521   \n",
       "483179         unknown           unknown                    0.169692   \n",
       "370156         unknown           unknown                    0.004736   \n",
       "409047         unknown           unknown                    0.011050   \n",
       "34135          unknown           unknown                    0.004736   \n",
       "408969         unknown           unknown                    0.002368   \n",
       "81900          unknown           unknown                    0.022889   \n",
       "15105          unknown           unknown                    0.003946   \n",
       "53716          unknown           unknown                    0.000000   \n",
       "4930           unknown           unknown                    0.115233   \n",
       "225835         unknown           unknown                    0.047356   \n",
       "409075         unknown           unknown                    0.072612   \n",
       "151252         unknown           unknown                    0.000789   \n",
       "53803          unknown           unknown                    0.001579   \n",
       "151236         unknown           unknown                    0.089187   \n",
       "197036         unknown           unknown                    0.140489   \n",
       "34125          unknown           unknown                    0.089976   \n",
       "197042         unknown           unknown                    0.033149   \n",
       "116515         unknown           unknown                    0.004736   \n",
       "369973         unknown           unknown                    0.101815   \n",
       "369662         unknown           unknown                    0.013418   \n",
       "369663         unknown           unknown                    0.004736   \n",
       "320790         unknown           unknown                    0.005525   \n",
       "4686           unknown           unknown                    0.160221   \n",
       "296144         unknown           unknown                    0.228887   \n",
       "...                ...               ...                         ...   \n",
       "245450         unknown      bactNOG98109                    0.053670   \n",
       "285369          K13734      bactNOG74050                    0.823204   \n",
       "101624         unknown      bactNOG23571                    0.009471   \n",
       "183807          K00558      bactNOG23571                    0.011050   \n",
       "59169           K07279           COG3468                    0.048934   \n",
       "267886          K15125          NOG12793                    0.013418   \n",
       "355168          K00599   COG4646;COG0827                    0.032360   \n",
       "215409         unknown          NOG12793                    0.022099   \n",
       "470957         unknown          NOG12793                    0.032360   \n",
       "189070         unknown          NOG12793                    0.047356   \n",
       "164915          K00558      bactNOG23571                    0.011839   \n",
       "425189          K00599   COG4646;COG0827                    0.004736   \n",
       "56580          unknown      bactNOG34255                    0.010260   \n",
       "366267         unknown   COG4646;COG0827                    0.011050   \n",
       "259192         unknown      bactNOG98109                    0.044988   \n",
       "363401         unknown      bactNOG98109                    0.004736   \n",
       "217952         unknown           unknown                    0.002368   \n",
       "399905         unknown           unknown                    0.074980   \n",
       "458845         unknown   COG4646;COG0827                    0.155485   \n",
       "339944          K13734         NOG244570                    0.060773   \n",
       "103410         unknown          NOG12793                    0.421468   \n",
       "498872         unknown           unknown                    0.095501   \n",
       "412707         unknown           unknown                    0.093923   \n",
       "126993         unknown          NOG12793                    0.029203   \n",
       "64334          unknown   COG4646;COG0827                    0.029203   \n",
       "18027          unknown           unknown                    0.006314   \n",
       "83989           K01317          NOG12793                    0.203631   \n",
       "53975          unknown          NOG12793                    0.036306   \n",
       "4901            K14194          NOG12793                    0.204420   \n",
       "336287          K13735           COG5184                    0.864246   \n",
       "\n",
       "        Individual Occurence Frequency        KEGG Functional Categories  \\\n",
       "88836                         0.005607                           unknown   \n",
       "476498                        0.020561                           unknown   \n",
       "151345                        0.014953                           unknown   \n",
       "34192                         0.012150                           unknown   \n",
       "331083                        0.061682                           unknown   \n",
       "280181                        0.023364                           unknown   \n",
       "483179                        0.157944                           unknown   \n",
       "370156                        0.005607                           unknown   \n",
       "409047                        0.007477                           unknown   \n",
       "34135                         0.005607                           unknown   \n",
       "408969                        0.001869                           unknown   \n",
       "81900                         0.023364                           unknown   \n",
       "15105                         0.004673                           unknown   \n",
       "53716                         0.000000                           unknown   \n",
       "4930                          0.116822                           unknown   \n",
       "225835                        0.043925                           unknown   \n",
       "409075                        0.065421                           unknown   \n",
       "151252                        0.000935                           unknown   \n",
       "53803                         0.001869                           unknown   \n",
       "151236                        0.091589                           unknown   \n",
       "197036                        0.115888                           unknown   \n",
       "34125                         0.088785                           unknown   \n",
       "197042                        0.036449                           unknown   \n",
       "116515                        0.003738                           unknown   \n",
       "369973                        0.079439                           unknown   \n",
       "369662                        0.014019                           unknown   \n",
       "369663                        0.005607                           unknown   \n",
       "320790                        0.005607                           unknown   \n",
       "4686                          0.163551                           unknown   \n",
       "296144                        0.232710                           unknown   \n",
       "...                                ...                               ...   \n",
       "245450                        0.055140                           unknown   \n",
       "285369                        0.835514               Infectious Diseases   \n",
       "101624                        0.010280                           unknown   \n",
       "183807                        0.013084             Amino Acid Metabolism   \n",
       "59169                         0.050467  Cellular Processes and Signaling   \n",
       "267886                        0.015888               Infectious Diseases   \n",
       "355168                        0.028037             Amino Acid Metabolism   \n",
       "215409                        0.026168                           unknown   \n",
       "470957                        0.023364                           unknown   \n",
       "189070                        0.042056                           unknown   \n",
       "164915                        0.013084             Amino Acid Metabolism   \n",
       "425189                        0.003738             Amino Acid Metabolism   \n",
       "56580                         0.012150                           unknown   \n",
       "366267                        0.007477                           unknown   \n",
       "259192                        0.053271                           unknown   \n",
       "363401                        0.005607                           unknown   \n",
       "217952                        0.002804                           unknown   \n",
       "399905                        0.076636                           unknown   \n",
       "458845                        0.157944                           unknown   \n",
       "339944                        0.070093               Infectious Diseases   \n",
       "103410                        0.402804                           unknown   \n",
       "498872                        0.095327                           unknown   \n",
       "412707                        0.092523                           unknown   \n",
       "126993                        0.032710                           unknown   \n",
       "64334                         0.021495                           unknown   \n",
       "18027                         0.007477                           unknown   \n",
       "83989                         0.212150                   Enzyme Families   \n",
       "53975                         0.038318                           unknown   \n",
       "4901                          0.207477               Infectious Diseases   \n",
       "336287                        0.873832               Infectious Diseases   \n",
       "\n",
       "                             eggNOG Functional Categories Cohort Assembled  \n",
       "88836                                             unknown              EUR  \n",
       "476498                                            unknown              EUR  \n",
       "151345                                            unknown              EUR  \n",
       "34192                                             unknown              USA  \n",
       "331083                                            unknown              EUR  \n",
       "280181                                            unknown              EUR  \n",
       "483179                                            unknown              USA  \n",
       "370156                                            unknown              EUR  \n",
       "409047                                            unknown              EUR  \n",
       "34135                                             unknown              CHN  \n",
       "408969                                            unknown              EUR  \n",
       "81900                                             unknown              EUR  \n",
       "15105                                             unknown              EUR  \n",
       "53716                                             unknown              USA  \n",
       "4930                                              unknown              CHN  \n",
       "225835                                            unknown              EUR  \n",
       "409075                                            unknown              CHN  \n",
       "151252                                            unknown              EUR  \n",
       "53803                                             unknown              EUR  \n",
       "151236                                            unknown              EUR  \n",
       "197036                                            unknown              EUR  \n",
       "34125                                             unknown              EUR  \n",
       "197042                                            unknown              EUR  \n",
       "116515                                            unknown              NaN  \n",
       "369973                                            unknown              EUR  \n",
       "369662                                            unknown              EUR  \n",
       "369663                                            unknown              EUR  \n",
       "320790                                            unknown              EUR  \n",
       "4686                                              unknown              CHN  \n",
       "296144                                            unknown              EUR  \n",
       "...                                                   ...              ...  \n",
       "245450                                   Function unknown              EUR  \n",
       "285369             Cell wall/membrane/envelope biogenesis      EUR;CHN;USA  \n",
       "101624                                   Function unknown              EUR  \n",
       "183807                                   Function unknown              EUR  \n",
       "59169   Cell wall/membrane/envelope biogenesis;Intrace...              USA  \n",
       "267886                                   Function unknown              EUR  \n",
       "355168  Transcription;Replication, recombination and r...          EUR;CHN  \n",
       "215409                                   Function unknown              EUR  \n",
       "470957                                   Function unknown              EUR  \n",
       "189070                                   Function unknown              EUR  \n",
       "164915                                   Function unknown              EUR  \n",
       "425189  Transcription;Replication, recombination and r...          EUR;CHN  \n",
       "56580                                    Function unknown              EUR  \n",
       "366267  Transcription;Replication, recombination and r...          EUR;USA  \n",
       "259192                                   Function unknown          EUR;CHN  \n",
       "363401                                   Function unknown              CHN  \n",
       "217952                                            unknown              EUR  \n",
       "399905                                            unknown              EUR  \n",
       "458845  Transcription;Replication, recombination and r...              EUR  \n",
       "339944                                   Function unknown          EUR;CHN  \n",
       "103410                                   Function unknown      EUR;CHN;USA  \n",
       "498872                                            unknown          EUR;USA  \n",
       "412707                                            unknown              EUR  \n",
       "126993                                   Function unknown              EUR  \n",
       "64334   Transcription;Replication, recombination and r...          EUR;USA  \n",
       "18027                                             unknown              EUR  \n",
       "83989                                    Function unknown          EUR;CHN  \n",
       "53975                                    Function unknown              EUR  \n",
       "4901                                     Function unknown          EUR;CHN  \n",
       "336287  Cell cycle control, cell division, chromosome ...      EUR;CHN;USA  \n",
       "\n",
       "[500000 rows x 14 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by='Gene Length', ascending=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Funciones utilizadas el días de hoy**\n",
    "\n",
    "* `df.head()`\n",
    "* `df.tail()`\n",
    "* `df.dtypes`\n",
    "* `df.shape`\n",
    "* `df.info()`\n",
    "* `df.describe()`\n",
    "* `df.T`\n",
    "* `df.index`\n",
    "* `df.columns`\n",
    "* `df.sort_index()`\n",
    "* `df.sort_values()`\n",
    "* `df.to_csv()`"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
